<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.45">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Philipp Sven Lars Schäfer">
<meta name="dcterms.date" content="2024-09-23">

<title>Batch Effects – PSLS @ 3rd CMI-PB Challenge</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">PSLS @ 3rd CMI-PB Challenge</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/psl-schaefer"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://x.com/psl_schaefer"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/philipp-sl-schaefer/"> <i class="bi bi-linkedin" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="1">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#packages" id="toc-packages" class="nav-link active" data-scroll-target="#packages"><span class="header-section-number">1</span> Packages</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2</span> Data</a></li>
  <li><a href="#conclusions" id="toc-conclusions" class="nav-link" data-scroll-target="#conclusions"><span class="header-section-number">3</span> Conclusions</a>
  <ul class="collapse">
  <li><a href="#pbmc-fractions" id="toc-pbmc-fractions" class="nav-link" data-scroll-target="#pbmc-fractions"><span class="header-section-number">3.1</span> PBMC Fractions</a></li>
  <li><a href="#pbmc-gene-expression" id="toc-pbmc-gene-expression" class="nav-link" data-scroll-target="#pbmc-gene-expression"><span class="header-section-number">3.2</span> PBMC Gene Expression</a></li>
  <li><a href="#plasma-antibody-levels" id="toc-plasma-antibody-levels" class="nav-link" data-scroll-target="#plasma-antibody-levels"><span class="header-section-number">3.3</span> Plasma Antibody Levels</a></li>
  <li><a href="#plasma-cytokine-concentration-by-legendplex" id="toc-plasma-cytokine-concentration-by-legendplex" class="nav-link" data-scroll-target="#plasma-cytokine-concentration-by-legendplex"><span class="header-section-number">3.4</span> Plasma Cytokine Concentration by Legendplex</a></li>
  <li><a href="#plasma-cytokine-concentration-by-olink" id="toc-plasma-cytokine-concentration-by-olink" class="nav-link" data-scroll-target="#plasma-cytokine-concentration-by-olink"><span class="header-section-number">3.5</span> Plasma Cytokine Concentration by Olink</a></li>
  <li><a href="#t-cell-activation" id="toc-t-cell-activation" class="nav-link" data-scroll-target="#t-cell-activation"><span class="header-section-number">3.6</span> T Cell Activation</a></li>
  <li><a href="#t-cell-polarization" id="toc-t-cell-polarization" class="nav-link" data-scroll-target="#t-cell-polarization"><span class="header-section-number">3.7</span> T Cell Polarization</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">4</span> Results</a>
  <ul class="collapse">
  <li><a href="#day_0" id="toc-day_0" class="nav-link" data-scroll-target="#day_0"><span class="header-section-number">4.1</span> day_0</a>
  <ul class="collapse">
  <li><a href="#pbmc_cell_frequency" id="toc-pbmc_cell_frequency" class="nav-link" data-scroll-target="#pbmc_cell_frequency"><span class="header-section-number">4.1.1</span> pbmc_cell_frequency</a></li>
  <li><a href="#pbmc_gene_expression" id="toc-pbmc_gene_expression" class="nav-link" data-scroll-target="#pbmc_gene_expression"><span class="header-section-number">4.1.2</span> pbmc_gene_expression</a></li>
  <li><a href="#plasma_ab_titer" id="toc-plasma_ab_titer" class="nav-link" data-scroll-target="#plasma_ab_titer"><span class="header-section-number">4.1.3</span> plasma_ab_titer</a></li>
  <li><a href="#plasma_cytokine_concentration_by_legendplex" id="toc-plasma_cytokine_concentration_by_legendplex" class="nav-link" data-scroll-target="#plasma_cytokine_concentration_by_legendplex"><span class="header-section-number">4.1.4</span> plasma_cytokine_concentration_by_legendplex</a></li>
  <li><a href="#plasma_cytokine_concentration_by_olink" id="toc-plasma_cytokine_concentration_by_olink" class="nav-link" data-scroll-target="#plasma_cytokine_concentration_by_olink"><span class="header-section-number">4.1.5</span> plasma_cytokine_concentration_by_olink</a></li>
  <li><a href="#t_cell_activation" id="toc-t_cell_activation" class="nav-link" data-scroll-target="#t_cell_activation"><span class="header-section-number">4.1.6</span> t_cell_activation</a></li>
  <li><a href="#t_cell_polarization" id="toc-t_cell_polarization" class="nav-link" data-scroll-target="#t_cell_polarization"><span class="header-section-number">4.1.7</span> t_cell_polarization</a></li>
  </ul></li>
  <li><a href="#day_1" id="toc-day_1" class="nav-link" data-scroll-target="#day_1"><span class="header-section-number">4.2</span> day_1</a>
  <ul class="collapse">
  <li><a href="#pbmc_cell_frequency-1" id="toc-pbmc_cell_frequency-1" class="nav-link" data-scroll-target="#pbmc_cell_frequency-1"><span class="header-section-number">4.2.1</span> pbmc_cell_frequency</a></li>
  <li><a href="#pbmc_gene_expression-1" id="toc-pbmc_gene_expression-1" class="nav-link" data-scroll-target="#pbmc_gene_expression-1"><span class="header-section-number">4.2.2</span> pbmc_gene_expression</a></li>
  <li><a href="#plasma_ab_titer-1" id="toc-plasma_ab_titer-1" class="nav-link" data-scroll-target="#plasma_ab_titer-1"><span class="header-section-number">4.2.3</span> plasma_ab_titer</a></li>
  <li><a href="#plasma_cytokine_concentration_by_legendplex-1" id="toc-plasma_cytokine_concentration_by_legendplex-1" class="nav-link" data-scroll-target="#plasma_cytokine_concentration_by_legendplex-1"><span class="header-section-number">4.2.4</span> plasma_cytokine_concentration_by_legendplex</a></li>
  <li><a href="#plasma_cytokine_concentration_by_olink-1" id="toc-plasma_cytokine_concentration_by_olink-1" class="nav-link" data-scroll-target="#plasma_cytokine_concentration_by_olink-1"><span class="header-section-number">4.2.5</span> plasma_cytokine_concentration_by_olink</a></li>
  <li><a href="#t_cell_activation-1" id="toc-t_cell_activation-1" class="nav-link" data-scroll-target="#t_cell_activation-1"><span class="header-section-number">4.2.6</span> t_cell_activation</a></li>
  <li><a href="#t_cell_polarization-1" id="toc-t_cell_polarization-1" class="nav-link" data-scroll-target="#t_cell_polarization-1"><span class="header-section-number">4.2.7</span> t_cell_polarization</a></li>
  </ul></li>
  <li><a href="#day_3" id="toc-day_3" class="nav-link" data-scroll-target="#day_3"><span class="header-section-number">4.3</span> day_3</a>
  <ul class="collapse">
  <li><a href="#pbmc_cell_frequency-2" id="toc-pbmc_cell_frequency-2" class="nav-link" data-scroll-target="#pbmc_cell_frequency-2"><span class="header-section-number">4.3.1</span> pbmc_cell_frequency</a></li>
  <li><a href="#pbmc_gene_expression-2" id="toc-pbmc_gene_expression-2" class="nav-link" data-scroll-target="#pbmc_gene_expression-2"><span class="header-section-number">4.3.2</span> pbmc_gene_expression</a></li>
  <li><a href="#plasma_ab_titer-2" id="toc-plasma_ab_titer-2" class="nav-link" data-scroll-target="#plasma_ab_titer-2"><span class="header-section-number">4.3.3</span> plasma_ab_titer</a></li>
  <li><a href="#plasma_cytokine_concentration_by_legendplex-2" id="toc-plasma_cytokine_concentration_by_legendplex-2" class="nav-link" data-scroll-target="#plasma_cytokine_concentration_by_legendplex-2"><span class="header-section-number">4.3.4</span> plasma_cytokine_concentration_by_legendplex</a></li>
  <li><a href="#plasma_cytokine_concentration_by_olink-2" id="toc-plasma_cytokine_concentration_by_olink-2" class="nav-link" data-scroll-target="#plasma_cytokine_concentration_by_olink-2"><span class="header-section-number">4.3.5</span> plasma_cytokine_concentration_by_olink</a></li>
  <li><a href="#t_cell_activation-2" id="toc-t_cell_activation-2" class="nav-link" data-scroll-target="#t_cell_activation-2"><span class="header-section-number">4.3.6</span> t_cell_activation</a></li>
  <li><a href="#t_cell_polarization-2" id="toc-t_cell_polarization-2" class="nav-link" data-scroll-target="#t_cell_polarization-2"><span class="header-section-number">4.3.7</span> t_cell_polarization</a></li>
  </ul></li>
  <li><a href="#day_14" id="toc-day_14" class="nav-link" data-scroll-target="#day_14"><span class="header-section-number">4.4</span> day_14</a>
  <ul class="collapse">
  <li><a href="#pbmc_cell_frequency-3" id="toc-pbmc_cell_frequency-3" class="nav-link" data-scroll-target="#pbmc_cell_frequency-3"><span class="header-section-number">4.4.1</span> pbmc_cell_frequency</a></li>
  <li><a href="#pbmc_gene_expression-3" id="toc-pbmc_gene_expression-3" class="nav-link" data-scroll-target="#pbmc_gene_expression-3"><span class="header-section-number">4.4.2</span> pbmc_gene_expression</a></li>
  <li><a href="#plasma_ab_titer-3" id="toc-plasma_ab_titer-3" class="nav-link" data-scroll-target="#plasma_ab_titer-3"><span class="header-section-number">4.4.3</span> plasma_ab_titer</a></li>
  <li><a href="#plasma_cytokine_concentration_by_legendplex-3" id="toc-plasma_cytokine_concentration_by_legendplex-3" class="nav-link" data-scroll-target="#plasma_cytokine_concentration_by_legendplex-3"><span class="header-section-number">4.4.4</span> plasma_cytokine_concentration_by_legendplex</a></li>
  <li><a href="#plasma_cytokine_concentration_by_olink-3" id="toc-plasma_cytokine_concentration_by_olink-3" class="nav-link" data-scroll-target="#plasma_cytokine_concentration_by_olink-3"><span class="header-section-number">4.4.5</span> plasma_cytokine_concentration_by_olink</a></li>
  <li><a href="#t_cell_activation-3" id="toc-t_cell_activation-3" class="nav-link" data-scroll-target="#t_cell_activation-3"><span class="header-section-number">4.4.6</span> t_cell_activation</a></li>
  <li><a href="#t_cell_polarization-3" id="toc-t_cell_polarization-3" class="nav-link" data-scroll-target="#t_cell_polarization-3"><span class="header-section-number">4.4.7</span> t_cell_polarization</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">5</span> Appendix</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Batch Effects</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Philipp Sven Lars Schäfer </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 23, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="packages" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Packages</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(vsn)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(lme4)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(DESeq2)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick) <span class="co"># formatting</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>knit_hooks<span class="sc">$</span><span class="fu">set</span>(<span class="at">crop =</span> knitr<span class="sc">::</span>hook_pdfcrop) <span class="co"># formatting</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"generate_targets.R"</span>))</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"model.R"</span>))</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"batch_effects.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Data</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>experimental_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>experimental_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(<span class="at">meta_data=</span>meta_data, </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">experimental_data=</span>experimental_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>pbmc_cell_frequency | Removed 550 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 6931 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 495 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>pbmc_cell_frequency | Removed 56 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 48 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 234 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>t_cell_polarization | Removed 3 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 300 features because qc warning</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 10540 measurements because wrong unit used</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 2400 measurements because wrong unit used</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_legendplex | Removed 8 because specimen is outlier</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>specimen_per_day <span class="ot">&lt;-</span> <span class="fu">get_specimen_per_day</span>(<span class="at">meta_data=</span>meta_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="conclusions" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Conclusions</h1>
<section id="pbmc-fractions" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="pbmc-fractions"><span class="header-section-number">3.1</span> PBMC Fractions</h2>
<ul>
<li>For the baseline specimen (day 0), we see a very strong batch effect, because the 2023 dataset is very different</li>
</ul>
</section>
<section id="pbmc-gene-expression" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="pbmc-gene-expression"><span class="header-section-number">3.2</span> PBMC Gene Expression</h2>
<ul>
<li>Very strong cohort-specific effects for baseline specimen and post-booster specimen</li>
</ul>
</section>
<section id="plasma-antibody-levels" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="plasma-antibody-levels"><span class="header-section-number">3.3</span> Plasma Antibody Levels</h2>
</section>
<section id="plasma-cytokine-concentration-by-legendplex" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="plasma-cytokine-concentration-by-legendplex"><span class="header-section-number">3.4</span> Plasma Cytokine Concentration by Legendplex</h2>
<ul>
<li>After removing outlier in the legendplex data…</li>
</ul>
</section>
<section id="plasma-cytokine-concentration-by-olink" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="plasma-cytokine-concentration-by-olink"><span class="header-section-number">3.5</span> Plasma Cytokine Concentration by Olink</h2>
<ul>
<li>At baseline: About 6.5% of the variance is explained by the dataset, probably also due to 3 outliers from the 2022 datasets. TODO: Investigate this in more detail.</li>
</ul>
</section>
<section id="t-cell-activation" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="t-cell-activation"><span class="header-section-number">3.6</span> T Cell Activation</h2>
<ul>
<li>Specimen are only available for baseline, no cohort-specific effect</li>
</ul>
</section>
<section id="t-cell-polarization" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="t-cell-polarization"><span class="header-section-number">3.7</span> T Cell Polarization</h2>
<ul>
<li>Specimen are only available for baseline, most of the variance is explained by interaction terms <code>age_at_boost:dataset</code> and <code>infancy_vac:dataset</code>. TODO: Think about what to make of this.</li>
</ul>
</section>
</section>
<section id="results" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Results</h1>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">##| crop: true # interfers with dynamic toc generation</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (day <span class="cf">in</span> <span class="fu">names</span>(specimen_per_day)) {</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># day &lt;- names(specimen_per_day)[1]</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"## "</span>, day))</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  specimen_df <span class="ot">&lt;-</span> specimen_per_day[[day]]</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (assay <span class="cf">in</span> <span class="fu">names</span>(experimental_data)) {</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># assay &lt;- names(experimental_data)[4]</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"### "</span>, assay))</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>    feature_col <span class="ot">&lt;-</span> experimental_data_settings[[assay]]<span class="sc">$</span>feature_col</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>    value_col <span class="ot">&lt;-</span> experimental_data_settings[[assay]]<span class="sc">$</span>value_col</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>    assay_df_long <span class="ot">&lt;-</span> experimental_data[[assay]]</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>    <span class="co">#str(assay_df_long)</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>    specimen_df_filtered <span class="ot">&lt;-</span> specimen_df <span class="sc">%&gt;%</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(specimen_id <span class="sc">%in%</span> <span class="fu">unique</span>(assay_df_long<span class="sc">$</span>specimen_id))</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(specimen_df_filtered) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"No specimen for this combination of assay and day available"</span>))</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>      assay_mtx_wide <span class="ot">&lt;-</span> assay_df_long <span class="sc">%&gt;%</span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"specimen_id"</span>, feature_col, value_col))) <span class="sc">%&gt;%</span></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>        tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(feature_col), </span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>                           <span class="at">values_from=</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(value_col)) <span class="sc">%&gt;%</span></span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>        tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">as.matrix</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>        .[<span class="fu">as.character</span>(specimen_df_filtered<span class="sc">$</span>specimen_id), ]</span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>      impute_mode <span class="ot">&lt;-</span> <span class="st">"knn"</span></span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stopifnot</span>(impute_mode <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"zero"</span>, <span class="st">"median_feature"</span>, <span class="st">"knn"</span>))</span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">any</span>(<span class="fu">is.na</span>(assay_mtx_wide))) {</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"Non-zero fraction of NAs: "</span>, </span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">round</span>(<span class="fu">mean</span>(<span class="fu">is.na</span>(assay_mtx_wide)), <span class="dv">4</span>)))</span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>        <span class="co"># data: matrix with genes in the rows, samples in the columns</span></span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (impute_mode <span class="sc">==</span> <span class="st">"zero"</span>) {</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>          assay_mtx_wide[<span class="fu">is.na</span>(assay_mtx_wide)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span> (impute_mode <span class="sc">==</span> <span class="st">"knn"</span>) {</span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>          assay_mtx_wide <span class="ot">&lt;-</span> <span class="fu">t</span>(impute<span class="sc">::</span><span class="fu">impute.knn</span>(<span class="at">data=</span><span class="fu">t</span>(assay_mtx_wide))<span class="sc">$</span>data)</span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span> (impute_mode <span class="sc">==</span> <span class="st">"median_feature"</span>) {</span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>          col_medians <span class="ot">&lt;-</span> matrixStats<span class="sc">::</span><span class="fu">colMedians</span>(assay_mtx_wide, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>          <span class="cf">for</span> (colname <span class="cf">in</span> <span class="fu">colnames</span>(assay_mtx_wide)) {</span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>            col_vec <span class="ot">&lt;-</span> assay_mtx_wide[, colname]</span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>            col_vec[<span class="fu">is.na</span>(col_vec)] <span class="ot">&lt;-</span> col_medians[colname]</span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>            assay_mtx_wide[, colname] <span class="ot">&lt;-</span> col_vec</span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stopifnot</span>(<span class="sc">!</span><span class="fu">any</span>(<span class="fu">is.na</span>(assay_mtx_wide)))</span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (assay <span class="sc">==</span> <span class="st">"pbmc_gene_expression"</span>) {</span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a>        <span class="co"># normalization using vst from DEseq2</span></span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a>        <span class="co">#str(assay_mtx_wide)</span></span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-65"><a href="#cb16-65" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create DESeq dataset object</span></span>
<span id="cb16-66"><a href="#cb16-66" aria-hidden="true" tabindex="-1"></a>        min_counts <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb16-67"><a href="#cb16-67" aria-hidden="true" tabindex="-1"></a>        min_samples <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb16-68"><a href="#cb16-68" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb16-69"><a href="#cb16-69" aria-hidden="true" tabindex="-1"></a>        genes_to_keep <span class="ot">&lt;-</span> <span class="fu">colnames</span>(assay_mtx_wide)[</span>
<span id="cb16-70"><a href="#cb16-70" aria-hidden="true" tabindex="-1"></a>          (<span class="fu">colSums</span>(assay_mtx_wide) <span class="sc">&gt;=</span> min_counts) <span class="sc">&amp;</span></span>
<span id="cb16-71"><a href="#cb16-71" aria-hidden="true" tabindex="-1"></a>            (<span class="fu">colSums</span>(assay_mtx_wide <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">&gt;=</span> min_samples)</span>
<span id="cb16-72"><a href="#cb16-72" aria-hidden="true" tabindex="-1"></a>        ]</span>
<span id="cb16-73"><a href="#cb16-73" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> assay_mtx_wide[ , genes_to_keep]</span>
<span id="cb16-74"><a href="#cb16-74" aria-hidden="true" tabindex="-1"></a>        <span class="co">#str(assay_mtx_wide)</span></span>
<span id="cb16-75"><a href="#cb16-75" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-76"><a href="#cb16-76" aria-hidden="true" tabindex="-1"></a>        <span class="co"># check that all column (gene) names are unique</span></span>
<span id="cb16-77"><a href="#cb16-77" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stopifnot</span>(<span class="fu">length</span>(<span class="fu">colnames</span>(assay_mtx_wide)) <span class="sc">==</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">colnames</span>(assay_mtx_wide))))</span>
<span id="cb16-78"><a href="#cb16-78" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-79"><a href="#cb16-79" aria-hidden="true" tabindex="-1"></a>        ensemble_to_gene <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">versioned_ensembl_gene_id =</span> <span class="fu">colnames</span>(assay_mtx_wide)) <span class="sc">%&gt;%</span></span>
<span id="cb16-80"><a href="#cb16-80" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene_meta, <span class="at">by=</span><span class="st">"versioned_ensembl_gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-81"><a href="#cb16-81" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_symbol) <span class="sc">%&gt;%</span></span>
<span id="cb16-82"><a href="#cb16-82" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb16-83"><a href="#cb16-83" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(n), gene_symbol) <span class="sc">%&gt;%</span></span>
<span id="cb16-84"><a href="#cb16-84" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-85"><a href="#cb16-85" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span>
<span id="cb16-86"><a href="#cb16-86" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-87"><a href="#cb16-87" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> assay_mtx_wide[ , ensemble_to_gene<span class="sc">$</span>versioned_ensembl_gene_id]</span>
<span id="cb16-88"><a href="#cb16-88" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(assay_mtx_wide) <span class="ot">&lt;-</span> ensemble_to_gene<span class="sc">$</span>gene_symbol</span>
<span id="cb16-89"><a href="#cb16-89" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-90"><a href="#cb16-90" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stopifnot</span>(<span class="fu">length</span>(<span class="fu">colnames</span>(assay_mtx_wide)) <span class="sc">==</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">colnames</span>(assay_mtx_wide))))</span>
<span id="cb16-91"><a href="#cb16-91" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-92"><a href="#cb16-92" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> </span>
<span id="cb16-93"><a href="#cb16-93" aria-hidden="true" tabindex="-1"></a>          DESeq2<span class="sc">::</span><span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">t</span>(assay_mtx_wide),</span>
<span id="cb16-94"><a href="#cb16-94" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">colData =</span> tibble<span class="sc">::</span><span class="fu">tibble</span>(<span class="at">specimen_id=</span><span class="fu">rownames</span>(assay_mtx_wide)),</span>
<span id="cb16-95"><a href="#cb16-95" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">design =</span> <span class="sc">~</span> <span class="dv">1</span>)</span>
<span id="cb16-96"><a href="#cb16-96" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb16-97"><a href="#cb16-97" aria-hidden="true" tabindex="-1"></a>        <span class="co"># DESeq2 normalization with variance stabilizing transformation (vst)</span></span>
<span id="cb16-98"><a href="#cb16-98" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">estimateSizeFactors</span>(assay_mtx_wide, <span class="at">quiet=</span><span class="cn">TRUE</span>) <span class="do">## estimate size factor</span></span>
<span id="cb16-99"><a href="#cb16-99" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">estimateDispersions</span>(assay_mtx_wide, <span class="at">quiet=</span><span class="cn">TRUE</span>)</span>
<span id="cb16-100"><a href="#cb16-100" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">varianceStabilizingTransformation</span>(assay_mtx_wide, <span class="at">blind=</span><span class="cn">TRUE</span>)</span>
<span id="cb16-101"><a href="#cb16-101" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">assay</span>(assay_mtx_wide))</span>
<span id="cb16-102"><a href="#cb16-102" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb16-103"><a href="#cb16-103" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb16-104"><a href="#cb16-104" aria-hidden="true" tabindex="-1"></a>      <span class="co"># dist_mtx &lt;- as.matrix(dist(assay_mtx_wide))</span></span>
<span id="cb16-105"><a href="#cb16-105" aria-hidden="true" tabindex="-1"></a>      <span class="co"># dist_mtx_median &lt;- median(dist_mtx[lower.tri(dist_mtx)])</span></span>
<span id="cb16-106"><a href="#cb16-106" aria-hidden="true" tabindex="-1"></a>      <span class="co"># dist_mtx_binary &lt;- dist_mtx &gt; (3 * dist_mtx_median)</span></span>
<span id="cb16-107"><a href="#cb16-107" aria-hidden="true" tabindex="-1"></a>      <span class="co"># outliers &lt;-</span></span>
<span id="cb16-108"><a href="#cb16-108" aria-hidden="true" tabindex="-1"></a>      <span class="co">#  which(rowSums(dist_mtx_binary) &gt; (10*median(rowSums(dist_mtx_binary))))</span></span>
<span id="cb16-109"><a href="#cb16-109" aria-hidden="true" tabindex="-1"></a>      <span class="co"># no_outliers &lt;- rownames(dist_mtx)[!rownames(dist_mtx) %in% names(outliers)]</span></span>
<span id="cb16-110"><a href="#cb16-110" aria-hidden="true" tabindex="-1"></a>      <span class="co"># assay_mtx_wide &lt;- assay_mtx_wide[no_outliers, ]</span></span>
<span id="cb16-111"><a href="#cb16-111" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb16-112"><a href="#cb16-112" aria-hidden="true" tabindex="-1"></a>      <span class="co">#str(assay_mtx_wide)</span></span>
<span id="cb16-113"><a href="#cb16-113" aria-hidden="true" tabindex="-1"></a>      <span class="fu">suppressMessages</span>({</span>
<span id="cb16-114"><a href="#cb16-114" aria-hidden="true" tabindex="-1"></a>        <span class="fu">suppressWarnings</span>({</span>
<span id="cb16-115"><a href="#cb16-115" aria-hidden="true" tabindex="-1"></a>          p1 <span class="ot">&lt;-</span> <span class="fu">pvca_analysis</span>(<span class="at">assay_mtx_wide=</span>assay_mtx_wide,</span>
<span id="cb16-116"><a href="#cb16-116" aria-hidden="true" tabindex="-1"></a>                              <span class="at">meta_data=</span>meta_data)<span class="sc">$</span>p1</span>
<span id="cb16-117"><a href="#cb16-117" aria-hidden="true" tabindex="-1"></a>        })</span>
<span id="cb16-118"><a href="#cb16-118" aria-hidden="true" tabindex="-1"></a>      })</span>
<span id="cb16-119"><a href="#cb16-119" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(p1)</span>
<span id="cb16-120"><a href="#cb16-120" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb16-121"><a href="#cb16-121" aria-hidden="true" tabindex="-1"></a>      plist <span class="ot">&lt;-</span> <span class="fu">generate_pca_plots</span>(<span class="at">assay_mtx_wide=</span>assay_mtx_wide,</span>
<span id="cb16-122"><a href="#cb16-122" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">meta_data=</span>meta_data)</span>
<span id="cb16-123"><a href="#cb16-123" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(plist<span class="sc">$</span>p1)</span>
<span id="cb16-124"><a href="#cb16-124" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(plist<span class="sc">$</span>p2)</span>
<span id="cb16-125"><a href="#cb16-125" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(plist<span class="sc">$</span>p3)</span>
<span id="cb16-126"><a href="#cb16-126" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb16-127"><a href="#cb16-127" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>))</span>
<span id="cb16-128"><a href="#cb16-128" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-129"><a href="#cb16-129" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<section id="day_0" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="day_0"><span class="header-section-number">4.1</span> day_0</h2>
<section id="pbmc_cell_frequency" class="level3" data-number="4.1.1">
<h3 data-number="4.1.1" class="anchored" data-anchor-id="pbmc_cell_frequency"><span class="header-section-number">4.1.1</span> pbmc_cell_frequency</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-2.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-3.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-4.png" class="img-fluid" width="672"></p>
</section>
<section id="pbmc_gene_expression" class="level3" data-number="4.1.2">
<h3 data-number="4.1.2" class="anchored" data-anchor-id="pbmc_gene_expression"><span class="header-section-number">4.1.2</span> pbmc_gene_expression</h3>
<pre><code>converting counts to integer mode</code></pre>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-5.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-6.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-7.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-8.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_ab_titer" class="level3" data-number="4.1.3">
<h3 data-number="4.1.3" class="anchored" data-anchor-id="plasma_ab_titer"><span class="header-section-number">4.1.3</span> plasma_ab_titer</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-9.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-10.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-11.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-12.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_cytokine_concentration_by_legendplex" class="level3" data-number="4.1.4">
<h3 data-number="4.1.4" class="anchored" data-anchor-id="plasma_cytokine_concentration_by_legendplex"><span class="header-section-number">4.1.4</span> plasma_cytokine_concentration_by_legendplex</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-13.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-14.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-15.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-16.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_cytokine_concentration_by_olink" class="level3" data-number="4.1.5">
<h3 data-number="4.1.5" class="anchored" data-anchor-id="plasma_cytokine_concentration_by_olink"><span class="header-section-number">4.1.5</span> plasma_cytokine_concentration_by_olink</h3>
<p>Non-zero fraction of NAs: 0.0039<img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-17.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-18.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-19.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-20.png" class="img-fluid" width="672"></p>
</section>
<section id="t_cell_activation" class="level3" data-number="4.1.6">
<h3 data-number="4.1.6" class="anchored" data-anchor-id="t_cell_activation"><span class="header-section-number">4.1.6</span> t_cell_activation</h3>
<p>Non-zero fraction of NAs: 0.059<img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-21.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-22.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-23.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-24.png" class="img-fluid" width="672"></p>
</section>
<section id="t_cell_polarization" class="level3" data-number="4.1.7">
<h3 data-number="4.1.7" class="anchored" data-anchor-id="t_cell_polarization"><span class="header-section-number">4.1.7</span> t_cell_polarization</h3>
<p>Non-zero fraction of NAs: 0.0106<img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-25.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-26.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-27.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-28.png" class="img-fluid" width="672"></p>
</section>
</section>
<section id="day_1" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="day_1"><span class="header-section-number">4.2</span> day_1</h2>
<section id="pbmc_cell_frequency-1" class="level3" data-number="4.2.1">
<h3 data-number="4.2.1" class="anchored" data-anchor-id="pbmc_cell_frequency-1"><span class="header-section-number">4.2.1</span> pbmc_cell_frequency</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-29.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-30.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-31.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-32.png" class="img-fluid" width="672"></p>
</section>
<section id="pbmc_gene_expression-1" class="level3" data-number="4.2.2">
<h3 data-number="4.2.2" class="anchored" data-anchor-id="pbmc_gene_expression-1"><span class="header-section-number">4.2.2</span> pbmc_gene_expression</h3>
<pre><code>converting counts to integer mode</code></pre>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-33.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-34.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-35.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-36.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_ab_titer-1" class="level3" data-number="4.2.3">
<h3 data-number="4.2.3" class="anchored" data-anchor-id="plasma_ab_titer-1"><span class="header-section-number">4.2.3</span> plasma_ab_titer</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-37.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-38.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-39.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-40.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_cytokine_concentration_by_legendplex-1" class="level3" data-number="4.2.4">
<h3 data-number="4.2.4" class="anchored" data-anchor-id="plasma_cytokine_concentration_by_legendplex-1"><span class="header-section-number">4.2.4</span> plasma_cytokine_concentration_by_legendplex</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-41.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-42.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-43.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-44.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_cytokine_concentration_by_olink-1" class="level3" data-number="4.2.5">
<h3 data-number="4.2.5" class="anchored" data-anchor-id="plasma_cytokine_concentration_by_olink-1"><span class="header-section-number">4.2.5</span> plasma_cytokine_concentration_by_olink</h3>
<p>Non-zero fraction of NAs: 0.014<img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-45.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-46.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-47.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-48.png" class="img-fluid" width="672"></p>
</section>
<section id="t_cell_activation-1" class="level3" data-number="4.2.6">
<h3 data-number="4.2.6" class="anchored" data-anchor-id="t_cell_activation-1"><span class="header-section-number">4.2.6</span> t_cell_activation</h3>
<p>No specimen for this combination of assay and day available</p>
</section>
<section id="t_cell_polarization-1" class="level3" data-number="4.2.7">
<h3 data-number="4.2.7" class="anchored" data-anchor-id="t_cell_polarization-1"><span class="header-section-number">4.2.7</span> t_cell_polarization</h3>
<p>No specimen for this combination of assay and day available</p>
</section>
</section>
<section id="day_3" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="day_3"><span class="header-section-number">4.3</span> day_3</h2>
<section id="pbmc_cell_frequency-2" class="level3" data-number="4.3.1">
<h3 data-number="4.3.1" class="anchored" data-anchor-id="pbmc_cell_frequency-2"><span class="header-section-number">4.3.1</span> pbmc_cell_frequency</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-49.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-50.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-51.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-52.png" class="img-fluid" width="672"></p>
</section>
<section id="pbmc_gene_expression-2" class="level3" data-number="4.3.2">
<h3 data-number="4.3.2" class="anchored" data-anchor-id="pbmc_gene_expression-2"><span class="header-section-number">4.3.2</span> pbmc_gene_expression</h3>
<pre><code>converting counts to integer mode</code></pre>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-53.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-54.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-55.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-56.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_ab_titer-2" class="level3" data-number="4.3.3">
<h3 data-number="4.3.3" class="anchored" data-anchor-id="plasma_ab_titer-2"><span class="header-section-number">4.3.3</span> plasma_ab_titer</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-57.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-58.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-59.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-60.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_cytokine_concentration_by_legendplex-2" class="level3" data-number="4.3.4">
<h3 data-number="4.3.4" class="anchored" data-anchor-id="plasma_cytokine_concentration_by_legendplex-2"><span class="header-section-number">4.3.4</span> plasma_cytokine_concentration_by_legendplex</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-61.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-62.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-63.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-64.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_cytokine_concentration_by_olink-2" class="level3" data-number="4.3.5">
<h3 data-number="4.3.5" class="anchored" data-anchor-id="plasma_cytokine_concentration_by_olink-2"><span class="header-section-number">4.3.5</span> plasma_cytokine_concentration_by_olink</h3>
<p>Non-zero fraction of NAs: 0.0199<img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-65.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-66.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-67.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-68.png" class="img-fluid" width="672"></p>
</section>
<section id="t_cell_activation-2" class="level3" data-number="4.3.6">
<h3 data-number="4.3.6" class="anchored" data-anchor-id="t_cell_activation-2"><span class="header-section-number">4.3.6</span> t_cell_activation</h3>
<p>No specimen for this combination of assay and day available</p>
</section>
<section id="t_cell_polarization-2" class="level3" data-number="4.3.7">
<h3 data-number="4.3.7" class="anchored" data-anchor-id="t_cell_polarization-2"><span class="header-section-number">4.3.7</span> t_cell_polarization</h3>
<p>No specimen for this combination of assay and day available</p>
</section>
</section>
<section id="day_14" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="day_14"><span class="header-section-number">4.4</span> day_14</h2>
<section id="pbmc_cell_frequency-3" class="level3" data-number="4.4.1">
<h3 data-number="4.4.1" class="anchored" data-anchor-id="pbmc_cell_frequency-3"><span class="header-section-number">4.4.1</span> pbmc_cell_frequency</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-69.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-70.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-71.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-72.png" class="img-fluid" width="672"></p>
</section>
<section id="pbmc_gene_expression-3" class="level3" data-number="4.4.2">
<h3 data-number="4.4.2" class="anchored" data-anchor-id="pbmc_gene_expression-3"><span class="header-section-number">4.4.2</span> pbmc_gene_expression</h3>
<pre><code>converting counts to integer mode</code></pre>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-73.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-74.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-75.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-76.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_ab_titer-3" class="level3" data-number="4.4.3">
<h3 data-number="4.4.3" class="anchored" data-anchor-id="plasma_ab_titer-3"><span class="header-section-number">4.4.3</span> plasma_ab_titer</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-77.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-78.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-79.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-80.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_cytokine_concentration_by_legendplex-3" class="level3" data-number="4.4.4">
<h3 data-number="4.4.4" class="anchored" data-anchor-id="plasma_cytokine_concentration_by_legendplex-3"><span class="header-section-number">4.4.4</span> plasma_cytokine_concentration_by_legendplex</h3>
<p><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-81.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-82.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-83.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-84.png" class="img-fluid" width="672"></p>
</section>
<section id="plasma_cytokine_concentration_by_olink-3" class="level3" data-number="4.4.5">
<h3 data-number="4.4.5" class="anchored" data-anchor-id="plasma_cytokine_concentration_by_olink-3"><span class="header-section-number">4.4.5</span> plasma_cytokine_concentration_by_olink</h3>
<p>Non-zero fraction of NAs: 0.0025<img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-85.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-86.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-87.png" class="img-fluid" width="672"><img src="batch_effects_raw_data_files/figure-html/unnamed-chunk-4-88.png" class="img-fluid" width="672"></p>
</section>
<section id="t_cell_activation-3" class="level3" data-number="4.4.6">
<h3 data-number="4.4.6" class="anchored" data-anchor-id="t_cell_activation-3"><span class="header-section-number">4.4.6</span> t_cell_activation</h3>
<p>No specimen for this combination of assay and day available</p>
</section>
<section id="t_cell_polarization-3" class="level3" data-number="4.4.7">
<h3 data-number="4.4.7" class="anchored" data-anchor-id="t_cell_polarization-3"><span class="header-section-number">4.4.7</span> t_cell_polarization</h3>
<p>No specimen for this combination of assay and day available</p>
</section>
</section>
</section>
<section id="appendix" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Appendix</h1>
<button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne">
Session Information
</button>
<div id="collapseOne" class="accordion-collapse collapse">
<div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 14.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] magick_2.8.3                FactoMineR_2.11            
 [3] factoextra_1.0.7            DESeq2_1.38.3              
 [5] SummarizedExperiment_1.28.0 MatrixGenerics_1.10.0      
 [7] matrixStats_1.4.1           GenomicRanges_1.50.2       
 [9] GenomeInfoDb_1.34.9         IRanges_2.32.0             
[11] S4Vectors_0.36.2            lme4_1.1-35.5              
[13] Matrix_1.6-5                vsn_3.66.0                 
[15] Biobase_2.58.0              BiocGenerics_0.44.0        
[17] ggdark_0.2.1                lubridate_1.9.3            
[19] forcats_1.0.0               stringr_1.5.1              
[21] dplyr_1.1.4                 purrr_1.0.2                
[23] readr_2.1.5                 tidyr_1.3.1                
[25] tibble_3.2.1                ggplot2_3.5.1              
[27] tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] TH.data_1.1-2          minqa_1.2.8            colorspace_2.1-1      
  [4] ggsignif_0.6.4         estimability_1.5.1     XVector_0.38.0        
  [7] ggpubr_0.6.0           farver_2.1.2           affyio_1.68.0         
 [10] DT_0.33                ggrepel_0.9.6          bit64_4.5.2           
 [13] AnnotationDbi_1.60.2   fansi_1.0.6            mvtnorm_1.3-1         
 [16] codetools_0.2-20       splines_4.2.3          leaps_3.2             
 [19] impute_1.72.3          cachem_1.1.0           geneplotter_1.76.0    
 [22] knitr_1.48             jsonlite_1.8.9         nloptr_2.1.1          
 [25] broom_1.0.6            annotate_1.76.0        cluster_2.1.6         
 [28] png_0.1-8              BiocManager_1.30.25    compiler_4.2.3        
 [31] httr_1.4.7             backports_1.5.0        emmeans_1.10.4        
 [34] fastmap_1.2.0          limma_3.54.2           cli_3.6.3             
 [37] htmltools_0.5.8.1      tools_4.2.3            coda_0.19-4.1         
 [40] gtable_0.3.5           glue_1.7.0             GenomeInfoDbData_1.2.9
 [43] affy_1.76.0            Rcpp_1.0.13            carData_3.0-5         
 [46] vctrs_0.6.5            Biostrings_2.66.0      preprocessCore_1.60.2 
 [49] nlme_3.1-166           xfun_0.47              timechange_0.3.0      
 [52] lifecycle_1.0.4        rstatix_0.7.2          XML_3.99-0.17         
 [55] zlibbioc_1.44.0        MASS_7.3-60.0.1        zoo_1.8-12            
 [58] scales_1.3.0           vroom_1.6.5            hms_1.1.3             
 [61] parallel_4.2.3         sandwich_3.1-1         RColorBrewer_1.1-3    
 [64] yaml_2.3.10            memoise_2.0.1          stringi_1.8.4         
 [67] RSQLite_2.3.7          boot_1.3-31            BiocParallel_1.32.6   
 [70] rlang_1.1.4            pkgconfig_2.0.3        bitops_1.0-8          
 [73] evaluate_1.0.0         lattice_0.22-6         labeling_0.4.3        
 [76] htmlwidgets_1.6.4      bit_4.5.0              tidyselect_1.2.1      
 [79] magrittr_2.0.3         R6_2.5.1               multcompView_0.1-10   
 [82] generics_0.1.3         multcomp_1.4-26        DelayedArray_0.24.0   
 [85] DBI_1.2.3              pillar_1.9.0           withr_3.0.1           
 [88] abind_1.4-8            scatterplot3d_0.3-44   survival_3.7-0        
 [91] KEGGREST_1.38.0        RCurl_1.98-1.14        car_3.1-2             
 [94] crayon_1.5.3           utf8_1.2.4             tzdb_0.4.0            
 [97] rmarkdown_2.28         locfit_1.5-9.10        grid_4.2.3            
[100] blob_1.2.4             flashClust_1.01-2      digest_0.6.37         
[103] xtable_1.8-4           munsell_0.5.1          viridisLite_0.4.2     </code></pre>
</div>
</div>
</div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb23" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Batch Effects"</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Philipp Sven Lars Schäfer"</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "`r format(Sys.time(), '%d %B, %Y')`"</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> source</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="an">engine:</span><span class="co"> knitr</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="fu"># Packages</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(vsn)</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(lme4)</span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(DESeq2)</span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb23-22"><a href="#cb23-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick) <span class="co"># formatting</span></span>
<span id="cb23-23"><a href="#cb23-23" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb23-24"><a href="#cb23-24" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>knit_hooks<span class="sc">$</span><span class="fu">set</span>(<span class="at">crop =</span> knitr<span class="sc">::</span>hook_pdfcrop) <span class="co"># formatting</span></span>
<span id="cb23-25"><a href="#cb23-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-26"><a href="#cb23-26" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb23-27"><a href="#cb23-27" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"generate_targets.R"</span>))</span>
<span id="cb23-28"><a href="#cb23-28" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"model.R"</span>))</span>
<span id="cb23-29"><a href="#cb23-29" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"batch_effects.R"</span>))</span>
<span id="cb23-30"><a href="#cb23-30" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb23-31"><a href="#cb23-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-32"><a href="#cb23-32" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data</span></span>
<span id="cb23-33"><a href="#cb23-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-36"><a href="#cb23-36" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb23-37"><a href="#cb23-37" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span>
<span id="cb23-38"><a href="#cb23-38" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb23-39"><a href="#cb23-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-42"><a href="#cb23-42" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb23-43"><a href="#cb23-43" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb23-44"><a href="#cb23-44" aria-hidden="true" tabindex="-1"></a>experimental_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb23-45"><a href="#cb23-45" aria-hidden="true" tabindex="-1"></a>experimental_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(<span class="at">meta_data=</span>meta_data, </span>
<span id="cb23-46"><a href="#cb23-46" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">experimental_data=</span>experimental_data)</span>
<span id="cb23-47"><a href="#cb23-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-48"><a href="#cb23-48" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb23-49"><a href="#cb23-49" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb23-50"><a href="#cb23-50" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span>
<span id="cb23-51"><a href="#cb23-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-52"><a href="#cb23-52" aria-hidden="true" tabindex="-1"></a>specimen_per_day <span class="ot">&lt;-</span> <span class="fu">get_specimen_per_day</span>(<span class="at">meta_data=</span>meta_data)</span>
<span id="cb23-53"><a href="#cb23-53" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb23-54"><a href="#cb23-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-55"><a href="#cb23-55" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusions</span></span>
<span id="cb23-56"><a href="#cb23-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-57"><a href="#cb23-57" aria-hidden="true" tabindex="-1"></a><span class="fu">## PBMC Fractions</span></span>
<span id="cb23-58"><a href="#cb23-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-59"><a href="#cb23-59" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>For the baseline specimen (day 0), we see a very strong batch effect, because the 2023 dataset is very different</span>
<span id="cb23-60"><a href="#cb23-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-61"><a href="#cb23-61" aria-hidden="true" tabindex="-1"></a><span class="fu">## PBMC Gene Expression</span></span>
<span id="cb23-62"><a href="#cb23-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-63"><a href="#cb23-63" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Very strong cohort-specific effects for baseline specimen and post-booster specimen</span>
<span id="cb23-64"><a href="#cb23-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-65"><a href="#cb23-65" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plasma Antibody Levels</span></span>
<span id="cb23-66"><a href="#cb23-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-67"><a href="#cb23-67" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plasma Cytokine Concentration by Legendplex</span></span>
<span id="cb23-68"><a href="#cb23-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-69"><a href="#cb23-69" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>After removing outlier in the legendplex data...</span>
<span id="cb23-70"><a href="#cb23-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-71"><a href="#cb23-71" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plasma Cytokine Concentration by Olink</span></span>
<span id="cb23-72"><a href="#cb23-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-73"><a href="#cb23-73" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>At baseline: About 6.5% of the variance is explained by the dataset, probably also due to 3 outliers from the 2022 datasets. TODO: Investigate this in more detail.</span>
<span id="cb23-74"><a href="#cb23-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-75"><a href="#cb23-75" aria-hidden="true" tabindex="-1"></a><span class="fu">## T Cell Activation</span></span>
<span id="cb23-76"><a href="#cb23-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-77"><a href="#cb23-77" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Specimen are only available for baseline, no cohort-specific effect</span>
<span id="cb23-78"><a href="#cb23-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-79"><a href="#cb23-79" aria-hidden="true" tabindex="-1"></a><span class="fu">## T Cell Polarization</span></span>
<span id="cb23-80"><a href="#cb23-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-81"><a href="#cb23-81" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Specimen are only available for baseline, most of the variance is explained by interaction terms <span class="in">`age_at_boost:dataset`</span> and <span class="in">`infancy_vac:dataset`</span>. TODO: Think about what to make of this.</span>
<span id="cb23-82"><a href="#cb23-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-83"><a href="#cb23-83" aria-hidden="true" tabindex="-1"></a><span class="fu"># Results</span></span>
<span id="cb23-84"><a href="#cb23-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-87"><a href="#cb23-87" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb23-88"><a href="#cb23-88" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: asis</span></span>
<span id="cb23-89"><a href="#cb23-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-90"><a href="#cb23-90" aria-hidden="true" tabindex="-1"></a><span class="do">##| crop: true # interfers with dynamic toc generation</span></span>
<span id="cb23-91"><a href="#cb23-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-92"><a href="#cb23-92" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (day <span class="cf">in</span> <span class="fu">names</span>(specimen_per_day)) {</span>
<span id="cb23-93"><a href="#cb23-93" aria-hidden="true" tabindex="-1"></a>  <span class="co"># day &lt;- names(specimen_per_day)[1]</span></span>
<span id="cb23-94"><a href="#cb23-94" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-95"><a href="#cb23-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb23-96"><a href="#cb23-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"## "</span>, day))</span>
<span id="cb23-97"><a href="#cb23-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb23-98"><a href="#cb23-98" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-99"><a href="#cb23-99" aria-hidden="true" tabindex="-1"></a>  specimen_df <span class="ot">&lt;-</span> specimen_per_day[[day]]</span>
<span id="cb23-100"><a href="#cb23-100" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-101"><a href="#cb23-101" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (assay <span class="cf">in</span> <span class="fu">names</span>(experimental_data)) {</span>
<span id="cb23-102"><a href="#cb23-102" aria-hidden="true" tabindex="-1"></a>    <span class="co"># assay &lt;- names(experimental_data)[4]</span></span>
<span id="cb23-103"><a href="#cb23-103" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-104"><a href="#cb23-104" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb23-105"><a href="#cb23-105" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"### "</span>, assay))</span>
<span id="cb23-106"><a href="#cb23-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb23-107"><a href="#cb23-107" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-108"><a href="#cb23-108" aria-hidden="true" tabindex="-1"></a>    feature_col <span class="ot">&lt;-</span> experimental_data_settings[[assay]]<span class="sc">$</span>feature_col</span>
<span id="cb23-109"><a href="#cb23-109" aria-hidden="true" tabindex="-1"></a>    value_col <span class="ot">&lt;-</span> experimental_data_settings[[assay]]<span class="sc">$</span>value_col</span>
<span id="cb23-110"><a href="#cb23-110" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-111"><a href="#cb23-111" aria-hidden="true" tabindex="-1"></a>    assay_df_long <span class="ot">&lt;-</span> experimental_data[[assay]]</span>
<span id="cb23-112"><a href="#cb23-112" aria-hidden="true" tabindex="-1"></a>    <span class="co">#str(assay_df_long)</span></span>
<span id="cb23-113"><a href="#cb23-113" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-114"><a href="#cb23-114" aria-hidden="true" tabindex="-1"></a>    specimen_df_filtered <span class="ot">&lt;-</span> specimen_df <span class="sc">%&gt;%</span></span>
<span id="cb23-115"><a href="#cb23-115" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(specimen_id <span class="sc">%in%</span> <span class="fu">unique</span>(assay_df_long<span class="sc">$</span>specimen_id))</span>
<span id="cb23-116"><a href="#cb23-116" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb23-117"><a href="#cb23-117" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(specimen_df_filtered) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb23-118"><a href="#cb23-118" aria-hidden="true" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"No specimen for this combination of assay and day available"</span>))</span>
<span id="cb23-119"><a href="#cb23-119" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb23-120"><a href="#cb23-120" aria-hidden="true" tabindex="-1"></a>      assay_mtx_wide <span class="ot">&lt;-</span> assay_df_long <span class="sc">%&gt;%</span></span>
<span id="cb23-121"><a href="#cb23-121" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"specimen_id"</span>, feature_col, value_col))) <span class="sc">%&gt;%</span></span>
<span id="cb23-122"><a href="#cb23-122" aria-hidden="true" tabindex="-1"></a>        tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(feature_col), </span>
<span id="cb23-123"><a href="#cb23-123" aria-hidden="true" tabindex="-1"></a>                           <span class="at">values_from=</span>dplyr<span class="sc">::</span><span class="fu">all_of</span>(value_col)) <span class="sc">%&gt;%</span></span>
<span id="cb23-124"><a href="#cb23-124" aria-hidden="true" tabindex="-1"></a>        tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-125"><a href="#cb23-125" aria-hidden="true" tabindex="-1"></a>        <span class="fu">as.matrix</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-126"><a href="#cb23-126" aria-hidden="true" tabindex="-1"></a>        .[<span class="fu">as.character</span>(specimen_df_filtered<span class="sc">$</span>specimen_id), ]</span>
<span id="cb23-127"><a href="#cb23-127" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb23-128"><a href="#cb23-128" aria-hidden="true" tabindex="-1"></a>      impute_mode <span class="ot">&lt;-</span> <span class="st">"knn"</span></span>
<span id="cb23-129"><a href="#cb23-129" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stopifnot</span>(impute_mode <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"zero"</span>, <span class="st">"median_feature"</span>, <span class="st">"knn"</span>))</span>
<span id="cb23-130"><a href="#cb23-130" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb23-131"><a href="#cb23-131" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">any</span>(<span class="fu">is.na</span>(assay_mtx_wide))) {</span>
<span id="cb23-132"><a href="#cb23-132" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"Non-zero fraction of NAs: "</span>, </span>
<span id="cb23-133"><a href="#cb23-133" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">round</span>(<span class="fu">mean</span>(<span class="fu">is.na</span>(assay_mtx_wide)), <span class="dv">4</span>)))</span>
<span id="cb23-134"><a href="#cb23-134" aria-hidden="true" tabindex="-1"></a>        <span class="co"># data: matrix with genes in the rows, samples in the columns</span></span>
<span id="cb23-135"><a href="#cb23-135" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (impute_mode <span class="sc">==</span> <span class="st">"zero"</span>) {</span>
<span id="cb23-136"><a href="#cb23-136" aria-hidden="true" tabindex="-1"></a>          assay_mtx_wide[<span class="fu">is.na</span>(assay_mtx_wide)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb23-137"><a href="#cb23-137" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span> (impute_mode <span class="sc">==</span> <span class="st">"knn"</span>) {</span>
<span id="cb23-138"><a href="#cb23-138" aria-hidden="true" tabindex="-1"></a>          assay_mtx_wide <span class="ot">&lt;-</span> <span class="fu">t</span>(impute<span class="sc">::</span><span class="fu">impute.knn</span>(<span class="at">data=</span><span class="fu">t</span>(assay_mtx_wide))<span class="sc">$</span>data)</span>
<span id="cb23-139"><a href="#cb23-139" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span> (impute_mode <span class="sc">==</span> <span class="st">"median_feature"</span>) {</span>
<span id="cb23-140"><a href="#cb23-140" aria-hidden="true" tabindex="-1"></a>          col_medians <span class="ot">&lt;-</span> matrixStats<span class="sc">::</span><span class="fu">colMedians</span>(assay_mtx_wide, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb23-141"><a href="#cb23-141" aria-hidden="true" tabindex="-1"></a>          <span class="cf">for</span> (colname <span class="cf">in</span> <span class="fu">colnames</span>(assay_mtx_wide)) {</span>
<span id="cb23-142"><a href="#cb23-142" aria-hidden="true" tabindex="-1"></a>            col_vec <span class="ot">&lt;-</span> assay_mtx_wide[, colname]</span>
<span id="cb23-143"><a href="#cb23-143" aria-hidden="true" tabindex="-1"></a>            col_vec[<span class="fu">is.na</span>(col_vec)] <span class="ot">&lt;-</span> col_medians[colname]</span>
<span id="cb23-144"><a href="#cb23-144" aria-hidden="true" tabindex="-1"></a>            assay_mtx_wide[, colname] <span class="ot">&lt;-</span> col_vec</span>
<span id="cb23-145"><a href="#cb23-145" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb23-146"><a href="#cb23-146" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb23-147"><a href="#cb23-147" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stopifnot</span>(<span class="sc">!</span><span class="fu">any</span>(<span class="fu">is.na</span>(assay_mtx_wide)))</span>
<span id="cb23-148"><a href="#cb23-148" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb23-149"><a href="#cb23-149" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb23-150"><a href="#cb23-150" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (assay <span class="sc">==</span> <span class="st">"pbmc_gene_expression"</span>) {</span>
<span id="cb23-151"><a href="#cb23-151" aria-hidden="true" tabindex="-1"></a>        <span class="co"># normalization using vst from DEseq2</span></span>
<span id="cb23-152"><a href="#cb23-152" aria-hidden="true" tabindex="-1"></a>        <span class="co">#str(assay_mtx_wide)</span></span>
<span id="cb23-153"><a href="#cb23-153" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb23-154"><a href="#cb23-154" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Create DESeq dataset object</span></span>
<span id="cb23-155"><a href="#cb23-155" aria-hidden="true" tabindex="-1"></a>        min_counts <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb23-156"><a href="#cb23-156" aria-hidden="true" tabindex="-1"></a>        min_samples <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb23-157"><a href="#cb23-157" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb23-158"><a href="#cb23-158" aria-hidden="true" tabindex="-1"></a>        genes_to_keep <span class="ot">&lt;-</span> <span class="fu">colnames</span>(assay_mtx_wide)[</span>
<span id="cb23-159"><a href="#cb23-159" aria-hidden="true" tabindex="-1"></a>          (<span class="fu">colSums</span>(assay_mtx_wide) <span class="sc">&gt;=</span> min_counts) <span class="sc">&amp;</span></span>
<span id="cb23-160"><a href="#cb23-160" aria-hidden="true" tabindex="-1"></a>            (<span class="fu">colSums</span>(assay_mtx_wide <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">&gt;=</span> min_samples)</span>
<span id="cb23-161"><a href="#cb23-161" aria-hidden="true" tabindex="-1"></a>        ]</span>
<span id="cb23-162"><a href="#cb23-162" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> assay_mtx_wide[ , genes_to_keep]</span>
<span id="cb23-163"><a href="#cb23-163" aria-hidden="true" tabindex="-1"></a>        <span class="co">#str(assay_mtx_wide)</span></span>
<span id="cb23-164"><a href="#cb23-164" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb23-165"><a href="#cb23-165" aria-hidden="true" tabindex="-1"></a>        <span class="co"># check that all column (gene) names are unique</span></span>
<span id="cb23-166"><a href="#cb23-166" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stopifnot</span>(<span class="fu">length</span>(<span class="fu">colnames</span>(assay_mtx_wide)) <span class="sc">==</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">colnames</span>(assay_mtx_wide))))</span>
<span id="cb23-167"><a href="#cb23-167" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb23-168"><a href="#cb23-168" aria-hidden="true" tabindex="-1"></a>        ensemble_to_gene <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">versioned_ensembl_gene_id =</span> <span class="fu">colnames</span>(assay_mtx_wide)) <span class="sc">%&gt;%</span></span>
<span id="cb23-169"><a href="#cb23-169" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene_meta, <span class="at">by=</span><span class="st">"versioned_ensembl_gene_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-170"><a href="#cb23-170" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">group_by</span>(gene_symbol) <span class="sc">%&gt;%</span></span>
<span id="cb23-171"><a href="#cb23-171" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb23-172"><a href="#cb23-172" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(n), gene_symbol) <span class="sc">%&gt;%</span></span>
<span id="cb23-173"><a href="#cb23-173" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">filter</span>(n<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-174"><a href="#cb23-174" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">ungroup</span>()</span>
<span id="cb23-175"><a href="#cb23-175" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb23-176"><a href="#cb23-176" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> assay_mtx_wide[ , ensemble_to_gene<span class="sc">$</span>versioned_ensembl_gene_id]</span>
<span id="cb23-177"><a href="#cb23-177" aria-hidden="true" tabindex="-1"></a>        <span class="fu">colnames</span>(assay_mtx_wide) <span class="ot">&lt;-</span> ensemble_to_gene<span class="sc">$</span>gene_symbol</span>
<span id="cb23-178"><a href="#cb23-178" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb23-179"><a href="#cb23-179" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stopifnot</span>(<span class="fu">length</span>(<span class="fu">colnames</span>(assay_mtx_wide)) <span class="sc">==</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">colnames</span>(assay_mtx_wide))))</span>
<span id="cb23-180"><a href="#cb23-180" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb23-181"><a href="#cb23-181" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> </span>
<span id="cb23-182"><a href="#cb23-182" aria-hidden="true" tabindex="-1"></a>          DESeq2<span class="sc">::</span><span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">t</span>(assay_mtx_wide),</span>
<span id="cb23-183"><a href="#cb23-183" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">colData =</span> tibble<span class="sc">::</span><span class="fu">tibble</span>(<span class="at">specimen_id=</span><span class="fu">rownames</span>(assay_mtx_wide)),</span>
<span id="cb23-184"><a href="#cb23-184" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">design =</span> <span class="sc">~</span> <span class="dv">1</span>)</span>
<span id="cb23-185"><a href="#cb23-185" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb23-186"><a href="#cb23-186" aria-hidden="true" tabindex="-1"></a>        <span class="co"># DESeq2 normalization with variance stabilizing transformation (vst)</span></span>
<span id="cb23-187"><a href="#cb23-187" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">estimateSizeFactors</span>(assay_mtx_wide, <span class="at">quiet=</span><span class="cn">TRUE</span>) <span class="do">## estimate size factor</span></span>
<span id="cb23-188"><a href="#cb23-188" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">estimateDispersions</span>(assay_mtx_wide, <span class="at">quiet=</span><span class="cn">TRUE</span>)</span>
<span id="cb23-189"><a href="#cb23-189" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">varianceStabilizingTransformation</span>(assay_mtx_wide, <span class="at">blind=</span><span class="cn">TRUE</span>)</span>
<span id="cb23-190"><a href="#cb23-190" aria-hidden="true" tabindex="-1"></a>        assay_mtx_wide <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">assay</span>(assay_mtx_wide))</span>
<span id="cb23-191"><a href="#cb23-191" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb23-192"><a href="#cb23-192" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb23-193"><a href="#cb23-193" aria-hidden="true" tabindex="-1"></a>      <span class="co"># dist_mtx &lt;- as.matrix(dist(assay_mtx_wide))</span></span>
<span id="cb23-194"><a href="#cb23-194" aria-hidden="true" tabindex="-1"></a>      <span class="co"># dist_mtx_median &lt;- median(dist_mtx[lower.tri(dist_mtx)])</span></span>
<span id="cb23-195"><a href="#cb23-195" aria-hidden="true" tabindex="-1"></a>      <span class="co"># dist_mtx_binary &lt;- dist_mtx &gt; (3 * dist_mtx_median)</span></span>
<span id="cb23-196"><a href="#cb23-196" aria-hidden="true" tabindex="-1"></a>      <span class="co"># outliers &lt;-</span></span>
<span id="cb23-197"><a href="#cb23-197" aria-hidden="true" tabindex="-1"></a>      <span class="co">#  which(rowSums(dist_mtx_binary) &gt; (10*median(rowSums(dist_mtx_binary))))</span></span>
<span id="cb23-198"><a href="#cb23-198" aria-hidden="true" tabindex="-1"></a>      <span class="co"># no_outliers &lt;- rownames(dist_mtx)[!rownames(dist_mtx) %in% names(outliers)]</span></span>
<span id="cb23-199"><a href="#cb23-199" aria-hidden="true" tabindex="-1"></a>      <span class="co"># assay_mtx_wide &lt;- assay_mtx_wide[no_outliers, ]</span></span>
<span id="cb23-200"><a href="#cb23-200" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb23-201"><a href="#cb23-201" aria-hidden="true" tabindex="-1"></a>      <span class="co">#str(assay_mtx_wide)</span></span>
<span id="cb23-202"><a href="#cb23-202" aria-hidden="true" tabindex="-1"></a>      <span class="fu">suppressMessages</span>({</span>
<span id="cb23-203"><a href="#cb23-203" aria-hidden="true" tabindex="-1"></a>        <span class="fu">suppressWarnings</span>({</span>
<span id="cb23-204"><a href="#cb23-204" aria-hidden="true" tabindex="-1"></a>          p1 <span class="ot">&lt;-</span> <span class="fu">pvca_analysis</span>(<span class="at">assay_mtx_wide=</span>assay_mtx_wide,</span>
<span id="cb23-205"><a href="#cb23-205" aria-hidden="true" tabindex="-1"></a>                              <span class="at">meta_data=</span>meta_data)<span class="sc">$</span>p1</span>
<span id="cb23-206"><a href="#cb23-206" aria-hidden="true" tabindex="-1"></a>        })</span>
<span id="cb23-207"><a href="#cb23-207" aria-hidden="true" tabindex="-1"></a>      })</span>
<span id="cb23-208"><a href="#cb23-208" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(p1)</span>
<span id="cb23-209"><a href="#cb23-209" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb23-210"><a href="#cb23-210" aria-hidden="true" tabindex="-1"></a>      plist <span class="ot">&lt;-</span> <span class="fu">generate_pca_plots</span>(<span class="at">assay_mtx_wide=</span>assay_mtx_wide,</span>
<span id="cb23-211"><a href="#cb23-211" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">meta_data=</span>meta_data)</span>
<span id="cb23-212"><a href="#cb23-212" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(plist<span class="sc">$</span>p1)</span>
<span id="cb23-213"><a href="#cb23-213" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(plist<span class="sc">$</span>p2)</span>
<span id="cb23-214"><a href="#cb23-214" aria-hidden="true" tabindex="-1"></a>      <span class="fu">print</span>(plist<span class="sc">$</span>p3)</span>
<span id="cb23-215"><a href="#cb23-215" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb23-216"><a href="#cb23-216" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>))</span>
<span id="cb23-217"><a href="#cb23-217" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb23-218"><a href="#cb23-218" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-219"><a href="#cb23-219" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb23-220"><a href="#cb23-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-221"><a href="#cb23-221" aria-hidden="true" tabindex="-1"></a><span class="fu"># Appendix</span></span>
<span id="cb23-222"><a href="#cb23-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-223"><a href="#cb23-223" aria-hidden="true" tabindex="-1"></a>&lt;button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne" &gt;Session Information&lt;/button&gt;&lt;div id="collapseOne" class="accordion-collapse collapse"&gt;&lt;div&gt;</span>
<span id="cb23-224"><a href="#cb23-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-227"><a href="#cb23-227" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb23-228"><a href="#cb23-228" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb23-229"><a href="#cb23-229" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb23-230"><a href="#cb23-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-231"><a href="#cb23-231" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;/div&gt;</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>