<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.45">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Philipp Sven Lars Schäfer">
<meta name="dcterms.date" content="2024-09-16">

<title>Data Overview – PSLS @ 3rd CMI-PB Challenge</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">PSLS @ 3rd CMI-PB Challenge</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/psl-schaefer"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://x.com/psl_schaefer"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/philipp-sl-schaefer/"> <i class="bi bi-linkedin" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#packages" id="toc-packages" class="nav-link active" data-scroll-target="#packages"><span class="header-section-number">1</span> Packages</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2</span> Data</a></li>
  <li><a href="#checks" id="toc-checks" class="nav-link" data-scroll-target="#checks"><span class="header-section-number">3</span> Checks</a></li>
  <li><a href="#demography-metadata" id="toc-demography-metadata" class="nav-link" data-scroll-target="#demography-metadata"><span class="header-section-number">4</span> Demography &amp; Metadata</a>
  <ul class="collapse">
  <li><a href="#how-many-subjects-do-we-have-per-dataset-per-partition" id="toc-how-many-subjects-do-we-have-per-dataset-per-partition" class="nav-link" data-scroll-target="#how-many-subjects-do-we-have-per-dataset-per-partition"><span class="header-section-number">4.1</span> How many subjects do we have per dataset / per partition?</a></li>
  <li><a href="#what-is-the-age-range" id="toc-what-is-the-age-range" class="nav-link" data-scroll-target="#what-is-the-age-range"><span class="header-section-number">4.2</span> What is the age range</a></li>
  </ul></li>
  <li><a href="#assays-timepoints-per-subject-dataset" id="toc-assays-timepoints-per-subject-dataset" class="nav-link" data-scroll-target="#assays-timepoints-per-subject-dataset"><span class="header-section-number">5</span> Assays &amp; Timepoints per Subject / Dataset</a>
  <ul class="collapse">
  <li><a href="#what-is-the-difference-between-planned-and-actual-booster-administration" id="toc-what-is-the-difference-between-planned-and-actual-booster-administration" class="nav-link" data-scroll-target="#what-is-the-difference-between-planned-and-actual-booster-administration"><span class="header-section-number">5.1</span> What is the difference between planned and actual booster administration</a></li>
  <li><a href="#what-is-the-difference-between-planned-and-actual-for-the-baseline" id="toc-what-is-the-difference-between-planned-and-actual-for-the-baseline" class="nav-link" data-scroll-target="#what-is-the-difference-between-planned-and-actual-for-the-baseline"><span class="header-section-number">5.2</span> What is the difference between planned and actual for the baseline</a></li>
  <li><a href="#how-many-time-points-do-we-have-per-subject" id="toc-how-many-time-points-do-we-have-per-subject" class="nav-link" data-scroll-target="#how-many-time-points-do-we-have-per-subject"><span class="header-section-number">5.3</span> How many time points do we have per subject?</a></li>
  <li><a href="#how-many-assays-do-we-have-for-the-baseline-measurement" id="toc-how-many-assays-do-we-have-for-the-baseline-measurement" class="nav-link" data-scroll-target="#how-many-assays-do-we-have-for-the-baseline-measurement"><span class="header-section-number">5.4</span> How many assays do we have for the baseline measurement?</a></li>
  </ul></li>
  <li><a href="#training-data" id="toc-training-data" class="nav-link" data-scroll-target="#training-data"><span class="header-section-number">6</span> Training Data</a>
  <ul class="collapse">
  <li><a href="#how-many-subjects-can-we-use-to-construct-training-data" id="toc-how-many-subjects-can-we-use-to-construct-training-data" class="nav-link" data-scroll-target="#how-many-subjects-can-we-use-to-construct-training-data"><span class="header-section-number">6.1</span> How many subjects can we use to construct training data?</a>
  <ul class="collapse">
  <li><a href="#antibody-level-tasks-anti-pt-14-days-after-booster" id="toc-antibody-level-tasks-anti-pt-14-days-after-booster" class="nav-link" data-scroll-target="#antibody-level-tasks-anti-pt-14-days-after-booster"><span class="header-section-number">6.1.1</span> 1) Antibody Level Tasks (anti-PT 14 days after booster)</a></li>
  <li><a href="#cell-frequency-tasks-monocytes-1-day-after-booster" id="toc-cell-frequency-tasks-monocytes-1-day-after-booster" class="nav-link" data-scroll-target="#cell-frequency-tasks-monocytes-1-day-after-booster"><span class="header-section-number">6.1.2</span> 2) Cell Frequency Tasks (Monocytes 1 day after booster)</a></li>
  <li><a href="#gene-expression-tasks-ccl3-expression-3-days-after-booster" id="toc-gene-expression-tasks-ccl3-expression-3-days-after-booster" class="nav-link" data-scroll-target="#gene-expression-tasks-ccl3-expression-3-days-after-booster"><span class="header-section-number">6.1.3</span> 3) Gene Expression Tasks (CCL3 expression 3 days after booster)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#missing-data" id="toc-missing-data" class="nav-link" data-scroll-target="#missing-data"><span class="header-section-number">7</span> Missing Data</a>
  <ul class="collapse">
  <li><a href="#how-many-na-values-do-we-have-per-assay" id="toc-how-many-na-values-do-we-have-per-assay" class="nav-link" data-scroll-target="#how-many-na-values-do-we-have-per-assay"><span class="header-section-number">7.1</span> How many NA values do we have per assay?</a></li>
  <li><a href="#what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies" id="toc-what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies" class="nav-link" data-scroll-target="#what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies"><span class="header-section-number">7.2</span> What is the fraction of NA values in the PBMC Cell Type Frequencies</a>
  <ul class="collapse">
  <li><a href="#per-cell-type" id="toc-per-cell-type" class="nav-link" data-scroll-target="#per-cell-type"><span class="header-section-number">7.2.1</span> Per Cell Type</a></li>
  <li><a href="#per-cell-specimen" id="toc-per-cell-specimen" class="nav-link" data-scroll-target="#per-cell-specimen"><span class="header-section-number">7.2.2</span> Per Cell Specimen</a></li>
  </ul></li>
  <li><a href="#what-is-the-fraction-of-na-values-in-the-olink-data" id="toc-what-is-the-fraction-of-na-values-in-the-olink-data" class="nav-link" data-scroll-target="#what-is-the-fraction-of-na-values-in-the-olink-data"><span class="header-section-number">7.3</span> What is the fraction of NA values in the Olink Data?</a>
  <ul class="collapse">
  <li><a href="#per-protein" id="toc-per-protein" class="nav-link" data-scroll-target="#per-protein"><span class="header-section-number">7.3.1</span> Per Protein</a></li>
  <li><a href="#per-specimen" id="toc-per-specimen" class="nav-link" data-scroll-target="#per-specimen"><span class="header-section-number">7.3.2</span> Per Specimen</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#question-for-assay-metadata" id="toc-question-for-assay-metadata" class="nav-link" data-scroll-target="#question-for-assay-metadata"><span class="header-section-number">8</span> Question for Assay Metadata</a>
  <ul class="collapse">
  <li><a href="#how-does-the-quality-control-for-olink-look-like" id="toc-how-does-the-quality-control-for-olink-look-like" class="nav-link" data-scroll-target="#how-does-the-quality-control-for-olink-look-like"><span class="header-section-number">8.1</span> How does the quality control for Olink look like?</a></li>
  <li><a href="#how-often-is-the-cytokine-concentration-below-the-lod-in-olink" id="toc-how-often-is-the-cytokine-concentration-below-the-lod-in-olink" class="nav-link" data-scroll-target="#how-often-is-the-cytokine-concentration-below-the-lod-in-olink"><span class="header-section-number">8.2</span> How often is the Cytokine Concentration below the LOD in Olink</a></li>
  <li><a href="#what-is-the-fraction-of-unspecific-antibody-measurements" id="toc-what-is-the-fraction-of-unspecific-antibody-measurements" class="nav-link" data-scroll-target="#what-is-the-fraction-of-unspecific-antibody-measurements"><span class="header-section-number">8.3</span> What is the fraction of unspecific antibody measurements</a>
  <ul class="collapse">
  <li><a href="#per-isotype-antigen" id="toc-per-isotype-antigen" class="nav-link" data-scroll-target="#per-isotype-antigen"><span class="header-section-number">8.3.1</span> Per Isotype-Antigen</a></li>
  <li><a href="#per-specimen-1" id="toc-per-specimen-1" class="nav-link" data-scroll-target="#per-specimen-1"><span class="header-section-number">8.3.2</span> Per Specimen</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#hierarchy-in-pbmc-frequency-data" id="toc-hierarchy-in-pbmc-frequency-data" class="nav-link" data-scroll-target="#hierarchy-in-pbmc-frequency-data"><span class="header-section-number">9</span> Hierarchy in PBMC Frequency Data</a>
  <ul class="collapse">
  <li><a href="#nk" id="toc-nk" class="nav-link" data-scroll-target="#nk"><span class="header-section-number">9.1</span> NK</a></li>
  <li><a href="#no-gating-info-for-2023" id="toc-no-gating-info-for-2023" class="nav-link" data-scroll-target="#no-gating-info-for-2023"><span class="header-section-number">9.2</span> No Gating Info for 2023</a></li>
  <li><a href="#no-gating-info-for-basophils-and-cd19-in-2020" id="toc-no-gating-info-for-basophils-and-cd19-in-2020" class="nav-link" data-scroll-target="#no-gating-info-for-basophils-and-cd19-in-2020"><span class="header-section-number">9.3</span> No Gating Info for Basophils and CD19 in 2020</a></li>
  <li><a href="#no-gating-info-for-non-pdcs-at-all" id="toc-no-gating-info-for-non-pdcs-at-all" class="nav-link" data-scroll-target="#no-gating-info-for-non-pdcs-at-all"><span class="header-section-number">9.4</span> No Gating Info for non-pDCs at all</a></li>
  <li><a href="#should-the-proportions-sum-up-to-1-at-level-0" id="toc-should-the-proportions-sum-up-to-1-at-level-0" class="nav-link" data-scroll-target="#should-the-proportions-sum-up-to-1-at-level-0"><span class="header-section-number">9.5</span> Should the Proportions sum up to 1 at level 0?</a></li>
  <li><a href="#difference-in-2023-dataset" id="toc-difference-in-2023-dataset" class="nav-link" data-scroll-target="#difference-in-2023-dataset"><span class="header-section-number">9.6</span> Difference in 2023 Dataset</a></li>
  </ul></li>
  <li><a href="#features-for-t-cell-activation-assay-aim" id="toc-features-for-t-cell-activation-assay-aim" class="nav-link" data-scroll-target="#features-for-t-cell-activation-assay-aim"><span class="header-section-number">10</span> Features for T cell Activation Assay (AIM)</a></li>
  <li><a href="#features-for-t-cell-polarization-assay-fluorospot" id="toc-features-for-t-cell-polarization-assay-fluorospot" class="nav-link" data-scroll-target="#features-for-t-cell-polarization-assay-fluorospot"><span class="header-section-number">11</span> Features for T cell Polarization Assay (FluoroSpot)</a></li>
  <li><a href="#other-notes" id="toc-other-notes" class="nav-link" data-scroll-target="#other-notes"><span class="header-section-number">12</span> Other Notes</a></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">13</span> Appendix</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Data Overview</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Philipp Sven Lars Schäfer </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 16, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="packages" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Packages</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(circlize)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Data</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co">#exp_data &lt;- read_harmonized_experimental_data(input_dir)</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(meta_data, exp_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>pbmc_cell_frequency | Removed 550 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 6931 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 495 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>pbmc_cell_frequency | Removed 56 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 48 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 233 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>t_cell_polarization | Removed 3 features because not in feature subset</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="checks" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Checks</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># only works due to filtering step above</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="sc">~</span> <span class="fu">all</span>(.x<span class="sc">$</span>specimen_id <span class="sc">%in%</span> meta_data<span class="sc">$</span>specimen_id)))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co"># no subject is recorded in more than one dataset</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(n) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="co"># we have the baseline specimen (planned_day_relative_to_boost = 0) for every subject</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, planned_day_relative_to_boost, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_baseline =</span> (planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present =</span> <span class="fu">any</span>(is_baseline),</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>                   <span class="at">dataset =</span> <span class="fu">first</span>(dataset)) <span class="sc">%&gt;%</span></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present_frac =</span> <span class="fu">mean</span>(baseline_present)) <span class="sc">%&gt;%</span></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(baseline_present_frac) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="demography-metadata" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Demography &amp; Metadata</h1>
<section id="how-many-subjects-do-we-have-per-dataset-per-partition" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="how-many-subjects-do-we-have-per-dataset-per-partition"><span class="header-section-number">4.1</span> How many subjects do we have per dataset / per partition?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, dataset, partition, infancy_vac) <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(dataset, partition, infancy_vac) <span class="sc">%&gt;%</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> n, <span class="at">y =</span> dataset, <span class="at">fill =</span> infancy_vac, <span class="at">color =</span> partition)) <span class="sc">+</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>), <span class="at">width =</span> <span class="fl">0.7</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span>  </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> n), </span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>),  </span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> infancy_vac_colors) <span class="sc">+</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> partition_colors) <span class="sc">+</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="what-is-the-age-range" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="what-is-the-age-range"><span class="header-section-number">4.2</span> What is the age range</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>age_at_boost, <span class="at">fill=</span>dataset, <span class="at">y=</span><span class="fu">after_stat</span>(density)), </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color=</span><span class="st">"black"</span>, <span class="at">position=</span><span class="st">"identity"</span>, <span class="at">bins=</span><span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors) <span class="sc">+</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="assays-timepoints-per-subject-dataset" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Assays &amp; Timepoints per Subject / Dataset</h1>
<section id="what-is-the-difference-between-planned-and-actual-booster-administration" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="what-is-the-difference-between-planned-and-actual-booster-administration"><span class="header-section-number">5.1</span> What is the difference between planned and actual booster administration</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost), <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="what-is-the-difference-between-planned-and-actual-for-the-baseline" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="what-is-the-difference-between-planned-and-actual-for-the-baseline"><span class="header-section-number">5.2</span> What is the difference between planned and actual for the baseline</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="st">`</span><span class="at">diff_&gt;_15</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">abs</span>(actual_day_relative_to_boost) <span class="sc">&gt;</span> <span class="dv">15</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>actual_day_relative_to_boost, <span class="at">fill=</span><span class="st">`</span><span class="at">diff_&gt;_15</span><span class="st">`</span>), <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="how-many-time-points-do-we-have-per-subject" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="how-many-time-points-do-we-have-per-subject"><span class="header-section-number">5.3</span> How many time points do we have per subject?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(dataset, subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">fill=</span>dataset),<span class="at">color=</span><span class="st">"black"</span>, <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset, <span class="at">ncol=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="how-many-assays-do-we-have-for-the-baseline-measurement" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="how-many-assays-do-we-have-for-the-baseline-measurement"><span class="header-section-number">5.4</span> How many assays do we have for the baseline measurement?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>assays_per_specimen <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">imap</span>(exp_data, <span class="sc">~</span> .x <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assay=</span>.y)) <span class="sc">%&gt;%</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, assay, subject_id, planned_day_relative_to_boost, infancy_vac, dataset)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>assays_per_specimen <span class="sc">%&gt;%</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, planned_day_relative_to_boost, dataset, specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(subject_id)) <span class="sc">%&gt;%</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">fill=</span>dataset), <span class="at">color=</span><span class="st">"black"</span>, <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset, <span class="at">ncol=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (day <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">14</span>)) {</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>annotation_data <span class="ot">&lt;-</span> meta_data <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, planned_day_relative_to_boost, infancy_vac, </span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>                dataset, biological_sex) <span class="sc">%&gt;%</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(specimen_id <span class="sc">%in%</span> assays_per_specimen<span class="sc">$</span>specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">==</span> <span class="sc">!!</span>day) <span class="sc">%&gt;%</span> </span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span> planned_day_relative_to_boost) <span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(dataset, infancy_vac, biological_sex) <span class="sc">%&gt;%</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(dataset, infancy_vac, biological_sex)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> assays_per_specimen <span class="sc">%&gt;%</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, assay) <span class="sc">%&gt;%</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">value =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"assay"</span>, <span class="at">values_from=</span><span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="at">.fns =</span> <span class="sc">~</span><span class="fu">replace_na</span>(.,<span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heatmap_data) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">colnames</span>(heatmap_data))</span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> heatmap_data[, <span class="fu">rownames</span>(annotation_data)]</span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>ht <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(heatmap_data, </span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>              <span class="at">name =</span> <span class="st">"heatmap"</span>, </span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>              <span class="at">row_title =</span> <span class="st">""</span>, </span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>              <span class="at">column_title =</span> <span class="fu">paste0</span>(<span class="st">"Specimens for Planned Day "</span>, day),</span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>              <span class="at">col =</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"black"</span>)),</span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a>              <span class="at">show_row_names =</span> <span class="cn">TRUE</span>, </span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>              <span class="at">show_column_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">width =</span> <span class="fu">unit</span>(<span class="dv">16</span>, <span class="st">"cm"</span>),</span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>              <span class="at">top_annotation =</span> ComplexHeatmap<span class="sc">::</span><span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> annotation_data,</span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">col =</span> <span class="fu">list</span>(</span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"dataset"</span> <span class="ot">=</span> dataset_colors,</span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"infancy_vac"</span> <span class="ot">=</span> infancy_vac_colors,</span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"biological_sex"</span> <span class="ot">=</span> sex_colors</span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>                                    ),</span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">which=</span><span class="st">"column"</span>)</span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht, </span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a>     <span class="at">heatmap_legend_side =</span> <span class="st">"top"</span>,</span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a>     <span class="at">annotation_legend_side =</span> <span class="st">"top"</span>,</span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a>     <span class="at">show_heatmap_legend =</span> <span class="cn">FALSE</span> <span class="co"># don't show colorbar</span></span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a>     )</span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-11-2.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-11-3.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-11-4.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="training-data" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Training Data</h1>
<section id="how-many-subjects-can-we-use-to-construct-training-data" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="how-many-subjects-can-we-use-to-construct-training-data"><span class="header-section-number">6.1</span> How many subjects can we use to construct training data?</h2>
<section id="antibody-level-tasks-anti-pt-14-days-after-booster" class="level3" data-number="6.1.1">
<h3 data-number="6.1.1" class="anchored" data-anchor-id="antibody-level-tasks-anti-pt-14-days-after-booster"><span class="header-section-number">6.1.1</span> 1) Antibody Level Tasks (anti-PT 14 days after booster)</h3>
<ul>
<li>All but 7 people were assayed approximately 14 days after the booster administration</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_1_possible =</span> </span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>plasma_ab_titer,</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                                                      <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(isotype_antigen<span class="sc">==</span><span class="st">"IgG_PT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_1_possible_total =</span> <span class="fu">sum</span>(task_1_possible), </span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_1_possible_fraction =</span> <span class="fu">mean</span>(task_1_possible))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["dataset"],"name":[1],"type":["chr"],"align":["left"]},{"label":["task_1_possible_total"],"name":[2],"type":["int"],"align":["right"]},{"label":["task_1_possible_fraction"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"2020_dataset","2":"57","3":"0.9500000"},{"1":"2021_dataset","2":"33","3":"0.9166667"},{"1":"2022_dataset","2":"21","3":"1.0000000"},{"1":"2023_dataset","2":"0","3":"0.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="cell-frequency-tasks-monocytes-1-day-after-booster" class="level3" data-number="6.1.2">
<h3 data-number="6.1.2" class="anchored" data-anchor-id="cell-frequency-tasks-monocytes-1-day-after-booster"><span class="header-section-number">6.1.2</span> 2) Cell Frequency Tasks (Monocytes 1 day after booster)</h3>
<ul>
<li>But there is one person where it does not really make sense</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_possible =</span> </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>pbmc_cell_frequency, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(cell_type_name<span class="sc">==</span><span class="st">"Monocytes"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_possible_total =</span> <span class="fu">sum</span>(task_possible), </span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_possible_fraction =</span> <span class="fu">mean</span>(task_possible))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["dataset"],"name":[1],"type":["chr"],"align":["left"]},{"label":["task_possible_total"],"name":[2],"type":["int"],"align":["right"]},{"label":["task_possible_fraction"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"2020_dataset","2":"20","3":"0.3333333"},{"1":"2021_dataset","2":"33","3":"0.9166667"},{"1":"2022_dataset","2":"21","3":"1.0000000"},{"1":"2023_dataset","2":"0","3":"0.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="gene-expression-tasks-ccl3-expression-3-days-after-booster" class="level3" data-number="6.1.3">
<h3 data-number="6.1.3" class="anchored" data-anchor-id="gene-expression-tasks-ccl3-expression-3-days-after-booster"><span class="header-section-number">6.1.3</span> 3) Gene Expression Tasks (CCL3 expression 3 days after booster)</h3>
<ul>
<li>There are 7 people for which this task does nto make sense</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(exp_data<span class="sc">$</span>pbmc_gene_expression<span class="sc">$</span>versioned_ensembl_gene_id)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>ccl3_ensembl <span class="ot">&lt;-</span> <span class="st">"ENSG00000277632"</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>ccl3_ensembl_versioned <span class="ot">&lt;-</span> all_genes[<span class="fu">str_starts</span>(all_genes, ccl3_ensembl)]</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_possible =</span> </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>pbmc_gene_expression, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(versioned_ensembl_gene_id<span class="sc">==</span>ccl3_ensembl_versioned) <span class="sc">%&gt;%</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_possible_total =</span> <span class="fu">sum</span>(task_possible), </span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_possible_fraction =</span> <span class="fu">mean</span>(task_possible))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["dataset"],"name":[1],"type":["chr"],"align":["left"]},{"label":["task_possible_total"],"name":[2],"type":["int"],"align":["right"]},{"label":["task_possible_fraction"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"2020_dataset","2":"36","3":"0.6"},{"1":"2021_dataset","2":"36","3":"1.0"},{"1":"2022_dataset","2":"21","3":"1.0"},{"1":"2023_dataset","2":"0","3":"0.0"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="missing-data" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Missing Data</h1>
<section id="how-many-na-values-do-we-have-per-assay" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="how-many-na-values-do-we-have-per-assay"><span class="header-section-number">7.1</span> How many NA values do we have per assay?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>value_vars_per_modality <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"pbmc_cell_frequency"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"percent_live_cell"</span>),</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"pbmc_gene_expression"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"raw_count"</span>, <span class="st">"tpm"</span>),</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"plasma_ab_titer"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"MFI"</span>, <span class="st">"MFI_normalised"</span>),</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"plasma_cytokine_concentration_by_legendplex"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"concentration"</span>),</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"plasma_cytokine_concentration_by_olink"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"concentration"</span>),</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"t_cell_activation"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"analyte_percentages"</span>),</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"t_cell_polarization"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"analyte_counts"</span>)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">imap_dfr</span>(exp_data, <span class="cf">function</span>(df, modality) {</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_dfr</span>(value_vars_per_modality[[modality]], </span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>                 <span class="sc">~</span> <span class="fu">tibble</span>(<span class="at">modality =</span> modality,</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>                          <span class="at">value_vars =</span> .x,</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sum_na =</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(df[[.x]]))))</span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["modality"],"name":[1],"type":["chr"],"align":["left"]},{"label":["value_vars"],"name":[2],"type":["chr"],"align":["left"]},{"label":["sum_na"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"pbmc_cell_frequency","2":"percent_live_cell","3":"0"},{"1":"pbmc_gene_expression","2":"raw_count","3":"0"},{"1":"pbmc_gene_expression","2":"tpm","3":"0"},{"1":"plasma_ab_titer","2":"MFI","3":"0"},{"1":"plasma_ab_titer","2":"MFI_normalised","3":"0"},{"1":"plasma_cytokine_concentration_by_legendplex","2":"concentration","3":"0"},{"1":"plasma_cytokine_concentration_by_olink","2":"concentration","3":"275"},{"1":"t_cell_activation","2":"analyte_percentages","3":"0"},{"1":"t_cell_polarization","2":"analyte_counts","3":"0"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies"><span class="header-section-number">7.2</span> What is the fraction of NA values in the PBMC Cell Type Frequencies</h2>
<section id="per-cell-type" class="level3" data-number="7.2.1">
<h3 data-number="7.2.1" class="anchored" data-anchor-id="per-cell-type"><span class="header-section-number">7.2.1</span> Per Cell Type</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(cell_type_name) <span class="sc">%&gt;%</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(frac_na)) <span class="sc">%&gt;%</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cell_type_name =</span> <span class="fu">factor</span>(cell_type_name, <span class="at">levels=</span><span class="fu">rev</span>(cell_type_name))) <span class="sc">%&gt;%</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">y=</span>cell_type_name, <span class="at">x=</span>frac_na)) <span class="sc">+</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Cell Type"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="per-cell-specimen" class="level3" data-number="7.2.2">
<h3 data-number="7.2.2" class="anchored" data-anchor-id="per-cell-specimen"><span class="header-section-number">7.2.2</span> Per Cell Specimen</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> frac_na)) <span class="sc">+</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add color for better visibility</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins =</span> <span class="dv">20</span>, <span class="at">geom =</span> <span class="st">"text"</span>, </span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Display bin counts at the top</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>So there are 5 specimen with more than 20% NA, which are all from the same subject!</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(frac_na <span class="sc">&gt;</span> <span class="fl">0.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["specimen_id"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["frac_na"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["subject_id"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["actual_day_relative_to_boost"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["planned_day_relative_to_boost"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["visit"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["infancy_vac"],"name":[7],"type":["chr"],"align":["left"]},{"label":["biological_sex"],"name":[8],"type":["chr"],"align":["left"]},{"label":["ethnicity"],"name":[9],"type":["chr"],"align":["left"]},{"label":["race"],"name":[10],"type":["chr"],"align":["left"]},{"label":["year_of_birth"],"name":[11],"type":["date"],"align":["right"]},{"label":["date_of_boost"],"name":[12],"type":["date"],"align":["right"]},{"label":["dataset"],"name":[13],"type":["chr"],"align":["left"]},{"label":["partition"],"name":[14],"type":["chr"],"align":["left"]},{"label":["age_at_boost"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["diff_relative_to_boost"],"name":[16],"type":["dbl"],"align":["right"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
</section>
<section id="what-is-the-fraction-of-na-values-in-the-olink-data" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="what-is-the-fraction-of-na-values-in-the-olink-data"><span class="header-section-number">7.3</span> What is the fraction of NA values in the Olink Data?</h2>
<section id="per-protein" class="level3" data-number="7.3.1">
<h3 data-number="7.3.1" class="anchored" data-anchor-id="per-protein"><span class="header-section-number">7.3.1</span> Per Protein</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(protein_id) <span class="sc">%&gt;%</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(frac_na)) <span class="sc">%&gt;%</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">protein_id =</span> <span class="fu">factor</span>(protein_id, <span class="at">levels=</span><span class="fu">rev</span>(protein_id))) <span class="sc">%&gt;%</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">y=</span>protein_id, <span class="at">x=</span>frac_na)) <span class="sc">+</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Protein"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="per-specimen" class="level3" data-number="7.3.2">
<h3 data-number="7.3.2" class="anchored" data-anchor-id="per-specimen"><span class="header-section-number">7.3.2</span> Per Specimen</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> frac_na)) <span class="sc">+</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add color for better visibility</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins =</span> <span class="dv">20</span>, <span class="at">geom =</span> <span class="st">"text"</span>, </span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Display bin counts at the top</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(frac_na <span class="sc">&gt;</span> <span class="fl">0.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, frac_na, subject_id, dataset, </span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>                actual_day_relative_to_boost, planned_day_relative_to_boost)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["specimen_id"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["frac_na"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["subject_id"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["dataset"],"name":[4],"type":["chr"],"align":["left"]},{"label":["actual_day_relative_to_boost"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["planned_day_relative_to_boost"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"750","2":"0.5333333","3":"99","4":"2022_dataset","5":"-26","6":"-30"},{"1":"760","2":"0.5555556","3":"100","4":"2022_dataset","5":"-31","6":"-30"},{"1":"824","2":"0.5333333","3":"106","4":"2022_dataset","5":"7","6":"7"},{"1":"833","2":"0.5555556","3":"107","4":"2022_dataset","5":"3","6":"3"},{"1":"894","2":"0.5111111","3":"114","4":"2022_dataset","5":"3","6":"3"},{"1":"903","2":"0.5777778","3":"115","4":"2022_dataset","5":"1","6":"1"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="question-for-assay-metadata" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Question for Assay Metadata</h1>
<section id="how-does-the-quality-control-for-olink-look-like" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="how-does-the-quality-control-for-olink-look-like"><span class="header-section-number">8.1</span> How does the quality control for Olink look like?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(quality_control)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["quality_control"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"PASS","2":"5625"},{"1":"Pass","2":"14775"},{"1":"Warning","2":"300"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["partition"],"name":[1],"type":["chr"],"align":["left"]},{"label":["dataset"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"challenge","2":"2023_dataset","3":"4275"},{"1":"train","2":"2020_dataset","3":"2700"},{"1":"train","2":"2021_dataset","3":"8100"},{"1":"train","2":"2022_dataset","3":"5625"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_legendplex <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["partition"],"name":[1],"type":["chr"],"align":["left"]},{"label":["dataset"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"challenge","2":"2023_dataset","3":"2268"},{"1":"train","2":"2021_dataset","3":"2282"},{"1":"train","2":"2022_dataset","3":"1764"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="how-often-is-the-cytokine-concentration-below-the-lod-in-olink" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="how-often-is-the-cytokine-concentration-below-the-lod-in-olink"><span class="header-section-number">8.2</span> How often is the Cytokine Concentration below the LOD in Olink</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check how often is it below the limit of detection?</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">below_lod =</span> concentration <span class="sc">&lt;</span> lower_limit_of_quantitation) <span class="sc">%&gt;%</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(below_lod)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["below_lod"],"name":[1],"type":["lgl"],"align":["right"]},{"label":["n"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"FALSE","2":"18504"},{"1":"TRUE","2":"1921"},{"1":"NA","2":"275"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="what-is-the-fraction-of-unspecific-antibody-measurements" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="what-is-the-fraction-of-unspecific-antibody-measurements"><span class="header-section-number">8.3</span> What is the fraction of unspecific antibody measurements</h2>
<section id="per-isotype-antigen" class="level3" data-number="8.3.1">
<h3 data-number="8.3.1" class="anchored" data-anchor-id="per-isotype-antigen"><span class="header-section-number">8.3.1</span> Per Isotype-Antigen</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_ab_titer <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(isotype_antigen) <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">is_antigen_specific_fraction =</span> <span class="fu">mean</span>(is_antigen_specific)) <span class="sc">%&gt;%</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>is_antigen_specific_fraction)) <span class="sc">+</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">30</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins=</span><span class="dv">30</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="per-specimen-1" class="level3" data-number="8.3.2">
<h3 data-number="8.3.2" class="anchored" data-anchor-id="per-specimen-1"><span class="header-section-number">8.3.2</span> Per Specimen</h3>
<p>So in some specimen we have non-specific antibody measurements? What does that mean?</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_ab_titer <span class="sc">%&gt;%</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">is_antigen_specific_fraction =</span> <span class="fu">mean</span>(is_antigen_specific)) <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>is_antigen_specific_fraction)) <span class="sc">+</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="hierarchy-in-pbmc-frequency-data" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Hierarchy in PBMC Frequency Data</h1>
<section id="nk" class="level2" data-number="9.1">
<h2 data-number="9.1" class="anchored" data-anchor-id="nk"><span class="header-section-number">9.1</span> NK</h2>
<p>There are 3 types of NK cells:</p>
<ul>
<li>NK cells (CD3-CD19-CD56+): CD19-CD3-CD56+/++</li>
<li>NK: CD19-CD3-CD56+</li>
<li>CD56high NK cells: CD19-CD3-CD56++</li>
</ul>
<p>And I don’t know what to do with that information!</p>
</section>
<section id="no-gating-info-for-2023" class="level2" data-number="9.2">
<h2 data-number="9.2" class="anchored" data-anchor-id="no-gating-info-for-2023"><span class="header-section-number">9.2</span> No Gating Info for 2023</h2>
<ul>
<li>I just assume it is the same as in 2022, but this might not be true since the data look so different</li>
</ul>
</section>
<section id="no-gating-info-for-basophils-and-cd19-in-2020" class="level2" data-number="9.3">
<h2 data-number="9.3" class="anchored" data-anchor-id="no-gating-info-for-basophils-and-cd19-in-2020"><span class="header-section-number">9.3</span> No Gating Info for Basophils and CD19 in 2020</h2>
<ul>
<li>I just assume it is the same as in 2021 and 2022</li>
</ul>
</section>
<section id="no-gating-info-for-non-pdcs-at-all" class="level2" data-number="9.4">
<h2 data-number="9.4" class="anchored" data-anchor-id="no-gating-info-for-non-pdcs-at-all"><span class="header-section-number">9.4</span> No Gating Info for non-pDCs at all</h2>
<p>Not sure how to fix this)</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(level)) <span class="sc">%&gt;%</span></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(dataset, cell_type_name) <span class="sc">%&gt;%</span></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["dataset"],"name":[1],"type":["chr"],"align":["left"]},{"label":["cell_type_name"],"name":[2],"type":["chr"],"align":["left"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="should-the-proportions-sum-up-to-1-at-level-0" class="level2" data-number="9.5">
<h2 data-number="9.5" class="anchored" data-anchor-id="should-the-proportions-sum-up-to-1-at-level-0"><span class="header-section-number">9.5</span> Should the Proportions sum up to 1 at level 0?</h2>
<ul>
<li>Trying to take the hierarchical information into account, it still does not make sense</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(level <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="difference-in-2023-dataset" class="level2" data-number="9.6">
<h2 data-number="9.6" class="anchored" data-anchor-id="difference-in-2023-dataset"><span class="header-section-number">9.6</span> Difference in 2023 Dataset</h2>
<ul>
<li>It seems like the 2023 data are very very different, which makes me hesitant to use it for any kind of predictions on the challenge data. E.g. just looking at level 0 which should make sense I get the following</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># first check for NAs for the levels</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(level))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["specimen_id"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["cell_type_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["percent_live_cell"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["dataset"],"name":[4],"type":["chr"],"align":["left"]},{"label":["level"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(level <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<ul>
<li>Especially, when completely ignoring the hierarchy in the gating information, I would expect to have more than 100% in <code>percelt_live_cell_sum</code>, but this is not true for the specimens from the 2023 dataset</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<ul>
<li>Or what I use a list of predefined cell types of interest?</li>
</ul>
</section>
</section>
<section id="features-for-t-cell-activation-assay-aim" class="level1" data-number="10">
<h1 data-number="10"><span class="header-section-number">10</span> Features for T cell Activation Assay (AIM)</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_activation <span class="sc">%&gt;%</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(stimulation) <span class="sc">%&gt;%</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["stimulation"],"name":[1],"type":["chr"],"align":["left"]}],"data":[{"1":"DMSO"},{"1":"PT"},{"1":"TT"},{"1":"PHA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_activation <span class="sc">%&gt;%</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>analyte_percentages), <span class="at">bins=</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>stimulation) <span class="sc">+</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="features-for-t-cell-polarization-assay-fluorospot" class="level1" data-number="11">
<h1 data-number="11"><span class="header-section-number">11</span> Features for T cell Polarization Assay (FluoroSpot)</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(protein_meta, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"protein_id"</span><span class="ot">=</span><span class="st">"uniprot_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(protein_id, cytokine) <span class="sc">%&gt;%</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["protein_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["cytokine"],"name":[2],"type":["chr"],"align":["left"]}],"data":[{"1":"P01579","2":"IFNG"},{"1":"Q16552","2":"IL17A"},{"1":"P05113","2":"IL5"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(stimulation) <span class="sc">%&gt;%</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["stimulation"],"name":[1],"type":["chr"],"align":["left"]}],"data":[{"1":"PHA"},{"1":"PT"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>where</p>
<ul>
<li><p>DMSO: “Negative” Control (solvent where peptides are dissolved in)</p></li>
<li><p>PHA (Phytohaemagglutinin): “Postive” Control (this should elicit strong T cell response?)</p></li>
<li><p>PT: Bordetella Pertussis Toxin</p></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(protein_meta, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"protein_id"</span><span class="ot">=</span><span class="st">"uniprot_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>analyte_counts), <span class="at">bins=</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows=</span><span class="fu">vars</span>(stimulation), <span class="at">cols=</span><span class="fu">vars</span>(cytokine)) <span class="sc">+</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="data_overview_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: the assay has been normalized relative to DMSO</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(stimulation<span class="sc">==</span><span class="st">"DMSO"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(analyte_counts) <span class="sc">%&gt;%</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">near</span>(., <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NaN</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(stimulation<span class="sc">!=</span><span class="st">"DMSO"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(stimulation_protein_id) <span class="sc">%&gt;%</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PHA_P01579" "PHA_Q16552" "PHA_P05113" "PT_P01579"  "PT_Q16552" 
[6] "PT_P05113" </code></pre>
</div>
</div>
</section>
<section id="other-notes" class="level1" data-number="12">
<h1 data-number="12"><span class="header-section-number">12</span> Other Notes</h1>
<ul>
<li>Only from 2022 onwards do we have time-course baseline measurements, i.e.&nbsp;<code>-30, -14/15, 0</code></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset, planned_day_relative_to_boost) <span class="sc">%&gt;%</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">&lt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["partition"],"name":[1],"type":["chr"],"align":["left"]},{"label":["dataset"],"name":[2],"type":["chr"],"align":["left"]},{"label":["planned_day_relative_to_boost"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["n"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"challenge","2":"2023_dataset","3":"-30","4":"54"},{"1":"challenge","2":"2023_dataset","3":"-14","4":"54"},{"1":"train","2":"2022_dataset","3":"-30","4":"21"},{"1":"train","2":"2022_dataset","3":"-15","4":"21"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="appendix" class="level1" data-number="13">
<h1 data-number="13"><span class="header-section-number">13</span> Appendix</h1>
<button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne">
Session Information
</button>
<div id="collapseOne" class="accordion-collapse collapse">
<div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 14.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] C

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] circlize_0.4.16       ComplexHeatmap_2.15.4 magick_2.8.3         
 [4] FactoMineR_2.11       factoextra_1.0.7      ggdark_0.2.1         
 [7] lubridate_1.9.3       forcats_1.0.0         stringr_1.5.1        
[10] dplyr_1.1.4           purrr_1.0.2           readr_2.1.5          
[13] tidyr_1.3.1           tibble_3.2.1          ggplot2_3.5.1        
[16] tidyverse_2.0.0      

loaded via a namespace (and not attached):
 [1] bit64_4.0.5          vroom_1.6.5          jsonlite_1.8.8      
 [4] splines_4.2.3        foreach_1.5.2        stats4_4.2.3        
 [7] yaml_2.3.10          ggrepel_0.9.5        pillar_1.9.0        
[10] lattice_0.22-6       glue_1.7.0           digest_0.6.37       
[13] RColorBrewer_1.1-3   colorspace_2.1-1     sandwich_3.1-0      
[16] htmltools_0.5.8.1    Matrix_1.6-5         pkgconfig_2.0.3     
[19] GetoptLong_1.0.5     xtable_1.8-4         mvtnorm_1.2-6       
[22] scales_1.3.0         tzdb_0.4.0           timechange_0.3.0    
[25] emmeans_1.10.4       farver_2.1.2         generics_0.1.3      
[28] IRanges_2.32.0       DT_0.33              TH.data_1.1-2       
[31] withr_3.0.1          BiocGenerics_0.44.0  cli_3.6.3           
[34] survival_3.7-0       magrittr_2.0.3       crayon_1.5.3        
[37] estimability_1.5.1   evaluate_0.24.0      fansi_1.0.6         
[40] doParallel_1.0.17    MASS_7.3-60.0.1      tools_4.2.3         
[43] hms_1.1.3            GlobalOptions_0.1.2  lifecycle_1.0.4     
[46] matrixStats_1.4.1    multcomp_1.4-26      S4Vectors_0.36.2    
[49] munsell_0.5.1        cluster_2.1.6        flashClust_1.01-2   
[52] compiler_4.2.3       multcompView_0.1-10  rlang_1.1.4         
[55] iterators_1.0.14     rjson_0.2.21         htmlwidgets_1.6.4   
[58] leaps_3.2            labeling_0.4.3       rmarkdown_2.28      
[61] gtable_0.3.5         codetools_0.2-20     R6_2.5.1            
[64] zoo_1.8-12           knitr_1.48           bit_4.0.5           
[67] fastmap_1.2.0        utf8_1.2.4           clue_0.3-65         
[70] shape_1.4.6.1        stringi_1.8.4        parallel_4.2.3      
[73] Rcpp_1.0.13          vctrs_0.6.5          png_0.1-8           
[76] scatterplot3d_0.3-44 tidyselect_1.2.1     xfun_0.47           
[79] coda_0.19-4.1       </code></pre>
</div>
</div>
</div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb56" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Data Overview"</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Philipp Sven Lars Schäfer"</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "`r format(Sys.time(), '%d %B, %Y')`"</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> source</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a><span class="an">engine:</span><span class="co"> knitr</span></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a><span class="fu"># Packages</span></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick)</span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(circlize)</span>
<span id="cb56-22"><a href="#cb56-22" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb56-23"><a href="#cb56-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-24"><a href="#cb56-24" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb56-25"><a href="#cb56-25" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span>
<span id="cb56-26"><a href="#cb56-26" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-27"><a href="#cb56-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-28"><a href="#cb56-28" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data</span></span>
<span id="cb56-29"><a href="#cb56-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-32"><a href="#cb56-32" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-33"><a href="#cb56-33" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span>
<span id="cb56-34"><a href="#cb56-34" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-35"><a href="#cb56-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-38"><a href="#cb56-38" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-39"><a href="#cb56-39" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb56-40"><a href="#cb56-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-41"><a href="#cb56-41" aria-hidden="true" tabindex="-1"></a><span class="co">#exp_data &lt;- read_harmonized_experimental_data(input_dir)</span></span>
<span id="cb56-42"><a href="#cb56-42" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb56-43"><a href="#cb56-43" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(meta_data, exp_data)</span>
<span id="cb56-44"><a href="#cb56-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-45"><a href="#cb56-45" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb56-46"><a href="#cb56-46" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb56-47"><a href="#cb56-47" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span>
<span id="cb56-48"><a href="#cb56-48" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-49"><a href="#cb56-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-50"><a href="#cb56-50" aria-hidden="true" tabindex="-1"></a><span class="fu"># Checks</span></span>
<span id="cb56-51"><a href="#cb56-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-54"><a href="#cb56-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-55"><a href="#cb56-55" aria-hidden="true" tabindex="-1"></a><span class="co"># only works due to filtering step above</span></span>
<span id="cb56-56"><a href="#cb56-56" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb56-57"><a href="#cb56-57" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="sc">~</span> <span class="fu">all</span>(.x<span class="sc">$</span>specimen_id <span class="sc">%in%</span> meta_data<span class="sc">$</span>specimen_id)))</span>
<span id="cb56-58"><a href="#cb56-58" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb56-59"><a href="#cb56-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-60"><a href="#cb56-60" aria-hidden="true" tabindex="-1"></a><span class="co"># no subject is recorded in more than one dataset</span></span>
<span id="cb56-61"><a href="#cb56-61" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-62"><a href="#cb56-62" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-63"><a href="#cb56-63" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-64"><a href="#cb56-64" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(n) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-65"><a href="#cb56-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-66"><a href="#cb56-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb56-67"><a href="#cb56-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-68"><a href="#cb56-68" aria-hidden="true" tabindex="-1"></a><span class="co"># we have the baseline specimen (planned_day_relative_to_boost = 0) for every subject</span></span>
<span id="cb56-69"><a href="#cb56-69" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-70"><a href="#cb56-70" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, planned_day_relative_to_boost, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-71"><a href="#cb56-71" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-72"><a href="#cb56-72" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_baseline =</span> (planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-73"><a href="#cb56-73" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-74"><a href="#cb56-74" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present =</span> <span class="fu">any</span>(is_baseline),</span>
<span id="cb56-75"><a href="#cb56-75" aria-hidden="true" tabindex="-1"></a>                   <span class="at">dataset =</span> <span class="fu">first</span>(dataset)) <span class="sc">%&gt;%</span></span>
<span id="cb56-76"><a href="#cb56-76" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-77"><a href="#cb56-77" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present_frac =</span> <span class="fu">mean</span>(baseline_present)) <span class="sc">%&gt;%</span></span>
<span id="cb56-78"><a href="#cb56-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(baseline_present_frac) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-79"><a href="#cb56-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-80"><a href="#cb56-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb56-81"><a href="#cb56-81" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-82"><a href="#cb56-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-83"><a href="#cb56-83" aria-hidden="true" tabindex="-1"></a><span class="fu"># Demography &amp; Metadata</span></span>
<span id="cb56-84"><a href="#cb56-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-85"><a href="#cb56-85" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many subjects do we have per dataset / per partition?</span></span>
<span id="cb56-86"><a href="#cb56-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-89"><a href="#cb56-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-90"><a href="#cb56-90" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-91"><a href="#cb56-91" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, dataset, partition, infancy_vac) <span class="sc">%&gt;%</span></span>
<span id="cb56-92"><a href="#cb56-92" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-93"><a href="#cb56-93" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(dataset, partition, infancy_vac) <span class="sc">%&gt;%</span></span>
<span id="cb56-94"><a href="#cb56-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> n, <span class="at">y =</span> dataset, <span class="at">fill =</span> infancy_vac, <span class="at">color =</span> partition)) <span class="sc">+</span></span>
<span id="cb56-95"><a href="#cb56-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>), <span class="at">width =</span> <span class="fl">0.7</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span>  </span>
<span id="cb56-96"><a href="#cb56-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> n), </span>
<span id="cb56-97"><a href="#cb56-97" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>),  </span>
<span id="cb56-98"><a href="#cb56-98" aria-hidden="true" tabindex="-1"></a>            <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb56-99"><a href="#cb56-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> infancy_vac_colors) <span class="sc">+</span></span>
<span id="cb56-100"><a href="#cb56-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> partition_colors) <span class="sc">+</span></span>
<span id="cb56-101"><a href="#cb56-101" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb56-102"><a href="#cb56-102" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-103"><a href="#cb56-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-104"><a href="#cb56-104" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the age range</span></span>
<span id="cb56-105"><a href="#cb56-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-108"><a href="#cb56-108" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-109"><a href="#cb56-109" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-110"><a href="#cb56-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb56-111"><a href="#cb56-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>age_at_boost, <span class="at">fill=</span>dataset, <span class="at">y=</span><span class="fu">after_stat</span>(density)), </span>
<span id="cb56-112"><a href="#cb56-112" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color=</span><span class="st">"black"</span>, <span class="at">position=</span><span class="st">"identity"</span>, <span class="at">bins=</span><span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb56-113"><a href="#cb56-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb56-114"><a href="#cb56-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors) <span class="sc">+</span></span>
<span id="cb56-115"><a href="#cb56-115" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb56-116"><a href="#cb56-116" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-117"><a href="#cb56-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-118"><a href="#cb56-118" aria-hidden="true" tabindex="-1"></a><span class="fu"># Assays &amp; Timepoints per Subject / Dataset</span></span>
<span id="cb56-119"><a href="#cb56-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-120"><a href="#cb56-120" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the difference between planned and actual booster administration</span></span>
<span id="cb56-121"><a href="#cb56-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-124"><a href="#cb56-124" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-125"><a href="#cb56-125" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-126"><a href="#cb56-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-127"><a href="#cb56-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost), <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb56-128"><a href="#cb56-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb56-129"><a href="#cb56-129" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb56-130"><a href="#cb56-130" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-131"><a href="#cb56-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-132"><a href="#cb56-132" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the difference between planned and actual for the baseline</span></span>
<span id="cb56-133"><a href="#cb56-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-136"><a href="#cb56-136" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-137"><a href="#cb56-137" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-138"><a href="#cb56-138" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-139"><a href="#cb56-139" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="st">`</span><span class="at">diff_&gt;_15</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">abs</span>(actual_day_relative_to_boost) <span class="sc">&gt;</span> <span class="dv">15</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-140"><a href="#cb56-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-141"><a href="#cb56-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>actual_day_relative_to_boost, <span class="at">fill=</span><span class="st">`</span><span class="at">diff_&gt;_15</span><span class="st">`</span>), <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb56-142"><a href="#cb56-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb56-143"><a href="#cb56-143" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb56-144"><a href="#cb56-144" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-145"><a href="#cb56-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-146"><a href="#cb56-146" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many time points do we have per subject?</span></span>
<span id="cb56-147"><a href="#cb56-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-150"><a href="#cb56-150" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-151"><a href="#cb56-151" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-152"><a href="#cb56-152" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(dataset, subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-153"><a href="#cb56-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-154"><a href="#cb56-154" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">fill=</span>dataset),<span class="at">color=</span><span class="st">"black"</span>, <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb56-155"><a href="#cb56-155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset, <span class="at">ncol=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb56-156"><a href="#cb56-156" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb56-157"><a href="#cb56-157" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb56-158"><a href="#cb56-158" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors)</span>
<span id="cb56-159"><a href="#cb56-159" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-160"><a href="#cb56-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-161"><a href="#cb56-161" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many assays do we have for the baseline measurement?</span></span>
<span id="cb56-162"><a href="#cb56-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-165"><a href="#cb56-165" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-166"><a href="#cb56-166" aria-hidden="true" tabindex="-1"></a>assays_per_specimen <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">imap</span>(exp_data, <span class="sc">~</span> .x <span class="sc">%&gt;%</span> </span>
<span id="cb56-167"><a href="#cb56-167" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-168"><a href="#cb56-168" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-169"><a href="#cb56-169" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assay=</span>.y)) <span class="sc">%&gt;%</span></span>
<span id="cb56-170"><a href="#cb56-170" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-171"><a href="#cb56-171" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-172"><a href="#cb56-172" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, assay, subject_id, planned_day_relative_to_boost, infancy_vac, dataset)</span>
<span id="cb56-173"><a href="#cb56-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-174"><a href="#cb56-174" aria-hidden="true" tabindex="-1"></a>assays_per_specimen <span class="sc">%&gt;%</span></span>
<span id="cb56-175"><a href="#cb56-175" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, planned_day_relative_to_boost, dataset, specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-176"><a href="#cb56-176" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(subject_id)) <span class="sc">%&gt;%</span></span>
<span id="cb56-177"><a href="#cb56-177" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-178"><a href="#cb56-178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-179"><a href="#cb56-179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">fill=</span>dataset), <span class="at">color=</span><span class="st">"black"</span>, <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb56-180"><a href="#cb56-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset, <span class="at">ncol=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb56-181"><a href="#cb56-181" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb56-182"><a href="#cb56-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb56-183"><a href="#cb56-183" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors)</span>
<span id="cb56-184"><a href="#cb56-184" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-185"><a href="#cb56-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-186"><a href="#cb56-186" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.height=6, fig.width=12}</span></span>
<span id="cb56-187"><a href="#cb56-187" aria-hidden="true" tabindex="-1"></a><span class="in">for (day in c(0, 1, 3, 14)) {</span></span>
<span id="cb56-188"><a href="#cb56-188" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb56-189"><a href="#cb56-189" aria-hidden="true" tabindex="-1"></a><span class="in">annotation_data &lt;- meta_data %&gt;%</span></span>
<span id="cb56-190"><a href="#cb56-190" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(specimen_id, planned_day_relative_to_boost, infancy_vac, </span></span>
<span id="cb56-191"><a href="#cb56-191" aria-hidden="true" tabindex="-1"></a><span class="in">                dataset, biological_sex) %&gt;%</span></span>
<span id="cb56-192"><a href="#cb56-192" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::filter(specimen_id %in% assays_per_specimen$specimen_id) %&gt;%</span></span>
<span id="cb56-193"><a href="#cb56-193" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::filter(planned_day_relative_to_boost == !!day) %&gt;% </span></span>
<span id="cb56-194"><a href="#cb56-194" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(- planned_day_relative_to_boost) %&gt;%</span></span>
<span id="cb56-195"><a href="#cb56-195" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::arrange(dataset, infancy_vac, biological_sex) %&gt;%</span></span>
<span id="cb56-196"><a href="#cb56-196" aria-hidden="true" tabindex="-1"></a><span class="in">  tibble::column_to_rownames("specimen_id") %&gt;%</span></span>
<span id="cb56-197"><a href="#cb56-197" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(dataset, infancy_vac, biological_sex)</span></span>
<span id="cb56-198"><a href="#cb56-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-199"><a href="#cb56-199" aria-hidden="true" tabindex="-1"></a><span class="in">heatmap_data &lt;- assays_per_specimen %&gt;%</span></span>
<span id="cb56-200"><a href="#cb56-200" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(specimen_id, assay) %&gt;%</span></span>
<span id="cb56-201"><a href="#cb56-201" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(value = 1) %&gt;%</span></span>
<span id="cb56-202"><a href="#cb56-202" aria-hidden="true" tabindex="-1"></a><span class="in">  tidyr::pivot_wider(names_from="assay", values_from="value") %&gt;%</span></span>
<span id="cb56-203"><a href="#cb56-203" aria-hidden="true" tabindex="-1"></a><span class="in">  mutate(across(everything(), .fns = ~replace_na(.,0))) %&gt;%</span></span>
<span id="cb56-204"><a href="#cb56-204" aria-hidden="true" tabindex="-1"></a><span class="in">  tibble::column_to_rownames("specimen_id") %&gt;%</span></span>
<span id="cb56-205"><a href="#cb56-205" aria-hidden="true" tabindex="-1"></a><span class="in">  as.matrix() %&gt;%</span></span>
<span id="cb56-206"><a href="#cb56-206" aria-hidden="true" tabindex="-1"></a><span class="in">  t()</span></span>
<span id="cb56-207"><a href="#cb56-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-208"><a href="#cb56-208" aria-hidden="true" tabindex="-1"></a><span class="in">colnames(heatmap_data) &lt;- as.character(colnames(heatmap_data))</span></span>
<span id="cb56-209"><a href="#cb56-209" aria-hidden="true" tabindex="-1"></a><span class="in">heatmap_data &lt;- heatmap_data[, rownames(annotation_data)]</span></span>
<span id="cb56-210"><a href="#cb56-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-211"><a href="#cb56-211" aria-hidden="true" tabindex="-1"></a><span class="in"># Create the heatmap</span></span>
<span id="cb56-212"><a href="#cb56-212" aria-hidden="true" tabindex="-1"></a><span class="in">ht &lt;- Heatmap(heatmap_data, </span></span>
<span id="cb56-213"><a href="#cb56-213" aria-hidden="true" tabindex="-1"></a><span class="in">              name = "heatmap", </span></span>
<span id="cb56-214"><a href="#cb56-214" aria-hidden="true" tabindex="-1"></a><span class="in">              row_title = "", </span></span>
<span id="cb56-215"><a href="#cb56-215" aria-hidden="true" tabindex="-1"></a><span class="in">              column_title = paste0("Specimens for Planned Day ", day),</span></span>
<span id="cb56-216"><a href="#cb56-216" aria-hidden="true" tabindex="-1"></a><span class="in">              col = colorRamp2(c(0, 1), c("white", "black")),</span></span>
<span id="cb56-217"><a href="#cb56-217" aria-hidden="true" tabindex="-1"></a><span class="in">              cluster_rows = FALSE,</span></span>
<span id="cb56-218"><a href="#cb56-218" aria-hidden="true" tabindex="-1"></a><span class="in">              cluster_columns = FALSE,</span></span>
<span id="cb56-219"><a href="#cb56-219" aria-hidden="true" tabindex="-1"></a><span class="in">              show_row_names = TRUE, </span></span>
<span id="cb56-220"><a href="#cb56-220" aria-hidden="true" tabindex="-1"></a><span class="in">              show_column_names = FALSE,</span></span>
<span id="cb56-221"><a href="#cb56-221" aria-hidden="true" tabindex="-1"></a><span class="in">              width = unit(16, "cm"),</span></span>
<span id="cb56-222"><a href="#cb56-222" aria-hidden="true" tabindex="-1"></a><span class="in">              top_annotation = ComplexHeatmap::HeatmapAnnotation(df = annotation_data,</span></span>
<span id="cb56-223"><a href="#cb56-223" aria-hidden="true" tabindex="-1"></a><span class="in">                                    col = list(</span></span>
<span id="cb56-224"><a href="#cb56-224" aria-hidden="true" tabindex="-1"></a><span class="in">                                      "dataset" = dataset_colors,</span></span>
<span id="cb56-225"><a href="#cb56-225" aria-hidden="true" tabindex="-1"></a><span class="in">                                      "infancy_vac" = infancy_vac_colors,</span></span>
<span id="cb56-226"><a href="#cb56-226" aria-hidden="true" tabindex="-1"></a><span class="in">                                      "biological_sex" = sex_colors</span></span>
<span id="cb56-227"><a href="#cb56-227" aria-hidden="true" tabindex="-1"></a><span class="in">                                    ),</span></span>
<span id="cb56-228"><a href="#cb56-228" aria-hidden="true" tabindex="-1"></a><span class="in">                                    which="column")</span></span>
<span id="cb56-229"><a href="#cb56-229" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb56-230"><a href="#cb56-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-231"><a href="#cb56-231" aria-hidden="true" tabindex="-1"></a><span class="in">draw(ht, </span></span>
<span id="cb56-232"><a href="#cb56-232" aria-hidden="true" tabindex="-1"></a><span class="in">     heatmap_legend_side = "top",</span></span>
<span id="cb56-233"><a href="#cb56-233" aria-hidden="true" tabindex="-1"></a><span class="in">     annotation_legend_side = "top",</span></span>
<span id="cb56-234"><a href="#cb56-234" aria-hidden="true" tabindex="-1"></a><span class="in">     show_heatmap_legend = FALSE # don't show colorbar</span></span>
<span id="cb56-235"><a href="#cb56-235" aria-hidden="true" tabindex="-1"></a><span class="in">     )</span></span>
<span id="cb56-236"><a href="#cb56-236" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb56-237"><a href="#cb56-237" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-238"><a href="#cb56-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-241"><a href="#cb56-241" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-242"><a href="#cb56-242" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-243"><a href="#cb56-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-244"><a href="#cb56-244" aria-hidden="true" tabindex="-1"></a><span class="fu"># Training Data</span></span>
<span id="cb56-245"><a href="#cb56-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-246"><a href="#cb56-246" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many subjects can we use to construct training data?</span></span>
<span id="cb56-247"><a href="#cb56-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-248"><a href="#cb56-248" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1) Antibody Level Tasks (anti-PT 14 days after booster)</span></span>
<span id="cb56-249"><a href="#cb56-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-250"><a href="#cb56-250" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>All but 7 people were assayed approximately 14 days after the booster administration</span>
<span id="cb56-251"><a href="#cb56-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-254"><a href="#cb56-254" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-255"><a href="#cb56-255" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-256"><a href="#cb56-256" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-257"><a href="#cb56-257" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb56-258"><a href="#cb56-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb56-259"><a href="#cb56-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb56-260"><a href="#cb56-260" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb56-261"><a href="#cb56-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span>
<span id="cb56-262"><a href="#cb56-262" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-263"><a href="#cb56-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-266"><a href="#cb56-266" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-267"><a href="#cb56-267" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-268"><a href="#cb56-268" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb56-269"><a href="#cb56-269" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-270"><a href="#cb56-270" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_1_possible =</span> </span>
<span id="cb56-271"><a href="#cb56-271" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-272"><a href="#cb56-272" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-273"><a href="#cb56-273" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>plasma_ab_titer,</span>
<span id="cb56-274"><a href="#cb56-274" aria-hidden="true" tabindex="-1"></a>                                                      <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-275"><a href="#cb56-275" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(isotype_antigen<span class="sc">==</span><span class="st">"IgG_PT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-276"><a href="#cb56-276" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-277"><a href="#cb56-277" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb56-278"><a href="#cb56-278" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb56-279"><a href="#cb56-279" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-280"><a href="#cb56-280" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_1_possible_total =</span> <span class="fu">sum</span>(task_1_possible), </span>
<span id="cb56-281"><a href="#cb56-281" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_1_possible_fraction =</span> <span class="fu">mean</span>(task_1_possible))</span>
<span id="cb56-282"><a href="#cb56-282" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-283"><a href="#cb56-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-284"><a href="#cb56-284" aria-hidden="true" tabindex="-1"></a><span class="fu">### 2) Cell Frequency Tasks (Monocytes 1 day after booster)</span></span>
<span id="cb56-285"><a href="#cb56-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-286"><a href="#cb56-286" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>But there is one person where it does not really make sense</span>
<span id="cb56-287"><a href="#cb56-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-290"><a href="#cb56-290" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-291"><a href="#cb56-291" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-292"><a href="#cb56-292" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-293"><a href="#cb56-293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb56-294"><a href="#cb56-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb56-295"><a href="#cb56-295" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb56-296"><a href="#cb56-296" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb56-297"><a href="#cb56-297" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span>
<span id="cb56-298"><a href="#cb56-298" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-299"><a href="#cb56-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-302"><a href="#cb56-302" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-303"><a href="#cb56-303" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-304"><a href="#cb56-304" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb56-305"><a href="#cb56-305" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-306"><a href="#cb56-306" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_possible =</span> </span>
<span id="cb56-307"><a href="#cb56-307" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-308"><a href="#cb56-308" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-309"><a href="#cb56-309" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>pbmc_cell_frequency, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-310"><a href="#cb56-310" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(cell_type_name<span class="sc">==</span><span class="st">"Monocytes"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-311"><a href="#cb56-311" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-312"><a href="#cb56-312" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb56-313"><a href="#cb56-313" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb56-314"><a href="#cb56-314" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-315"><a href="#cb56-315" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_possible_total =</span> <span class="fu">sum</span>(task_possible), </span>
<span id="cb56-316"><a href="#cb56-316" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_possible_fraction =</span> <span class="fu">mean</span>(task_possible))</span>
<span id="cb56-317"><a href="#cb56-317" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-318"><a href="#cb56-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-319"><a href="#cb56-319" aria-hidden="true" tabindex="-1"></a><span class="fu">### 3) Gene Expression Tasks (CCL3 expression 3 days after booster)</span></span>
<span id="cb56-320"><a href="#cb56-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-321"><a href="#cb56-321" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>There are 7 people for which this task does nto make sense</span>
<span id="cb56-322"><a href="#cb56-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-325"><a href="#cb56-325" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-326"><a href="#cb56-326" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-327"><a href="#cb56-327" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-328"><a href="#cb56-328" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb56-329"><a href="#cb56-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb56-330"><a href="#cb56-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb56-331"><a href="#cb56-331" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb56-332"><a href="#cb56-332" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span>
<span id="cb56-333"><a href="#cb56-333" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-334"><a href="#cb56-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-337"><a href="#cb56-337" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-338"><a href="#cb56-338" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(exp_data<span class="sc">$</span>pbmc_gene_expression<span class="sc">$</span>versioned_ensembl_gene_id)</span>
<span id="cb56-339"><a href="#cb56-339" aria-hidden="true" tabindex="-1"></a>ccl3_ensembl <span class="ot">&lt;-</span> <span class="st">"ENSG00000277632"</span></span>
<span id="cb56-340"><a href="#cb56-340" aria-hidden="true" tabindex="-1"></a>ccl3_ensembl_versioned <span class="ot">&lt;-</span> all_genes[<span class="fu">str_starts</span>(all_genes, ccl3_ensembl)]</span>
<span id="cb56-341"><a href="#cb56-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-342"><a href="#cb56-342" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-343"><a href="#cb56-343" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb56-344"><a href="#cb56-344" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-345"><a href="#cb56-345" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_possible =</span> </span>
<span id="cb56-346"><a href="#cb56-346" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-347"><a href="#cb56-347" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-348"><a href="#cb56-348" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>pbmc_gene_expression, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-349"><a href="#cb56-349" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(versioned_ensembl_gene_id<span class="sc">==</span>ccl3_ensembl_versioned) <span class="sc">%&gt;%</span></span>
<span id="cb56-350"><a href="#cb56-350" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-351"><a href="#cb56-351" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb56-352"><a href="#cb56-352" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb56-353"><a href="#cb56-353" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-354"><a href="#cb56-354" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_possible_total =</span> <span class="fu">sum</span>(task_possible), </span>
<span id="cb56-355"><a href="#cb56-355" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_possible_fraction =</span> <span class="fu">mean</span>(task_possible))</span>
<span id="cb56-356"><a href="#cb56-356" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-357"><a href="#cb56-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-358"><a href="#cb56-358" aria-hidden="true" tabindex="-1"></a><span class="fu"># Missing Data</span></span>
<span id="cb56-359"><a href="#cb56-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-360"><a href="#cb56-360" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many NA values do we have per assay?</span></span>
<span id="cb56-361"><a href="#cb56-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-364"><a href="#cb56-364" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-365"><a href="#cb56-365" aria-hidden="true" tabindex="-1"></a>value_vars_per_modality <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb56-366"><a href="#cb56-366" aria-hidden="true" tabindex="-1"></a>  <span class="st">"pbmc_cell_frequency"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"percent_live_cell"</span>),</span>
<span id="cb56-367"><a href="#cb56-367" aria-hidden="true" tabindex="-1"></a>  <span class="st">"pbmc_gene_expression"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"raw_count"</span>, <span class="st">"tpm"</span>),</span>
<span id="cb56-368"><a href="#cb56-368" aria-hidden="true" tabindex="-1"></a>  <span class="st">"plasma_ab_titer"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"MFI"</span>, <span class="st">"MFI_normalised"</span>),</span>
<span id="cb56-369"><a href="#cb56-369" aria-hidden="true" tabindex="-1"></a>  <span class="st">"plasma_cytokine_concentration_by_legendplex"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"concentration"</span>),</span>
<span id="cb56-370"><a href="#cb56-370" aria-hidden="true" tabindex="-1"></a>  <span class="st">"plasma_cytokine_concentration_by_olink"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"concentration"</span>),</span>
<span id="cb56-371"><a href="#cb56-371" aria-hidden="true" tabindex="-1"></a>  <span class="st">"t_cell_activation"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"analyte_percentages"</span>),</span>
<span id="cb56-372"><a href="#cb56-372" aria-hidden="true" tabindex="-1"></a>  <span class="st">"t_cell_polarization"</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"analyte_counts"</span>)</span>
<span id="cb56-373"><a href="#cb56-373" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb56-374"><a href="#cb56-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-375"><a href="#cb56-375" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">imap_dfr</span>(exp_data, <span class="cf">function</span>(df, modality) {</span>
<span id="cb56-376"><a href="#cb56-376" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_dfr</span>(value_vars_per_modality[[modality]], </span>
<span id="cb56-377"><a href="#cb56-377" aria-hidden="true" tabindex="-1"></a>                 <span class="sc">~</span> <span class="fu">tibble</span>(<span class="at">modality =</span> modality,</span>
<span id="cb56-378"><a href="#cb56-378" aria-hidden="true" tabindex="-1"></a>                          <span class="at">value_vars =</span> .x,</span>
<span id="cb56-379"><a href="#cb56-379" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sum_na =</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(df[[.x]]))))</span>
<span id="cb56-380"><a href="#cb56-380" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb56-381"><a href="#cb56-381" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-382"><a href="#cb56-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-383"><a href="#cb56-383" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the fraction of NA values in the PBMC Cell Type Frequencies</span></span>
<span id="cb56-384"><a href="#cb56-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-385"><a href="#cb56-385" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Cell Type</span></span>
<span id="cb56-386"><a href="#cb56-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-389"><a href="#cb56-389" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-390"><a href="#cb56-390" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb56-391"><a href="#cb56-391" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb56-392"><a href="#cb56-392" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(cell_type_name) <span class="sc">%&gt;%</span></span>
<span id="cb56-393"><a href="#cb56-393" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb56-394"><a href="#cb56-394" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(frac_na)) <span class="sc">%&gt;%</span></span>
<span id="cb56-395"><a href="#cb56-395" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cell_type_name =</span> <span class="fu">factor</span>(cell_type_name, <span class="at">levels=</span><span class="fu">rev</span>(cell_type_name))) <span class="sc">%&gt;%</span></span>
<span id="cb56-396"><a href="#cb56-396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-397"><a href="#cb56-397" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">y=</span>cell_type_name, <span class="at">x=</span>frac_na)) <span class="sc">+</span></span>
<span id="cb56-398"><a href="#cb56-398" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb56-399"><a href="#cb56-399" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Cell Type"</span>)</span>
<span id="cb56-400"><a href="#cb56-400" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-401"><a href="#cb56-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-402"><a href="#cb56-402" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Cell Specimen</span></span>
<span id="cb56-403"><a href="#cb56-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-406"><a href="#cb56-406" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-407"><a href="#cb56-407" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb56-408"><a href="#cb56-408" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb56-409"><a href="#cb56-409" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-410"><a href="#cb56-410" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb56-411"><a href="#cb56-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> frac_na)) <span class="sc">+</span></span>
<span id="cb56-412"><a href="#cb56-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add color for better visibility</span></span>
<span id="cb56-413"><a href="#cb56-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins =</span> <span class="dv">20</span>, <span class="at">geom =</span> <span class="st">"text"</span>, </span>
<span id="cb56-414"><a href="#cb56-414" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Display bin counts at the top</span></span>
<span id="cb56-415"><a href="#cb56-415" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb56-416"><a href="#cb56-416" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span>
<span id="cb56-417"><a href="#cb56-417" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-418"><a href="#cb56-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-419"><a href="#cb56-419" aria-hidden="true" tabindex="-1"></a>So there are 5 specimen with more than 20% NA, which are all from the same subject!</span>
<span id="cb56-420"><a href="#cb56-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-423"><a href="#cb56-423" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-424"><a href="#cb56-424" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb56-425"><a href="#cb56-425" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb56-426"><a href="#cb56-426" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-427"><a href="#cb56-427" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb56-428"><a href="#cb56-428" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(frac_na <span class="sc">&gt;</span> <span class="fl">0.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-429"><a href="#cb56-429" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>)</span>
<span id="cb56-430"><a href="#cb56-430" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-431"><a href="#cb56-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-432"><a href="#cb56-432" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=12, include=FALSE}</span></span>
<span id="cb56-433"><a href="#cb56-433" aria-hidden="true" tabindex="-1"></a><span class="in">exp_data$pbmc_cell_frequency %&gt;%</span></span>
<span id="cb56-434"><a href="#cb56-434" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::filter(specimen_id %in% c(537, 538, 539, 540, 541)) %&gt;%</span></span>
<span id="cb56-435"><a href="#cb56-435" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::left_join(meta_data, by="specimen_id") %&gt;%</span></span>
<span id="cb56-436"><a href="#cb56-436" aria-hidden="true" tabindex="-1"></a><span class="in">  tidyr::drop_na() %&gt;%</span></span>
<span id="cb56-437"><a href="#cb56-437" aria-hidden="true" tabindex="-1"></a><span class="in">  ggplot() +</span></span>
<span id="cb56-438"><a href="#cb56-438" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_line(aes(x=planned_day_relative_to_boost, y=percent_live_cell, color=cell_type_name)) +</span></span>
<span id="cb56-439"><a href="#cb56-439" aria-hidden="true" tabindex="-1"></a><span class="in">  ggdark::dark_mode(verbose=FALSE)</span></span>
<span id="cb56-440"><a href="#cb56-440" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-441"><a href="#cb56-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-442"><a href="#cb56-442" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the fraction of NA values in the Olink Data?</span></span>
<span id="cb56-443"><a href="#cb56-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-444"><a href="#cb56-444" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Protein</span></span>
<span id="cb56-445"><a href="#cb56-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-446"><a href="#cb56-446" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.height=6}</span></span>
<span id="cb56-447"><a href="#cb56-447" aria-hidden="true" tabindex="-1"></a><span class="in">exp_data$plasma_cytokine_concentration_by_olink %&gt;%</span></span>
<span id="cb56-448"><a href="#cb56-448" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(is_na = is.na(concentration)) %&gt;%</span></span>
<span id="cb56-449"><a href="#cb56-449" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::group_by(protein_id) %&gt;%</span></span>
<span id="cb56-450"><a href="#cb56-450" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::summarise(frac_na = sum(is_na) / n()) %&gt;%</span></span>
<span id="cb56-451"><a href="#cb56-451" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::arrange(desc(frac_na)) %&gt;%</span></span>
<span id="cb56-452"><a href="#cb56-452" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(protein_id = factor(protein_id, levels=rev(protein_id))) %&gt;%</span></span>
<span id="cb56-453"><a href="#cb56-453" aria-hidden="true" tabindex="-1"></a><span class="in">  ggplot() +</span></span>
<span id="cb56-454"><a href="#cb56-454" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_col(aes(y=protein_id, x=frac_na)) +</span></span>
<span id="cb56-455"><a href="#cb56-455" aria-hidden="true" tabindex="-1"></a><span class="in">  ggdark::dark_mode(verbose=FALSE) +</span></span>
<span id="cb56-456"><a href="#cb56-456" aria-hidden="true" tabindex="-1"></a><span class="in">  labs(x="Fraction of NA Values per Protein")</span></span>
<span id="cb56-457"><a href="#cb56-457" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-458"><a href="#cb56-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-459"><a href="#cb56-459" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Specimen</span></span>
<span id="cb56-460"><a href="#cb56-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-463"><a href="#cb56-463" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-464"><a href="#cb56-464" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb56-465"><a href="#cb56-465" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb56-466"><a href="#cb56-466" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-467"><a href="#cb56-467" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb56-468"><a href="#cb56-468" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> frac_na)) <span class="sc">+</span></span>
<span id="cb56-469"><a href="#cb56-469" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add color for better visibility</span></span>
<span id="cb56-470"><a href="#cb56-470" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins =</span> <span class="dv">20</span>, <span class="at">geom =</span> <span class="st">"text"</span>, </span>
<span id="cb56-471"><a href="#cb56-471" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Display bin counts at the top</span></span>
<span id="cb56-472"><a href="#cb56-472" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb56-473"><a href="#cb56-473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span>
<span id="cb56-474"><a href="#cb56-474" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-475"><a href="#cb56-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-478"><a href="#cb56-478" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-479"><a href="#cb56-479" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb56-480"><a href="#cb56-480" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb56-481"><a href="#cb56-481" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-482"><a href="#cb56-482" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb56-483"><a href="#cb56-483" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(frac_na <span class="sc">&gt;</span> <span class="fl">0.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-484"><a href="#cb56-484" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-485"><a href="#cb56-485" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, frac_na, subject_id, dataset, </span>
<span id="cb56-486"><a href="#cb56-486" aria-hidden="true" tabindex="-1"></a>                actual_day_relative_to_boost, planned_day_relative_to_boost)</span>
<span id="cb56-487"><a href="#cb56-487" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-488"><a href="#cb56-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-489"><a href="#cb56-489" aria-hidden="true" tabindex="-1"></a><span class="fu"># Question for Assay Metadata</span></span>
<span id="cb56-490"><a href="#cb56-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-491"><a href="#cb56-491" aria-hidden="true" tabindex="-1"></a><span class="fu">## How does the quality control for Olink look like?</span></span>
<span id="cb56-492"><a href="#cb56-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-495"><a href="#cb56-495" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-496"><a href="#cb56-496" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb56-497"><a href="#cb56-497" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(quality_control)</span>
<span id="cb56-498"><a href="#cb56-498" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-499"><a href="#cb56-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-500"><a href="#cb56-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-503"><a href="#cb56-503" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-504"><a href="#cb56-504" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb56-505"><a href="#cb56-505" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-506"><a href="#cb56-506" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset)</span>
<span id="cb56-507"><a href="#cb56-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-508"><a href="#cb56-508" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_legendplex <span class="sc">%&gt;%</span></span>
<span id="cb56-509"><a href="#cb56-509" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-510"><a href="#cb56-510" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset)</span>
<span id="cb56-511"><a href="#cb56-511" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-512"><a href="#cb56-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-513"><a href="#cb56-513" aria-hidden="true" tabindex="-1"></a><span class="fu">## How often is the Cytokine Concentration below the LOD in Olink</span></span>
<span id="cb56-514"><a href="#cb56-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-517"><a href="#cb56-517" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-518"><a href="#cb56-518" aria-hidden="true" tabindex="-1"></a><span class="co"># check how often is it below the limit of detection?</span></span>
<span id="cb56-519"><a href="#cb56-519" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb56-520"><a href="#cb56-520" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">below_lod =</span> concentration <span class="sc">&lt;</span> lower_limit_of_quantitation) <span class="sc">%&gt;%</span></span>
<span id="cb56-521"><a href="#cb56-521" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(below_lod)</span>
<span id="cb56-522"><a href="#cb56-522" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-523"><a href="#cb56-523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-524"><a href="#cb56-524" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the fraction of unspecific antibody measurements</span></span>
<span id="cb56-525"><a href="#cb56-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-526"><a href="#cb56-526" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Isotype-Antigen</span></span>
<span id="cb56-527"><a href="#cb56-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-530"><a href="#cb56-530" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-531"><a href="#cb56-531" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_ab_titer <span class="sc">%&gt;%</span></span>
<span id="cb56-532"><a href="#cb56-532" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(isotype_antigen) <span class="sc">%&gt;%</span></span>
<span id="cb56-533"><a href="#cb56-533" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">is_antigen_specific_fraction =</span> <span class="fu">mean</span>(is_antigen_specific)) <span class="sc">%&gt;%</span></span>
<span id="cb56-534"><a href="#cb56-534" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>is_antigen_specific_fraction)) <span class="sc">+</span></span>
<span id="cb56-535"><a href="#cb56-535" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">30</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb56-536"><a href="#cb56-536" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins=</span><span class="dv">30</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb56-537"><a href="#cb56-537" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb56-538"><a href="#cb56-538" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-539"><a href="#cb56-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-540"><a href="#cb56-540" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Specimen</span></span>
<span id="cb56-541"><a href="#cb56-541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-542"><a href="#cb56-542" aria-hidden="true" tabindex="-1"></a>So in some specimen we have non-specific antibody measurements? What does that mean?</span>
<span id="cb56-543"><a href="#cb56-543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-546"><a href="#cb56-546" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-547"><a href="#cb56-547" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_ab_titer <span class="sc">%&gt;%</span></span>
<span id="cb56-548"><a href="#cb56-548" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-549"><a href="#cb56-549" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">is_antigen_specific_fraction =</span> <span class="fu">mean</span>(is_antigen_specific)) <span class="sc">%&gt;%</span></span>
<span id="cb56-550"><a href="#cb56-550" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>is_antigen_specific_fraction)) <span class="sc">+</span></span>
<span id="cb56-551"><a href="#cb56-551" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb56-552"><a href="#cb56-552" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb56-553"><a href="#cb56-553" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb56-554"><a href="#cb56-554" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-555"><a href="#cb56-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-556"><a href="#cb56-556" aria-hidden="true" tabindex="-1"></a><span class="fu"># Hierarchy in PBMC Frequency Data</span></span>
<span id="cb56-557"><a href="#cb56-557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-558"><a href="#cb56-558" aria-hidden="true" tabindex="-1"></a><span class="fu">## NK</span></span>
<span id="cb56-559"><a href="#cb56-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-560"><a href="#cb56-560" aria-hidden="true" tabindex="-1"></a>There are 3 types of NK cells:</span>
<span id="cb56-561"><a href="#cb56-561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-562"><a href="#cb56-562" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>NK cells (CD3-CD19-CD56+): CD19-CD3-CD56+/++</span>
<span id="cb56-563"><a href="#cb56-563" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>NK: CD19-CD3-CD56+</span>
<span id="cb56-564"><a href="#cb56-564" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>CD56high NK cells: CD19-CD3-CD56++</span>
<span id="cb56-565"><a href="#cb56-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-566"><a href="#cb56-566" aria-hidden="true" tabindex="-1"></a>And I don't know what to do with that information!</span>
<span id="cb56-567"><a href="#cb56-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-568"><a href="#cb56-568" aria-hidden="true" tabindex="-1"></a><span class="fu">## No Gating Info for 2023</span></span>
<span id="cb56-569"><a href="#cb56-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-570"><a href="#cb56-570" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>I just assume it is the same as in 2022, but this might not be true since the data look so different</span>
<span id="cb56-571"><a href="#cb56-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-572"><a href="#cb56-572" aria-hidden="true" tabindex="-1"></a><span class="fu">## No Gating Info for Basophils and CD19 in 2020</span></span>
<span id="cb56-573"><a href="#cb56-573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-574"><a href="#cb56-574" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>I just assume it is the same as in 2021 and 2022</span>
<span id="cb56-575"><a href="#cb56-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-576"><a href="#cb56-576" aria-hidden="true" tabindex="-1"></a><span class="fu">## No Gating Info for non-pDCs at all </span></span>
<span id="cb56-577"><a href="#cb56-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-578"><a href="#cb56-578" aria-hidden="true" tabindex="-1"></a>Not sure how to fix this)</span>
<span id="cb56-579"><a href="#cb56-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-582"><a href="#cb56-582" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-583"><a href="#cb56-583" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb56-584"><a href="#cb56-584" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-585"><a href="#cb56-585" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-586"><a href="#cb56-586" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-587"><a href="#cb56-587" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-588"><a href="#cb56-588" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb56-589"><a href="#cb56-589" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb56-590"><a href="#cb56-590" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-591"><a href="#cb56-591" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-592"><a href="#cb56-592" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(level)) <span class="sc">%&gt;%</span></span>
<span id="cb56-593"><a href="#cb56-593" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(dataset, cell_type_name) <span class="sc">%&gt;%</span></span>
<span id="cb56-594"><a href="#cb56-594" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb56-595"><a href="#cb56-595" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-596"><a href="#cb56-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-597"><a href="#cb56-597" aria-hidden="true" tabindex="-1"></a><span class="fu">## Should the Proportions sum up to 1 at level 0?</span></span>
<span id="cb56-598"><a href="#cb56-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-599"><a href="#cb56-599" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Trying to take the hierarchical information into account, it still does not make sense</span>
<span id="cb56-600"><a href="#cb56-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-603"><a href="#cb56-603" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-604"><a href="#cb56-604" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb56-605"><a href="#cb56-605" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-606"><a href="#cb56-606" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-607"><a href="#cb56-607" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-608"><a href="#cb56-608" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-609"><a href="#cb56-609" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb56-610"><a href="#cb56-610" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb56-611"><a href="#cb56-611" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-612"><a href="#cb56-612" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-613"><a href="#cb56-613" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(level <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-614"><a href="#cb56-614" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-615"><a href="#cb56-615" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-616"><a href="#cb56-616" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-617"><a href="#cb56-617" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-618"><a href="#cb56-618" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb56-619"><a href="#cb56-619" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb56-620"><a href="#cb56-620" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-621"><a href="#cb56-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-622"><a href="#cb56-622" aria-hidden="true" tabindex="-1"></a><span class="fu">## Difference in 2023 Dataset</span></span>
<span id="cb56-623"><a href="#cb56-623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-624"><a href="#cb56-624" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>It seems like the 2023 data are very very different, which makes me hesitant to use it for any kind of predictions on the challenge data. E.g. just looking at level 0 which should make sense I get the following</span>
<span id="cb56-625"><a href="#cb56-625" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-628"><a href="#cb56-628" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-629"><a href="#cb56-629" aria-hidden="true" tabindex="-1"></a><span class="co"># first check for NAs for the levels</span></span>
<span id="cb56-630"><a href="#cb56-630" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb56-631"><a href="#cb56-631" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-632"><a href="#cb56-632" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-633"><a href="#cb56-633" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-634"><a href="#cb56-634" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-635"><a href="#cb56-635" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb56-636"><a href="#cb56-636" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb56-637"><a href="#cb56-637" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-638"><a href="#cb56-638" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-639"><a href="#cb56-639" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(level))</span>
<span id="cb56-640"><a href="#cb56-640" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-641"><a href="#cb56-641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-644"><a href="#cb56-644" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-645"><a href="#cb56-645" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb56-646"><a href="#cb56-646" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-647"><a href="#cb56-647" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-648"><a href="#cb56-648" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-649"><a href="#cb56-649" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-650"><a href="#cb56-650" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb56-651"><a href="#cb56-651" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb56-652"><a href="#cb56-652" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-653"><a href="#cb56-653" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-654"><a href="#cb56-654" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(level <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-655"><a href="#cb56-655" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-656"><a href="#cb56-656" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-657"><a href="#cb56-657" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-658"><a href="#cb56-658" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-659"><a href="#cb56-659" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb56-660"><a href="#cb56-660" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb56-661"><a href="#cb56-661" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-662"><a href="#cb56-662" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-663"><a href="#cb56-663" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Especially, when completely ignoring the hierarchy in the gating information, I would expect to have more than 100% in <span class="in">`percelt_live_cell_sum`</span>, but this is not true for the specimens from the 2023 dataset</span>
<span id="cb56-664"><a href="#cb56-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-667"><a href="#cb56-667" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-668"><a href="#cb56-668" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb56-669"><a href="#cb56-669" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-670"><a href="#cb56-670" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb56-671"><a href="#cb56-671" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-672"><a href="#cb56-672" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-673"><a href="#cb56-673" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb56-674"><a href="#cb56-674" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb56-675"><a href="#cb56-675" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb56-676"><a href="#cb56-676" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-677"><a href="#cb56-677" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-678"><a href="#cb56-678" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-679"><a href="#cb56-679" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-680"><a href="#cb56-680" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-681"><a href="#cb56-681" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb56-682"><a href="#cb56-682" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb56-683"><a href="#cb56-683" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-684"><a href="#cb56-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-685"><a href="#cb56-685" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Or what I use a list of predefined cell types of interest?</span>
<span id="cb56-686"><a href="#cb56-686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-687"><a href="#cb56-687" aria-hidden="true" tabindex="-1"></a><span class="fu"># Features for T cell Activation Assay (AIM)</span></span>
<span id="cb56-688"><a href="#cb56-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-691"><a href="#cb56-691" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-692"><a href="#cb56-692" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_activation <span class="sc">%&gt;%</span></span>
<span id="cb56-693"><a href="#cb56-693" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(stimulation) <span class="sc">%&gt;%</span></span>
<span id="cb56-694"><a href="#cb56-694" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb56-695"><a href="#cb56-695" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-696"><a href="#cb56-696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-699"><a href="#cb56-699" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-700"><a href="#cb56-700" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_activation <span class="sc">%&gt;%</span></span>
<span id="cb56-701"><a href="#cb56-701" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-702"><a href="#cb56-702" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>analyte_percentages), <span class="at">bins=</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb56-703"><a href="#cb56-703" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>stimulation) <span class="sc">+</span></span>
<span id="cb56-704"><a href="#cb56-704" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span>
<span id="cb56-705"><a href="#cb56-705" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-706"><a href="#cb56-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-707"><a href="#cb56-707" aria-hidden="true" tabindex="-1"></a><span class="fu"># Features for T cell Polarization Assay (FluoroSpot)</span></span>
<span id="cb56-708"><a href="#cb56-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-711"><a href="#cb56-711" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-712"><a href="#cb56-712" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb56-713"><a href="#cb56-713" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(protein_meta, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"protein_id"</span><span class="ot">=</span><span class="st">"uniprot_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-714"><a href="#cb56-714" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(protein_id, cytokine) <span class="sc">%&gt;%</span></span>
<span id="cb56-715"><a href="#cb56-715" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb56-716"><a href="#cb56-716" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-717"><a href="#cb56-717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-720"><a href="#cb56-720" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-721"><a href="#cb56-721" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb56-722"><a href="#cb56-722" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(stimulation) <span class="sc">%&gt;%</span></span>
<span id="cb56-723"><a href="#cb56-723" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb56-724"><a href="#cb56-724" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-725"><a href="#cb56-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-726"><a href="#cb56-726" aria-hidden="true" tabindex="-1"></a>where</span>
<span id="cb56-727"><a href="#cb56-727" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-728"><a href="#cb56-728" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>DMSO: "Negative" Control (solvent where peptides are dissolved in)</span>
<span id="cb56-729"><a href="#cb56-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-730"><a href="#cb56-730" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>PHA (Phytohaemagglutinin): "Postive" Control (this should elicit strong T cell response?)</span>
<span id="cb56-731"><a href="#cb56-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-732"><a href="#cb56-732" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>PT: Bordetella Pertussis Toxin</span>
<span id="cb56-733"><a href="#cb56-733" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-736"><a href="#cb56-736" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-737"><a href="#cb56-737" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb56-738"><a href="#cb56-738" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(protein_meta, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"protein_id"</span><span class="ot">=</span><span class="st">"uniprot_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-739"><a href="#cb56-739" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-740"><a href="#cb56-740" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>analyte_counts), <span class="at">bins=</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb56-741"><a href="#cb56-741" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows=</span><span class="fu">vars</span>(stimulation), <span class="at">cols=</span><span class="fu">vars</span>(cytokine)) <span class="sc">+</span></span>
<span id="cb56-742"><a href="#cb56-742" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb56-743"><a href="#cb56-743" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span>
<span id="cb56-744"><a href="#cb56-744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-745"><a href="#cb56-745" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: the assay has been normalized relative to DMSO</span></span>
<span id="cb56-746"><a href="#cb56-746" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb56-747"><a href="#cb56-747" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(stimulation<span class="sc">==</span><span class="st">"DMSO"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-748"><a href="#cb56-748" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(analyte_counts) <span class="sc">%&gt;%</span></span>
<span id="cb56-749"><a href="#cb56-749" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">near</span>(., <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-750"><a href="#cb56-750" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mean</span>()</span>
<span id="cb56-751"><a href="#cb56-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-752"><a href="#cb56-752" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb56-753"><a href="#cb56-753" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(stimulation<span class="sc">!=</span><span class="st">"DMSO"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-754"><a href="#cb56-754" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(stimulation_protein_id) <span class="sc">%&gt;%</span></span>
<span id="cb56-755"><a href="#cb56-755" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb56-756"><a href="#cb56-756" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-757"><a href="#cb56-757" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-758"><a href="#cb56-758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-759"><a href="#cb56-759" aria-hidden="true" tabindex="-1"></a><span class="fu"># Other Notes</span></span>
<span id="cb56-760"><a href="#cb56-760" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-761"><a href="#cb56-761" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Only from 2022 onwards do we have time-course baseline measurements, i.e. <span class="in">`-30, -14/15, 0`</span></span>
<span id="cb56-762"><a href="#cb56-762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-765"><a href="#cb56-765" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-766"><a href="#cb56-766" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb56-767"><a href="#cb56-767" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset, planned_day_relative_to_boost) <span class="sc">%&gt;%</span></span>
<span id="cb56-768"><a href="#cb56-768" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">&lt;</span> <span class="dv">0</span>)</span>
<span id="cb56-769"><a href="#cb56-769" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-770"><a href="#cb56-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-771"><a href="#cb56-771" aria-hidden="true" tabindex="-1"></a><span class="fu"># Appendix</span></span>
<span id="cb56-772"><a href="#cb56-772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-773"><a href="#cb56-773" aria-hidden="true" tabindex="-1"></a>&lt;button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne" &gt;Session Information&lt;/button&gt;&lt;div id="collapseOne" class="accordion-collapse collapse"&gt;&lt;div&gt;</span>
<span id="cb56-774"><a href="#cb56-774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-777"><a href="#cb56-777" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb56-778"><a href="#cb56-778" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb56-779"><a href="#cb56-779" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb56-780"><a href="#cb56-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-781"><a href="#cb56-781" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;/div&gt;</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>