<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.45">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Philipp Sven Lars Schäfer">
<meta name="dcterms.date" content="2024-11-12">

<title>Data Overview – PSLS @ 3rd CMI-PB Challenge</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">PSLS @ 3rd CMI-PB Challenge</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/psl-schaefer/cmi_pb_third_challenge"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://x.com/psl_schaefer"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/philipp-sl-schaefer/"> <i class="bi bi-linkedin" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="1">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#packages" id="toc-packages" class="nav-link active" data-scroll-target="#packages"><span class="header-section-number">1</span> Packages</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2</span> Data</a></li>
  <li><a href="#checks" id="toc-checks" class="nav-link" data-scroll-target="#checks"><span class="header-section-number">3</span> Checks</a></li>
  <li><a href="#demography-metadata" id="toc-demography-metadata" class="nav-link" data-scroll-target="#demography-metadata"><span class="header-section-number">4</span> Demography &amp; Metadata</a>
  <ul class="collapse">
  <li><a href="#how-many-subjects-do-we-have-per-dataset-per-partition" id="toc-how-many-subjects-do-we-have-per-dataset-per-partition" class="nav-link" data-scroll-target="#how-many-subjects-do-we-have-per-dataset-per-partition"><span class="header-section-number">4.1</span> How many subjects do we have per dataset / per partition?</a></li>
  <li><a href="#what-is-the-age-range" id="toc-what-is-the-age-range" class="nav-link" data-scroll-target="#what-is-the-age-range"><span class="header-section-number">4.2</span> What is the age range</a></li>
  </ul></li>
  <li><a href="#assays-timepoints-per-subject-dataset" id="toc-assays-timepoints-per-subject-dataset" class="nav-link" data-scroll-target="#assays-timepoints-per-subject-dataset"><span class="header-section-number">5</span> Assays &amp; Timepoints per Subject / Dataset</a>
  <ul class="collapse">
  <li><a href="#what-is-the-difference-between-planned-and-actual-booster-administration" id="toc-what-is-the-difference-between-planned-and-actual-booster-administration" class="nav-link" data-scroll-target="#what-is-the-difference-between-planned-and-actual-booster-administration"><span class="header-section-number">5.1</span> What is the difference between planned and actual booster administration</a></li>
  <li><a href="#what-is-the-difference-between-planned-and-actual-for-the-baseline" id="toc-what-is-the-difference-between-planned-and-actual-for-the-baseline" class="nav-link" data-scroll-target="#what-is-the-difference-between-planned-and-actual-for-the-baseline"><span class="header-section-number">5.2</span> What is the difference between planned and actual for the baseline</a></li>
  <li><a href="#how-many-time-points-do-we-have-per-subject" id="toc-how-many-time-points-do-we-have-per-subject" class="nav-link" data-scroll-target="#how-many-time-points-do-we-have-per-subject"><span class="header-section-number">5.3</span> How many time points do we have per subject?</a></li>
  <li><a href="#how-many-assays-do-we-have-for-the-baseline-measurement" id="toc-how-many-assays-do-we-have-for-the-baseline-measurement" class="nav-link" data-scroll-target="#how-many-assays-do-we-have-for-the-baseline-measurement"><span class="header-section-number">5.4</span> How many assays do we have for the baseline measurement?</a></li>
  </ul></li>
  <li><a href="#training-data" id="toc-training-data" class="nav-link" data-scroll-target="#training-data"><span class="header-section-number">6</span> Training Data</a>
  <ul class="collapse">
  <li><a href="#how-many-subjects-can-we-use-to-construct-training-data" id="toc-how-many-subjects-can-we-use-to-construct-training-data" class="nav-link" data-scroll-target="#how-many-subjects-can-we-use-to-construct-training-data"><span class="header-section-number">6.1</span> How many subjects can we use to construct training data?</a>
  <ul class="collapse">
  <li><a href="#antibody-level-tasks-anti-pt-14-days-after-booster" id="toc-antibody-level-tasks-anti-pt-14-days-after-booster" class="nav-link" data-scroll-target="#antibody-level-tasks-anti-pt-14-days-after-booster"><span class="header-section-number">6.1.1</span> 1) Antibody Level Tasks (anti-PT 14 days after booster)</a></li>
  <li><a href="#cell-frequency-tasks-monocytes-1-day-after-booster" id="toc-cell-frequency-tasks-monocytes-1-day-after-booster" class="nav-link" data-scroll-target="#cell-frequency-tasks-monocytes-1-day-after-booster"><span class="header-section-number">6.1.2</span> 2) Cell Frequency Tasks (Monocytes 1 day after booster)</a></li>
  <li><a href="#gene-expression-tasks-ccl3-expression-3-days-after-booster" id="toc-gene-expression-tasks-ccl3-expression-3-days-after-booster" class="nav-link" data-scroll-target="#gene-expression-tasks-ccl3-expression-3-days-after-booster"><span class="header-section-number">6.1.3</span> 3) Gene Expression Tasks (CCL3 expression 3 days after booster)</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#missing-data" id="toc-missing-data" class="nav-link" data-scroll-target="#missing-data"><span class="header-section-number">7</span> Missing Data</a>
  <ul class="collapse">
  <li><a href="#how-many-na-values-do-we-have-per-assay" id="toc-how-many-na-values-do-we-have-per-assay" class="nav-link" data-scroll-target="#how-many-na-values-do-we-have-per-assay"><span class="header-section-number">7.1</span> How many NA values do we have per assay?</a></li>
  <li><a href="#wide-format" id="toc-wide-format" class="nav-link" data-scroll-target="#wide-format"><span class="header-section-number">7.2</span> Wide format</a></li>
  <li><a href="#what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies" id="toc-what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies" class="nav-link" data-scroll-target="#what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies"><span class="header-section-number">7.3</span> What is the fraction of NA values in the PBMC Cell Type Frequencies</a>
  <ul class="collapse">
  <li><a href="#per-cell-type" id="toc-per-cell-type" class="nav-link" data-scroll-target="#per-cell-type"><span class="header-section-number">7.3.1</span> Per Cell Type</a></li>
  <li><a href="#per-cell-specimen" id="toc-per-cell-specimen" class="nav-link" data-scroll-target="#per-cell-specimen"><span class="header-section-number">7.3.2</span> Per Cell Specimen</a></li>
  </ul></li>
  <li><a href="#what-is-the-fraction-of-na-values-in-the-olink-data" id="toc-what-is-the-fraction-of-na-values-in-the-olink-data" class="nav-link" data-scroll-target="#what-is-the-fraction-of-na-values-in-the-olink-data"><span class="header-section-number">7.4</span> What is the fraction of NA values in the Olink Data?</a>
  <ul class="collapse">
  <li><a href="#per-protein" id="toc-per-protein" class="nav-link" data-scroll-target="#per-protein"><span class="header-section-number">7.4.1</span> Per Protein</a></li>
  <li><a href="#per-specimen" id="toc-per-specimen" class="nav-link" data-scroll-target="#per-specimen"><span class="header-section-number">7.4.2</span> Per Specimen</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#question-for-assay-metadata" id="toc-question-for-assay-metadata" class="nav-link" data-scroll-target="#question-for-assay-metadata"><span class="header-section-number">8</span> Question for Assay Metadata</a>
  <ul class="collapse">
  <li><a href="#how-does-the-quality-control-for-olink-look-like" id="toc-how-does-the-quality-control-for-olink-look-like" class="nav-link" data-scroll-target="#how-does-the-quality-control-for-olink-look-like"><span class="header-section-number">8.1</span> How does the quality control for Olink look like?</a></li>
  <li><a href="#how-often-is-the-cytokine-concentration-below-the-lod-in-olink" id="toc-how-often-is-the-cytokine-concentration-below-the-lod-in-olink" class="nav-link" data-scroll-target="#how-often-is-the-cytokine-concentration-below-the-lod-in-olink"><span class="header-section-number">8.2</span> How often is the Cytokine Concentration below the LOD in Olink</a></li>
  <li><a href="#what-is-the-fraction-of-unspecific-antibody-measurements" id="toc-what-is-the-fraction-of-unspecific-antibody-measurements" class="nav-link" data-scroll-target="#what-is-the-fraction-of-unspecific-antibody-measurements"><span class="header-section-number">8.3</span> What is the fraction of unspecific antibody measurements</a>
  <ul class="collapse">
  <li><a href="#per-isotype-antigen" id="toc-per-isotype-antigen" class="nav-link" data-scroll-target="#per-isotype-antigen"><span class="header-section-number">8.3.1</span> Per Isotype-Antigen</a></li>
  <li><a href="#per-specimen-1" id="toc-per-specimen-1" class="nav-link" data-scroll-target="#per-specimen-1"><span class="header-section-number">8.3.2</span> Per Specimen</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#hierarchy-in-pbmc-frequency-data" id="toc-hierarchy-in-pbmc-frequency-data" class="nav-link" data-scroll-target="#hierarchy-in-pbmc-frequency-data"><span class="header-section-number">9</span> Hierarchy in PBMC Frequency Data</a>
  <ul class="collapse">
  <li><a href="#nk" id="toc-nk" class="nav-link" data-scroll-target="#nk"><span class="header-section-number">9.1</span> NK</a></li>
  <li><a href="#no-gating-info-for-2023" id="toc-no-gating-info-for-2023" class="nav-link" data-scroll-target="#no-gating-info-for-2023"><span class="header-section-number">9.2</span> No Gating Info for 2023</a></li>
  <li><a href="#no-gating-info-for-basophils-and-cd19-in-2020" id="toc-no-gating-info-for-basophils-and-cd19-in-2020" class="nav-link" data-scroll-target="#no-gating-info-for-basophils-and-cd19-in-2020"><span class="header-section-number">9.3</span> No Gating Info for Basophils and CD19 in 2020</a></li>
  <li><a href="#no-gating-info-for-non-pdcs-at-all" id="toc-no-gating-info-for-non-pdcs-at-all" class="nav-link" data-scroll-target="#no-gating-info-for-non-pdcs-at-all"><span class="header-section-number">9.4</span> No Gating Info for non-pDCs at all</a></li>
  <li><a href="#should-the-proportions-sum-up-to-1-at-level-0" id="toc-should-the-proportions-sum-up-to-1-at-level-0" class="nav-link" data-scroll-target="#should-the-proportions-sum-up-to-1-at-level-0"><span class="header-section-number">9.5</span> Should the Proportions sum up to 1 at level 0?</a></li>
  <li><a href="#difference-in-2023-dataset" id="toc-difference-in-2023-dataset" class="nav-link" data-scroll-target="#difference-in-2023-dataset"><span class="header-section-number">9.6</span> Difference in 2023 Dataset</a></li>
  </ul></li>
  <li><a href="#features-for-t-cell-activation-assay-aim" id="toc-features-for-t-cell-activation-assay-aim" class="nav-link" data-scroll-target="#features-for-t-cell-activation-assay-aim"><span class="header-section-number">10</span> Features for T cell Activation Assay (AIM)</a></li>
  <li><a href="#features-for-t-cell-polarization-assay-fluorospot" id="toc-features-for-t-cell-polarization-assay-fluorospot" class="nav-link" data-scroll-target="#features-for-t-cell-polarization-assay-fluorospot"><span class="header-section-number">11</span> Features for T cell Polarization Assay (FluoroSpot)</a></li>
  <li><a href="#other-notes" id="toc-other-notes" class="nav-link" data-scroll-target="#other-notes"><span class="header-section-number">12</span> Other Notes</a></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">13</span> Appendix</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Data Overview</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Philipp Sven Lars Schäfer </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 12, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="packages" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Packages</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(circlize)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Data</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#exp_data &lt;- read_harmonized_experimental_data(input_dir)</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(meta_data, exp_data, <span class="at">gene_meta=</span>gene_meta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>pbmc_cell_frequency | Removed 550 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 6931 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 495 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>pbmc_cell_frequency | Removed 56 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 48 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 234 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>t_cell_polarization | Removed 3 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 300 features because qc warning</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 10540 measurements because wrong unit used</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 2400 measurements because wrong unit used</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_legendplex | Removed 8 because specimen is outlier</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed specimen 750, 760, 824, 833, 894, 903 because fraction of measurements below LOQ &gt; 50%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed specimen 674, 675, 676 because fraction of measurements below LOD &gt; 50%</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="checks" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Checks</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># only works due to filtering step above</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="sc">~</span> <span class="fu">all</span>(.x<span class="sc">$</span>specimen_id <span class="sc">%in%</span> meta_data<span class="sc">$</span>specimen_id)))</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co"># no subject is recorded in more than one dataset</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(n) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="co"># we have the baseline specimen (planned_day_relative_to_boost = 0) for every subject</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, planned_day_relative_to_boost, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_baseline =</span> (planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present =</span> <span class="fu">any</span>(is_baseline),</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>                   <span class="at">dataset =</span> <span class="fu">first</span>(dataset)) <span class="sc">%&gt;%</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present_frac =</span> <span class="fu">mean</span>(baseline_present)) <span class="sc">%&gt;%</span></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(baseline_present_frac) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="cf">function</span>(df) {</span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="st">"unit"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(df)) {</span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">length</span>(<span class="fu">unique</span>(df[[<span class="st">"unit"</span>]])) <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="cn">TRUE</span>)</span>
<span id="cb18-34"><a href="#cb18-34" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb18-35"><a href="#cb18-35" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb18-36"><a href="#cb18-36" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="demography-metadata" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Demography &amp; Metadata</h1>
<section id="how-many-subjects-do-we-have-per-dataset-per-partition" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="how-many-subjects-do-we-have-per-dataset-per-partition"><span class="header-section-number">4.1</span> How many subjects do we have per dataset / per partition?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, dataset, partition, infancy_vac) <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(dataset, partition, infancy_vac) <span class="sc">%&gt;%</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> n, <span class="at">y =</span> dataset, <span class="at">fill =</span> infancy_vac, <span class="at">color =</span> partition)) <span class="sc">+</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>), <span class="at">width =</span> <span class="fl">0.7</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span>  </span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> n), </span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>),  </span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> infancy_vac_colors) <span class="sc">+</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> partition_colors) <span class="sc">+</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="what-is-the-age-range" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="what-is-the-age-range"><span class="header-section-number">4.2</span> What is the age range</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>age_at_boost, <span class="at">fill=</span>dataset, <span class="at">y=</span><span class="fu">after_stat</span>(density)), </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color=</span><span class="st">"black"</span>, <span class="at">position=</span><span class="st">"identity"</span>, <span class="at">bins=</span><span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors) <span class="sc">+</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="assays-timepoints-per-subject-dataset" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Assays &amp; Timepoints per Subject / Dataset</h1>
<section id="what-is-the-difference-between-planned-and-actual-booster-administration" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="what-is-the-difference-between-planned-and-actual-booster-administration"><span class="header-section-number">5.1</span> What is the difference between planned and actual booster administration</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost), <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="what-is-the-difference-between-planned-and-actual-for-the-baseline" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="what-is-the-difference-between-planned-and-actual-for-the-baseline"><span class="header-section-number">5.2</span> What is the difference between planned and actual for the baseline</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="st">`</span><span class="at">diff_&gt;_15</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">abs</span>(actual_day_relative_to_boost) <span class="sc">&gt;</span> <span class="dv">15</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>actual_day_relative_to_boost, <span class="at">fill=</span><span class="st">`</span><span class="at">diff_&gt;_15</span><span class="st">`</span>), <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="how-many-time-points-do-we-have-per-subject" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="how-many-time-points-do-we-have-per-subject"><span class="header-section-number">5.3</span> How many time points do we have per subject?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(dataset, subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">fill=</span>dataset),<span class="at">color=</span><span class="st">"black"</span>, <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset, <span class="at">ncol=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="how-many-assays-do-we-have-for-the-baseline-measurement" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="how-many-assays-do-we-have-for-the-baseline-measurement"><span class="header-section-number">5.4</span> How many assays do we have for the baseline measurement?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>assays_per_specimen <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">imap</span>(exp_data, <span class="sc">~</span> .x <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assay=</span>.y)) <span class="sc">%&gt;%</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, assay, subject_id, planned_day_relative_to_boost, infancy_vac, dataset)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>assays_per_specimen <span class="sc">%&gt;%</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, planned_day_relative_to_boost, dataset, specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(subject_id)) <span class="sc">%&gt;%</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">fill=</span>dataset), <span class="at">color=</span><span class="st">"black"</span>, <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset, <span class="at">ncol=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (day <span class="cf">in</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">14</span>)) {</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>annotation_data <span class="ot">&lt;-</span> meta_data <span class="sc">%&gt;%</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, planned_day_relative_to_boost, infancy_vac, </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>                dataset, biological_sex) <span class="sc">%&gt;%</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(specimen_id <span class="sc">%in%</span> assays_per_specimen<span class="sc">$</span>specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">==</span> <span class="sc">!!</span>day) <span class="sc">%&gt;%</span> </span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span> planned_day_relative_to_boost) <span class="sc">%&gt;%</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(dataset, infancy_vac, biological_sex) <span class="sc">%&gt;%</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(dataset, infancy_vac, biological_sex)</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> assays_per_specimen <span class="sc">%&gt;%</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, assay) <span class="sc">%&gt;%</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">value =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"assay"</span>, <span class="at">values_from=</span><span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="at">.fns =</span> <span class="sc">~</span><span class="fu">replace_na</span>(.,<span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>() <span class="sc">%&gt;%</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(heatmap_data) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">colnames</span>(heatmap_data))</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> heatmap_data[, <span class="fu">rownames</span>(annotation_data)]</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the heatmap</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>ht <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(heatmap_data, </span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>              <span class="at">name =</span> <span class="st">"heatmap"</span>, </span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>              <span class="at">row_title =</span> <span class="st">""</span>, </span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>              <span class="at">column_title =</span> <span class="fu">paste0</span>(<span class="st">"Specimens for Planned Day "</span>, day),</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>              <span class="at">col =</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"black"</span>)),</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>              <span class="at">show_row_names =</span> <span class="cn">TRUE</span>, </span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>              <span class="at">show_column_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>              <span class="at">width =</span> <span class="fu">unit</span>(<span class="dv">16</span>, <span class="st">"cm"</span>),</span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a>              <span class="at">top_annotation =</span> ComplexHeatmap<span class="sc">::</span><span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> annotation_data,</span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">col =</span> <span class="fu">list</span>(</span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"dataset"</span> <span class="ot">=</span> dataset_colors,</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"infancy_vac"</span> <span class="ot">=</span> infancy_vac_colors,</span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"biological_sex"</span> <span class="ot">=</span> sex_colors</span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a>                                    ),</span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">which=</span><span class="st">"column"</span>)</span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht, </span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a>     <span class="at">heatmap_legend_side =</span> <span class="st">"top"</span>,</span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a>     <span class="at">annotation_legend_side =</span> <span class="st">"top"</span>,</span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a>     <span class="at">show_heatmap_legend =</span> <span class="cn">FALSE</span> <span class="co"># don't show colorbar</span></span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a>     )</span>
<span id="cb25-50"><a href="#cb25-50" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-11-2.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-11-3.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-11-4.png" class="img-fluid figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="training-data" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Training Data</h1>
<section id="how-many-subjects-can-we-use-to-construct-training-data" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="how-many-subjects-can-we-use-to-construct-training-data"><span class="header-section-number">6.1</span> How many subjects can we use to construct training data?</h2>
<section id="antibody-level-tasks-anti-pt-14-days-after-booster" class="level3" data-number="6.1.1">
<h3 data-number="6.1.1" class="anchored" data-anchor-id="antibody-level-tasks-anti-pt-14-days-after-booster"><span class="header-section-number">6.1.1</span> 1) Antibody Level Tasks (anti-PT 14 days after booster)</h3>
<ul>
<li>All but 7 people were assayed approximately 14 days after the booster administration</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_1_possible =</span> </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>plasma_ab_titer,</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>                                                      <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(isotype_antigen<span class="sc">==</span><span class="st">"IgG_PT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_1_possible_total =</span> <span class="fu">sum</span>(task_1_possible), </span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_1_possible_fraction =</span> <span class="fu">mean</span>(task_1_possible))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["dataset"],"name":[1],"type":["chr"],"align":["left"]},{"label":["task_1_possible_total"],"name":[2],"type":["int"],"align":["right"]},{"label":["task_1_possible_fraction"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"2020_dataset","2":"0","3":"0.0000000"},{"1":"2021_dataset","2":"33","3":"0.9166667"},{"1":"2022_dataset","2":"21","3":"1.0000000"},{"1":"2023_dataset","2":"0","3":"0.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="cell-frequency-tasks-monocytes-1-day-after-booster" class="level3" data-number="6.1.2">
<h3 data-number="6.1.2" class="anchored" data-anchor-id="cell-frequency-tasks-monocytes-1-day-after-booster"><span class="header-section-number">6.1.2</span> 2) Cell Frequency Tasks (Monocytes 1 day after booster)</h3>
<ul>
<li>But there is one person where it does not really make sense</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_possible =</span> </span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>pbmc_cell_frequency, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(cell_type_name<span class="sc">==</span><span class="st">"Monocytes"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_possible_total =</span> <span class="fu">sum</span>(task_possible), </span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_possible_fraction =</span> <span class="fu">mean</span>(task_possible))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["dataset"],"name":[1],"type":["chr"],"align":["left"]},{"label":["task_possible_total"],"name":[2],"type":["int"],"align":["right"]},{"label":["task_possible_fraction"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"2020_dataset","2":"20","3":"0.3333333"},{"1":"2021_dataset","2":"33","3":"0.9166667"},{"1":"2022_dataset","2":"21","3":"1.0000000"},{"1":"2023_dataset","2":"0","3":"0.0000000"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="gene-expression-tasks-ccl3-expression-3-days-after-booster" class="level3" data-number="6.1.3">
<h3 data-number="6.1.3" class="anchored" data-anchor-id="gene-expression-tasks-ccl3-expression-3-days-after-booster"><span class="header-section-number">6.1.3</span> 3) Gene Expression Tasks (CCL3 expression 3 days after booster)</h3>
<ul>
<li>There are 7 people for which this task does nto make sense</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(exp_data<span class="sc">$</span>pbmc_gene_expression<span class="sc">$</span>versioned_ensembl_gene_id)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>ccl3_ensembl <span class="ot">&lt;-</span> <span class="st">"ENSG00000277632"</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>ccl3_ensembl_versioned <span class="ot">&lt;-</span> all_genes[<span class="fu">str_starts</span>(all_genes, ccl3_ensembl)]</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_possible =</span> </span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>pbmc_gene_expression, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(versioned_ensembl_gene_id<span class="sc">==</span>ccl3_ensembl_versioned) <span class="sc">%&gt;%</span></span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_possible_total =</span> <span class="fu">sum</span>(task_possible), </span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_possible_fraction =</span> <span class="fu">mean</span>(task_possible))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["dataset"],"name":[1],"type":["chr"],"align":["left"]},{"label":["task_possible_total"],"name":[2],"type":["int"],"align":["right"]},{"label":["task_possible_fraction"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"2020_dataset","2":"36","3":"0.6"},{"1":"2021_dataset","2":"36","3":"1.0"},{"1":"2022_dataset","2":"21","3":"1.0"},{"1":"2023_dataset","2":"0","3":"0.0"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="missing-data" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Missing Data</h1>
<section id="how-many-na-values-do-we-have-per-assay" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="how-many-na-values-do-we-have-per-assay"><span class="header-section-number">7.1</span> How many NA values do we have per assay?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">imap</span>(exp_data, <span class="cf">function</span>(df, modality) {</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  feature_col <span class="ot">&lt;-</span> experimental_data_settings[[modality]]<span class="sc">$</span>feature_col</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  value_col <span class="ot">&lt;-</span> experimental_data_settings[[modality]]<span class="sc">$</span>value_col</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">modality =</span> modality, <span class="at">feature_col =</span> feature_col, <span class="at">sum_na =</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(df[[value_col]])))</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["modality"],"name":[1],"type":["chr"],"align":["left"]},{"label":["feature_col"],"name":[2],"type":["chr"],"align":["left"]},{"label":["sum_na"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"pbmc_cell_frequency","2":"cell_type_name","3":"0"},{"1":"pbmc_gene_expression","2":"versioned_ensembl_gene_id","3":"0"},{"1":"plasma_ab_titer","2":"isotype_antigen","3":"0"},{"1":"plasma_cytokine_concentration_by_legendplex","2":"protein_id","3":"0"},{"1":"plasma_cytokine_concentration_by_olink","2":"protein_id","3":"50"},{"1":"t_cell_activation","2":"stimulation","3":"0"},{"1":"t_cell_polarization","2":"stimulation_protein_id","3":"0"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="wide-format" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="wide-format"><span class="header-section-number">7.2</span> Wide format</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">imap_dfr</span>(<span class="fu">generate_wide_experimental_data</span>(<span class="at">experimental_data=</span>exp_data, </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">impute=</span><span class="cn">NULL</span>),</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(mtx, modality) {</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>                    rmeans <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">is.na</span>(mtx))</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">tibble</span>(<span class="at">modality=</span>modality, </span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">subject_id =</span> <span class="fu">names</span>(rmeans), </span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">na_frac=</span>rmeans)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>           }) <span class="sc">%&gt;%</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(modality) <span class="sc">%&gt;%</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean_na_frac =</span> <span class="fu">mean</span>(na_frac))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>t_cell_activation | NA Fraction: 0.0576923076923077 | Removed samples: 681</code></pre>
</div>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["modality"],"name":[1],"type":["chr"],"align":["left"]},{"label":["mean_na_frac"],"name":[2],"type":["dbl"],"align":["right"]}],"data":[{"1":"pbmc_cell_frequency","2":"0.000000000"},{"1":"pbmc_gene_expression","2":"0.000000000"},{"1":"plasma_ab_titer","2":"0.000000000"},{"1":"plasma_cytokine_concentration_by_legendplex","2":"0.000000000"},{"1":"plasma_cytokine_concentration_by_olink","2":"0.002884172"},{"1":"t_cell_activation","2":"0.055369128"},{"1":"t_cell_polarization","2":"0.013409962"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="what-is-the-fraction-of-na-values-in-the-pbmc-cell-type-frequencies"><span class="header-section-number">7.3</span> What is the fraction of NA values in the PBMC Cell Type Frequencies</h2>
<p>With the current selection of cell types, there is no specimen with NA values.</p>
<section id="per-cell-type" class="level3" data-number="7.3.1">
<h3 data-number="7.3.1" class="anchored" data-anchor-id="per-cell-type"><span class="header-section-number">7.3.1</span> Per Cell Type</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(cell_type_name) <span class="sc">%&gt;%</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(frac_na)) <span class="sc">%&gt;%</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cell_type_name =</span> <span class="fu">factor</span>(cell_type_name, <span class="at">levels=</span><span class="fu">rev</span>(cell_type_name))) <span class="sc">%&gt;%</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">y=</span>cell_type_name, <span class="at">x=</span>frac_na)) <span class="sc">+</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Cell Type"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="per-cell-specimen" class="level3" data-number="7.3.2">
<h3 data-number="7.3.2" class="anchored" data-anchor-id="per-cell-specimen"><span class="header-section-number">7.3.2</span> Per Cell Specimen</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> frac_na)) <span class="sc">+</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add color for better visibility</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins =</span> <span class="dv">20</span>, <span class="at">geom =</span> <span class="st">"text"</span>, </span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Display bin counts at the top</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="what-is-the-fraction-of-na-values-in-the-olink-data" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="what-is-the-fraction-of-na-values-in-the-olink-data"><span class="header-section-number">7.4</span> What is the fraction of NA values in the Olink Data?</h2>
<section id="per-protein" class="level3" data-number="7.4.1">
<h3 data-number="7.4.1" class="anchored" data-anchor-id="per-protein"><span class="header-section-number">7.4.1</span> Per Protein</h3>
<p>After excluding Q969D9, the largest fraction of NA values is about 5% which is acceptable I guess.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(protein_id) <span class="sc">%&gt;%</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(frac_na)) <span class="sc">%&gt;%</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">protein_id =</span> <span class="fu">factor</span>(protein_id, <span class="at">levels=</span><span class="fu">rev</span>(protein_id))) <span class="sc">%&gt;%</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">y=</span>protein_id, <span class="at">x=</span>frac_na)) <span class="sc">+</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Protein"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="per-specimen" class="level3" data-number="7.4.2">
<h3 data-number="7.4.2" class="anchored" data-anchor-id="per-specimen"><span class="header-section-number">7.4.2</span> Per Specimen</h3>
<p>There are 6 specimen with more than 50% NA values, I am not sure whether I should exlcude those?</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> frac_na)) <span class="sc">+</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add color for better visibility</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins =</span> <span class="dv">20</span>, <span class="at">geom =</span> <span class="st">"text"</span>, </span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Display bin counts at the top</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(frac_na <span class="sc">&gt;</span> <span class="fl">0.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, frac_na, subject_id, dataset, </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>                actual_day_relative_to_boost, planned_day_relative_to_boost)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["specimen_id"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["frac_na"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["subject_id"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["dataset"],"name":[4],"type":["chr"],"align":["left"]},{"label":["actual_day_relative_to_boost"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["planned_day_relative_to_boost"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="question-for-assay-metadata" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Question for Assay Metadata</h1>
<section id="how-does-the-quality-control-for-olink-look-like" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="how-does-the-quality-control-for-olink-look-like"><span class="header-section-number">8.1</span> How does the quality control for Olink look like?</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(quality_control)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["quality_control"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"PASS","2":"5236"},{"1":"Pass","2":"12100"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["partition"],"name":[1],"type":["chr"],"align":["left"]},{"label":["dataset"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"challenge","2":"2023_dataset","3":"4180"},{"1":"train","2":"2021_dataset","3":"7920"},{"1":"train","2":"2022_dataset","3":"5236"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_legendplex <span class="sc">%&gt;%</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["partition"],"name":[1],"type":["chr"],"align":["left"]},{"label":["dataset"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"challenge","2":"2023_dataset","3":"2268"},{"1":"train","2":"2021_dataset","3":"2198"},{"1":"train","2":"2022_dataset","3":"1750"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="how-often-is-the-cytokine-concentration-below-the-lod-in-olink" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="how-often-is-the-cytokine-concentration-below-the-lod-in-olink"><span class="header-section-number">8.2</span> How often is the Cytokine Concentration below the LOD in Olink</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># check how often is it below the limit of detection?</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">below_lod =</span> concentration <span class="sc">&lt;</span> lower_limit_of_quantitation) <span class="sc">%&gt;%</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(below_lod)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["below_lod"],"name":[1],"type":["lgl"],"align":["right"]},{"label":["n"],"name":[2],"type":["int"],"align":["right"]}],"data":[{"1":"FALSE","2":"15676"},{"1":"TRUE","2":"1610"},{"1":"NA","2":"50"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">below_lod =</span> concentration <span class="sc">&lt;</span> lower_limit_of_quantitation) <span class="sc">%&gt;%</span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">qc_warning =</span> quality_control <span class="sc">==</span> <span class="st">"Warning"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(protein_id) <span class="sc">%&gt;%</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_below_lod =</span> <span class="fu">mean</span>(below_lod, <span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">mean_qc_warning =</span> <span class="fu">mean</span>(qc_warning, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(cytokine_uniprot_mapping, <span class="at">by=</span><span class="st">"protein_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_below_lod, <span class="at">y =</span> mean_qc_warning)) <span class="sc">+</span></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_below_lod, <span class="at">y =</span> mean_qc_warning, </span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>                      <span class="at">label =</span> <span class="fu">ifelse</span>(mean_below_lod <span class="sc">&gt;</span> <span class="fl">0.2</span>, protein_id, <span class="st">''</span>)),</span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">max.overlaps =</span> <span class="cn">Inf</span>, <span class="at">color=</span><span class="st">"grey"</span>) <span class="sc">+</span> <span class="co"># This ensures all labels are shown</span></span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">below_lod =</span> concentration <span class="sc">&lt;</span> lower_limit_of_quantitation) <span class="sc">%&gt;%</span></span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">qc_warning =</span> quality_control <span class="sc">==</span> <span class="st">"Warning"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(protein_id) <span class="sc">%&gt;%</span></span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">mean_below_lod =</span> <span class="fu">mean</span>(below_lod, <span class="at">na.rm=</span><span class="cn">TRUE</span>),</span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a>                   <span class="at">mean_qc_warning =</span> <span class="fu">mean</span>(qc_warning, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(cytokine_uniprot_mapping, <span class="at">by=</span><span class="st">"protein_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_below_lod, <span class="at">y =</span> mean_qc_warning)) <span class="sc">+</span></span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_below_lod, <span class="at">y =</span> mean_qc_warning, </span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a>                      <span class="at">label =</span> <span class="fu">ifelse</span>(mean_below_lod <span class="sc">&gt;</span> <span class="fl">0.2</span>, cytokine, <span class="st">''</span>)),</span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a>                  <span class="at">max.overlaps =</span> <span class="cn">Inf</span>, <span class="at">color=</span><span class="st">"grey"</span>) <span class="sc">+</span> <span class="co"># This ensures all labels are shown</span></span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span>
<span id="cb44-29"><a href="#cb44-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-30"><a href="#cb44-30" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, <span class="at">ncol=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="what-is-the-fraction-of-unspecific-antibody-measurements" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="what-is-the-fraction-of-unspecific-antibody-measurements"><span class="header-section-number">8.3</span> What is the fraction of unspecific antibody measurements</h2>
<section id="per-isotype-antigen" class="level3" data-number="8.3.1">
<h3 data-number="8.3.1" class="anchored" data-anchor-id="per-isotype-antigen"><span class="header-section-number">8.3.1</span> Per Isotype-Antigen</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_ab_titer <span class="sc">%&gt;%</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(isotype_antigen) <span class="sc">%&gt;%</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">is_antigen_specific_fraction =</span> <span class="fu">mean</span>(is_antigen_specific)) <span class="sc">%&gt;%</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>is_antigen_specific_fraction)) <span class="sc">+</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">30</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins=</span><span class="dv">30</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="per-specimen-1" class="level3" data-number="8.3.2">
<h3 data-number="8.3.2" class="anchored" data-anchor-id="per-specimen-1"><span class="header-section-number">8.3.2</span> Per Specimen</h3>
<p>So in some specimen we have non-specific antibody measurements? What does that mean?</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_ab_titer <span class="sc">%&gt;%</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">is_antigen_specific_fraction =</span> <span class="fu">mean</span>(is_antigen_specific)) <span class="sc">%&gt;%</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>is_antigen_specific_fraction)) <span class="sc">+</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="hierarchy-in-pbmc-frequency-data" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Hierarchy in PBMC Frequency Data</h1>
<section id="nk" class="level2" data-number="9.1">
<h2 data-number="9.1" class="anchored" data-anchor-id="nk"><span class="header-section-number">9.1</span> NK</h2>
<p>There are 3 types of NK cells:</p>
<ul>
<li>NK cells (CD3-CD19-CD56+): CD19-CD3-CD56+/++</li>
<li>NK: CD19-CD3-CD56+</li>
<li>CD56high NK cells: CD19-CD3-CD56++</li>
</ul>
<p>And I don’t know what to do with that information!</p>
</section>
<section id="no-gating-info-for-2023" class="level2" data-number="9.2">
<h2 data-number="9.2" class="anchored" data-anchor-id="no-gating-info-for-2023"><span class="header-section-number">9.2</span> No Gating Info for 2023</h2>
<ul>
<li>I just assume it is the same as in 2022, but this might not be true since the data look so different</li>
</ul>
</section>
<section id="no-gating-info-for-basophils-and-cd19-in-2020" class="level2" data-number="9.3">
<h2 data-number="9.3" class="anchored" data-anchor-id="no-gating-info-for-basophils-and-cd19-in-2020"><span class="header-section-number">9.3</span> No Gating Info for Basophils and CD19 in 2020</h2>
<ul>
<li>I just assume it is the same as in 2021 and 2022</li>
</ul>
</section>
<section id="no-gating-info-for-non-pdcs-at-all" class="level2" data-number="9.4">
<h2 data-number="9.4" class="anchored" data-anchor-id="no-gating-info-for-non-pdcs-at-all"><span class="header-section-number">9.4</span> No Gating Info for non-pDCs at all</h2>
<p>Not sure how to fix this)</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(level)) <span class="sc">%&gt;%</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(dataset, cell_type_name) <span class="sc">%&gt;%</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["dataset"],"name":[1],"type":["chr"],"align":["left"]},{"label":["cell_type_name"],"name":[2],"type":["chr"],"align":["left"]}],"data":[{"1":"2023_dataset","2":"Non_N_Classical_Monocytes"},{"1":"2020_dataset","2":"Non_N_Classical_Monocytes"},{"1":"2021_dataset","2":"Non_N_Classical_Monocytes"},{"1":"2022_dataset","2":"Non_N_Classical_Monocytes"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="should-the-proportions-sum-up-to-1-at-level-0" class="level2" data-number="9.5">
<h2 data-number="9.5" class="anchored" data-anchor-id="should-the-proportions-sum-up-to-1-at-level-0"><span class="header-section-number">9.5</span> Should the Proportions sum up to 1 at level 0?</h2>
<ul>
<li>Trying to take the hierarchical information into account, it still does not make sense</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(level <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="difference-in-2023-dataset" class="level2" data-number="9.6">
<h2 data-number="9.6" class="anchored" data-anchor-id="difference-in-2023-dataset"><span class="header-section-number">9.6</span> Difference in 2023 Dataset</h2>
<ul>
<li>It seems like the 2023 data are very very different, which makes me hesitant to use it for any kind of predictions on the challenge data. E.g. just looking at level 0 which should make sense I get the following</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># first check for NAs for the levels</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(level))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["specimen_id"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["cell_type_name"],"name":[2],"type":["chr"],"align":["left"]},{"label":["percent_live_cell"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["dataset"],"name":[4],"type":["chr"],"align":["left"]},{"label":["level"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"1474","2":"Non_N_Classical_Monocytes","3":"0.7636364","4":"2023_dataset","5":"NA"},{"1":"1037","2":"Non_N_Classical_Monocytes","3":"1.2229730","4":"2023_dataset","5":"NA"},{"1":"1310","2":"Non_N_Classical_Monocytes","3":"2.5405405","4":"2023_dataset","5":"NA"},{"1":"1181","2":"Non_N_Classical_Monocytes","3":"0.5675676","4":"2023_dataset","5":"NA"},{"1":"1300","2":"Non_N_Classical_Monocytes","3":"0.5135135","4":"2023_dataset","5":"NA"},{"1":"1205","2":"Non_N_Classical_Monocytes","3":"0.6756757","4":"2023_dataset","5":"NA"},{"1":"1437","2":"Non_N_Classical_Monocytes","3":"1.3818182","4":"2023_dataset","5":"NA"},{"1":"1362","2":"Non_N_Classical_Monocytes","3":"1.5272727","4":"2023_dataset","5":"NA"},{"1":"1395","2":"Non_N_Classical_Monocytes","3":"1.5181818","4":"2023_dataset","5":"NA"},{"1":"1108","2":"Non_N_Classical_Monocytes","3":"1.6418919","4":"2023_dataset","5":"NA"},{"1":"1351","2":"Non_N_Classical_Monocytes","3":"0.7972973","4":"2023_dataset","5":"NA"},{"1":"1183","2":"Non_N_Classical_Monocytes","3":"1.0270270","4":"2023_dataset","5":"NA"},{"1":"1414","2":"Non_N_Classical_Monocytes","3":"1.9636364","4":"2023_dataset","5":"NA"},{"1":"1234","2":"Non_N_Classical_Monocytes","3":"0.9729730","4":"2023_dataset","5":"NA"},{"1":"1226","2":"Non_N_Classical_Monocytes","3":"1.2500000","4":"2023_dataset","5":"NA"},{"1":"1195","2":"Non_N_Classical_Monocytes","3":"0.7770270","4":"2023_dataset","5":"NA"},{"1":"996","2":"Non_N_Classical_Monocytes","3":"0.5454545","4":"2023_dataset","5":"NA"},{"1":"943","2":"Non_N_Classical_Monocytes","3":"1.5454545","4":"2023_dataset","5":"NA"},{"1":"1494","2":"Non_N_Classical_Monocytes","3":"1.0090909","4":"2023_dataset","5":"NA"},{"1":"1174","2":"Non_N_Classical_Monocytes","3":"1.1148649","4":"2023_dataset","5":"NA"},{"1":"1056","2":"Non_N_Classical_Monocytes","3":"0.6636364","4":"2023_dataset","5":"NA"},{"1":"1248","2":"Non_N_Classical_Monocytes","3":"0.5067568","4":"2023_dataset","5":"NA"},{"1":"1466","2":"Non_N_Classical_Monocytes","3":"0.3636364","4":"2023_dataset","5":"NA"},{"1":"1045","2":"Non_N_Classical_Monocytes","3":"0.9391892","4":"2023_dataset","5":"NA"},{"1":"1331","2":"Non_N_Classical_Monocytes","3":"1.0818182","4":"2023_dataset","5":"NA"},{"1":"942","2":"Non_N_Classical_Monocytes","3":"1.7181818","4":"2023_dataset","5":"NA"},{"1":"1363","2":"Non_N_Classical_Monocytes","3":"3.1272727","4":"2023_dataset","5":"NA"},{"1":"1307","2":"Non_N_Classical_Monocytes","3":"2.2094595","4":"2023_dataset","5":"NA"},{"1":"976","2":"Non_N_Classical_Monocytes","3":"1.6959459","4":"2023_dataset","5":"NA"},{"1":"957","2":"Non_N_Classical_Monocytes","3":"1.3545455","4":"2023_dataset","5":"NA"},{"1":"1067","2":"Non_N_Classical_Monocytes","3":"0.9454545","4":"2023_dataset","5":"NA"},{"1":"1299","2":"Non_N_Classical_Monocytes","3":"0.7432432","4":"2023_dataset","5":"NA"},{"1":"1096","2":"Non_N_Classical_Monocytes","3":"1.1351351","4":"2023_dataset","5":"NA"},{"1":"1202","2":"Non_N_Classical_Monocytes","3":"0.7635135","4":"2023_dataset","5":"NA"},{"1":"1064","2":"Non_N_Classical_Monocytes","3":"0.5272727","4":"2023_dataset","5":"NA"},{"1":"1359","2":"Non_N_Classical_Monocytes","3":"1.8000000","4":"2023_dataset","5":"NA"},{"1":"940","2":"Non_N_Classical_Monocytes","3":"1.3818182","4":"2023_dataset","5":"NA"},{"1":"1152","2":"Non_N_Classical_Monocytes","3":"3.3243243","4":"2023_dataset","5":"NA"},{"1":"1384","2":"Non_N_Classical_Monocytes","3":"0.6909091","4":"2023_dataset","5":"NA"},{"1":"1018","2":"Non_N_Classical_Monocytes","3":"1.5472973","4":"2023_dataset","5":"NA"},{"1":"1085","2":"Non_N_Classical_Monocytes","3":"1.6621622","4":"2023_dataset","5":"NA"},{"1":"1456","2":"Non_N_Classical_Monocytes","3":"0.5909091","4":"2023_dataset","5":"NA"},{"1":"1255","2":"Non_N_Classical_Monocytes","3":"0.4189189","4":"2023_dataset","5":"NA"},{"1":"1141","2":"Non_N_Classical_Monocytes","3":"1.0675676","4":"2023_dataset","5":"NA"},{"1":"1447","2":"Non_N_Classical_Monocytes","3":"0.8000000","4":"2023_dataset","5":"NA"},{"1":"1329","2":"Non_N_Classical_Monocytes","3":"0.6363636","4":"2023_dataset","5":"NA"},{"1":"1407","2":"Non_N_Classical_Monocytes","3":"0.9181818","4":"2023_dataset","5":"NA"},{"1":"1426","2":"Non_N_Classical_Monocytes","3":"1.2090909","4":"2023_dataset","5":"NA"},{"1":"1057","2":"Non_N_Classical_Monocytes","3":"0.2727273","4":"2023_dataset","5":"NA"},{"1":"1467","2":"Non_N_Classical_Monocytes","3":"0.5363636","4":"2023_dataset","5":"NA"},{"1":"1396","2":"Non_N_Classical_Monocytes","3":"1.2363636","4":"2023_dataset","5":"NA"},{"1":"1258","2":"Non_N_Classical_Monocytes","3":"0.5540541","4":"2023_dataset","5":"NA"},{"1":"1184","2":"Non_N_Classical_Monocytes","3":"0.3648649","4":"2023_dataset","5":"NA"},{"1":"1066","2":"Non_N_Classical_Monocytes","3":"0.8272727","4":"2023_dataset","5":"NA"},{"1":"1444","2":"Non_N_Classical_Monocytes","3":"0.8181818","4":"2023_dataset","5":"NA"},{"1":"1289","2":"Non_N_Classical_Monocytes","3":"0.4729730","4":"2023_dataset","5":"NA"},{"1":"973","2":"Non_N_Classical_Monocytes","3":"1.1621622","4":"2023_dataset","5":"NA"},{"1":"1170","2":"Non_N_Classical_Monocytes","3":"1.2972973","4":"2023_dataset","5":"NA"},{"1":"1130","2":"Non_N_Classical_Monocytes","3":"0.9932432","4":"2023_dataset","5":"NA"},{"1":"1129","2":"Non_N_Classical_Monocytes","3":"1.3851351","4":"2023_dataset","5":"NA"},{"1":"1159","2":"Non_N_Classical_Monocytes","3":"1.5067568","4":"2023_dataset","5":"NA"},{"1":"1434","2":"Non_N_Classical_Monocytes","3":"1.4545455","4":"2023_dataset","5":"NA"},{"1":"965","2":"Non_N_Classical_Monocytes","3":"1.2432432","4":"2023_dataset","5":"NA"},{"1":"1088","2":"Non_N_Classical_Monocytes","3":"1.4932432","4":"2023_dataset","5":"NA"},{"1":"1119","2":"Non_N_Classical_Monocytes","3":"1.3243243","4":"2023_dataset","5":"NA"},{"1":"1416","2":"Non_N_Classical_Monocytes","3":"1.3454545","4":"2023_dataset","5":"NA"},{"1":"1038","2":"Non_N_Classical_Monocytes","3":"1.5270270","4":"2023_dataset","5":"NA"},{"1":"1194","2":"Non_N_Classical_Monocytes","3":"0.8716216","4":"2023_dataset","5":"NA"},{"1":"1290","2":"Non_N_Classical_Monocytes","3":"0.7567568","4":"2023_dataset","5":"NA"},{"1":"1089","2":"Non_N_Classical_Monocytes","3":"1.2094595","4":"2023_dataset","5":"NA"},{"1":"1227","2":"Non_N_Classical_Monocytes","3":"2.3175676","4":"2023_dataset","5":"NA"},{"1":"1385","2":"Non_N_Classical_Monocytes","3":"1.3454545","4":"2023_dataset","5":"NA"},{"1":"1436","2":"Non_N_Classical_Monocytes","3":"0.9909091","4":"2023_dataset","5":"NA"},{"1":"1077","2":"Non_N_Classical_Monocytes","3":"0.3918919","4":"2023_dataset","5":"NA"},{"1":"997","2":"Non_N_Classical_Monocytes","3":"0.6818182","4":"2023_dataset","5":"NA"},{"1":"1162","2":"Non_N_Classical_Monocytes","3":"2.1486486","4":"2023_dataset","5":"NA"},{"1":"1464","2":"Non_N_Classical_Monocytes","3":"0.5909091","4":"2023_dataset","5":"NA"},{"1":"1126","2":"Non_N_Classical_Monocytes","3":"0.9391892","4":"2023_dataset","5":"NA"},{"1":"1029","2":"Non_N_Classical_Monocytes","3":"0.3513514","4":"2023_dataset","5":"NA"},{"1":"1486","2":"Non_N_Classical_Monocytes","3":"1.8909091","4":"2023_dataset","5":"NA"},{"1":"1247","2":"Non_N_Classical_Monocytes","3":"0.4324324","4":"2023_dataset","5":"NA"},{"1":"1028","2":"Non_N_Classical_Monocytes","3":"0.4189189","4":"2023_dataset","5":"NA"},{"1":"984","2":"Non_N_Classical_Monocytes","3":"0.4363636","4":"2023_dataset","5":"NA"},{"1":"1245","2":"Non_N_Classical_Monocytes","3":"0.5067568","4":"2023_dataset","5":"NA"},{"1":"1427","2":"Non_N_Classical_Monocytes","3":"1.1636364","4":"2023_dataset","5":"NA"},{"1":"1173","2":"Non_N_Classical_Monocytes","3":"1.4054054","4":"2023_dataset","5":"NA"},{"1":"1484","2":"Non_N_Classical_Monocytes","3":"1.5181818","4":"2023_dataset","5":"NA"},{"1":"1496","2":"Non_N_Classical_Monocytes","3":"0.9545455","4":"2023_dataset","5":"NA"},{"1":"1446","2":"Non_N_Classical_Monocytes","3":"0.8090909","4":"2023_dataset","5":"NA"},{"1":"956","2":"Non_N_Classical_Monocytes","3":"1.0181818","4":"2023_dataset","5":"NA"},{"1":"1406","2":"Non_N_Classical_Monocytes","3":"0.8818182","4":"2023_dataset","5":"NA"},{"1":"1017","2":"Non_N_Classical_Monocytes","3":"1.4391892","4":"2023_dataset","5":"NA"},{"1":"955","2":"Non_N_Classical_Monocytes","3":"1.4181818","4":"2023_dataset","5":"NA"},{"1":"1163","2":"Non_N_Classical_Monocytes","3":"2.7972973","4":"2023_dataset","5":"NA"},{"1":"1297","2":"Non_N_Classical_Monocytes","3":"0.9121622","4":"2023_dataset","5":"NA"},{"1":"1148","2":"Non_N_Classical_Monocytes","3":"1.1554054","4":"2023_dataset","5":"NA"},{"1":"1278","2":"Non_N_Classical_Monocytes","3":"0.5878378","4":"2023_dataset","5":"NA"},{"1":"1487","2":"Non_N_Classical_Monocytes","3":"2.0272727","4":"2023_dataset","5":"NA"},{"1":"1224","2":"Non_N_Classical_Monocytes","3":"1.6554054","4":"2023_dataset","5":"NA"},{"1":"1118","2":"Non_N_Classical_Monocytes","3":"1.3648649","4":"2023_dataset","5":"NA"},{"1":"1115","2":"Non_N_Classical_Monocytes","3":"1.0067568","4":"2023_dataset","5":"NA"},{"1":"1058","2":"Non_N_Classical_Monocytes","3":"1.0363636","4":"2023_dataset","5":"NA"},{"1":"1477","2":"Non_N_Classical_Monocytes","3":"1.7181818","4":"2023_dataset","5":"NA"},{"1":"1025","2":"Non_N_Classical_Monocytes","3":"0.3918919","4":"2023_dataset","5":"NA"},{"1":"1424","2":"Non_N_Classical_Monocytes","3":"1.9090909","4":"2023_dataset","5":"NA"},{"1":"1151","2":"Non_N_Classical_Monocytes","3":"2.2027027","4":"2023_dataset","5":"NA"},{"1":"1352","2":"Non_N_Classical_Monocytes","3":"1.0810811","4":"2023_dataset","5":"NA"},{"1":"1311","2":"Non_N_Classical_Monocytes","3":"1.0608108","4":"2023_dataset","5":"NA"},{"1":"1286","2":"Non_N_Classical_Monocytes","3":"0.4256757","4":"2023_dataset","5":"NA"},{"1":"1237","2":"Non_N_Classical_Monocytes","3":"0.8040541","4":"2023_dataset","5":"NA"},{"1":"1137","2":"Non_N_Classical_Monocytes","3":"1.0067568","4":"2023_dataset","5":"NA"},{"1":"1097","2":"Non_N_Classical_Monocytes","3":"1.6081081","4":"2023_dataset","5":"NA"},{"1":"1048","2":"Non_N_Classical_Monocytes","3":"0.8108108","4":"2023_dataset","5":"NA"},{"1":"1078","2":"Non_N_Classical_Monocytes","3":"0.1689189","4":"2023_dataset","5":"NA"},{"1":"1049","2":"Non_N_Classical_Monocytes","3":"0.6216216","4":"2023_dataset","5":"NA"},{"1":"1140","2":"Non_N_Classical_Monocytes","3":"0.6216216","4":"2023_dataset","5":"NA"},{"1":"1279","2":"Non_N_Classical_Monocytes","3":"0.6689189","4":"2023_dataset","5":"NA"},{"1":"1403","2":"Non_N_Classical_Monocytes","3":"0.6818182","4":"2023_dataset","5":"NA"},{"1":"1476","2":"Non_N_Classical_Monocytes","3":"1.3636364","4":"2023_dataset","5":"NA"},{"1":"1381","2":"Non_N_Classical_Monocytes","3":"0.8454545","4":"2023_dataset","5":"NA"},{"1":"1238","2":"Non_N_Classical_Monocytes","3":"1.0067568","4":"2023_dataset","5":"NA"},{"1":"1497","2":"Non_N_Classical_Monocytes","3":"1.0636364","4":"2023_dataset","5":"NA"},{"1":"1098","2":"Non_N_Classical_Monocytes","3":"1.0608108","4":"2023_dataset","5":"NA"},{"1":"966","2":"Non_N_Classical_Monocytes","3":"1.7297297","4":"2023_dataset","5":"NA"},{"1":"994","2":"Non_N_Classical_Monocytes","3":"0.6454545","4":"2023_dataset","5":"NA"},{"1":"987","2":"Non_N_Classical_Monocytes","3":"0.3545455","4":"2023_dataset","5":"NA"},{"1":"1257","2":"Non_N_Classical_Monocytes","3":"0.5000000","4":"2023_dataset","5":"NA"},{"1":"1457","2":"Non_N_Classical_Monocytes","3":"0.8181818","4":"2023_dataset","5":"NA"},{"1":"986","2":"Non_N_Classical_Monocytes","3":"0.3090909","4":"2023_dataset","5":"NA"},{"1":"1454","2":"Non_N_Classical_Monocytes","3":"0.8363636","4":"2023_dataset","5":"NA"},{"1":"1191","2":"Non_N_Classical_Monocytes","3":"0.7702703","4":"2023_dataset","5":"NA"},{"1":"1105","2":"Non_N_Classical_Monocytes","3":"0.6891892","4":"2023_dataset","5":"NA"},{"1":"1350","2":"Non_N_Classical_Monocytes","3":"0.5743243","4":"2023_dataset","5":"NA"},{"1":"1332","2":"Non_N_Classical_Monocytes","3":"0.9181818","4":"2023_dataset","5":"NA"},{"1":"1014","2":"Non_N_Classical_Monocytes","3":"1.0675676","4":"2023_dataset","5":"NA"},{"1":"1417","2":"Non_N_Classical_Monocytes","3":"1.0727273","4":"2023_dataset","5":"NA"},{"1":"1206","2":"Non_N_Classical_Monocytes","3":"0.9391892","4":"2023_dataset","5":"NA"},{"1":"1074","2":"Non_N_Classical_Monocytes","3":"0.5743243","4":"2023_dataset","5":"NA"},{"1":"962","2":"Non_N_Classical_Monocytes","3":"1.0000000","4":"2023_dataset","5":"NA"},{"1":"977","2":"Non_N_Classical_Monocytes","3":"1.3581081","4":"2023_dataset","5":"NA"},{"1":"1276","2":"Non_N_Classical_Monocytes","3":"0.6824324","4":"2023_dataset","5":"NA"},{"1":"1107","2":"Non_N_Classical_Monocytes","3":"0.7567568","4":"2023_dataset","5":"NA"},{"1":"1392","2":"Non_N_Classical_Monocytes","3":"1.1272727","4":"2023_dataset","5":"NA"},{"1":"223","2":"Non_N_Classical_Monocytes","3":"0.4882124","4":"2020_dataset","5":"NA"},{"1":"224","2":"Non_N_Classical_Monocytes","3":"0.4807948","4":"2020_dataset","5":"NA"},{"1":"225","2":"Non_N_Classical_Monocytes","3":"1.0272108","4":"2020_dataset","5":"NA"},{"1":"226","2":"Non_N_Classical_Monocytes","3":"0.6337061","4":"2020_dataset","5":"NA"},{"1":"227","2":"Non_N_Classical_Monocytes","3":"0.9884798","4":"2020_dataset","5":"NA"},{"1":"281","2":"Non_N_Classical_Monocytes","3":"0.5509326","4":"2020_dataset","5":"NA"},{"1":"282","2":"Non_N_Classical_Monocytes","3":"0.6628495","4":"2020_dataset","5":"NA"},{"1":"283","2":"Non_N_Classical_Monocytes","3":"1.7883709","4":"2020_dataset","5":"NA"},{"1":"284","2":"Non_N_Classical_Monocytes","3":"1.1092570","4":"2020_dataset","5":"NA"},{"1":"285","2":"Non_N_Classical_Monocytes","3":"0.6366015","4":"2020_dataset","5":"NA"},{"1":"342","2":"Non_N_Classical_Monocytes","3":"1.1008569","4":"2020_dataset","5":"NA"},{"1":"343","2":"Non_N_Classical_Monocytes","3":"1.7971761","4":"2020_dataset","5":"NA"},{"1":"344","2":"Non_N_Classical_Monocytes","3":"1.8134807","4":"2020_dataset","5":"NA"},{"1":"345","2":"Non_N_Classical_Monocytes","3":"1.0351924","4":"2020_dataset","5":"NA"},{"1":"346","2":"Non_N_Classical_Monocytes","3":"1.1056551","4":"2020_dataset","5":"NA"},{"1":"349","2":"Non_N_Classical_Monocytes","3":"0.5898065","4":"2020_dataset","5":"NA"},{"1":"350","2":"Non_N_Classical_Monocytes","3":"0.3581645","4":"2020_dataset","5":"NA"},{"1":"351","2":"Non_N_Classical_Monocytes","3":"0.5063506","4":"2020_dataset","5":"NA"},{"1":"352","2":"Non_N_Classical_Monocytes","3":"0.6179106","4":"2020_dataset","5":"NA"},{"1":"353","2":"Non_N_Classical_Monocytes","3":"0.3076885","4":"2020_dataset","5":"NA"},{"1":"355","2":"Non_N_Classical_Monocytes","3":"1.8618655","4":"2020_dataset","5":"NA"},{"1":"356","2":"Non_N_Classical_Monocytes","3":"2.3353791","4":"2020_dataset","5":"NA"},{"1":"357","2":"Non_N_Classical_Monocytes","3":"1.1625781","4":"2020_dataset","5":"NA"},{"1":"358","2":"Non_N_Classical_Monocytes","3":"1.2486784","4":"2020_dataset","5":"NA"},{"1":"359","2":"Non_N_Classical_Monocytes","3":"1.3860664","4":"2020_dataset","5":"NA"},{"1":"360","2":"Non_N_Classical_Monocytes","3":"0.2840816","4":"2020_dataset","5":"NA"},{"1":"361","2":"Non_N_Classical_Monocytes","3":"0.1975181","4":"2020_dataset","5":"NA"},{"1":"362","2":"Non_N_Classical_Monocytes","3":"0.1432798","4":"2020_dataset","5":"NA"},{"1":"363","2":"Non_N_Classical_Monocytes","3":"0.1665483","4":"2020_dataset","5":"NA"},{"1":"364","2":"Non_N_Classical_Monocytes","3":"0.1396853","4":"2020_dataset","5":"NA"},{"1":"369","2":"Non_N_Classical_Monocytes","3":"2.5759281","4":"2020_dataset","5":"NA"},{"1":"370","2":"Non_N_Classical_Monocytes","3":"2.7915030","4":"2020_dataset","5":"NA"},{"1":"371","2":"Non_N_Classical_Monocytes","3":"1.7691656","4":"2020_dataset","5":"NA"},{"1":"372","2":"Non_N_Classical_Monocytes","3":"1.3906429","4":"2020_dataset","5":"NA"},{"1":"373","2":"Non_N_Classical_Monocytes","3":"1.4807622","4":"2020_dataset","5":"NA"},{"1":"376","2":"Non_N_Classical_Monocytes","3":"0.4366070","4":"2020_dataset","5":"NA"},{"1":"377","2":"Non_N_Classical_Monocytes","3":"0.0000000","4":"2020_dataset","5":"NA"},{"1":"378","2":"Non_N_Classical_Monocytes","3":"0.0000000","4":"2020_dataset","5":"NA"},{"1":"379","2":"Non_N_Classical_Monocytes","3":"0.0000000","4":"2020_dataset","5":"NA"},{"1":"380","2":"Non_N_Classical_Monocytes","3":"0.2457957","4":"2020_dataset","5":"NA"},{"1":"397","2":"Non_N_Classical_Monocytes","3":"3.0664982","4":"2020_dataset","5":"NA"},{"1":"398","2":"Non_N_Classical_Monocytes","3":"1.8395523","4":"2020_dataset","5":"NA"},{"1":"399","2":"Non_N_Classical_Monocytes","3":"1.1156506","4":"2020_dataset","5":"NA"},{"1":"400","2":"Non_N_Classical_Monocytes","3":"1.4550231","4":"2020_dataset","5":"NA"},{"1":"401","2":"Non_N_Classical_Monocytes","3":"1.3831045","4":"2020_dataset","5":"NA"},{"1":"419","2":"Non_N_Classical_Monocytes","3":"0.8040812","4":"2020_dataset","5":"NA"},{"1":"420","2":"Non_N_Classical_Monocytes","3":"1.4411041","4":"2020_dataset","5":"NA"},{"1":"421","2":"Non_N_Classical_Monocytes","3":"0.9212763","4":"2020_dataset","5":"NA"},{"1":"422","2":"Non_N_Classical_Monocytes","3":"0.8026872","4":"2020_dataset","5":"NA"},{"1":"423","2":"Non_N_Classical_Monocytes","3":"0.5390037","4":"2020_dataset","5":"NA"},{"1":"27","2":"Non_N_Classical_Monocytes","3":"0.3624285","4":"2020_dataset","5":"NA"},{"1":"28","2":"Non_N_Classical_Monocytes","3":"1.1244326","4":"2020_dataset","5":"NA"},{"1":"29","2":"Non_N_Classical_Monocytes","3":"0.0000000","4":"2020_dataset","5":"NA"},{"1":"30","2":"Non_N_Classical_Monocytes","3":"0.3715464","4":"2020_dataset","5":"NA"},{"1":"31","2":"Non_N_Classical_Monocytes","3":"0.8090510","4":"2020_dataset","5":"NA"},{"1":"45","2":"Non_N_Classical_Monocytes","3":"2.3900098","4":"2020_dataset","5":"NA"},{"1":"46","2":"Non_N_Classical_Monocytes","3":"1.1560703","4":"2020_dataset","5":"NA"},{"1":"47","2":"Non_N_Classical_Monocytes","3":"1.5308889","4":"2020_dataset","5":"NA"},{"1":"48","2":"Non_N_Classical_Monocytes","3":"0.4880895","4":"2020_dataset","5":"NA"},{"1":"49","2":"Non_N_Classical_Monocytes","3":"0.9056320","4":"2020_dataset","5":"NA"},{"1":"87","2":"Non_N_Classical_Monocytes","3":"0.5228240","4":"2020_dataset","5":"NA"},{"1":"88","2":"Non_N_Classical_Monocytes","3":"0.0000000","4":"2020_dataset","5":"NA"},{"1":"89","2":"Non_N_Classical_Monocytes","3":"0.0000000","4":"2020_dataset","5":"NA"},{"1":"90","2":"Non_N_Classical_Monocytes","3":"0.3361388","4":"2020_dataset","5":"NA"},{"1":"91","2":"Non_N_Classical_Monocytes","3":"0.4938365","4":"2020_dataset","5":"NA"},{"1":"114","2":"Non_N_Classical_Monocytes","3":"1.6133068","4":"2020_dataset","5":"NA"},{"1":"115","2":"Non_N_Classical_Monocytes","3":"2.5669301","4":"2020_dataset","5":"NA"},{"1":"116","2":"Non_N_Classical_Monocytes","3":"2.2762978","4":"2020_dataset","5":"NA"},{"1":"117","2":"Non_N_Classical_Monocytes","3":"2.5908431","4":"2020_dataset","5":"NA"},{"1":"118","2":"Non_N_Classical_Monocytes","3":"2.6497073","4":"2020_dataset","5":"NA"},{"1":"131","2":"Non_N_Classical_Monocytes","3":"0.7296364","4":"2020_dataset","5":"NA"},{"1":"132","2":"Non_N_Classical_Monocytes","3":"1.6369036","4":"2020_dataset","5":"NA"},{"1":"133","2":"Non_N_Classical_Monocytes","3":"1.4393730","4":"2020_dataset","5":"NA"},{"1":"134","2":"Non_N_Classical_Monocytes","3":"0.6341864","4":"2020_dataset","5":"NA"},{"1":"135","2":"Non_N_Classical_Monocytes","3":"1.7959778","4":"2020_dataset","5":"NA"},{"1":"153","2":"Non_N_Classical_Monocytes","3":"1.1604044","4":"2020_dataset","5":"NA"},{"1":"154","2":"Non_N_Classical_Monocytes","3":"2.0923029","4":"2020_dataset","5":"NA"},{"1":"155","2":"Non_N_Classical_Monocytes","3":"1.7520952","4":"2020_dataset","5":"NA"},{"1":"156","2":"Non_N_Classical_Monocytes","3":"1.7880049","4":"2020_dataset","5":"NA"},{"1":"157","2":"Non_N_Classical_Monocytes","3":"1.8406396","4":"2020_dataset","5":"NA"},{"1":"160","2":"Non_N_Classical_Monocytes","3":"1.7596720","4":"2020_dataset","5":"NA"},{"1":"161","2":"Non_N_Classical_Monocytes","3":"3.4497391","4":"2020_dataset","5":"NA"},{"1":"162","2":"Non_N_Classical_Monocytes","3":"5.1788956","4":"2020_dataset","5":"NA"},{"1":"163","2":"Non_N_Classical_Monocytes","3":"3.7010864","4":"2020_dataset","5":"NA"},{"1":"164","2":"Non_N_Classical_Monocytes","3":"2.6227634","4":"2020_dataset","5":"NA"},{"1":"201","2":"Non_N_Classical_Monocytes","3":"0.8766447","4":"2020_dataset","5":"NA"},{"1":"202","2":"Non_N_Classical_Monocytes","3":"1.1512878","4":"2020_dataset","5":"NA"},{"1":"203","2":"Non_N_Classical_Monocytes","3":"1.5537527","4":"2020_dataset","5":"NA"},{"1":"204","2":"Non_N_Classical_Monocytes","3":"1.2683911","4":"2020_dataset","5":"NA"},{"1":"205","2":"Non_N_Classical_Monocytes","3":"1.4380761","4":"2020_dataset","5":"NA"},{"1":"241","2":"Non_N_Classical_Monocytes","3":"0.6380212","4":"2020_dataset","5":"NA"},{"1":"242","2":"Non_N_Classical_Monocytes","3":"0.4584027","4":"2020_dataset","5":"NA"},{"1":"243","2":"Non_N_Classical_Monocytes","3":"0.6591111","4":"2020_dataset","5":"NA"},{"1":"244","2":"Non_N_Classical_Monocytes","3":"1.4569725","4":"2020_dataset","5":"NA"},{"1":"245","2":"Non_N_Classical_Monocytes","3":"1.1584413","4":"2020_dataset","5":"NA"},{"1":"255","2":"Non_N_Classical_Monocytes","3":"2.0082137","4":"2020_dataset","5":"NA"},{"1":"256","2":"Non_N_Classical_Monocytes","3":"2.5387938","4":"2020_dataset","5":"NA"},{"1":"257","2":"Non_N_Classical_Monocytes","3":"3.5289149","4":"2020_dataset","5":"NA"},{"1":"258","2":"Non_N_Classical_Monocytes","3":"3.0525747","4":"2020_dataset","5":"NA"},{"1":"259","2":"Non_N_Classical_Monocytes","3":"2.5670383","4":"2020_dataset","5":"NA"},{"1":"483","2":"Non_N_Classical_Monocytes","3":"0.5000000","4":"2021_dataset","5":"NA"},{"1":"484","2":"Non_N_Classical_Monocytes","3":"0.4700000","4":"2021_dataset","5":"NA"},{"1":"485","2":"Non_N_Classical_Monocytes","3":"0.5000000","4":"2021_dataset","5":"NA"},{"1":"486","2":"Non_N_Classical_Monocytes","3":"0.5600000","4":"2021_dataset","5":"NA"},{"1":"487","2":"Non_N_Classical_Monocytes","3":"0.5200000","4":"2021_dataset","5":"NA"},{"1":"490","2":"Non_N_Classical_Monocytes","3":"1.3600000","4":"2021_dataset","5":"NA"},{"1":"491","2":"Non_N_Classical_Monocytes","3":"2.6800000","4":"2021_dataset","5":"NA"},{"1":"492","2":"Non_N_Classical_Monocytes","3":"1.8500000","4":"2021_dataset","5":"NA"},{"1":"493","2":"Non_N_Classical_Monocytes","3":"1.6400000","4":"2021_dataset","5":"NA"},{"1":"494","2":"Non_N_Classical_Monocytes","3":"0.9800000","4":"2021_dataset","5":"NA"},{"1":"498","2":"Non_N_Classical_Monocytes","3":"0.4100000","4":"2021_dataset","5":"NA"},{"1":"499","2":"Non_N_Classical_Monocytes","3":"0.6300000","4":"2021_dataset","5":"NA"},{"1":"500","2":"Non_N_Classical_Monocytes","3":"0.7700000","4":"2021_dataset","5":"NA"},{"1":"501","2":"Non_N_Classical_Monocytes","3":"0.5900000","4":"2021_dataset","5":"NA"},{"1":"502","2":"Non_N_Classical_Monocytes","3":"0.5000000","4":"2021_dataset","5":"NA"},{"1":"506","2":"Non_N_Classical_Monocytes","3":"2.2400000","4":"2021_dataset","5":"NA"},{"1":"507","2":"Non_N_Classical_Monocytes","3":"1.8000000","4":"2021_dataset","5":"NA"},{"1":"508","2":"Non_N_Classical_Monocytes","3":"2.5100000","4":"2021_dataset","5":"NA"},{"1":"509","2":"Non_N_Classical_Monocytes","3":"1.5000000","4":"2021_dataset","5":"NA"},{"1":"510","2":"Non_N_Classical_Monocytes","3":"2.5200000","4":"2021_dataset","5":"NA"},{"1":"513","2":"Non_N_Classical_Monocytes","3":"2.2500000","4":"2021_dataset","5":"NA"},{"1":"514","2":"Non_N_Classical_Monocytes","3":"1.5400000","4":"2021_dataset","5":"NA"},{"1":"515","2":"Non_N_Classical_Monocytes","3":"1.6100000","4":"2021_dataset","5":"NA"},{"1":"516","2":"Non_N_Classical_Monocytes","3":"2.4100000","4":"2021_dataset","5":"NA"},{"1":"517","2":"Non_N_Classical_Monocytes","3":"1.9300000","4":"2021_dataset","5":"NA"},{"1":"521","2":"Non_N_Classical_Monocytes","3":"2.4000000","4":"2021_dataset","5":"NA"},{"1":"522","2":"Non_N_Classical_Monocytes","3":"1.9800000","4":"2021_dataset","5":"NA"},{"1":"523","2":"Non_N_Classical_Monocytes","3":"2.8600000","4":"2021_dataset","5":"NA"},{"1":"524","2":"Non_N_Classical_Monocytes","3":"2.0600000","4":"2021_dataset","5":"NA"},{"1":"525","2":"Non_N_Classical_Monocytes","3":"1.6300000","4":"2021_dataset","5":"NA"},{"1":"529","2":"Non_N_Classical_Monocytes","3":"0.2900000","4":"2021_dataset","5":"NA"},{"1":"530","2":"Non_N_Classical_Monocytes","3":"0.5700000","4":"2021_dataset","5":"NA"},{"1":"531","2":"Non_N_Classical_Monocytes","3":"0.4700000","4":"2021_dataset","5":"NA"},{"1":"532","2":"Non_N_Classical_Monocytes","3":"0.5100000","4":"2021_dataset","5":"NA"},{"1":"533","2":"Non_N_Classical_Monocytes","3":"0.0640000","4":"2021_dataset","5":"NA"},{"1":"537","2":"Non_N_Classical_Monocytes","3":"0.9400000","4":"2021_dataset","5":"NA"},{"1":"538","2":"Non_N_Classical_Monocytes","3":"0.8200000","4":"2021_dataset","5":"NA"},{"1":"539","2":"Non_N_Classical_Monocytes","3":"1.0500000","4":"2021_dataset","5":"NA"},{"1":"540","2":"Non_N_Classical_Monocytes","3":"0.5200000","4":"2021_dataset","5":"NA"},{"1":"541","2":"Non_N_Classical_Monocytes","3":"0.9500000","4":"2021_dataset","5":"NA"},{"1":"546","2":"Non_N_Classical_Monocytes","3":"1.2800000","4":"2021_dataset","5":"NA"},{"1":"547","2":"Non_N_Classical_Monocytes","3":"1.3400000","4":"2021_dataset","5":"NA"},{"1":"548","2":"Non_N_Classical_Monocytes","3":"1.2100000","4":"2021_dataset","5":"NA"},{"1":"549","2":"Non_N_Classical_Monocytes","3":"1.1100000","4":"2021_dataset","5":"NA"},{"1":"550","2":"Non_N_Classical_Monocytes","3":"1.5100000","4":"2021_dataset","5":"NA"},{"1":"554","2":"Non_N_Classical_Monocytes","3":"1.2000000","4":"2021_dataset","5":"NA"},{"1":"555","2":"Non_N_Classical_Monocytes","3":"1.3200000","4":"2021_dataset","5":"NA"},{"1":"556","2":"Non_N_Classical_Monocytes","3":"0.7500000","4":"2021_dataset","5":"NA"},{"1":"557","2":"Non_N_Classical_Monocytes","3":"0.8800000","4":"2021_dataset","5":"NA"},{"1":"558","2":"Non_N_Classical_Monocytes","3":"0.9300000","4":"2021_dataset","5":"NA"},{"1":"562","2":"Non_N_Classical_Monocytes","3":"1.1800000","4":"2021_dataset","5":"NA"},{"1":"563","2":"Non_N_Classical_Monocytes","3":"0.8200000","4":"2021_dataset","5":"NA"},{"1":"564","2":"Non_N_Classical_Monocytes","3":"0.7700000","4":"2021_dataset","5":"NA"},{"1":"565","2":"Non_N_Classical_Monocytes","3":"0.6200000","4":"2021_dataset","5":"NA"},{"1":"566","2":"Non_N_Classical_Monocytes","3":"0.4900000","4":"2021_dataset","5":"NA"},{"1":"569","2":"Non_N_Classical_Monocytes","3":"0.9100000","4":"2021_dataset","5":"NA"},{"1":"570","2":"Non_N_Classical_Monocytes","3":"1.2600000","4":"2021_dataset","5":"NA"},{"1":"571","2":"Non_N_Classical_Monocytes","3":"0.8000000","4":"2021_dataset","5":"NA"},{"1":"572","2":"Non_N_Classical_Monocytes","3":"1.3100000","4":"2021_dataset","5":"NA"},{"1":"573","2":"Non_N_Classical_Monocytes","3":"0.2900000","4":"2021_dataset","5":"NA"},{"1":"585","2":"Non_N_Classical_Monocytes","3":"1.4400000","4":"2021_dataset","5":"NA"},{"1":"586","2":"Non_N_Classical_Monocytes","3":"0.9500000","4":"2021_dataset","5":"NA"},{"1":"587","2":"Non_N_Classical_Monocytes","3":"1.6600000","4":"2021_dataset","5":"NA"},{"1":"588","2":"Non_N_Classical_Monocytes","3":"2.7900000","4":"2021_dataset","5":"NA"},{"1":"589","2":"Non_N_Classical_Monocytes","3":"2.0500000","4":"2021_dataset","5":"NA"},{"1":"593","2":"Non_N_Classical_Monocytes","3":"2.6400000","4":"2021_dataset","5":"NA"},{"1":"594","2":"Non_N_Classical_Monocytes","3":"1.7400000","4":"2021_dataset","5":"NA"},{"1":"595","2":"Non_N_Classical_Monocytes","3":"1.5000000","4":"2021_dataset","5":"NA"},{"1":"596","2":"Non_N_Classical_Monocytes","3":"1.1400000","4":"2021_dataset","5":"NA"},{"1":"597","2":"Non_N_Classical_Monocytes","3":"2.2300000","4":"2021_dataset","5":"NA"},{"1":"601","2":"Non_N_Classical_Monocytes","3":"2.0600000","4":"2021_dataset","5":"NA"},{"1":"602","2":"Non_N_Classical_Monocytes","3":"2.0100000","4":"2021_dataset","5":"NA"},{"1":"603","2":"Non_N_Classical_Monocytes","3":"2.6200000","4":"2021_dataset","5":"NA"},{"1":"604","2":"Non_N_Classical_Monocytes","3":"1.9900000","4":"2021_dataset","5":"NA"},{"1":"605","2":"Non_N_Classical_Monocytes","3":"2.2900000","4":"2021_dataset","5":"NA"},{"1":"608","2":"Non_N_Classical_Monocytes","3":"6.0900000","4":"2021_dataset","5":"NA"},{"1":"609","2":"Non_N_Classical_Monocytes","3":"4.7300000","4":"2021_dataset","5":"NA"},{"1":"610","2":"Non_N_Classical_Monocytes","3":"5.5900000","4":"2021_dataset","5":"NA"},{"1":"611","2":"Non_N_Classical_Monocytes","3":"6.6000000","4":"2021_dataset","5":"NA"},{"1":"612","2":"Non_N_Classical_Monocytes","3":"2.5900000","4":"2021_dataset","5":"NA"},{"1":"616","2":"Non_N_Classical_Monocytes","3":"2.9700000","4":"2021_dataset","5":"NA"},{"1":"617","2":"Non_N_Classical_Monocytes","3":"2.9200000","4":"2021_dataset","5":"NA"},{"1":"618","2":"Non_N_Classical_Monocytes","3":"2.2100000","4":"2021_dataset","5":"NA"},{"1":"619","2":"Non_N_Classical_Monocytes","3":"2.0900000","4":"2021_dataset","5":"NA"},{"1":"620","2":"Non_N_Classical_Monocytes","3":"1.7900000","4":"2021_dataset","5":"NA"},{"1":"623","2":"Non_N_Classical_Monocytes","3":"1.8800000","4":"2021_dataset","5":"NA"},{"1":"624","2":"Non_N_Classical_Monocytes","3":"1.2500000","4":"2021_dataset","5":"NA"},{"1":"625","2":"Non_N_Classical_Monocytes","3":"0.7600000","4":"2021_dataset","5":"NA"},{"1":"626","2":"Non_N_Classical_Monocytes","3":"0.8700000","4":"2021_dataset","5":"NA"},{"1":"627","2":"Non_N_Classical_Monocytes","3":"2.0100000","4":"2021_dataset","5":"NA"},{"1":"630","2":"Non_N_Classical_Monocytes","3":"1.7200000","4":"2021_dataset","5":"NA"},{"1":"631","2":"Non_N_Classical_Monocytes","3":"1.7100000","4":"2021_dataset","5":"NA"},{"1":"632","2":"Non_N_Classical_Monocytes","3":"1.3600000","4":"2021_dataset","5":"NA"},{"1":"633","2":"Non_N_Classical_Monocytes","3":"1.0300000","4":"2021_dataset","5":"NA"},{"1":"634","2":"Non_N_Classical_Monocytes","3":"1.1500000","4":"2021_dataset","5":"NA"},{"1":"636","2":"Non_N_Classical_Monocytes","3":"1.7100000","4":"2021_dataset","5":"NA"},{"1":"637","2":"Non_N_Classical_Monocytes","3":"1.4100000","4":"2021_dataset","5":"NA"},{"1":"638","2":"Non_N_Classical_Monocytes","3":"1.5400000","4":"2021_dataset","5":"NA"},{"1":"639","2":"Non_N_Classical_Monocytes","3":"1.1500000","4":"2021_dataset","5":"NA"},{"1":"640","2":"Non_N_Classical_Monocytes","3":"0.9200000","4":"2021_dataset","5":"NA"},{"1":"643","2":"Non_N_Classical_Monocytes","3":"1.3400000","4":"2021_dataset","5":"NA"},{"1":"644","2":"Non_N_Classical_Monocytes","3":"0.8100000","4":"2021_dataset","5":"NA"},{"1":"645","2":"Non_N_Classical_Monocytes","3":"1.6700000","4":"2021_dataset","5":"NA"},{"1":"646","2":"Non_N_Classical_Monocytes","3":"1.8000000","4":"2021_dataset","5":"NA"},{"1":"647","2":"Non_N_Classical_Monocytes","3":"1.7900000","4":"2021_dataset","5":"NA"},{"1":"650","2":"Non_N_Classical_Monocytes","3":"0.8200000","4":"2021_dataset","5":"NA"},{"1":"651","2":"Non_N_Classical_Monocytes","3":"0.6800000","4":"2021_dataset","5":"NA"},{"1":"652","2":"Non_N_Classical_Monocytes","3":"0.8400000","4":"2021_dataset","5":"NA"},{"1":"653","2":"Non_N_Classical_Monocytes","3":"0.7000000","4":"2021_dataset","5":"NA"},{"1":"654","2":"Non_N_Classical_Monocytes","3":"0.6600000","4":"2021_dataset","5":"NA"},{"1":"657","2":"Non_N_Classical_Monocytes","3":"2.7500000","4":"2021_dataset","5":"NA"},{"1":"658","2":"Non_N_Classical_Monocytes","3":"1.6300000","4":"2021_dataset","5":"NA"},{"1":"659","2":"Non_N_Classical_Monocytes","3":"1.2200000","4":"2021_dataset","5":"NA"},{"1":"660","2":"Non_N_Classical_Monocytes","3":"0.9900000","4":"2021_dataset","5":"NA"},{"1":"661","2":"Non_N_Classical_Monocytes","3":"1.1100000","4":"2021_dataset","5":"NA"},{"1":"664","2":"Non_N_Classical_Monocytes","3":"1.7300000","4":"2021_dataset","5":"NA"},{"1":"665","2":"Non_N_Classical_Monocytes","3":"1.5900000","4":"2021_dataset","5":"NA"},{"1":"666","2":"Non_N_Classical_Monocytes","3":"2.2200000","4":"2021_dataset","5":"NA"},{"1":"667","2":"Non_N_Classical_Monocytes","3":"1.2500000","4":"2021_dataset","5":"NA"},{"1":"668","2":"Non_N_Classical_Monocytes","3":"0.9000000","4":"2021_dataset","5":"NA"},{"1":"669","2":"Non_N_Classical_Monocytes","3":"1.3200000","4":"2021_dataset","5":"NA"},{"1":"670","2":"Non_N_Classical_Monocytes","3":"1.0300000","4":"2021_dataset","5":"NA"},{"1":"671","2":"Non_N_Classical_Monocytes","3":"1.4700000","4":"2021_dataset","5":"NA"},{"1":"672","2":"Non_N_Classical_Monocytes","3":"0.9300000","4":"2021_dataset","5":"NA"},{"1":"673","2":"Non_N_Classical_Monocytes","3":"0.9300000","4":"2021_dataset","5":"NA"},{"1":"674","2":"Non_N_Classical_Monocytes","3":"1.8800000","4":"2021_dataset","5":"NA"},{"1":"675","2":"Non_N_Classical_Monocytes","3":"1.9200000","4":"2021_dataset","5":"NA"},{"1":"676","2":"Non_N_Classical_Monocytes","3":"1.5500000","4":"2021_dataset","5":"NA"},{"1":"677","2":"Non_N_Classical_Monocytes","3":"1.3200000","4":"2021_dataset","5":"NA"},{"1":"678","2":"Non_N_Classical_Monocytes","3":"2.3800000","4":"2021_dataset","5":"NA"},{"1":"681","2":"Non_N_Classical_Monocytes","3":"0.7600000","4":"2021_dataset","5":"NA"},{"1":"682","2":"Non_N_Classical_Monocytes","3":"0.8100000","4":"2021_dataset","5":"NA"},{"1":"683","2":"Non_N_Classical_Monocytes","3":"0.4300000","4":"2021_dataset","5":"NA"},{"1":"684","2":"Non_N_Classical_Monocytes","3":"0.6200000","4":"2021_dataset","5":"NA"},{"1":"685","2":"Non_N_Classical_Monocytes","3":"1.1700000","4":"2021_dataset","5":"NA"},{"1":"688","2":"Non_N_Classical_Monocytes","3":"0.7800000","4":"2021_dataset","5":"NA"},{"1":"689","2":"Non_N_Classical_Monocytes","3":"0.9400000","4":"2021_dataset","5":"NA"},{"1":"690","2":"Non_N_Classical_Monocytes","3":"1.5500000","4":"2021_dataset","5":"NA"},{"1":"691","2":"Non_N_Classical_Monocytes","3":"1.2700000","4":"2021_dataset","5":"NA"},{"1":"692","2":"Non_N_Classical_Monocytes","3":"0.8800000","4":"2021_dataset","5":"NA"},{"1":"695","2":"Non_N_Classical_Monocytes","3":"0.4100000","4":"2021_dataset","5":"NA"},{"1":"696","2":"Non_N_Classical_Monocytes","3":"0.8800000","4":"2021_dataset","5":"NA"},{"1":"697","2":"Non_N_Classical_Monocytes","3":"1.1300000","4":"2021_dataset","5":"NA"},{"1":"698","2":"Non_N_Classical_Monocytes","3":"0.5000000","4":"2021_dataset","5":"NA"},{"1":"699","2":"Non_N_Classical_Monocytes","3":"0.8400000","4":"2021_dataset","5":"NA"},{"1":"702","2":"Non_N_Classical_Monocytes","3":"1.3200000","4":"2021_dataset","5":"NA"},{"1":"703","2":"Non_N_Classical_Monocytes","3":"0.9600000","4":"2021_dataset","5":"NA"},{"1":"704","2":"Non_N_Classical_Monocytes","3":"1.2500000","4":"2021_dataset","5":"NA"},{"1":"705","2":"Non_N_Classical_Monocytes","3":"2.4100000","4":"2021_dataset","5":"NA"},{"1":"706","2":"Non_N_Classical_Monocytes","3":"2.2100000","4":"2021_dataset","5":"NA"},{"1":"709","2":"Non_N_Classical_Monocytes","3":"1.6100000","4":"2021_dataset","5":"NA"},{"1":"710","2":"Non_N_Classical_Monocytes","3":"1.9400000","4":"2021_dataset","5":"NA"},{"1":"711","2":"Non_N_Classical_Monocytes","3":"2.0300000","4":"2021_dataset","5":"NA"},{"1":"712","2":"Non_N_Classical_Monocytes","3":"2.1600000","4":"2021_dataset","5":"NA"},{"1":"713","2":"Non_N_Classical_Monocytes","3":"1.7200000","4":"2021_dataset","5":"NA"},{"1":"716","2":"Non_N_Classical_Monocytes","3":"1.3600000","4":"2021_dataset","5":"NA"},{"1":"717","2":"Non_N_Classical_Monocytes","3":"1.0300000","4":"2021_dataset","5":"NA"},{"1":"718","2":"Non_N_Classical_Monocytes","3":"1.6500000","4":"2021_dataset","5":"NA"},{"1":"719","2":"Non_N_Classical_Monocytes","3":"1.4100000","4":"2021_dataset","5":"NA"},{"1":"720","2":"Non_N_Classical_Monocytes","3":"2.2900000","4":"2021_dataset","5":"NA"},{"1":"723","2":"Non_N_Classical_Monocytes","3":"3.6800000","4":"2021_dataset","5":"NA"},{"1":"724","2":"Non_N_Classical_Monocytes","3":"2.1700000","4":"2021_dataset","5":"NA"},{"1":"725","2":"Non_N_Classical_Monocytes","3":"2.2800000","4":"2021_dataset","5":"NA"},{"1":"726","2":"Non_N_Classical_Monocytes","3":"1.7500000","4":"2021_dataset","5":"NA"},{"1":"727","2":"Non_N_Classical_Monocytes","3":"2.3600000","4":"2021_dataset","5":"NA"},{"1":"924","2":"Non_N_Classical_Monocytes","3":"2.0600000","4":"2022_dataset","5":"NA"},{"1":"872","2":"Non_N_Classical_Monocytes","3":"1.3500000","4":"2022_dataset","5":"NA"},{"1":"740","2":"Non_N_Classical_Monocytes","3":"1.9800000","4":"2022_dataset","5":"NA"},{"1":"892","2":"Non_N_Classical_Monocytes","3":"2.2800000","4":"2022_dataset","5":"NA"},{"1":"792","2":"Non_N_Classical_Monocytes","3":"2.8400000","4":"2022_dataset","5":"NA"},{"1":"756","2":"Non_N_Classical_Monocytes","3":"1.0900000","4":"2022_dataset","5":"NA"},{"1":"933","2":"Non_N_Classical_Monocytes","3":"1.1000000","4":"2022_dataset","5":"NA"},{"1":"925","2":"Non_N_Classical_Monocytes","3":"1.4200000","4":"2022_dataset","5":"NA"},{"1":"771","2":"Non_N_Classical_Monocytes","3":"0.9300000","4":"2022_dataset","5":"NA"},{"1":"869","2":"Non_N_Classical_Monocytes","3":"1.0900000","4":"2022_dataset","5":"NA"},{"1":"730","2":"Non_N_Classical_Monocytes","3":"1.5100000","4":"2022_dataset","5":"NA"},{"1":"922","2":"Non_N_Classical_Monocytes","3":"1.9400000","4":"2022_dataset","5":"NA"},{"1":"864","2":"Non_N_Classical_Monocytes","3":"1.0700000","4":"2022_dataset","5":"NA"},{"1":"760","2":"Non_N_Classical_Monocytes","3":"3.3700000","4":"2022_dataset","5":"NA"},{"1":"852","2":"Non_N_Classical_Monocytes","3":"1.2500000","4":"2022_dataset","5":"NA"},{"1":"923","2":"Non_N_Classical_Monocytes","3":"1.9400000","4":"2022_dataset","5":"NA"},{"1":"890","2":"Non_N_Classical_Monocytes","3":"1.9100000","4":"2022_dataset","5":"NA"},{"1":"731","2":"Non_N_Classical_Monocytes","3":"3.2200000","4":"2022_dataset","5":"NA"},{"1":"935","2":"Non_N_Classical_Monocytes","3":"0.4300000","4":"2022_dataset","5":"NA"},{"1":"901","2":"Non_N_Classical_Monocytes","3":"0.9100000","4":"2022_dataset","5":"NA"},{"1":"750","2":"Non_N_Classical_Monocytes","3":"1.1300000","4":"2022_dataset","5":"NA"},{"1":"810","2":"Non_N_Classical_Monocytes","3":"0.7900000","4":"2022_dataset","5":"NA"},{"1":"885","2":"Non_N_Classical_Monocytes","3":"0.9000000","4":"2022_dataset","5":"NA"},{"1":"742","2":"Non_N_Classical_Monocytes","3":"1.7800000","4":"2022_dataset","5":"NA"},{"1":"912","2":"Non_N_Classical_Monocytes","3":"1.7000000","4":"2022_dataset","5":"NA"},{"1":"931","2":"Non_N_Classical_Monocytes","3":"0.8400000","4":"2022_dataset","5":"NA"},{"1":"811","2":"Non_N_Classical_Monocytes","3":"1.1500000","4":"2022_dataset","5":"NA"},{"1":"795","2":"Non_N_Classical_Monocytes","3":"1.8700000","4":"2022_dataset","5":"NA"},{"1":"809","2":"Non_N_Classical_Monocytes","3":"1.1700000","4":"2022_dataset","5":"NA"},{"1":"790","2":"Non_N_Classical_Monocytes","3":"2.6100000","4":"2022_dataset","5":"NA"},{"1":"934","2":"Non_N_Classical_Monocytes","3":"0.9700000","4":"2022_dataset","5":"NA"},{"1":"814","2":"Non_N_Classical_Monocytes","3":"0.8400000","4":"2022_dataset","5":"NA"},{"1":"850","2":"Non_N_Classical_Monocytes","3":"1.0600000","4":"2022_dataset","5":"NA"},{"1":"753","2":"Non_N_Classical_Monocytes","3":"0.7300000","4":"2022_dataset","5":"NA"},{"1":"765","2":"Non_N_Classical_Monocytes","3":"1.4500000","4":"2022_dataset","5":"NA"},{"1":"781","2":"Non_N_Classical_Monocytes","3":"1.3400000","4":"2022_dataset","5":"NA"},{"1":"773","2":"Non_N_Classical_Monocytes","3":"1.3600000","4":"2022_dataset","5":"NA"},{"1":"839","2":"Non_N_Classical_Monocytes","3":"0.6800000","4":"2022_dataset","5":"NA"},{"1":"732","2":"Non_N_Classical_Monocytes","3":"3.2800000","4":"2022_dataset","5":"NA"},{"1":"930","2":"Non_N_Classical_Monocytes","3":"0.8000000","4":"2022_dataset","5":"NA"},{"1":"861","2":"Non_N_Classical_Monocytes","3":"1.0900000","4":"2022_dataset","5":"NA"},{"1":"853","2":"Non_N_Classical_Monocytes","3":"0.8800000","4":"2022_dataset","5":"NA"},{"1":"819","2":"Non_N_Classical_Monocytes","3":"1.5400000","4":"2022_dataset","5":"NA"},{"1":"874","2":"Non_N_Classical_Monocytes","3":"0.8500000","4":"2022_dataset","5":"NA"},{"1":"900","2":"Non_N_Classical_Monocytes","3":"0.9500000","4":"2022_dataset","5":"NA"},{"1":"854","2":"Non_N_Classical_Monocytes","3":"0.7800000","4":"2022_dataset","5":"NA"},{"1":"911","2":"Non_N_Classical_Monocytes","3":"1.8900000","4":"2022_dataset","5":"NA"},{"1":"762","2":"Non_N_Classical_Monocytes","3":"1.5500000","4":"2022_dataset","5":"NA"},{"1":"734","2":"Non_N_Classical_Monocytes","3":"2.8000000","4":"2022_dataset","5":"NA"},{"1":"916","2":"Non_N_Classical_Monocytes","3":"2.0800000","4":"2022_dataset","5":"NA"},{"1":"751","2":"Non_N_Classical_Monocytes","3":"1.0800000","4":"2022_dataset","5":"NA"},{"1":"849","2":"Non_N_Classical_Monocytes","3":"1.0200000","4":"2022_dataset","5":"NA"},{"1":"863","2":"Non_N_Classical_Monocytes","3":"0.0820000","4":"2022_dataset","5":"NA"},{"1":"870","2":"Non_N_Classical_Monocytes","3":"1.0300000","4":"2022_dataset","5":"NA"},{"1":"910","2":"Non_N_Classical_Monocytes","3":"1.0200000","4":"2022_dataset","5":"NA"},{"1":"770","2":"Non_N_Classical_Monocytes","3":"0.4600000","4":"2022_dataset","5":"NA"},{"1":"754","2":"Non_N_Classical_Monocytes","3":"1.9700000","4":"2022_dataset","5":"NA"},{"1":"737","2":"Non_N_Classical_Monocytes","3":"3.3000000","4":"2022_dataset","5":"NA"},{"1":"782","2":"Non_N_Classical_Monocytes","3":"1.1900000","4":"2022_dataset","5":"NA"},{"1":"932","2":"Non_N_Classical_Monocytes","3":"1.4000000","4":"2022_dataset","5":"NA"},{"1":"784","2":"Non_N_Classical_Monocytes","3":"1.6200000","4":"2022_dataset","5":"NA"},{"1":"764","2":"Non_N_Classical_Monocytes","3":"1.2200000","4":"2022_dataset","5":"NA"},{"1":"780","2":"Non_N_Classical_Monocytes","3":"1.0300000","4":"2022_dataset","5":"NA"},{"1":"830","2":"Non_N_Classical_Monocytes","3":"1.2100000","4":"2022_dataset","5":"NA"},{"1":"791","2":"Non_N_Classical_Monocytes","3":"2.2200000","4":"2022_dataset","5":"NA"},{"1":"775","2":"Non_N_Classical_Monocytes","3":"1.3600000","4":"2022_dataset","5":"NA"},{"1":"744","2":"Non_N_Classical_Monocytes","3":"1.0400000","4":"2022_dataset","5":"NA"},{"1":"733","2":"Non_N_Classical_Monocytes","3":"3.2900000","4":"2022_dataset","5":"NA"},{"1":"821","2":"Non_N_Classical_Monocytes","3":"2.3300000","4":"2022_dataset","5":"NA"},{"1":"763","2":"Non_N_Classical_Monocytes","3":"1.7400000","4":"2022_dataset","5":"NA"},{"1":"860","2":"Non_N_Classical_Monocytes","3":"1.5900000","4":"2022_dataset","5":"NA"},{"1":"903","2":"Non_N_Classical_Monocytes","3":"1.7500000","4":"2022_dataset","5":"NA"},{"1":"755","2":"Non_N_Classical_Monocytes","3":"0.6600000","4":"2022_dataset","5":"NA"},{"1":"905","2":"Non_N_Classical_Monocytes","3":"1.7300000","4":"2022_dataset","5":"NA"},{"1":"741","2":"Non_N_Classical_Monocytes","3":"1.4100000","4":"2022_dataset","5":"NA"},{"1":"736","2":"Non_N_Classical_Monocytes","3":"2.8000000","4":"2022_dataset","5":"NA"},{"1":"802","2":"Non_N_Classical_Monocytes","3":"1.6100000","4":"2022_dataset","5":"NA"},{"1":"815","2":"Non_N_Classical_Monocytes","3":"0.7600000","4":"2022_dataset","5":"NA"},{"1":"843","2":"Non_N_Classical_Monocytes","3":"0.6900000","4":"2022_dataset","5":"NA"},{"1":"794","2":"Non_N_Classical_Monocytes","3":"2.4100000","4":"2022_dataset","5":"NA"},{"1":"914","2":"Non_N_Classical_Monocytes","3":"1.0200000","4":"2022_dataset","5":"NA"},{"1":"859","2":"Non_N_Classical_Monocytes","3":"1.0600000","4":"2022_dataset","5":"NA"},{"1":"842","2":"Non_N_Classical_Monocytes","3":"1.4500000","4":"2022_dataset","5":"NA"},{"1":"812","2":"Non_N_Classical_Monocytes","3":"1.2100000","4":"2022_dataset","5":"NA"},{"1":"766","2":"Non_N_Classical_Monocytes","3":"1.4300000","4":"2022_dataset","5":"NA"},{"1":"824","2":"Non_N_Classical_Monocytes","3":"0.7300000","4":"2022_dataset","5":"NA"},{"1":"840","2":"Non_N_Classical_Monocytes","3":"1.4900000","4":"2022_dataset","5":"NA"},{"1":"832","2":"Non_N_Classical_Monocytes","3":"1.5300000","4":"2022_dataset","5":"NA"},{"1":"800","2":"Non_N_Classical_Monocytes","3":"1.6500000","4":"2022_dataset","5":"NA"},{"1":"921","2":"Non_N_Classical_Monocytes","3":"1.9300000","4":"2022_dataset","5":"NA"},{"1":"913","2":"Non_N_Classical_Monocytes","3":"0.8100000","4":"2022_dataset","5":"NA"},{"1":"774","2":"Non_N_Classical_Monocytes","3":"1.3200000","4":"2022_dataset","5":"NA"},{"1":"746","2":"Non_N_Classical_Monocytes","3":"1.8100000","4":"2022_dataset","5":"NA"},{"1":"806","2":"Non_N_Classical_Monocytes","3":"1.5400000","4":"2022_dataset","5":"NA"},{"1":"834","2":"Non_N_Classical_Monocytes","3":"0.7500000","4":"2022_dataset","5":"NA"},{"1":"796","2":"Non_N_Classical_Monocytes","3":"1.7500000","4":"2022_dataset","5":"NA"},{"1":"831","2":"Non_N_Classical_Monocytes","3":"1.1500000","4":"2022_dataset","5":"NA"},{"1":"841","2":"Non_N_Classical_Monocytes","3":"1.4400000","4":"2022_dataset","5":"NA"},{"1":"813","2":"Non_N_Classical_Monocytes","3":"1.7500000","4":"2022_dataset","5":"NA"},{"1":"786","2":"Non_N_Classical_Monocytes","3":"0.8700000","4":"2022_dataset","5":"NA"},{"1":"820","2":"Non_N_Classical_Monocytes","3":"2.0600000","4":"2022_dataset","5":"NA"},{"1":"807","2":"Non_N_Classical_Monocytes","3":"2.2800000","4":"2022_dataset","5":"NA"},{"1":"803","2":"Non_N_Classical_Monocytes","3":"1.4600000","4":"2022_dataset","5":"NA"},{"1":"894","2":"Non_N_Classical_Monocytes","3":"2.2100000","4":"2022_dataset","5":"NA"},{"1":"743","2":"Non_N_Classical_Monocytes","3":"1.2500000","4":"2022_dataset","5":"NA"},{"1":"761","2":"Non_N_Classical_Monocytes","3":"1.0700000","4":"2022_dataset","5":"NA"},{"1":"844","2":"Non_N_Classical_Monocytes","3":"1.4200000","4":"2022_dataset","5":"NA"},{"1":"895","2":"Non_N_Classical_Monocytes","3":"1.2400000","4":"2022_dataset","5":"NA"},{"1":"904","2":"Non_N_Classical_Monocytes","3":"1.5800000","4":"2022_dataset","5":"NA"},{"1":"829","2":"Non_N_Classical_Monocytes","3":"1.4700000","4":"2022_dataset","5":"NA"},{"1":"785","2":"Non_N_Classical_Monocytes","3":"1.5600000","4":"2022_dataset","5":"NA"},{"1":"804","2":"Non_N_Classical_Monocytes","3":"2.9900000","4":"2022_dataset","5":"NA"},{"1":"882","2":"Non_N_Classical_Monocytes","3":"0.8000000","4":"2022_dataset","5":"NA"},{"1":"783","2":"Non_N_Classical_Monocytes","3":"1.9700000","4":"2022_dataset","5":"NA"},{"1":"891","2":"Non_N_Classical_Monocytes","3":"2.6600000","4":"2022_dataset","5":"NA"},{"1":"823","2":"Non_N_Classical_Monocytes","3":"0.9700000","4":"2022_dataset","5":"NA"},{"1":"822","2":"Non_N_Classical_Monocytes","3":"1.3300000","4":"2022_dataset","5":"NA"},{"1":"801","2":"Non_N_Classical_Monocytes","3":"1.5200000","4":"2022_dataset","5":"NA"},{"1":"920","2":"Non_N_Classical_Monocytes","3":"1.5700000","4":"2022_dataset","5":"NA"},{"1":"833","2":"Non_N_Classical_Monocytes","3":"0.8800000","4":"2022_dataset","5":"NA"},{"1":"776","2":"Non_N_Classical_Monocytes","3":"1.0100000","4":"2022_dataset","5":"NA"},{"1":"752","2":"Non_N_Classical_Monocytes","3":"1.1300000","4":"2022_dataset","5":"NA"},{"1":"862","2":"Non_N_Classical_Monocytes","3":"0.9700000","4":"2022_dataset","5":"NA"},{"1":"873","2":"Non_N_Classical_Monocytes","3":"0.8800000","4":"2022_dataset","5":"NA"},{"1":"884","2":"Non_N_Classical_Monocytes","3":"0.8100000","4":"2022_dataset","5":"NA"},{"1":"793","2":"Non_N_Classical_Monocytes","3":"2.2300000","4":"2022_dataset","5":"NA"},{"1":"915","2":"Non_N_Classical_Monocytes","3":"0.9800000","4":"2022_dataset","5":"NA"},{"1":"851","2":"Non_N_Classical_Monocytes","3":"1.9100000","4":"2022_dataset","5":"NA"},{"1":"893","2":"Non_N_Classical_Monocytes","3":"1.1200000","4":"2022_dataset","5":"NA"},{"1":"735","2":"Non_N_Classical_Monocytes","3":"1.7500000","4":"2022_dataset","5":"NA"},{"1":"871","2":"Non_N_Classical_Monocytes","3":"1.0000000","4":"2022_dataset","5":"NA"},{"1":"881","2":"Non_N_Classical_Monocytes","3":"0.7400000","4":"2022_dataset","5":"NA"},{"1":"906","2":"Non_N_Classical_Monocytes","3":"1.8100000","4":"2022_dataset","5":"NA"},{"1":"902","2":"Non_N_Classical_Monocytes","3":"1.9600000","4":"2022_dataset","5":"NA"},{"1":"883","2":"Non_N_Classical_Monocytes","3":"0.8700000","4":"2022_dataset","5":"NA"},{"1":"772","2":"Non_N_Classical_Monocytes","3":"1.3000000","4":"2022_dataset","5":"NA"},{"1":"745","2":"Non_N_Classical_Monocytes","3":"1.4300000","4":"2022_dataset","5":"NA"},{"1":"805","2":"Non_N_Classical_Monocytes","3":"1.3700000","4":"2022_dataset","5":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(level <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<ul>
<li>Especially, when completely ignoring the hierarchy in the gating information, I would expect to have more than 100% in <code>percelt_live_cell_sum</code>, but this is not true for the specimens from the 2023 dataset</li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<ul>
<li>Or what I use a list of predefined cell types of interest?</li>
</ul>
</section>
</section>
<section id="features-for-t-cell-activation-assay-aim" class="level1" data-number="10">
<h1 data-number="10"><span class="header-section-number">10</span> Features for T cell Activation Assay (AIM)</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_activation <span class="sc">%&gt;%</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(stimulation) <span class="sc">%&gt;%</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["stimulation"],"name":[1],"type":["chr"],"align":["left"]}],"data":[{"1":"DMSO"},{"1":"PT"},{"1":"TT"},{"1":"PHA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_activation <span class="sc">%&gt;%</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>analyte_percentages), <span class="at">bins=</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>stimulation) <span class="sc">+</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="features-for-t-cell-polarization-assay-fluorospot" class="level1" data-number="11">
<h1 data-number="11"><span class="header-section-number">11</span> Features for T cell Polarization Assay (FluoroSpot)</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(protein_meta, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"protein_id"</span><span class="ot">=</span><span class="st">"uniprot_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(protein_id, cytokine) <span class="sc">%&gt;%</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["protein_id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["cytokine"],"name":[2],"type":["chr"],"align":["left"]}],"data":[{"1":"P01579","2":"IFNG"},{"1":"Q16552","2":"IL17A"},{"1":"P05113","2":"IL5"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(stimulation) <span class="sc">%&gt;%</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["stimulation"],"name":[1],"type":["chr"],"align":["left"]}],"data":[{"1":"PHA"},{"1":"PT"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>where</p>
<ul>
<li><p>DMSO: “Negative” Control (solvent where peptides are dissolved in)</p></li>
<li><p>PHA (Phytohaemagglutinin): “Postive” Control (this should elicit strong T cell response?)</p></li>
<li><p>PT: Bordetella Pertussis Toxin</p></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(protein_meta, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"protein_id"</span><span class="ot">=</span><span class="st">"uniprot_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>analyte_counts), <span class="at">bins=</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows=</span><span class="fu">vars</span>(stimulation), <span class="at">cols=</span><span class="fu">vars</span>(cytokine)) <span class="sc">+</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="filtered_data_overview_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: the assay has been normalized relative to DMSO</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(stimulation<span class="sc">==</span><span class="st">"DMSO"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(analyte_counts) <span class="sc">%&gt;%</span></span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">near</span>(., <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NaN</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(stimulation<span class="sc">!=</span><span class="st">"DMSO"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(stimulation_protein_id) <span class="sc">%&gt;%</span></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "PHA_P01579" "PHA_Q16552" "PHA_P05113" "PT_P01579"  "PT_Q16552" 
[6] "PT_P05113" </code></pre>
</div>
</div>
</section>
<section id="other-notes" class="level1" data-number="12">
<h1 data-number="12"><span class="header-section-number">12</span> Other Notes</h1>
<ul>
<li>Only from 2022 onwards do we have time-course baseline measurements, i.e.&nbsp;<code>-30, -14/15, 0</code></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset, planned_day_relative_to_boost) <span class="sc">%&gt;%</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">&lt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["partition"],"name":[1],"type":["chr"],"align":["left"]},{"label":["dataset"],"name":[2],"type":["chr"],"align":["left"]},{"label":["planned_day_relative_to_boost"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["n"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"challenge","2":"2023_dataset","3":"-30","4":"54"},{"1":"challenge","2":"2023_dataset","3":"-14","4":"54"},{"1":"train","2":"2022_dataset","3":"-30","4":"21"},{"1":"train","2":"2022_dataset","3":"-15","4":"21"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="appendix" class="level1" data-number="13">
<h1 data-number="13"><span class="header-section-number">13</span> Appendix</h1>
<button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne">
Session Information
</button>
<div id="collapseOne" class="accordion-collapse collapse">
<div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 14.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] C

attached base packages:
[1] grid      stats     graphics  grDevices datasets  utils     methods  
[8] base     

other attached packages:
 [1] circlize_0.4.16       ComplexHeatmap_2.15.4 magick_2.8.3         
 [4] FactoMineR_2.11       factoextra_1.0.7      ggdark_0.2.1         
 [7] lubridate_1.9.3       forcats_1.0.0         stringr_1.5.1        
[10] dplyr_1.1.4           purrr_1.0.2           readr_2.1.5          
[13] tidyr_1.3.1           tibble_3.2.1          ggplot2_3.5.1        
[16] tidyverse_2.0.0      

loaded via a namespace (and not attached):
 [1] bit64_4.5.2          vroom_1.6.5          jsonlite_1.8.9      
 [4] foreach_1.5.2        BiocManager_1.30.25  stats4_4.2.3        
 [7] renv_1.0.11          yaml_2.3.10          ggrepel_0.9.6       
[10] pillar_1.9.0         lattice_0.22-6       glue_1.8.0          
[13] digest_0.6.37        RColorBrewer_1.1-3   colorspace_2.1-1    
[16] cowplot_1.1.3        htmltools_0.5.8.1    pkgconfig_2.0.3     
[19] GetoptLong_1.0.5     xtable_1.8-4         mvtnorm_1.3-1       
[22] scales_1.3.0         tzdb_0.4.0           timechange_0.3.0    
[25] emmeans_1.10.5       farver_2.1.2         generics_0.1.3      
[28] IRanges_2.32.0       DT_0.33              withr_3.0.2         
[31] BiocGenerics_0.44.0  cli_3.6.3            magrittr_2.0.3      
[34] crayon_1.5.3         estimability_1.5.1   evaluate_1.0.1      
[37] fansi_1.0.6          doParallel_1.0.17    MASS_7.3-60.0.1     
[40] tools_4.2.3          hms_1.1.3            GlobalOptions_0.1.2 
[43] lifecycle_1.0.4      matrixStats_1.4.1    S4Vectors_0.36.2    
[46] munsell_0.5.1        cluster_2.1.6        flashClust_1.01-2   
[49] compiler_4.2.3       multcompView_0.1-10  rlang_1.1.4         
[52] iterators_1.0.14     rjson_0.2.23         htmlwidgets_1.6.4   
[55] leaps_3.2            labeling_0.4.3       rmarkdown_2.29      
[58] gtable_0.3.6         codetools_0.2-20     R6_2.5.1            
[61] knitr_1.49           bit_4.5.0            fastmap_1.2.0       
[64] utf8_1.2.4           clue_0.3-65          shape_1.4.6.1       
[67] stringi_1.8.4        parallel_4.2.3       Rcpp_1.0.13-1       
[70] vctrs_0.6.5          png_0.1-8            scatterplot3d_0.3-44
[73] tidyselect_1.2.1     xfun_0.49            coda_0.19-4.1       </code></pre>
</div>
</div>
</div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb64" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Data Overview"</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Philipp Sven Lars Schäfer"</span></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "`r format(Sys.time(), '%d %B, %Y')`"</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> source</span></span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a><span class="an">engine:</span><span class="co"> knitr</span></span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a><span class="fu"># Packages</span></span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-13"><a href="#cb64-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-14"><a href="#cb64-14" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb64-15"><a href="#cb64-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb64-16"><a href="#cb64-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb64-17"><a href="#cb64-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb64-18"><a href="#cb64-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb64-19"><a href="#cb64-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick)</span>
<span id="cb64-20"><a href="#cb64-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb64-21"><a href="#cb64-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(circlize)</span>
<span id="cb64-22"><a href="#cb64-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb64-23"><a href="#cb64-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span>
<span id="cb64-24"><a href="#cb64-24" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb64-25"><a href="#cb64-25" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-26"><a href="#cb64-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-27"><a href="#cb64-27" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data</span></span>
<span id="cb64-28"><a href="#cb64-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-31"><a href="#cb64-31" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-32"><a href="#cb64-32" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span>
<span id="cb64-33"><a href="#cb64-33" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-34"><a href="#cb64-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-37"><a href="#cb64-37" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-38"><a href="#cb64-38" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb64-39"><a href="#cb64-39" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb64-40"><a href="#cb64-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-41"><a href="#cb64-41" aria-hidden="true" tabindex="-1"></a><span class="co">#exp_data &lt;- read_harmonized_experimental_data(input_dir)</span></span>
<span id="cb64-42"><a href="#cb64-42" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb64-43"><a href="#cb64-43" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(meta_data, exp_data, <span class="at">gene_meta=</span>gene_meta)</span>
<span id="cb64-44"><a href="#cb64-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-45"><a href="#cb64-45" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb64-46"><a href="#cb64-46" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb64-47"><a href="#cb64-47" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span>
<span id="cb64-48"><a href="#cb64-48" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-49"><a href="#cb64-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-50"><a href="#cb64-50" aria-hidden="true" tabindex="-1"></a><span class="fu"># Checks</span></span>
<span id="cb64-51"><a href="#cb64-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-54"><a href="#cb64-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-55"><a href="#cb64-55" aria-hidden="true" tabindex="-1"></a><span class="co"># only works due to filtering step above</span></span>
<span id="cb64-56"><a href="#cb64-56" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb64-57"><a href="#cb64-57" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="sc">~</span> <span class="fu">all</span>(.x<span class="sc">$</span>specimen_id <span class="sc">%in%</span> meta_data<span class="sc">$</span>specimen_id)))</span>
<span id="cb64-58"><a href="#cb64-58" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb64-59"><a href="#cb64-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-60"><a href="#cb64-60" aria-hidden="true" tabindex="-1"></a><span class="co"># no subject is recorded in more than one dataset</span></span>
<span id="cb64-61"><a href="#cb64-61" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-62"><a href="#cb64-62" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-63"><a href="#cb64-63" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-64"><a href="#cb64-64" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(n) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-65"><a href="#cb64-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-66"><a href="#cb64-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb64-67"><a href="#cb64-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-68"><a href="#cb64-68" aria-hidden="true" tabindex="-1"></a><span class="co"># we have the baseline specimen (planned_day_relative_to_boost = 0) for every subject</span></span>
<span id="cb64-69"><a href="#cb64-69" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-70"><a href="#cb64-70" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, planned_day_relative_to_boost, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-71"><a href="#cb64-71" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-72"><a href="#cb64-72" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_baseline =</span> (planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-73"><a href="#cb64-73" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-74"><a href="#cb64-74" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present =</span> <span class="fu">any</span>(is_baseline),</span>
<span id="cb64-75"><a href="#cb64-75" aria-hidden="true" tabindex="-1"></a>                   <span class="at">dataset =</span> <span class="fu">first</span>(dataset)) <span class="sc">%&gt;%</span></span>
<span id="cb64-76"><a href="#cb64-76" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-77"><a href="#cb64-77" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present_frac =</span> <span class="fu">mean</span>(baseline_present)) <span class="sc">%&gt;%</span></span>
<span id="cb64-78"><a href="#cb64-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(baseline_present_frac) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-79"><a href="#cb64-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-80"><a href="#cb64-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb64-81"><a href="#cb64-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-82"><a href="#cb64-82" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb64-83"><a href="#cb64-83" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="cf">function</span>(df) {</span>
<span id="cb64-84"><a href="#cb64-84" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="st">"unit"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(df)) {</span>
<span id="cb64-85"><a href="#cb64-85" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">length</span>(<span class="fu">unique</span>(df[[<span class="st">"unit"</span>]])) <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb64-86"><a href="#cb64-86" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb64-87"><a href="#cb64-87" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="cn">TRUE</span>)</span>
<span id="cb64-88"><a href="#cb64-88" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb64-89"><a href="#cb64-89" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb64-90"><a href="#cb64-90" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb64-91"><a href="#cb64-91" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-92"><a href="#cb64-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-93"><a href="#cb64-93" aria-hidden="true" tabindex="-1"></a><span class="fu"># Demography &amp; Metadata</span></span>
<span id="cb64-94"><a href="#cb64-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-95"><a href="#cb64-95" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many subjects do we have per dataset / per partition?</span></span>
<span id="cb64-96"><a href="#cb64-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-99"><a href="#cb64-99" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-100"><a href="#cb64-100" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-101"><a href="#cb64-101" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, dataset, partition, infancy_vac) <span class="sc">%&gt;%</span></span>
<span id="cb64-102"><a href="#cb64-102" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-103"><a href="#cb64-103" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(dataset, partition, infancy_vac) <span class="sc">%&gt;%</span></span>
<span id="cb64-104"><a href="#cb64-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> n, <span class="at">y =</span> dataset, <span class="at">fill =</span> infancy_vac, <span class="at">color =</span> partition)) <span class="sc">+</span></span>
<span id="cb64-105"><a href="#cb64-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>), <span class="at">width =</span> <span class="fl">0.7</span>, <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span>  </span>
<span id="cb64-106"><a href="#cb64-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> n), </span>
<span id="cb64-107"><a href="#cb64-107" aria-hidden="true" tabindex="-1"></a>            <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>),  </span>
<span id="cb64-108"><a href="#cb64-108" aria-hidden="true" tabindex="-1"></a>            <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb64-109"><a href="#cb64-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> infancy_vac_colors) <span class="sc">+</span></span>
<span id="cb64-110"><a href="#cb64-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> partition_colors) <span class="sc">+</span></span>
<span id="cb64-111"><a href="#cb64-111" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb64-112"><a href="#cb64-112" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-113"><a href="#cb64-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-114"><a href="#cb64-114" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the age range</span></span>
<span id="cb64-115"><a href="#cb64-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-118"><a href="#cb64-118" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-119"><a href="#cb64-119" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-120"><a href="#cb64-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb64-121"><a href="#cb64-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>age_at_boost, <span class="at">fill=</span>dataset, <span class="at">y=</span><span class="fu">after_stat</span>(density)), </span>
<span id="cb64-122"><a href="#cb64-122" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color=</span><span class="st">"black"</span>, <span class="at">position=</span><span class="st">"identity"</span>, <span class="at">bins=</span><span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb64-123"><a href="#cb64-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb64-124"><a href="#cb64-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors) <span class="sc">+</span></span>
<span id="cb64-125"><a href="#cb64-125" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-126"><a href="#cb64-126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-127"><a href="#cb64-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-128"><a href="#cb64-128" aria-hidden="true" tabindex="-1"></a><span class="fu"># Assays &amp; Timepoints per Subject / Dataset</span></span>
<span id="cb64-129"><a href="#cb64-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-130"><a href="#cb64-130" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the difference between planned and actual booster administration</span></span>
<span id="cb64-131"><a href="#cb64-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-134"><a href="#cb64-134" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-135"><a href="#cb64-135" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-136"><a href="#cb64-136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-137"><a href="#cb64-137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost), <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb64-138"><a href="#cb64-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb64-139"><a href="#cb64-139" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-140"><a href="#cb64-140" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-141"><a href="#cb64-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-142"><a href="#cb64-142" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the difference between planned and actual for the baseline</span></span>
<span id="cb64-143"><a href="#cb64-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-146"><a href="#cb64-146" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-147"><a href="#cb64-147" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-148"><a href="#cb64-148" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-149"><a href="#cb64-149" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="st">`</span><span class="at">diff_&gt;_15</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">abs</span>(actual_day_relative_to_boost) <span class="sc">&gt;</span> <span class="dv">15</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-150"><a href="#cb64-150" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-151"><a href="#cb64-151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>actual_day_relative_to_boost, <span class="at">fill=</span><span class="st">`</span><span class="at">diff_&gt;_15</span><span class="st">`</span>), <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb64-152"><a href="#cb64-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset) <span class="sc">+</span></span>
<span id="cb64-153"><a href="#cb64-153" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-154"><a href="#cb64-154" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-155"><a href="#cb64-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-156"><a href="#cb64-156" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many time points do we have per subject?</span></span>
<span id="cb64-157"><a href="#cb64-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-160"><a href="#cb64-160" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-161"><a href="#cb64-161" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-162"><a href="#cb64-162" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(dataset, subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-163"><a href="#cb64-163" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-164"><a href="#cb64-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">fill=</span>dataset),<span class="at">color=</span><span class="st">"black"</span>, <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb64-165"><a href="#cb64-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset, <span class="at">ncol=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb64-166"><a href="#cb64-166" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb64-167"><a href="#cb64-167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb64-168"><a href="#cb64-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors)</span>
<span id="cb64-169"><a href="#cb64-169" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-170"><a href="#cb64-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-171"><a href="#cb64-171" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many assays do we have for the baseline measurement?</span></span>
<span id="cb64-172"><a href="#cb64-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-175"><a href="#cb64-175" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-176"><a href="#cb64-176" aria-hidden="true" tabindex="-1"></a>assays_per_specimen <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">imap</span>(exp_data, <span class="sc">~</span> .x <span class="sc">%&gt;%</span> </span>
<span id="cb64-177"><a href="#cb64-177" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-178"><a href="#cb64-178" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-179"><a href="#cb64-179" aria-hidden="true" tabindex="-1"></a>              dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">assay=</span>.y)) <span class="sc">%&gt;%</span></span>
<span id="cb64-180"><a href="#cb64-180" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-181"><a href="#cb64-181" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-182"><a href="#cb64-182" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, assay, subject_id, planned_day_relative_to_boost, infancy_vac, dataset)</span>
<span id="cb64-183"><a href="#cb64-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-184"><a href="#cb64-184" aria-hidden="true" tabindex="-1"></a>assays_per_specimen <span class="sc">%&gt;%</span></span>
<span id="cb64-185"><a href="#cb64-185" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, planned_day_relative_to_boost, dataset, specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-186"><a href="#cb64-186" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(subject_id)) <span class="sc">%&gt;%</span></span>
<span id="cb64-187"><a href="#cb64-187" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-188"><a href="#cb64-188" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-189"><a href="#cb64-189" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>n, <span class="at">fill=</span>dataset), <span class="at">color=</span><span class="st">"black"</span>, <span class="at">binwidth=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb64-190"><a href="#cb64-190" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>dataset, <span class="at">ncol=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb64-191"><a href="#cb64-191" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb64-192"><a href="#cb64-192" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">8</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb64-193"><a href="#cb64-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>dataset_colors)</span>
<span id="cb64-194"><a href="#cb64-194" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-195"><a href="#cb64-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-196"><a href="#cb64-196" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.height=6, fig.width=12}</span></span>
<span id="cb64-197"><a href="#cb64-197" aria-hidden="true" tabindex="-1"></a><span class="in">for (day in c(0, 1, 3, 14)) {</span></span>
<span id="cb64-198"><a href="#cb64-198" aria-hidden="true" tabindex="-1"></a><span class="in">  </span></span>
<span id="cb64-199"><a href="#cb64-199" aria-hidden="true" tabindex="-1"></a><span class="in">annotation_data &lt;- meta_data %&gt;%</span></span>
<span id="cb64-200"><a href="#cb64-200" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(specimen_id, planned_day_relative_to_boost, infancy_vac, </span></span>
<span id="cb64-201"><a href="#cb64-201" aria-hidden="true" tabindex="-1"></a><span class="in">                dataset, biological_sex) %&gt;%</span></span>
<span id="cb64-202"><a href="#cb64-202" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::filter(specimen_id %in% assays_per_specimen$specimen_id) %&gt;%</span></span>
<span id="cb64-203"><a href="#cb64-203" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::filter(planned_day_relative_to_boost == !!day) %&gt;% </span></span>
<span id="cb64-204"><a href="#cb64-204" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(- planned_day_relative_to_boost) %&gt;%</span></span>
<span id="cb64-205"><a href="#cb64-205" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::arrange(dataset, infancy_vac, biological_sex) %&gt;%</span></span>
<span id="cb64-206"><a href="#cb64-206" aria-hidden="true" tabindex="-1"></a><span class="in">  tibble::column_to_rownames("specimen_id") %&gt;%</span></span>
<span id="cb64-207"><a href="#cb64-207" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(dataset, infancy_vac, biological_sex)</span></span>
<span id="cb64-208"><a href="#cb64-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-209"><a href="#cb64-209" aria-hidden="true" tabindex="-1"></a><span class="in">heatmap_data &lt;- assays_per_specimen %&gt;%</span></span>
<span id="cb64-210"><a href="#cb64-210" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(specimen_id, assay) %&gt;%</span></span>
<span id="cb64-211"><a href="#cb64-211" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(value = 1) %&gt;%</span></span>
<span id="cb64-212"><a href="#cb64-212" aria-hidden="true" tabindex="-1"></a><span class="in">  tidyr::pivot_wider(names_from="assay", values_from="value") %&gt;%</span></span>
<span id="cb64-213"><a href="#cb64-213" aria-hidden="true" tabindex="-1"></a><span class="in">  mutate(across(everything(), .fns = ~replace_na(.,0))) %&gt;%</span></span>
<span id="cb64-214"><a href="#cb64-214" aria-hidden="true" tabindex="-1"></a><span class="in">  tibble::column_to_rownames("specimen_id") %&gt;%</span></span>
<span id="cb64-215"><a href="#cb64-215" aria-hidden="true" tabindex="-1"></a><span class="in">  as.matrix() %&gt;%</span></span>
<span id="cb64-216"><a href="#cb64-216" aria-hidden="true" tabindex="-1"></a><span class="in">  t()</span></span>
<span id="cb64-217"><a href="#cb64-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-218"><a href="#cb64-218" aria-hidden="true" tabindex="-1"></a><span class="in">colnames(heatmap_data) &lt;- as.character(colnames(heatmap_data))</span></span>
<span id="cb64-219"><a href="#cb64-219" aria-hidden="true" tabindex="-1"></a><span class="in">heatmap_data &lt;- heatmap_data[, rownames(annotation_data)]</span></span>
<span id="cb64-220"><a href="#cb64-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-221"><a href="#cb64-221" aria-hidden="true" tabindex="-1"></a><span class="in"># Create the heatmap</span></span>
<span id="cb64-222"><a href="#cb64-222" aria-hidden="true" tabindex="-1"></a><span class="in">ht &lt;- Heatmap(heatmap_data, </span></span>
<span id="cb64-223"><a href="#cb64-223" aria-hidden="true" tabindex="-1"></a><span class="in">              name = "heatmap", </span></span>
<span id="cb64-224"><a href="#cb64-224" aria-hidden="true" tabindex="-1"></a><span class="in">              row_title = "", </span></span>
<span id="cb64-225"><a href="#cb64-225" aria-hidden="true" tabindex="-1"></a><span class="in">              column_title = paste0("Specimens for Planned Day ", day),</span></span>
<span id="cb64-226"><a href="#cb64-226" aria-hidden="true" tabindex="-1"></a><span class="in">              col = colorRamp2(c(0, 1), c("white", "black")),</span></span>
<span id="cb64-227"><a href="#cb64-227" aria-hidden="true" tabindex="-1"></a><span class="in">              cluster_rows = FALSE,</span></span>
<span id="cb64-228"><a href="#cb64-228" aria-hidden="true" tabindex="-1"></a><span class="in">              cluster_columns = FALSE,</span></span>
<span id="cb64-229"><a href="#cb64-229" aria-hidden="true" tabindex="-1"></a><span class="in">              show_row_names = TRUE, </span></span>
<span id="cb64-230"><a href="#cb64-230" aria-hidden="true" tabindex="-1"></a><span class="in">              show_column_names = FALSE,</span></span>
<span id="cb64-231"><a href="#cb64-231" aria-hidden="true" tabindex="-1"></a><span class="in">              width = unit(16, "cm"),</span></span>
<span id="cb64-232"><a href="#cb64-232" aria-hidden="true" tabindex="-1"></a><span class="in">              top_annotation = ComplexHeatmap::HeatmapAnnotation(df = annotation_data,</span></span>
<span id="cb64-233"><a href="#cb64-233" aria-hidden="true" tabindex="-1"></a><span class="in">                                    col = list(</span></span>
<span id="cb64-234"><a href="#cb64-234" aria-hidden="true" tabindex="-1"></a><span class="in">                                      "dataset" = dataset_colors,</span></span>
<span id="cb64-235"><a href="#cb64-235" aria-hidden="true" tabindex="-1"></a><span class="in">                                      "infancy_vac" = infancy_vac_colors,</span></span>
<span id="cb64-236"><a href="#cb64-236" aria-hidden="true" tabindex="-1"></a><span class="in">                                      "biological_sex" = sex_colors</span></span>
<span id="cb64-237"><a href="#cb64-237" aria-hidden="true" tabindex="-1"></a><span class="in">                                    ),</span></span>
<span id="cb64-238"><a href="#cb64-238" aria-hidden="true" tabindex="-1"></a><span class="in">                                    which="column")</span></span>
<span id="cb64-239"><a href="#cb64-239" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb64-240"><a href="#cb64-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-241"><a href="#cb64-241" aria-hidden="true" tabindex="-1"></a><span class="in">draw(ht, </span></span>
<span id="cb64-242"><a href="#cb64-242" aria-hidden="true" tabindex="-1"></a><span class="in">     heatmap_legend_side = "top",</span></span>
<span id="cb64-243"><a href="#cb64-243" aria-hidden="true" tabindex="-1"></a><span class="in">     annotation_legend_side = "top",</span></span>
<span id="cb64-244"><a href="#cb64-244" aria-hidden="true" tabindex="-1"></a><span class="in">     show_heatmap_legend = FALSE # don't show colorbar</span></span>
<span id="cb64-245"><a href="#cb64-245" aria-hidden="true" tabindex="-1"></a><span class="in">     )</span></span>
<span id="cb64-246"><a href="#cb64-246" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb64-247"><a href="#cb64-247" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-248"><a href="#cb64-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-251"><a href="#cb64-251" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-252"><a href="#cb64-252" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-253"><a href="#cb64-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-254"><a href="#cb64-254" aria-hidden="true" tabindex="-1"></a><span class="fu"># Training Data</span></span>
<span id="cb64-255"><a href="#cb64-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-256"><a href="#cb64-256" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many subjects can we use to construct training data?</span></span>
<span id="cb64-257"><a href="#cb64-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-258"><a href="#cb64-258" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1) Antibody Level Tasks (anti-PT 14 days after booster)</span></span>
<span id="cb64-259"><a href="#cb64-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-260"><a href="#cb64-260" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>All but 7 people were assayed approximately 14 days after the booster administration</span>
<span id="cb64-261"><a href="#cb64-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-264"><a href="#cb64-264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-265"><a href="#cb64-265" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-266"><a href="#cb64-266" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-267"><a href="#cb64-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb64-268"><a href="#cb64-268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb64-269"><a href="#cb64-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb64-270"><a href="#cb64-270" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-271"><a href="#cb64-271" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-272"><a href="#cb64-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-275"><a href="#cb64-275" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-276"><a href="#cb64-276" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-277"><a href="#cb64-277" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb64-278"><a href="#cb64-278" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-279"><a href="#cb64-279" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_1_possible =</span> </span>
<span id="cb64-280"><a href="#cb64-280" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-281"><a href="#cb64-281" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">14</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-282"><a href="#cb64-282" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>plasma_ab_titer,</span>
<span id="cb64-283"><a href="#cb64-283" aria-hidden="true" tabindex="-1"></a>                                                      <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-284"><a href="#cb64-284" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(isotype_antigen<span class="sc">==</span><span class="st">"IgG_PT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-285"><a href="#cb64-285" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-286"><a href="#cb64-286" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb64-287"><a href="#cb64-287" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb64-288"><a href="#cb64-288" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-289"><a href="#cb64-289" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_1_possible_total =</span> <span class="fu">sum</span>(task_1_possible), </span>
<span id="cb64-290"><a href="#cb64-290" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_1_possible_fraction =</span> <span class="fu">mean</span>(task_1_possible))</span>
<span id="cb64-291"><a href="#cb64-291" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-292"><a href="#cb64-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-293"><a href="#cb64-293" aria-hidden="true" tabindex="-1"></a><span class="fu">### 2) Cell Frequency Tasks (Monocytes 1 day after booster)</span></span>
<span id="cb64-294"><a href="#cb64-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-295"><a href="#cb64-295" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>But there is one person where it does not really make sense</span>
<span id="cb64-296"><a href="#cb64-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-299"><a href="#cb64-299" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-300"><a href="#cb64-300" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-301"><a href="#cb64-301" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-302"><a href="#cb64-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb64-303"><a href="#cb64-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb64-304"><a href="#cb64-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb64-305"><a href="#cb64-305" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-306"><a href="#cb64-306" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-307"><a href="#cb64-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-310"><a href="#cb64-310" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-311"><a href="#cb64-311" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-312"><a href="#cb64-312" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb64-313"><a href="#cb64-313" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-314"><a href="#cb64-314" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_possible =</span> </span>
<span id="cb64-315"><a href="#cb64-315" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-316"><a href="#cb64-316" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-317"><a href="#cb64-317" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>pbmc_cell_frequency, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-318"><a href="#cb64-318" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(cell_type_name<span class="sc">==</span><span class="st">"Monocytes"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-319"><a href="#cb64-319" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-320"><a href="#cb64-320" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb64-321"><a href="#cb64-321" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb64-322"><a href="#cb64-322" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-323"><a href="#cb64-323" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_possible_total =</span> <span class="fu">sum</span>(task_possible), </span>
<span id="cb64-324"><a href="#cb64-324" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_possible_fraction =</span> <span class="fu">mean</span>(task_possible))</span>
<span id="cb64-325"><a href="#cb64-325" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-326"><a href="#cb64-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-327"><a href="#cb64-327" aria-hidden="true" tabindex="-1"></a><span class="fu">### 3) Gene Expression Tasks (CCL3 expression 3 days after booster)</span></span>
<span id="cb64-328"><a href="#cb64-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-329"><a href="#cb64-329" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>There are 7 people for which this task does nto make sense</span>
<span id="cb64-330"><a href="#cb64-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-333"><a href="#cb64-333" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-334"><a href="#cb64-334" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-335"><a href="#cb64-335" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-336"><a href="#cb64-336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>diff_relative_to_boost)) <span class="sc">+</span></span>
<span id="cb64-337"><a href="#cb64-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb64-338"><a href="#cb64-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb64-339"><a href="#cb64-339" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-340"><a href="#cb64-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-341"><a href="#cb64-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-344"><a href="#cb64-344" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-345"><a href="#cb64-345" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">unique</span>(exp_data<span class="sc">$</span>pbmc_gene_expression<span class="sc">$</span>versioned_ensembl_gene_id)</span>
<span id="cb64-346"><a href="#cb64-346" aria-hidden="true" tabindex="-1"></a>ccl3_ensembl <span class="ot">&lt;-</span> <span class="st">"ENSG00000277632"</span></span>
<span id="cb64-347"><a href="#cb64-347" aria-hidden="true" tabindex="-1"></a>ccl3_ensembl_versioned <span class="ot">&lt;-</span> all_genes[<span class="fu">str_starts</span>(all_genes, ccl3_ensembl)]</span>
<span id="cb64-348"><a href="#cb64-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-349"><a href="#cb64-349" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-350"><a href="#cb64-350" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, dataset, biological_sex, ethnicity, race, year_of_birth) <span class="sc">%&gt;%</span></span>
<span id="cb64-351"><a href="#cb64-351" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-352"><a href="#cb64-352" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task_possible =</span> </span>
<span id="cb64-353"><a href="#cb64-353" aria-hidden="true" tabindex="-1"></a>                  subject_id <span class="sc">%in%</span> (meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-354"><a href="#cb64-354" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-355"><a href="#cb64-355" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">left_join</span>(exp_data<span class="sc">$</span>pbmc_gene_expression, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-356"><a href="#cb64-356" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">filter</span>(versioned_ensembl_gene_id<span class="sc">==</span>ccl3_ensembl_versioned) <span class="sc">%&gt;%</span></span>
<span id="cb64-357"><a href="#cb64-357" aria-hidden="true" tabindex="-1"></a>                                     tidyr<span class="sc">::</span><span class="fu">drop_na</span>() <span class="sc">%&gt;%</span></span>
<span id="cb64-358"><a href="#cb64-358" aria-hidden="true" tabindex="-1"></a>                                     dplyr<span class="sc">::</span><span class="fu">pull</span>(subject_id))</span>
<span id="cb64-359"><a href="#cb64-359" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb64-360"><a href="#cb64-360" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-361"><a href="#cb64-361" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">task_possible_total =</span> <span class="fu">sum</span>(task_possible), </span>
<span id="cb64-362"><a href="#cb64-362" aria-hidden="true" tabindex="-1"></a>                   <span class="at">task_possible_fraction =</span> <span class="fu">mean</span>(task_possible))</span>
<span id="cb64-363"><a href="#cb64-363" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-364"><a href="#cb64-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-365"><a href="#cb64-365" aria-hidden="true" tabindex="-1"></a><span class="fu"># Missing Data</span></span>
<span id="cb64-366"><a href="#cb64-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-367"><a href="#cb64-367" aria-hidden="true" tabindex="-1"></a><span class="fu">## How many NA values do we have per assay?</span></span>
<span id="cb64-368"><a href="#cb64-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-371"><a href="#cb64-371" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-372"><a href="#cb64-372" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">imap</span>(exp_data, <span class="cf">function</span>(df, modality) {</span>
<span id="cb64-373"><a href="#cb64-373" aria-hidden="true" tabindex="-1"></a>  feature_col <span class="ot">&lt;-</span> experimental_data_settings[[modality]]<span class="sc">$</span>feature_col</span>
<span id="cb64-374"><a href="#cb64-374" aria-hidden="true" tabindex="-1"></a>  value_col <span class="ot">&lt;-</span> experimental_data_settings[[modality]]<span class="sc">$</span>value_col</span>
<span id="cb64-375"><a href="#cb64-375" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">modality =</span> modality, <span class="at">feature_col =</span> feature_col, <span class="at">sum_na =</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(df[[value_col]])))</span>
<span id="cb64-376"><a href="#cb64-376" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb64-377"><a href="#cb64-377" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb64-378"><a href="#cb64-378" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-379"><a href="#cb64-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-380"><a href="#cb64-380" aria-hidden="true" tabindex="-1"></a><span class="fu">## Wide format</span></span>
<span id="cb64-381"><a href="#cb64-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-384"><a href="#cb64-384" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-385"><a href="#cb64-385" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">imap_dfr</span>(<span class="fu">generate_wide_experimental_data</span>(<span class="at">experimental_data=</span>exp_data, </span>
<span id="cb64-386"><a href="#cb64-386" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">impute=</span><span class="cn">NULL</span>),</span>
<span id="cb64-387"><a href="#cb64-387" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(mtx, modality) {</span>
<span id="cb64-388"><a href="#cb64-388" aria-hidden="true" tabindex="-1"></a>                    rmeans <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">is.na</span>(mtx))</span>
<span id="cb64-389"><a href="#cb64-389" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">tibble</span>(<span class="at">modality=</span>modality, </span>
<span id="cb64-390"><a href="#cb64-390" aria-hidden="true" tabindex="-1"></a>                           <span class="at">subject_id =</span> <span class="fu">names</span>(rmeans), </span>
<span id="cb64-391"><a href="#cb64-391" aria-hidden="true" tabindex="-1"></a>                           <span class="at">na_frac=</span>rmeans)</span>
<span id="cb64-392"><a href="#cb64-392" aria-hidden="true" tabindex="-1"></a>           }) <span class="sc">%&gt;%</span></span>
<span id="cb64-393"><a href="#cb64-393" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(modality) <span class="sc">%&gt;%</span></span>
<span id="cb64-394"><a href="#cb64-394" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">mean_na_frac =</span> <span class="fu">mean</span>(na_frac))</span>
<span id="cb64-395"><a href="#cb64-395" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-396"><a href="#cb64-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-397"><a href="#cb64-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-398"><a href="#cb64-398" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the fraction of NA values in the PBMC Cell Type Frequencies</span></span>
<span id="cb64-399"><a href="#cb64-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-400"><a href="#cb64-400" aria-hidden="true" tabindex="-1"></a>With the current selection of cell types, there is no specimen with NA values.</span>
<span id="cb64-401"><a href="#cb64-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-402"><a href="#cb64-402" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Cell Type</span></span>
<span id="cb64-403"><a href="#cb64-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-406"><a href="#cb64-406" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-407"><a href="#cb64-407" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb64-408"><a href="#cb64-408" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb64-409"><a href="#cb64-409" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(cell_type_name) <span class="sc">%&gt;%</span></span>
<span id="cb64-410"><a href="#cb64-410" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb64-411"><a href="#cb64-411" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(frac_na)) <span class="sc">%&gt;%</span></span>
<span id="cb64-412"><a href="#cb64-412" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">cell_type_name =</span> <span class="fu">factor</span>(cell_type_name, <span class="at">levels=</span><span class="fu">rev</span>(cell_type_name))) <span class="sc">%&gt;%</span></span>
<span id="cb64-413"><a href="#cb64-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-414"><a href="#cb64-414" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">y=</span>cell_type_name, <span class="at">x=</span>frac_na)) <span class="sc">+</span></span>
<span id="cb64-415"><a href="#cb64-415" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb64-416"><a href="#cb64-416" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Cell Type"</span>)</span>
<span id="cb64-417"><a href="#cb64-417" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-418"><a href="#cb64-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-419"><a href="#cb64-419" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Cell Specimen</span></span>
<span id="cb64-420"><a href="#cb64-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-423"><a href="#cb64-423" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-424"><a href="#cb64-424" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb64-425"><a href="#cb64-425" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(percent_live_cell)) <span class="sc">%&gt;%</span></span>
<span id="cb64-426"><a href="#cb64-426" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-427"><a href="#cb64-427" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb64-428"><a href="#cb64-428" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> frac_na)) <span class="sc">+</span></span>
<span id="cb64-429"><a href="#cb64-429" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add color for better visibility</span></span>
<span id="cb64-430"><a href="#cb64-430" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins =</span> <span class="dv">20</span>, <span class="at">geom =</span> <span class="st">"text"</span>, </span>
<span id="cb64-431"><a href="#cb64-431" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Display bin counts at the top</span></span>
<span id="cb64-432"><a href="#cb64-432" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb64-433"><a href="#cb64-433" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span>
<span id="cb64-434"><a href="#cb64-434" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-435"><a href="#cb64-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-436"><a href="#cb64-436" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the fraction of NA values in the Olink Data?</span></span>
<span id="cb64-437"><a href="#cb64-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-438"><a href="#cb64-438" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Protein</span></span>
<span id="cb64-439"><a href="#cb64-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-440"><a href="#cb64-440" aria-hidden="true" tabindex="-1"></a>After excluding Q969D9, the largest fraction of NA values is about 5% which is acceptable I guess.</span>
<span id="cb64-441"><a href="#cb64-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-442"><a href="#cb64-442" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.height=6}</span></span>
<span id="cb64-443"><a href="#cb64-443" aria-hidden="true" tabindex="-1"></a><span class="in">exp_data$plasma_cytokine_concentration_by_olink %&gt;%</span></span>
<span id="cb64-444"><a href="#cb64-444" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(is_na = is.na(concentration)) %&gt;%</span></span>
<span id="cb64-445"><a href="#cb64-445" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::group_by(protein_id) %&gt;%</span></span>
<span id="cb64-446"><a href="#cb64-446" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::summarise(frac_na = sum(is_na) / n()) %&gt;%</span></span>
<span id="cb64-447"><a href="#cb64-447" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::arrange(desc(frac_na)) %&gt;%</span></span>
<span id="cb64-448"><a href="#cb64-448" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(protein_id = factor(protein_id, levels=rev(protein_id))) %&gt;%</span></span>
<span id="cb64-449"><a href="#cb64-449" aria-hidden="true" tabindex="-1"></a><span class="in">  ggplot() +</span></span>
<span id="cb64-450"><a href="#cb64-450" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_col(aes(y=protein_id, x=frac_na)) +</span></span>
<span id="cb64-451"><a href="#cb64-451" aria-hidden="true" tabindex="-1"></a><span class="in">  ggdark::dark_mode(verbose=FALSE) +</span></span>
<span id="cb64-452"><a href="#cb64-452" aria-hidden="true" tabindex="-1"></a><span class="in">  labs(x="Fraction of NA Values per Protein")</span></span>
<span id="cb64-453"><a href="#cb64-453" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-454"><a href="#cb64-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-455"><a href="#cb64-455" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Specimen</span></span>
<span id="cb64-456"><a href="#cb64-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-457"><a href="#cb64-457" aria-hidden="true" tabindex="-1"></a>There are 6 specimen with more than 50% NA values, I am not sure whether I should exlcude those?</span>
<span id="cb64-458"><a href="#cb64-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-461"><a href="#cb64-461" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-462"><a href="#cb64-462" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb64-463"><a href="#cb64-463" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb64-464"><a href="#cb64-464" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-465"><a href="#cb64-465" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb64-466"><a href="#cb64-466" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> frac_na)) <span class="sc">+</span></span>
<span id="cb64-467"><a href="#cb64-467" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add color for better visibility</span></span>
<span id="cb64-468"><a href="#cb64-468" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins =</span> <span class="dv">20</span>, <span class="at">geom =</span> <span class="st">"text"</span>, </span>
<span id="cb64-469"><a href="#cb64-469" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Display bin counts at the top</span></span>
<span id="cb64-470"><a href="#cb64-470" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb64-471"><a href="#cb64-471" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Fraction of NA Values per Specimen"</span>)</span>
<span id="cb64-472"><a href="#cb64-472" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-473"><a href="#cb64-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-476"><a href="#cb64-476" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-477"><a href="#cb64-477" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb64-478"><a href="#cb64-478" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_na =</span> <span class="fu">is.na</span>(concentration)) <span class="sc">%&gt;%</span></span>
<span id="cb64-479"><a href="#cb64-479" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-480"><a href="#cb64-480" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">frac_na =</span> <span class="fu">sum</span>(is_na) <span class="sc">/</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb64-481"><a href="#cb64-481" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(frac_na <span class="sc">&gt;</span> <span class="fl">0.2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-482"><a href="#cb64-482" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-483"><a href="#cb64-483" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, frac_na, subject_id, dataset, </span>
<span id="cb64-484"><a href="#cb64-484" aria-hidden="true" tabindex="-1"></a>                actual_day_relative_to_boost, planned_day_relative_to_boost)</span>
<span id="cb64-485"><a href="#cb64-485" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-486"><a href="#cb64-486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-487"><a href="#cb64-487" aria-hidden="true" tabindex="-1"></a><span class="fu"># Question for Assay Metadata</span></span>
<span id="cb64-488"><a href="#cb64-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-489"><a href="#cb64-489" aria-hidden="true" tabindex="-1"></a><span class="fu">## How does the quality control for Olink look like?</span></span>
<span id="cb64-490"><a href="#cb64-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-493"><a href="#cb64-493" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-494"><a href="#cb64-494" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb64-495"><a href="#cb64-495" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(quality_control)</span>
<span id="cb64-496"><a href="#cb64-496" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-497"><a href="#cb64-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-500"><a href="#cb64-500" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-501"><a href="#cb64-501" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb64-502"><a href="#cb64-502" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-503"><a href="#cb64-503" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset)</span>
<span id="cb64-504"><a href="#cb64-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-505"><a href="#cb64-505" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_legendplex <span class="sc">%&gt;%</span></span>
<span id="cb64-506"><a href="#cb64-506" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-507"><a href="#cb64-507" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset)</span>
<span id="cb64-508"><a href="#cb64-508" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-509"><a href="#cb64-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-510"><a href="#cb64-510" aria-hidden="true" tabindex="-1"></a><span class="fu">## How often is the Cytokine Concentration below the LOD in Olink</span></span>
<span id="cb64-511"><a href="#cb64-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-514"><a href="#cb64-514" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-515"><a href="#cb64-515" aria-hidden="true" tabindex="-1"></a><span class="co"># check how often is it below the limit of detection?</span></span>
<span id="cb64-516"><a href="#cb64-516" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_cytokine_concentration_by_olink <span class="sc">%&gt;%</span></span>
<span id="cb64-517"><a href="#cb64-517" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">below_lod =</span> concentration <span class="sc">&lt;</span> lower_limit_of_quantitation) <span class="sc">%&gt;%</span></span>
<span id="cb64-518"><a href="#cb64-518" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(below_lod)</span>
<span id="cb64-519"><a href="#cb64-519" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-520"><a href="#cb64-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-521"><a href="#cb64-521" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=10, fig.height=4}</span></span>
<span id="cb64-522"><a href="#cb64-522" aria-hidden="true" tabindex="-1"></a><span class="in">set.seed(1)</span></span>
<span id="cb64-523"><a href="#cb64-523" aria-hidden="true" tabindex="-1"></a><span class="in">p1 &lt;- exp_data$plasma_cytokine_concentration_by_olink %&gt;%</span></span>
<span id="cb64-524"><a href="#cb64-524" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(below_lod = concentration &lt; lower_limit_of_quantitation) %&gt;%</span></span>
<span id="cb64-525"><a href="#cb64-525" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(qc_warning = quality_control == "Warning") %&gt;%</span></span>
<span id="cb64-526"><a href="#cb64-526" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::group_by(protein_id) %&gt;%</span></span>
<span id="cb64-527"><a href="#cb64-527" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::summarize(mean_below_lod = mean(below_lod, na.rm=TRUE),</span></span>
<span id="cb64-528"><a href="#cb64-528" aria-hidden="true" tabindex="-1"></a><span class="in">                   mean_qc_warning = mean(qc_warning, na.rm=TRUE)) %&gt;%</span></span>
<span id="cb64-529"><a href="#cb64-529" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::left_join(cytokine_uniprot_mapping, by="protein_id") %&gt;%</span></span>
<span id="cb64-530"><a href="#cb64-530" aria-hidden="true" tabindex="-1"></a><span class="in">  ggplot() +</span></span>
<span id="cb64-531"><a href="#cb64-531" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_point(aes(x = mean_below_lod, y = mean_qc_warning)) +</span></span>
<span id="cb64-532"><a href="#cb64-532" aria-hidden="true" tabindex="-1"></a><span class="in">  ggrepel::geom_text_repel(aes(x = mean_below_lod, y = mean_qc_warning, </span></span>
<span id="cb64-533"><a href="#cb64-533" aria-hidden="true" tabindex="-1"></a><span class="in">                      label = ifelse(mean_below_lod &gt; 0.2, protein_id, '')),</span></span>
<span id="cb64-534"><a href="#cb64-534" aria-hidden="true" tabindex="-1"></a><span class="in">                  max.overlaps = Inf, color="grey") + # This ensures all labels are shown</span></span>
<span id="cb64-535"><a href="#cb64-535" aria-hidden="true" tabindex="-1"></a><span class="in">  ggdark::dark_mode()</span></span>
<span id="cb64-536"><a href="#cb64-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-537"><a href="#cb64-537" aria-hidden="true" tabindex="-1"></a><span class="in">p2 &lt;- exp_data$plasma_cytokine_concentration_by_olink %&gt;%</span></span>
<span id="cb64-538"><a href="#cb64-538" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(below_lod = concentration &lt; lower_limit_of_quantitation) %&gt;%</span></span>
<span id="cb64-539"><a href="#cb64-539" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::mutate(qc_warning = quality_control == "Warning") %&gt;%</span></span>
<span id="cb64-540"><a href="#cb64-540" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::group_by(protein_id) %&gt;%</span></span>
<span id="cb64-541"><a href="#cb64-541" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::summarize(mean_below_lod = mean(below_lod, na.rm=TRUE),</span></span>
<span id="cb64-542"><a href="#cb64-542" aria-hidden="true" tabindex="-1"></a><span class="in">                   mean_qc_warning = mean(qc_warning, na.rm=TRUE)) %&gt;%</span></span>
<span id="cb64-543"><a href="#cb64-543" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::left_join(cytokine_uniprot_mapping, by="protein_id") %&gt;%</span></span>
<span id="cb64-544"><a href="#cb64-544" aria-hidden="true" tabindex="-1"></a><span class="in">  ggplot() +</span></span>
<span id="cb64-545"><a href="#cb64-545" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_point(aes(x = mean_below_lod, y = mean_qc_warning)) +</span></span>
<span id="cb64-546"><a href="#cb64-546" aria-hidden="true" tabindex="-1"></a><span class="in">  ggrepel::geom_text_repel(aes(x = mean_below_lod, y = mean_qc_warning, </span></span>
<span id="cb64-547"><a href="#cb64-547" aria-hidden="true" tabindex="-1"></a><span class="in">                      label = ifelse(mean_below_lod &gt; 0.2, cytokine, '')),</span></span>
<span id="cb64-548"><a href="#cb64-548" aria-hidden="true" tabindex="-1"></a><span class="in">                  max.overlaps = Inf, color="grey") + # This ensures all labels are shown</span></span>
<span id="cb64-549"><a href="#cb64-549" aria-hidden="true" tabindex="-1"></a><span class="in">  ggdark::dark_mode()</span></span>
<span id="cb64-550"><a href="#cb64-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-551"><a href="#cb64-551" aria-hidden="true" tabindex="-1"></a><span class="in">cowplot::plot_grid(p1, p2, ncol=2)</span></span>
<span id="cb64-552"><a href="#cb64-552" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-553"><a href="#cb64-553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-556"><a href="#cb64-556" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-557"><a href="#cb64-557" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-558"><a href="#cb64-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-559"><a href="#cb64-559" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is the fraction of unspecific antibody measurements</span></span>
<span id="cb64-560"><a href="#cb64-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-561"><a href="#cb64-561" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Isotype-Antigen</span></span>
<span id="cb64-562"><a href="#cb64-562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-565"><a href="#cb64-565" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-566"><a href="#cb64-566" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_ab_titer <span class="sc">%&gt;%</span></span>
<span id="cb64-567"><a href="#cb64-567" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(isotype_antigen) <span class="sc">%&gt;%</span></span>
<span id="cb64-568"><a href="#cb64-568" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">is_antigen_specific_fraction =</span> <span class="fu">mean</span>(is_antigen_specific)) <span class="sc">%&gt;%</span></span>
<span id="cb64-569"><a href="#cb64-569" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>is_antigen_specific_fraction)) <span class="sc">+</span></span>
<span id="cb64-570"><a href="#cb64-570" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">30</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb64-571"><a href="#cb64-571" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">bins=</span><span class="dv">30</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb64-572"><a href="#cb64-572" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-573"><a href="#cb64-573" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-574"><a href="#cb64-574" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-575"><a href="#cb64-575" aria-hidden="true" tabindex="-1"></a><span class="fu">### Per Specimen</span></span>
<span id="cb64-576"><a href="#cb64-576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-577"><a href="#cb64-577" aria-hidden="true" tabindex="-1"></a>So in some specimen we have non-specific antibody measurements? What does that mean?</span>
<span id="cb64-578"><a href="#cb64-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-581"><a href="#cb64-581" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-582"><a href="#cb64-582" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>plasma_ab_titer <span class="sc">%&gt;%</span></span>
<span id="cb64-583"><a href="#cb64-583" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-584"><a href="#cb64-584" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">is_antigen_specific_fraction =</span> <span class="fu">mean</span>(is_antigen_specific)) <span class="sc">%&gt;%</span></span>
<span id="cb64-585"><a href="#cb64-585" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>is_antigen_specific_fraction)) <span class="sc">+</span></span>
<span id="cb64-586"><a href="#cb64-586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb64-587"><a href="#cb64-587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_bin</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">after_stat</span>(count)), <span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">geom =</span> <span class="st">"text"</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  </span>
<span id="cb64-588"><a href="#cb64-588" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-589"><a href="#cb64-589" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-590"><a href="#cb64-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-591"><a href="#cb64-591" aria-hidden="true" tabindex="-1"></a><span class="fu"># Hierarchy in PBMC Frequency Data</span></span>
<span id="cb64-592"><a href="#cb64-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-593"><a href="#cb64-593" aria-hidden="true" tabindex="-1"></a><span class="fu">## NK</span></span>
<span id="cb64-594"><a href="#cb64-594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-595"><a href="#cb64-595" aria-hidden="true" tabindex="-1"></a>There are 3 types of NK cells:</span>
<span id="cb64-596"><a href="#cb64-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-597"><a href="#cb64-597" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>NK cells (CD3-CD19-CD56+): CD19-CD3-CD56+/++</span>
<span id="cb64-598"><a href="#cb64-598" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>NK: CD19-CD3-CD56+</span>
<span id="cb64-599"><a href="#cb64-599" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>CD56high NK cells: CD19-CD3-CD56++</span>
<span id="cb64-600"><a href="#cb64-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-601"><a href="#cb64-601" aria-hidden="true" tabindex="-1"></a>And I don't know what to do with that information!</span>
<span id="cb64-602"><a href="#cb64-602" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-603"><a href="#cb64-603" aria-hidden="true" tabindex="-1"></a><span class="fu">## No Gating Info for 2023</span></span>
<span id="cb64-604"><a href="#cb64-604" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-605"><a href="#cb64-605" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>I just assume it is the same as in 2022, but this might not be true since the data look so different</span>
<span id="cb64-606"><a href="#cb64-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-607"><a href="#cb64-607" aria-hidden="true" tabindex="-1"></a><span class="fu">## No Gating Info for Basophils and CD19 in 2020</span></span>
<span id="cb64-608"><a href="#cb64-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-609"><a href="#cb64-609" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>I just assume it is the same as in 2021 and 2022</span>
<span id="cb64-610"><a href="#cb64-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-611"><a href="#cb64-611" aria-hidden="true" tabindex="-1"></a><span class="fu">## No Gating Info for non-pDCs at all </span></span>
<span id="cb64-612"><a href="#cb64-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-613"><a href="#cb64-613" aria-hidden="true" tabindex="-1"></a>Not sure how to fix this)</span>
<span id="cb64-614"><a href="#cb64-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-617"><a href="#cb64-617" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-618"><a href="#cb64-618" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb64-619"><a href="#cb64-619" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-620"><a href="#cb64-620" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-621"><a href="#cb64-621" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-622"><a href="#cb64-622" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-623"><a href="#cb64-623" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb64-624"><a href="#cb64-624" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb64-625"><a href="#cb64-625" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-626"><a href="#cb64-626" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-627"><a href="#cb64-627" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(level)) <span class="sc">%&gt;%</span></span>
<span id="cb64-628"><a href="#cb64-628" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(dataset, cell_type_name) <span class="sc">%&gt;%</span></span>
<span id="cb64-629"><a href="#cb64-629" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb64-630"><a href="#cb64-630" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-631"><a href="#cb64-631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-632"><a href="#cb64-632" aria-hidden="true" tabindex="-1"></a><span class="fu">## Should the Proportions sum up to 1 at level 0?</span></span>
<span id="cb64-633"><a href="#cb64-633" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-634"><a href="#cb64-634" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Trying to take the hierarchical information into account, it still does not make sense</span>
<span id="cb64-635"><a href="#cb64-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-638"><a href="#cb64-638" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-639"><a href="#cb64-639" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb64-640"><a href="#cb64-640" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-641"><a href="#cb64-641" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-642"><a href="#cb64-642" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-643"><a href="#cb64-643" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-644"><a href="#cb64-644" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb64-645"><a href="#cb64-645" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb64-646"><a href="#cb64-646" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-647"><a href="#cb64-647" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-648"><a href="#cb64-648" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(level <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-649"><a href="#cb64-649" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-650"><a href="#cb64-650" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-651"><a href="#cb64-651" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-652"><a href="#cb64-652" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-653"><a href="#cb64-653" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb64-654"><a href="#cb64-654" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-655"><a href="#cb64-655" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-656"><a href="#cb64-656" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-657"><a href="#cb64-657" aria-hidden="true" tabindex="-1"></a><span class="fu">## Difference in 2023 Dataset</span></span>
<span id="cb64-658"><a href="#cb64-658" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-659"><a href="#cb64-659" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>It seems like the 2023 data are very very different, which makes me hesitant to use it for any kind of predictions on the challenge data. E.g. just looking at level 0 which should make sense I get the following</span>
<span id="cb64-660"><a href="#cb64-660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-663"><a href="#cb64-663" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-664"><a href="#cb64-664" aria-hidden="true" tabindex="-1"></a><span class="co"># first check for NAs for the levels</span></span>
<span id="cb64-665"><a href="#cb64-665" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb64-666"><a href="#cb64-666" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-667"><a href="#cb64-667" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-668"><a href="#cb64-668" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-669"><a href="#cb64-669" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-670"><a href="#cb64-670" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb64-671"><a href="#cb64-671" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb64-672"><a href="#cb64-672" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-673"><a href="#cb64-673" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-674"><a href="#cb64-674" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(level))</span>
<span id="cb64-675"><a href="#cb64-675" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-676"><a href="#cb64-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-679"><a href="#cb64-679" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-680"><a href="#cb64-680" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb64-681"><a href="#cb64-681" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-682"><a href="#cb64-682" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-683"><a href="#cb64-683" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-684"><a href="#cb64-684" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-685"><a href="#cb64-685" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb64-686"><a href="#cb64-686" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb64-687"><a href="#cb64-687" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-688"><a href="#cb64-688" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-689"><a href="#cb64-689" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(level <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-690"><a href="#cb64-690" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-691"><a href="#cb64-691" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-692"><a href="#cb64-692" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-693"><a href="#cb64-693" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-694"><a href="#cb64-694" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb64-695"><a href="#cb64-695" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-696"><a href="#cb64-696" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-697"><a href="#cb64-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-698"><a href="#cb64-698" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Especially, when completely ignoring the hierarchy in the gating information, I would expect to have more than 100% in <span class="in">`percelt_live_cell_sum`</span>, but this is not true for the specimens from the 2023 dataset</span>
<span id="cb64-699"><a href="#cb64-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-702"><a href="#cb64-702" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-703"><a href="#cb64-703" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>pbmc_cell_frequency <span class="sc">%&gt;%</span></span>
<span id="cb64-704"><a href="#cb64-704" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-705"><a href="#cb64-705" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(specimen_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb64-706"><a href="#cb64-706" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-707"><a href="#cb64-707" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-708"><a href="#cb64-708" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((celltype_meta <span class="sc">%&gt;%</span></span>
<span id="cb64-709"><a href="#cb64-709" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(cell_type_name, dataset, level) <span class="sc">%&gt;%</span></span>
<span id="cb64-710"><a href="#cb64-710" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">distinct</span>()),</span>
<span id="cb64-711"><a href="#cb64-711" aria-hidden="true" tabindex="-1"></a>                    <span class="at">by=</span><span class="fu">c</span>(<span class="st">"dataset"</span>, <span class="st">"cell_type_name"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-712"><a href="#cb64-712" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(specimen_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-713"><a href="#cb64-713" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">percent_live_cell_sum =</span> <span class="fu">sum</span>(percent_live_cell, <span class="at">na.rm=</span><span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-714"><a href="#cb64-714" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-715"><a href="#cb64-715" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-716"><a href="#cb64-716" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="fu">aes</span>(<span class="at">x=</span>dataset, <span class="at">y=</span>percent_live_cell_sum)) <span class="sc">+</span></span>
<span id="cb64-717"><a href="#cb64-717" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb64-718"><a href="#cb64-718" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-719"><a href="#cb64-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-720"><a href="#cb64-720" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Or what I use a list of predefined cell types of interest?</span>
<span id="cb64-721"><a href="#cb64-721" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-722"><a href="#cb64-722" aria-hidden="true" tabindex="-1"></a><span class="fu"># Features for T cell Activation Assay (AIM)</span></span>
<span id="cb64-723"><a href="#cb64-723" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-726"><a href="#cb64-726" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-727"><a href="#cb64-727" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_activation <span class="sc">%&gt;%</span></span>
<span id="cb64-728"><a href="#cb64-728" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(stimulation) <span class="sc">%&gt;%</span></span>
<span id="cb64-729"><a href="#cb64-729" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb64-730"><a href="#cb64-730" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-731"><a href="#cb64-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-734"><a href="#cb64-734" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-735"><a href="#cb64-735" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_activation <span class="sc">%&gt;%</span></span>
<span id="cb64-736"><a href="#cb64-736" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-737"><a href="#cb64-737" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>analyte_percentages), <span class="at">bins=</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb64-738"><a href="#cb64-738" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>stimulation) <span class="sc">+</span></span>
<span id="cb64-739"><a href="#cb64-739" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span>
<span id="cb64-740"><a href="#cb64-740" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-741"><a href="#cb64-741" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-742"><a href="#cb64-742" aria-hidden="true" tabindex="-1"></a><span class="fu"># Features for T cell Polarization Assay (FluoroSpot)</span></span>
<span id="cb64-743"><a href="#cb64-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-746"><a href="#cb64-746" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-747"><a href="#cb64-747" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb64-748"><a href="#cb64-748" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(protein_meta, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"protein_id"</span><span class="ot">=</span><span class="st">"uniprot_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-749"><a href="#cb64-749" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(protein_id, cytokine) <span class="sc">%&gt;%</span></span>
<span id="cb64-750"><a href="#cb64-750" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb64-751"><a href="#cb64-751" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-752"><a href="#cb64-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-755"><a href="#cb64-755" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-756"><a href="#cb64-756" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb64-757"><a href="#cb64-757" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(stimulation) <span class="sc">%&gt;%</span></span>
<span id="cb64-758"><a href="#cb64-758" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb64-759"><a href="#cb64-759" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-760"><a href="#cb64-760" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-761"><a href="#cb64-761" aria-hidden="true" tabindex="-1"></a>where</span>
<span id="cb64-762"><a href="#cb64-762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-763"><a href="#cb64-763" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>DMSO: "Negative" Control (solvent where peptides are dissolved in)</span>
<span id="cb64-764"><a href="#cb64-764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-765"><a href="#cb64-765" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>PHA (Phytohaemagglutinin): "Postive" Control (this should elicit strong T cell response?)</span>
<span id="cb64-766"><a href="#cb64-766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-767"><a href="#cb64-767" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>PT: Bordetella Pertussis Toxin</span>
<span id="cb64-768"><a href="#cb64-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-771"><a href="#cb64-771" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-772"><a href="#cb64-772" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb64-773"><a href="#cb64-773" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(protein_meta, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"protein_id"</span><span class="ot">=</span><span class="st">"uniprot_id"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb64-774"><a href="#cb64-774" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb64-775"><a href="#cb64-775" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>analyte_counts), <span class="at">bins=</span><span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb64-776"><a href="#cb64-776" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows=</span><span class="fu">vars</span>(stimulation), <span class="at">cols=</span><span class="fu">vars</span>(cytokine)) <span class="sc">+</span></span>
<span id="cb64-777"><a href="#cb64-777" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb64-778"><a href="#cb64-778" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span>
<span id="cb64-779"><a href="#cb64-779" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-780"><a href="#cb64-780" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: the assay has been normalized relative to DMSO</span></span>
<span id="cb64-781"><a href="#cb64-781" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb64-782"><a href="#cb64-782" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(stimulation<span class="sc">==</span><span class="st">"DMSO"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-783"><a href="#cb64-783" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(analyte_counts) <span class="sc">%&gt;%</span></span>
<span id="cb64-784"><a href="#cb64-784" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">near</span>(., <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-785"><a href="#cb64-785" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mean</span>()</span>
<span id="cb64-786"><a href="#cb64-786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-787"><a href="#cb64-787" aria-hidden="true" tabindex="-1"></a>exp_data<span class="sc">$</span>t_cell_polarization <span class="sc">%&gt;%</span></span>
<span id="cb64-788"><a href="#cb64-788" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(stimulation<span class="sc">!=</span><span class="st">"DMSO"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb64-789"><a href="#cb64-789" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(stimulation_protein_id) <span class="sc">%&gt;%</span></span>
<span id="cb64-790"><a href="#cb64-790" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb64-791"><a href="#cb64-791" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-792"><a href="#cb64-792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-793"><a href="#cb64-793" aria-hidden="true" tabindex="-1"></a><span class="fu"># Other Notes</span></span>
<span id="cb64-794"><a href="#cb64-794" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-795"><a href="#cb64-795" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Only from 2022 onwards do we have time-course baseline measurements, i.e. <span class="in">`-30, -14/15, 0`</span></span>
<span id="cb64-796"><a href="#cb64-796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-799"><a href="#cb64-799" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-800"><a href="#cb64-800" aria-hidden="true" tabindex="-1"></a>meta_data <span class="sc">%&gt;%</span></span>
<span id="cb64-801"><a href="#cb64-801" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(partition, dataset, planned_day_relative_to_boost) <span class="sc">%&gt;%</span></span>
<span id="cb64-802"><a href="#cb64-802" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">&lt;</span> <span class="dv">0</span>)</span>
<span id="cb64-803"><a href="#cb64-803" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-804"><a href="#cb64-804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-805"><a href="#cb64-805" aria-hidden="true" tabindex="-1"></a><span class="fu"># Appendix</span></span>
<span id="cb64-806"><a href="#cb64-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-807"><a href="#cb64-807" aria-hidden="true" tabindex="-1"></a>&lt;button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#collapseOne" &gt;Session Information&lt;/button&gt;&lt;div id="collapseOne" class="accordion-collapse collapse"&gt;&lt;div&gt;</span>
<span id="cb64-808"><a href="#cb64-808" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-811"><a href="#cb64-811" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb64-812"><a href="#cb64-812" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb64-813"><a href="#cb64-813" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb64-814"><a href="#cb64-814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-815"><a href="#cb64-815" aria-hidden="true" tabindex="-1"></a>&lt;/div&gt;&lt;/div&gt;</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>