<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.45">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Philipp Sven Lars Schäfer">
<meta name="dcterms.date" content="2024-10-14">

<title>MOFA Test – PSLS @ 3rd CMI-PB Challenge</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">PSLS @ 3rd CMI-PB Challenge</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/psl-schaefer"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://x.com/psl_schaefer"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/philipp-sl-schaefer/"> <i class="bi bi-linkedin" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="1">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#packages" id="toc-packages" class="nav-link active" data-scroll-target="#packages"><span class="header-section-number">1</span> Packages</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2</span> Data</a></li>
  <li><a href="#checks" id="toc-checks" class="nav-link" data-scroll-target="#checks"><span class="header-section-number">3</span> Checks</a></li>
  <li><a href="#mofa" id="toc-mofa" class="nav-link" data-scroll-target="#mofa"><span class="header-section-number">4</span> MOFA</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">MOFA Test</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Philipp Sven Lars Schäfer </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 14, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="packages" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Packages</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(reticulate)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  reticulate<span class="sc">::</span><span class="fu">use_condaenv</span>(<span class="st">"scanpy"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(MOFA2)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Data</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#exp_data &lt;- read_harmonized_experimental_data(input_dir)</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(meta_data, exp_data, gene_meta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>pbmc_cell_frequency | Removed 550 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 6931 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 495 specimens because missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>pbmc_cell_frequency | Removed 56 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 48 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 234 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>t_cell_polarization | Removed 3 features because not in feature subset</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 300 features because qc warning</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_ab_titer | Removed 10540 measurements because wrong unit used</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_olink | Removed 2400 measurements because wrong unit used</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>plasma_cytokine_concentration_by_legendplex | Removed 8 because specimen is outlier</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>wide_exp_data <span class="ot">&lt;-</span> <span class="fu">generate_wide_experimental_data</span>(<span class="at">experimental_data=</span>exp_data, <span class="at">impute=</span><span class="cn">NULL</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="checks" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Checks</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># only works due to filtering step above</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="sc">~</span> <span class="fu">all</span>(.x<span class="sc">$</span>specimen_id <span class="sc">%in%</span> meta_data<span class="sc">$</span>specimen_id)))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co"># no subject is recorded in more than one dataset</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(n) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="co"># we have the baseline specimen (planned_day_relative_to_boost = 0) for every subject</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, planned_day_relative_to_boost, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_baseline =</span> (planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present =</span> <span class="fu">any</span>(is_baseline),</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>                   <span class="at">dataset =</span> <span class="fu">first</span>(dataset)) <span class="sc">%&gt;%</span></span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present_frac =</span> <span class="fu">mean</span>(baseline_present)) <span class="sc">%&gt;%</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(baseline_present_frac) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="cf">function</span>(df) {</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="st">"unit"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(df)) {</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">length</span>(<span class="fu">unique</span>(df[[<span class="st">"unit"</span>]])) <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="cn">TRUE</span>)</span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="mofa" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> MOFA</h1>
<ul>
<li><p>https://biofam.github.io/MOFA2/index.html</p></li>
<li><p>https://raw.githack.com/bioFAM/MOFA2_tutorials/master/R_tutorials/getting_started_R.html</p></li>
<li><p>https://raw.githack.com/bioFAM/MOFA2_tutorials/master/R_tutorials/downstream_analysis.html</p></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># filter genes and normalize the gene expression data</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>min_counts <span class="ot">&lt;-</span> <span class="dv">200</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>min_samples <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>genes_to_keep <span class="ot">&lt;-</span> <span class="fu">colnames</span>(wide_exp_data<span class="sc">$</span>pbmc_gene_expression)[</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  (<span class="fu">colSums</span>(wide_exp_data<span class="sc">$</span>pbmc_gene_expression) <span class="sc">&gt;=</span> min_counts) <span class="sc">&amp;</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>    (<span class="fu">colSums</span>(wide_exp_data<span class="sc">$</span>pbmc_gene_expression <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">&gt;=</span> min_samples)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>target_sum <span class="ot">&lt;-</span> <span class="fl">1e6</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>pbmc_gene_expression_norm <span class="ot">&lt;-</span> </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  wide_exp_data<span class="sc">$</span>pbmc_gene_expression <span class="sc">/</span> <span class="fu">rowSums</span>(wide_exp_data<span class="sc">$</span>pbmc_gene_expression) <span class="sc">*</span> target_sum</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(dplyr<span class="sc">::</span><span class="fu">near</span>(<span class="fu">rowSums</span>(pbmc_gene_expression_norm), target_sum, <span class="at">tol=</span><span class="fl">0.1</span>)))</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>pbmc_gene_expression_norm <span class="ot">&lt;-</span> pbmc_gene_expression_norm[ , genes_to_keep]</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>pbmc_gene_expression_log <span class="ot">&lt;-</span> <span class="fu">log1p</span>(pbmc_gene_expression_norm)</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>wide_exp_data<span class="sc">$</span>pbmc_gene_expression <span class="ot">&lt;-</span> pbmc_gene_expression_log</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>baseline_specimen <span class="ot">&lt;-</span> meta_data <span class="sc">%&gt;%</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(specimen_id)</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>assay_specimen <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(wide_exp_data, <span class="sc">~</span> <span class="fu">rownames</span>(.x)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>intersect_specimen <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">intersect</span>(baseline_specimen, assay_specimen)</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>group_vec <span class="ot">&lt;-</span> (meta_data <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"specimen_id"</span>))[intersect_specimen, ]<span class="sc">$</span>dataset</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>group_vec <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(group_vec, <span class="st">"_dataset"</span>, <span class="st">"_d"</span>)</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>mofa_data <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">imap</span>(wide_exp_data, <span class="cf">function</span>(mtx, modality) {</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mtx &lt;- wide_exp_data$pbmc_cell_frequency</span></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>  specimen_in_modality <span class="ot">&lt;-</span> intersect_specimen[intersect_specimen <span class="sc">%in%</span> <span class="fu">rownames</span>(mtx)]</span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>  out_mtx <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">data=</span><span class="cn">NA</span>, <span class="at">nrow=</span><span class="fu">ncol</span>(mtx), <span class="at">ncol=</span><span class="fu">length</span>(intersect_specimen))</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(out_mtx) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(mtx)</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(out_mtx) <span class="ot">&lt;-</span> intersect_specimen</span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>  out_mtx[, specimen_in_modality] <span class="ot">&lt;-</span> <span class="fu">t</span>(mtx[specimen_in_modality, ])</span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(modality, <span class="st">" | Fraction of NAs: "</span>, <span class="fu">mean</span>(<span class="fu">is.na</span>(out_mtx))))</span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(out_mtx)</span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "pbmc_cell_frequency | Fraction of NAs: 0.19205298013245"
[1] "pbmc_gene_expression | Fraction of NAs: 0.033112582781457"
[1] "plasma_ab_titer | Fraction of NAs: 0.28476821192053"
[1] "plasma_cytokine_concentration_by_legendplex | Fraction of NAs: 0.317880794701987"
[1] "plasma_cytokine_concentration_by_olink | Fraction of NAs: 0.426098735701385"
[1] "t_cell_activation | Fraction of NAs: 0.339403973509934"
[1] "t_cell_polarization | Fraction of NAs: 0.384105960264901"</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>MOFAobject <span class="ot">&lt;-</span> MOFA2<span class="sc">::</span><span class="fu">create_mofa_from_matrix</span>(mofa_data, <span class="at">groups=</span>group_vec)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .rename_duplicated_features(object): There are duplicated features names across different views. We will add the suffix *_view* only for those features 
            Example: if you have both TP53 in mRNA and mutation data it will be renamed to TP53_mRNA, TP53_mutation</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_data_overview</span>(MOFAobject) <span class="sc">+</span> ggdark<span class="sc">::</span><span class="fu">dark_mode</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Inverted geom defaults of fill and color/colour.
To change them back, use invert_geom_defaults().</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="mofa_test_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="1344"></p>
</figure>
</div>
</div>
</div>
<ul>
<li>Define data options
<ul>
<li><code>scale_groups</code>: if groups have different ranges/variances, it is good practice to scale each group to unit variance. Default is FALSE</li>
<li><code>scale_views</code>: if views have different ranges/variances, it is good practice to scale each view to unit variance. Default is FALSE</li>
</ul></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>data_opts <span class="ot">&lt;-</span> <span class="fu">get_default_data_options</span>(MOFAobject)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(data_opts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$scale_views
[1] FALSE

$scale_groups
[1] FALSE

$center_groups
[1] TRUE

$use_float32
[1] TRUE

$views
[1] "pbmc_cell_frequency"                        
[2] "pbmc_gene_expression"                       
[3] "plasma_ab_titer"                            
[4] "plasma_cytokine_concentration_by_legendplex"
[5] "plasma_cytokine_concentration_by_olink"     
[6] "t_cell_activation"                          
[7] "t_cell_polarization"                        

$groups
[1] "2020_d" "2021_d" "2022_d" "2023_d"</code></pre>
</div>
</div>
<ul>
<li>Define model options
<ul>
<li><code>num_factors</code>: number of factors</li>
<li><code>likelihoods</code>: likelihood per view (options are “gaussian”, “poisson”, “bernoulli”). By default they are learnt automatically. We advise users to use “gaussian” whenever possible!</li>
<li><code>spikeslab_factors</code>: use spike-slab sparsity prior in the factors? default is FALSE.</li>
<li><code>spikeslab_weights</code>: use spike-slab sparsity prior in the weights? default is TRUE.</li>
<li><code>ard_factors</code>: use ARD prior in the factors? Default is TRUE if using multiple groups.</li>
<li><code>ard_weights</code>: use ARD prior in the weights? Default is TRUE if using multiple views.</li>
</ul></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>model_opts <span class="ot">&lt;-</span> <span class="fu">get_default_model_options</span>(MOFAobject)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(model_opts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$likelihoods
                        pbmc_cell_frequency 
                                 "gaussian" 
                       pbmc_gene_expression 
                                 "gaussian" 
                            plasma_ab_titer 
                                 "gaussian" 
plasma_cytokine_concentration_by_legendplex 
                                 "gaussian" 
     plasma_cytokine_concentration_by_olink 
                                 "gaussian" 
                          t_cell_activation 
                                 "gaussian" 
                        t_cell_polarization 
                                 "gaussian" 

$num_factors
[1] 15

$spikeslab_factors
[1] FALSE

$spikeslab_weights
[1] FALSE

$ard_factors
[1] TRUE

$ard_weights
[1] TRUE</code></pre>
</div>
</div>
<ul>
<li>Define train options
<ul>
<li><code>maxiter</code>: number of iterations. Default is 1000.</li>
<li><code>convergence_mode</code>: “fast”, “medium”, “slow”. For exploration, the fast mode is good enough.</li>
<li><code>startELBO</code>: initial iteration to compute the ELBO (the objective function used to assess convergence).</li>
<li><code>freqELBO</code>: frequency of computations of the ELBO.</li>
<li><code>gpu_mode</code>: use GPU mode? (needs cupy installed and a functional GPU).</li>
<li><code>stochastic</code>: use stochastic inference? (default is FALSE).</li>
<li><code>verbose</code>: verbose mode?</li>
<li><code>seed</code>: random seed</li>
</ul></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>train_opts <span class="ot">&lt;-</span> <span class="fu">get_default_training_options</span>(MOFAobject)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(train_opts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$maxiter
[1] 1000

$convergence_mode
[1] "fast"

$drop_factor_threshold
[1] -1

$verbose
[1] FALSE

$startELBO
[1] 1

$freqELBO
[1] 5</code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>MOFAobject <span class="ot">&lt;-</span> <span class="fu">prepare_mofa</span>(</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">object =</span> MOFAobject,</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data_options =</span> data_opts,</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">model_options =</span> model_opts,</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">training_options =</span> train_opts</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in prepare_mofa(object = MOFAobject, data_options = data_opts,
model_options = model_opts, : Some view(s) have less than 15 features, MOFA
will have little power to to learn meaningful factors for these view(s)....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in prepare_mofa(object = MOFAobject, data_options = data_opts,
model_options = model_opts, : Some view(s) have a lot of features, it is
recommended to perform a more stringent feature selection before creating the
MOFA object....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
# Multi-group mode requested.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
This is an advanced option, if this is the first time that you are running MOFA, we suggest that you try do some exploration first without specifying groups. Two important remarks:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
 - The aim of the multi-group framework is to identify the sources of variability *within* the groups. If your aim is to find a factor that 'separates' the groups, you DO NOT want to use the multi-group framework. Please see the FAQ on the MOFA2 webpage.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
 - It is important to account for the group effect before selecting highly variable features (HVFs). We suggest that either you calculate HVFs per group and then take the union, or regress out the group effect before HVF selection</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Checking data options...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Checking training options...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Checking model options...</code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>outfile <span class="ot">=</span> <span class="fu">file.path</span>(<span class="fu">getwd</span>(), <span class="st">"model.hdf5"</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>MOFAobject.trained <span class="ot">&lt;-</span> <span class="fu">run_mofa</span>(MOFAobject, outfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Output file /Users/pschafer/Projects/cmi_pb_third_challenge/eda/model.hdf5 already exists, it will be replaced</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Connecting to the mofapy2 python package using reticulate (use_basilisk = FALSE)... 
    Please make sure to manually specify the right python binary when loading R with reticulate::use_python(..., force=TRUE) or the right conda environment with reticulate::use_condaenv(..., force=TRUE)
    If you prefer to let us automatically install a conda environment with 'mofapy2' installed using the 'basilisk' package, please use the argument 'use_basilisk = TRUE'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in run_mofa(MOFAobject, outfile): The latest mofapy2 version is 0.7.0,
you are using 0.7.1. Please upgrade with 'pip install mofapy2'</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
        #########################################################
        ###           __  __  ____  ______                    ### 
        ###          |  \/  |/ __ \|  ____/\    _             ### 
        ###          | \  / | |  | | |__ /  \ _| |_           ### 
        ###          | |\/| | |  | |  __/ /\ \_   _|          ###
        ###          | |  | | |__| | | / ____ \|_|            ###
        ###          |_|  |_|\____/|_|/_/    \_\              ###
        ###                                                   ### 
        ######################################################### 
       
 
        
use_float32 set to True: replacing float64 arrays by float32 arrays to speed up computations...

Successfully loaded view='pbmc_cell_frequency' group='2020_d' with N=40 samples and D=10 features...
Successfully loaded view='pbmc_cell_frequency' group='2021_d' with N=36 samples and D=10 features...
Successfully loaded view='pbmc_cell_frequency' group='2022_d' with N=21 samples and D=10 features...
Successfully loaded view='pbmc_cell_frequency' group='2023_d' with N=54 samples and D=10 features...
Successfully loaded view='pbmc_gene_expression' group='2020_d' with N=40 samples and D=30441 features...
Successfully loaded view='pbmc_gene_expression' group='2021_d' with N=36 samples and D=30441 features...
Successfully loaded view='pbmc_gene_expression' group='2022_d' with N=21 samples and D=30441 features...
Successfully loaded view='pbmc_gene_expression' group='2023_d' with N=54 samples and D=30441 features...
Successfully loaded view='plasma_ab_titer' group='2020_d' with N=40 samples and D=35 features...
Successfully loaded view='plasma_ab_titer' group='2021_d' with N=36 samples and D=35 features...
Successfully loaded view='plasma_ab_titer' group='2022_d' with N=21 samples and D=35 features...
Successfully loaded view='plasma_ab_titer' group='2023_d' with N=54 samples and D=35 features...
Successfully loaded view='plasma_cytokine_concentration_by_legendplex' group='2020_d' with N=40 samples and D=14 features...
Successfully loaded view='plasma_cytokine_concentration_by_legendplex' group='2021_d' with N=36 samples and D=14 features...
Successfully loaded view='plasma_cytokine_concentration_by_legendplex' group='2022_d' with N=21 samples and D=14 features...
Successfully loaded view='plasma_cytokine_concentration_by_legendplex' group='2023_d' with N=54 samples and D=14 features...
Successfully loaded view='plasma_cytokine_concentration_by_olink' group='2020_d' with N=40 samples and D=44 features...
Successfully loaded view='plasma_cytokine_concentration_by_olink' group='2021_d' with N=36 samples and D=44 features...
Successfully loaded view='plasma_cytokine_concentration_by_olink' group='2022_d' with N=21 samples and D=44 features...
Successfully loaded view='plasma_cytokine_concentration_by_olink' group='2023_d' with N=54 samples and D=44 features...
Successfully loaded view='t_cell_activation' group='2020_d' with N=40 samples and D=4 features...
Successfully loaded view='t_cell_activation' group='2021_d' with N=36 samples and D=4 features...
Successfully loaded view='t_cell_activation' group='2022_d' with N=21 samples and D=4 features...
Successfully loaded view='t_cell_activation' group='2023_d' with N=54 samples and D=4 features...
Successfully loaded view='t_cell_polarization' group='2020_d' with N=40 samples and D=6 features...
Successfully loaded view='t_cell_polarization' group='2021_d' with N=36 samples and D=6 features...
Successfully loaded view='t_cell_polarization' group='2022_d' with N=21 samples and D=6 features...
Successfully loaded view='t_cell_polarization' group='2023_d' with N=54 samples and D=6 features...


Model options:
- Automatic Relevance Determination prior on the factors: True
- Automatic Relevance Determination prior on the weights: True
- Spike-and-slab prior on the factors: False
- Spike-and-slab prior on the weights: False
Likelihoods:
- View 0 (pbmc_cell_frequency): gaussian
- View 1 (pbmc_gene_expression): gaussian
- View 2 (plasma_ab_titer): gaussian
- View 3 (plasma_cytokine_concentration_by_legendplex): gaussian
- View 4 (plasma_cytokine_concentration_by_olink): gaussian
- View 5 (t_cell_activation): gaussian
- View 6 (t_cell_polarization): gaussian



Warning: some view(s) have less than 15 features, MOFA won't be able to learn meaningful factors for these view(s)...



######################################
## Training the model with seed 42 ##
######################################


ELBO before training: -18287496579.50 

Iteration 1: time=0.38, ELBO=-5474940.22, deltaELBO=18282021639.276 (99.97006184%), Factors=15
Iteration 2: time=0.39, Factors=15
Iteration 3: time=0.40, Factors=15
Iteration 4: time=0.44, Factors=15
Iteration 5: time=0.39, Factors=15
Iteration 6: time=0.39, ELBO=716274.38, deltaELBO=6191214.600 (0.03385491%), Factors=15
Iteration 7: time=0.37, Factors=15
Iteration 8: time=0.39, Factors=15
Iteration 9: time=0.38, Factors=15
Iteration 10: time=0.37, Factors=15
Iteration 11: time=0.42, ELBO=860389.15, deltaELBO=144114.775 (0.00078805%), Factors=15
Iteration 12: time=0.39, Factors=15
Iteration 13: time=0.37, Factors=15
Iteration 14: time=0.38, Factors=15
Iteration 15: time=0.37, Factors=15
Iteration 16: time=0.40, ELBO=886660.34, deltaELBO=26271.186 (0.00014366%), Factors=15
Iteration 17: time=0.38, Factors=15
Iteration 18: time=0.38, Factors=15
Iteration 19: time=0.37, Factors=15
Iteration 20: time=0.37, Factors=15
Iteration 21: time=0.38, ELBO=894420.16, deltaELBO=7759.820 (0.00004243%), Factors=15

Converged!



#######################
## Training finished ##
#######################


Warning: Output file /Users/pschafer/Projects/cmi_pb_third_challenge/eda/model.hdf5 already exists, it will be replaced
Saving model in /Users/pschafer/Projects/cmi_pb_third_challenge/eda/model.hdf5...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .quality_control(object, verbose = verbose): Factor(s) 2 are strongly correlated with the total number of expressed features for at least one of your omics. Such factors appear when there are differences in the total 'levels' between your samples, *sometimes* because of poor normalisation in the preprocessing steps.</code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">load_model</span>(outfile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .quality_control(object, verbose = verbose): Factor(s) 2 are strongly correlated with the total number of expressed features for at least one of your omics. Such factors appear when there are differences in the total 'levels' between your samples, *sometimes* because of poor normalisation in the preprocessing steps.</code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_variance_explained</span>(model, <span class="at">x=</span><span class="st">"view"</span>, <span class="at">y=</span><span class="st">"factor"</span>) <span class="sc">+</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="mofa_test_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="1344"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_variance_explained</span>(model, <span class="at">x=</span><span class="st">"group"</span>, <span class="at">y=</span><span class="st">"factor"</span>, <span class="at">plot_total =</span> T)[[<span class="dv">2</span>]] <span class="sc">+</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 5 rows containing missing values or values outside the scale range
(`geom_bar()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="mofa_test_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="1536"></p>
</figure>
</div>
</div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb50" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "MOFA Test"</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Philipp Sven Lars Schäfer"</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "`r format(Sys.time(), '%d %B, %Y')`"</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> source</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="an">engine:</span><span class="co"> knitr</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a><span class="fu"># Packages</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick)</span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(reticulate)</span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a>  reticulate<span class="sc">::</span><span class="fu">use_condaenv</span>(<span class="st">"scanpy"</span>)</span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(MOFA2)</span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data</span></span>
<span id="cb50-30"><a href="#cb50-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span>
<span id="cb50-35"><a href="#cb50-35" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-36"><a href="#cb50-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-39"><a href="#cb50-39" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-40"><a href="#cb50-40" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb50-41"><a href="#cb50-41" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb50-42"><a href="#cb50-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-43"><a href="#cb50-43" aria-hidden="true" tabindex="-1"></a><span class="co">#exp_data &lt;- read_harmonized_experimental_data(input_dir)</span></span>
<span id="cb50-44"><a href="#cb50-44" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb50-45"><a href="#cb50-45" aria-hidden="true" tabindex="-1"></a>exp_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(meta_data, exp_data, gene_meta)</span>
<span id="cb50-46"><a href="#cb50-46" aria-hidden="true" tabindex="-1"></a>wide_exp_data <span class="ot">&lt;-</span> <span class="fu">generate_wide_experimental_data</span>(<span class="at">experimental_data=</span>exp_data, <span class="at">impute=</span><span class="cn">NULL</span>)</span>
<span id="cb50-47"><a href="#cb50-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-48"><a href="#cb50-48" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb50-49"><a href="#cb50-49" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta</span>(input_dir)</span>
<span id="cb50-50"><a href="#cb50-50" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span>
<span id="cb50-51"><a href="#cb50-51" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-52"><a href="#cb50-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-53"><a href="#cb50-53" aria-hidden="true" tabindex="-1"></a><span class="fu"># Checks</span></span>
<span id="cb50-54"><a href="#cb50-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-57"><a href="#cb50-57" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-58"><a href="#cb50-58" aria-hidden="true" tabindex="-1"></a><span class="co"># only works due to filtering step above</span></span>
<span id="cb50-59"><a href="#cb50-59" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb50-60"><a href="#cb50-60" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="sc">~</span> <span class="fu">all</span>(.x<span class="sc">$</span>specimen_id <span class="sc">%in%</span> meta_data<span class="sc">$</span>specimen_id)))</span>
<span id="cb50-61"><a href="#cb50-61" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-62"><a href="#cb50-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-63"><a href="#cb50-63" aria-hidden="true" tabindex="-1"></a><span class="co"># no subject is recorded in more than one dataset</span></span>
<span id="cb50-64"><a href="#cb50-64" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb50-65"><a href="#cb50-65" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb50-66"><a href="#cb50-66" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb50-67"><a href="#cb50-67" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(n) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-68"><a href="#cb50-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb50-69"><a href="#cb50-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb50-70"><a href="#cb50-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-71"><a href="#cb50-71" aria-hidden="true" tabindex="-1"></a><span class="co"># we have the baseline specimen (planned_day_relative_to_boost = 0) for every subject</span></span>
<span id="cb50-72"><a href="#cb50-72" aria-hidden="true" tabindex="-1"></a>(meta_data <span class="sc">%&gt;%</span></span>
<span id="cb50-73"><a href="#cb50-73" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, planned_day_relative_to_boost, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb50-74"><a href="#cb50-74" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb50-75"><a href="#cb50-75" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">is_baseline =</span> (planned_day_relative_to_boost<span class="sc">==</span><span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb50-76"><a href="#cb50-76" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(subject_id) <span class="sc">%&gt;%</span></span>
<span id="cb50-77"><a href="#cb50-77" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present =</span> <span class="fu">any</span>(is_baseline),</span>
<span id="cb50-78"><a href="#cb50-78" aria-hidden="true" tabindex="-1"></a>                   <span class="at">dataset =</span> <span class="fu">first</span>(dataset)) <span class="sc">%&gt;%</span></span>
<span id="cb50-79"><a href="#cb50-79" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(dataset) <span class="sc">%&gt;%</span></span>
<span id="cb50-80"><a href="#cb50-80" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">baseline_present_frac =</span> <span class="fu">mean</span>(baseline_present)) <span class="sc">%&gt;%</span></span>
<span id="cb50-81"><a href="#cb50-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(baseline_present_frac) <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-82"><a href="#cb50-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">all</span>() <span class="sc">%&gt;%</span></span>
<span id="cb50-83"><a href="#cb50-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>()</span>
<span id="cb50-84"><a href="#cb50-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-85"><a href="#cb50-85" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(</span>
<span id="cb50-86"><a href="#cb50-86" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_lgl</span>(exp_data, <span class="cf">function</span>(df) {</span>
<span id="cb50-87"><a href="#cb50-87" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="st">"unit"</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(df)) {</span>
<span id="cb50-88"><a href="#cb50-88" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">length</span>(<span class="fu">unique</span>(df[[<span class="st">"unit"</span>]])) <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb50-89"><a href="#cb50-89" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb50-90"><a href="#cb50-90" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(<span class="cn">TRUE</span>)</span>
<span id="cb50-91"><a href="#cb50-91" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb50-92"><a href="#cb50-92" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb50-93"><a href="#cb50-93" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb50-94"><a href="#cb50-94" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-95"><a href="#cb50-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-96"><a href="#cb50-96" aria-hidden="true" tabindex="-1"></a><span class="fu"># MOFA</span></span>
<span id="cb50-97"><a href="#cb50-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-98"><a href="#cb50-98" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>https://biofam.github.io/MOFA2/index.html</span>
<span id="cb50-99"><a href="#cb50-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-100"><a href="#cb50-100" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>https://raw.githack.com/bioFAM/MOFA2_tutorials/master/R_tutorials/getting_started_R.html</span>
<span id="cb50-101"><a href="#cb50-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-102"><a href="#cb50-102" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>https://raw.githack.com/bioFAM/MOFA2_tutorials/master/R_tutorials/downstream_analysis.html</span>
<span id="cb50-103"><a href="#cb50-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-104"><a href="#cb50-104" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=14, fig.height=10}</span></span>
<span id="cb50-105"><a href="#cb50-105" aria-hidden="true" tabindex="-1"></a><span class="in"># filter genes and normalize the gene expression data</span></span>
<span id="cb50-106"><a href="#cb50-106" aria-hidden="true" tabindex="-1"></a><span class="in">min_counts &lt;- 200</span></span>
<span id="cb50-107"><a href="#cb50-107" aria-hidden="true" tabindex="-1"></a><span class="in">min_samples &lt;- 20</span></span>
<span id="cb50-108"><a href="#cb50-108" aria-hidden="true" tabindex="-1"></a><span class="in">genes_to_keep &lt;- colnames(wide_exp_data$pbmc_gene_expression)[</span></span>
<span id="cb50-109"><a href="#cb50-109" aria-hidden="true" tabindex="-1"></a><span class="in">  (colSums(wide_exp_data$pbmc_gene_expression) &gt;= min_counts) &amp;</span></span>
<span id="cb50-110"><a href="#cb50-110" aria-hidden="true" tabindex="-1"></a><span class="in">    (colSums(wide_exp_data$pbmc_gene_expression &gt; 0) &gt;= min_samples)</span></span>
<span id="cb50-111"><a href="#cb50-111" aria-hidden="true" tabindex="-1"></a><span class="in">]</span></span>
<span id="cb50-112"><a href="#cb50-112" aria-hidden="true" tabindex="-1"></a><span class="in">target_sum &lt;- 1e6</span></span>
<span id="cb50-113"><a href="#cb50-113" aria-hidden="true" tabindex="-1"></a><span class="in">pbmc_gene_expression_norm &lt;- </span></span>
<span id="cb50-114"><a href="#cb50-114" aria-hidden="true" tabindex="-1"></a><span class="in">  wide_exp_data$pbmc_gene_expression / rowSums(wide_exp_data$pbmc_gene_expression) * target_sum</span></span>
<span id="cb50-115"><a href="#cb50-115" aria-hidden="true" tabindex="-1"></a><span class="in">stopifnot(all(dplyr::near(rowSums(pbmc_gene_expression_norm), target_sum, tol=0.1)))</span></span>
<span id="cb50-116"><a href="#cb50-116" aria-hidden="true" tabindex="-1"></a><span class="in">pbmc_gene_expression_norm &lt;- pbmc_gene_expression_norm[ , genes_to_keep]</span></span>
<span id="cb50-117"><a href="#cb50-117" aria-hidden="true" tabindex="-1"></a><span class="in">pbmc_gene_expression_log &lt;- log1p(pbmc_gene_expression_norm)</span></span>
<span id="cb50-118"><a href="#cb50-118" aria-hidden="true" tabindex="-1"></a><span class="in">wide_exp_data$pbmc_gene_expression &lt;- pbmc_gene_expression_log</span></span>
<span id="cb50-119"><a href="#cb50-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-120"><a href="#cb50-120" aria-hidden="true" tabindex="-1"></a><span class="in">baseline_specimen &lt;- meta_data %&gt;%</span></span>
<span id="cb50-121"><a href="#cb50-121" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::filter(planned_day_relative_to_boost==0) %&gt;%</span></span>
<span id="cb50-122"><a href="#cb50-122" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::pull(specimen_id)</span></span>
<span id="cb50-123"><a href="#cb50-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-124"><a href="#cb50-124" aria-hidden="true" tabindex="-1"></a><span class="in">assay_specimen &lt;- purrr::map(wide_exp_data, ~ rownames(.x)) %&gt;% </span></span>
<span id="cb50-125"><a href="#cb50-125" aria-hidden="true" tabindex="-1"></a><span class="in">  unlist() %&gt;% </span></span>
<span id="cb50-126"><a href="#cb50-126" aria-hidden="true" tabindex="-1"></a><span class="in">  unique()</span></span>
<span id="cb50-127"><a href="#cb50-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-128"><a href="#cb50-128" aria-hidden="true" tabindex="-1"></a><span class="in">intersect_specimen &lt;- dplyr::intersect(baseline_specimen, assay_specimen)</span></span>
<span id="cb50-129"><a href="#cb50-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-130"><a href="#cb50-130" aria-hidden="true" tabindex="-1"></a><span class="in">group_vec &lt;- (meta_data %&gt;% column_to_rownames(var="specimen_id"))[intersect_specimen, ]$dataset</span></span>
<span id="cb50-131"><a href="#cb50-131" aria-hidden="true" tabindex="-1"></a><span class="in">group_vec &lt;- str_replace(group_vec, "_dataset", "_d")</span></span>
<span id="cb50-132"><a href="#cb50-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-133"><a href="#cb50-133" aria-hidden="true" tabindex="-1"></a><span class="in">mofa_data &lt;- purrr::imap(wide_exp_data, function(mtx, modality) {</span></span>
<span id="cb50-134"><a href="#cb50-134" aria-hidden="true" tabindex="-1"></a><span class="in">  # mtx &lt;- wide_exp_data$pbmc_cell_frequency</span></span>
<span id="cb50-135"><a href="#cb50-135" aria-hidden="true" tabindex="-1"></a><span class="in">  specimen_in_modality &lt;- intersect_specimen[intersect_specimen %in% rownames(mtx)]</span></span>
<span id="cb50-136"><a href="#cb50-136" aria-hidden="true" tabindex="-1"></a><span class="in">  out_mtx &lt;- matrix(data=NA, nrow=ncol(mtx), ncol=length(intersect_specimen))</span></span>
<span id="cb50-137"><a href="#cb50-137" aria-hidden="true" tabindex="-1"></a><span class="in">  rownames(out_mtx) &lt;- colnames(mtx)</span></span>
<span id="cb50-138"><a href="#cb50-138" aria-hidden="true" tabindex="-1"></a><span class="in">  colnames(out_mtx) &lt;- intersect_specimen</span></span>
<span id="cb50-139"><a href="#cb50-139" aria-hidden="true" tabindex="-1"></a><span class="in">  out_mtx[, specimen_in_modality] &lt;- t(mtx[specimen_in_modality, ])</span></span>
<span id="cb50-140"><a href="#cb50-140" aria-hidden="true" tabindex="-1"></a><span class="in">  print(paste0(modality, " | Fraction of NAs: ", mean(is.na(out_mtx))))</span></span>
<span id="cb50-141"><a href="#cb50-141" aria-hidden="true" tabindex="-1"></a><span class="in">  return(out_mtx)</span></span>
<span id="cb50-142"><a href="#cb50-142" aria-hidden="true" tabindex="-1"></a><span class="in">})</span></span>
<span id="cb50-143"><a href="#cb50-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-144"><a href="#cb50-144" aria-hidden="true" tabindex="-1"></a><span class="in">MOFAobject &lt;- MOFA2::create_mofa_from_matrix(mofa_data, groups=group_vec)</span></span>
<span id="cb50-145"><a href="#cb50-145" aria-hidden="true" tabindex="-1"></a><span class="in">plot_data_overview(MOFAobject) + ggdark::dark_mode()</span></span>
<span id="cb50-146"><a href="#cb50-146" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-147"><a href="#cb50-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-148"><a href="#cb50-148" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Define data options</span>
<span id="cb50-149"><a href="#cb50-149" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`scale_groups`</span>: if groups have different ranges/variances, it is good practice to scale each group to unit variance. Default is FALSE</span>
<span id="cb50-150"><a href="#cb50-150" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`scale_views`</span>: if views have different ranges/variances, it is good practice to scale each view to unit variance. Default is FALSE</span>
<span id="cb50-151"><a href="#cb50-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-154"><a href="#cb50-154" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-155"><a href="#cb50-155" aria-hidden="true" tabindex="-1"></a>data_opts <span class="ot">&lt;-</span> <span class="fu">get_default_data_options</span>(MOFAobject)</span>
<span id="cb50-156"><a href="#cb50-156" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(data_opts)</span>
<span id="cb50-157"><a href="#cb50-157" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-158"><a href="#cb50-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-159"><a href="#cb50-159" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Define model options</span>
<span id="cb50-160"><a href="#cb50-160" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`num_factors`</span>: number of factors</span>
<span id="cb50-161"><a href="#cb50-161" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`likelihoods`</span>: likelihood per view (options are “gaussian”, “poisson”, “bernoulli”). By default they are learnt automatically. We advise users to use “gaussian” whenever possible!</span>
<span id="cb50-162"><a href="#cb50-162" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`spikeslab_factors`</span>: use spike-slab sparsity prior in the factors? default is FALSE.</span>
<span id="cb50-163"><a href="#cb50-163" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`spikeslab_weights`</span>: use spike-slab sparsity prior in the weights? default is TRUE.</span>
<span id="cb50-164"><a href="#cb50-164" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`ard_factors`</span>: use ARD prior in the factors? Default is TRUE if using multiple groups.</span>
<span id="cb50-165"><a href="#cb50-165" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`ard_weights`</span>: use ARD prior in the weights? Default is TRUE if using multiple views.</span>
<span id="cb50-166"><a href="#cb50-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-169"><a href="#cb50-169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-170"><a href="#cb50-170" aria-hidden="true" tabindex="-1"></a>model_opts <span class="ot">&lt;-</span> <span class="fu">get_default_model_options</span>(MOFAobject)</span>
<span id="cb50-171"><a href="#cb50-171" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(model_opts)</span>
<span id="cb50-172"><a href="#cb50-172" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-173"><a href="#cb50-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-174"><a href="#cb50-174" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Define train options</span>
<span id="cb50-175"><a href="#cb50-175" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`maxiter`</span>: number of iterations. Default is 1000.</span>
<span id="cb50-176"><a href="#cb50-176" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`convergence_mode`</span>: “fast”, “medium”, “slow”. For exploration, the fast mode is good enough.</span>
<span id="cb50-177"><a href="#cb50-177" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`startELBO`</span>: initial iteration to compute the ELBO (the objective function used to assess convergence).</span>
<span id="cb50-178"><a href="#cb50-178" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`freqELBO`</span>: frequency of computations of the ELBO.</span>
<span id="cb50-179"><a href="#cb50-179" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`gpu_mode`</span>: use GPU mode? (needs cupy installed and a functional GPU).</span>
<span id="cb50-180"><a href="#cb50-180" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`stochastic`</span>: use stochastic inference? (default is FALSE).</span>
<span id="cb50-181"><a href="#cb50-181" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`verbose`</span>: verbose mode?</span>
<span id="cb50-182"><a href="#cb50-182" aria-hidden="true" tabindex="-1"></a><span class="ss">  * </span><span class="in">`seed`</span>: random seed</span>
<span id="cb50-183"><a href="#cb50-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-186"><a href="#cb50-186" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-187"><a href="#cb50-187" aria-hidden="true" tabindex="-1"></a>train_opts <span class="ot">&lt;-</span> <span class="fu">get_default_training_options</span>(MOFAobject)</span>
<span id="cb50-188"><a href="#cb50-188" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(train_opts)</span>
<span id="cb50-189"><a href="#cb50-189" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-190"><a href="#cb50-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-193"><a href="#cb50-193" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-194"><a href="#cb50-194" aria-hidden="true" tabindex="-1"></a>MOFAobject <span class="ot">&lt;-</span> <span class="fu">prepare_mofa</span>(</span>
<span id="cb50-195"><a href="#cb50-195" aria-hidden="true" tabindex="-1"></a>  <span class="at">object =</span> MOFAobject,</span>
<span id="cb50-196"><a href="#cb50-196" aria-hidden="true" tabindex="-1"></a>  <span class="at">data_options =</span> data_opts,</span>
<span id="cb50-197"><a href="#cb50-197" aria-hidden="true" tabindex="-1"></a>  <span class="at">model_options =</span> model_opts,</span>
<span id="cb50-198"><a href="#cb50-198" aria-hidden="true" tabindex="-1"></a>  <span class="at">training_options =</span> train_opts</span>
<span id="cb50-199"><a href="#cb50-199" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-200"><a href="#cb50-200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-201"><a href="#cb50-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-204"><a href="#cb50-204" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-205"><a href="#cb50-205" aria-hidden="true" tabindex="-1"></a>outfile <span class="ot">=</span> <span class="fu">file.path</span>(<span class="fu">getwd</span>(), <span class="st">"model.hdf5"</span>)</span>
<span id="cb50-206"><a href="#cb50-206" aria-hidden="true" tabindex="-1"></a>MOFAobject.trained <span class="ot">&lt;-</span> <span class="fu">run_mofa</span>(MOFAobject, outfile)</span>
<span id="cb50-207"><a href="#cb50-207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-208"><a href="#cb50-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-211"><a href="#cb50-211" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-212"><a href="#cb50-212" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">load_model</span>(outfile)</span>
<span id="cb50-213"><a href="#cb50-213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-214"><a href="#cb50-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-215"><a href="#cb50-215" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=14, fig.height=8}</span></span>
<span id="cb50-216"><a href="#cb50-216" aria-hidden="true" tabindex="-1"></a><span class="in">plot_variance_explained(model, x="view", y="factor") +</span></span>
<span id="cb50-217"><a href="#cb50-217" aria-hidden="true" tabindex="-1"></a><span class="in">  ggdark::dark_mode(verbose=FALSE) +</span></span>
<span id="cb50-218"><a href="#cb50-218" aria-hidden="true" tabindex="-1"></a><span class="in">  theme(axis.text.x = element_text(angle=90, hjust=1))</span></span>
<span id="cb50-219"><a href="#cb50-219" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-220"><a href="#cb50-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-221"><a href="#cb50-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=16, fig.height=6}</span></span>
<span id="cb50-222"><a href="#cb50-222" aria-hidden="true" tabindex="-1"></a><span class="in">plot_variance_explained(model, x="group", y="factor", plot_total = T)[[2]] +</span></span>
<span id="cb50-223"><a href="#cb50-223" aria-hidden="true" tabindex="-1"></a><span class="in">  ggdark::dark_mode(verbose=FALSE)</span></span>
<span id="cb50-224"><a href="#cb50-224" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>