<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.45">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Philipp Sven Lars Schäfer">
<meta name="dcterms.date" content="2024-10-14">

<title>Validation Test – PSLS @ 3rd CMI-PB Challenge</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">PSLS @ 3rd CMI-PB Challenge</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/psl-schaefer"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://x.com/psl_schaefer"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/philipp-sl-schaefer/"> <i class="bi bi-linkedin" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="1">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#packages" id="toc-packages" class="nav-link active" data-scroll-target="#packages"><span class="header-section-number">1</span> Packages</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2</span> Data</a></li>
  <li><a href="#conclusions" id="toc-conclusions" class="nav-link" data-scroll-target="#conclusions"><span class="header-section-number">3</span> Conclusions</a></li>
  <li><a href="#construct-feature-matrix" id="toc-construct-feature-matrix" class="nav-link" data-scroll-target="#construct-feature-matrix"><span class="header-section-number">4</span> Construct Feature Matrix</a></li>
  <li><a href="#tasks" id="toc-tasks" class="nav-link" data-scroll-target="#tasks"><span class="header-section-number">5</span> Tasks</a></li>
  <li><a href="#loop" id="toc-loop" class="nav-link" data-scroll-target="#loop"><span class="header-section-number">6</span> Loop</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Validation Test</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Philipp Sven Lars Schäfer </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 14, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="packages" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Packages</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(flextable)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"generate_targets.R"</span>))</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"model.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Data</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta_plus</span>(input_dir)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>specimen_per_day <span class="ot">&lt;-</span> <span class="fu">get_specimen_per_day</span>(<span class="at">meta_data=</span>meta_data)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>RECOMPUTE <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (RECOMPUTE) {</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"normalize_integrate.R"</span>))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  raw_experimental_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  filtered_experimental_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">meta_data=</span>meta_data, </span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">experimental_data=</span>raw_experimental_data,</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_meta=</span>gene_meta)</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_rds</span>(filtered_experimental_data, </span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"filtered_experimental_data.RDS"</span>))</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  normalized_experimental_data <span class="ot">&lt;-</span> <span class="fu">normalize_experimental_data</span>(</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">meta_data=</span>meta_data, </span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw_experimental_data=</span>filtered_experimental_data,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_meta=</span>gene_meta)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_rds</span>(normalized_experimental_data, </span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"normalized_experimental_data.RDS"</span>))</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>  integrated_experimental_data <span class="ot">&lt;-</span> <span class="fu">integrate_experimental_data</span>(</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">meta_data=</span>meta_data, </span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">normalized_experimental_data=</span>normalized_experimental_data)</span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_rds</span>(integrated_experimental_data, </span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"integrated_experimental_data.RDS"</span>))</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># use raw/filtered experimental data for computation of targets</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  target_list <span class="ot">&lt;-</span> <span class="fu">generate_all_targets</span>(</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">meta_data=</span>meta_data, </span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">experimental_data=</span>filtered_experimental_data, </span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">experimental_data_settings=</span>experimental_data_settings, </span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_meta=</span>gene_meta,</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">protein_meta=</span>protein_meta</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_rds</span>(target_list, </span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"target_list.RDS"</span>))</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(raw_experimental_data, filtered_experimental_data)</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  experimental_data <span class="ot">&lt;-</span> integrated_experimental_data</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>  experimental_data <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>                                                            <span class="st">"integrated_experimental_data.RDS"</span>))</span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>  experimental_data <span class="ot">&lt;-</span> experimental_data[<span class="sc">-</span><span class="fu">which</span>(<span class="fu">names</span>(experimental_data) <span class="sc">==</span></span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">"pbmc_gene_expression_counts"</span>)]</span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>  target_list <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>                                           <span class="st">"target_list.RDS"</span>))</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="conclusions" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Conclusions</h1>
<ul>
<li><p>For tasks 1.1, 2.1, 3.1, 4.1 the prediction based on baseline is pretty good</p></li>
<li><p>For task 1.2, (-1) times the baseline seems to be a super good predictors</p></li>
<li><p>In tasks 1.2, 2.2, 3.2 my models looks pretty shitty</p></li>
</ul>
</section>
<section id="construct-feature-matrix" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Construct Feature Matrix</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>df_boruta <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"df_boruta.RDS"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">modality_feature =</span> <span class="fu">paste0</span>(modality, <span class="st">"_"</span>, feature))</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>boruta_features <span class="ot">&lt;-</span> <span class="fu">unique</span>(df_boruta<span class="sc">$</span>modality_feature)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>experimental_predictors <span class="ot">&lt;-</span> <span class="fu">generate_wide_experimental_data</span>(<span class="at">experimental_data=</span>experimental_data,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">impute=</span><span class="st">"median"</span>, </span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">verbose=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">imap</span>(<span class="cf">function</span>(df, modality) {</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(modality, <span class="st">"_"</span>, <span class="fu">colnames</span>(df))</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>    df</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>all_specimens <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(experimental_predictors, <span class="sc">~</span> <span class="fu">rownames</span>(.x)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">reduce</span>(<span class="at">.x=</span>., <span class="at">.f=</span>dplyr<span class="sc">::</span>union)</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>all_features <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(experimental_predictors, <span class="sc">~</span> <span class="fu">colnames</span>(.x)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">reduce</span>(<span class="at">.x=</span>., <span class="at">.f=</span>dplyr<span class="sc">::</span>union)</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>feature_mtx <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">data=</span><span class="cn">NA</span>, <span class="at">nrow=</span><span class="fu">length</span>(all_specimens), <span class="at">ncol=</span><span class="fu">length</span>(all_features))</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(feature_mtx) <span class="ot">&lt;-</span> all_specimens; <span class="fu">colnames</span>(feature_mtx) <span class="ot">&lt;-</span> all_features</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (modality_data <span class="cf">in</span> experimental_predictors) {</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  feature_mtx[<span class="fu">rownames</span>(modality_data), <span class="fu">colnames</span>(modality_data)] <span class="ot">&lt;-</span></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>    modality_data</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(boruta_features <span class="sc">%in%</span> <span class="fu">colnames</span>(feature_mtx)))</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>specimen_day_0 <span class="ot">&lt;-</span> <span class="fu">as.character</span>(specimen_per_day<span class="sc">$</span>day_0<span class="sc">$</span>specimen_id)[</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.character</span>(specimen_per_day<span class="sc">$</span>day_0<span class="sc">$</span>specimen_id) <span class="sc">%in%</span> <span class="fu">rownames</span>(feature_mtx)]</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>feature_mtx <span class="ot">&lt;-</span> feature_mtx[specimen_day_0, boruta_features]</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(feature_mtx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> num [1:138, 1:363] NA NA 0.356 NA 1.477 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:138] "1" "19" "27" "37" ...
  ..$ : chr [1:363] "pbmc_cell_frequency_Monocytes" "pbmc_cell_frequency_Basophils" "pbmc_cell_frequency_Bcells" "pbmc_cell_frequency_CD4Tcells" ...</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>feature_mtx <span class="ot">&lt;-</span> feature_mtx[<span class="fu">rowMeans</span>(<span class="fu">is.na</span>(feature_mtx)) <span class="sc">&lt;</span> <span class="fl">0.5</span>, ]</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(feature_mtx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> num [1:136, 1:363] NA NA 0.356 NA 1.477 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:136] "1" "19" "27" "37" ...
  ..$ : chr [1:363] "pbmc_cell_frequency_Monocytes" "pbmc_cell_frequency_Basophils" "pbmc_cell_frequency_Bcells" "pbmc_cell_frequency_CD4Tcells" ...</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>feature_mtx <span class="ot">&lt;-</span> <span class="fu">t</span>(impute<span class="sc">::</span><span class="fu">impute.knn</span>(<span class="at">data=</span><span class="fu">t</span>(feature_mtx))<span class="sc">$</span>data)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(feature_mtx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> num [1:136, 1:363] 6.48 6.459 0.356 6.422 1.477 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:136] "1" "19" "27" "37" ...
  ..$ : chr [1:363] "pbmc_cell_frequency_Monocytes" "pbmc_cell_frequency_Basophils" "pbmc_cell_frequency_Bcells" "pbmc_cell_frequency_CD4Tcells" ...</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>feature_df <span class="ot">&lt;-</span> feature_mtx <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">specimen_id =</span> <span class="fu">as.numeric</span>(specimen_id)) <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((specimen_per_day<span class="sc">$</span>day_0 <span class="sc">%&gt;%</span> </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, specimen_id, dataset)),</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(specimen_id, dataset))</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="sc">!</span><span class="fu">any</span>(<span class="fu">is.na</span>(feature_df<span class="sc">$</span>subject_id)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="tasks" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Tasks</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>task_meta <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_11 =</span> <span class="fu">list</span>(</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_11"</span>,</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 1.1"</span>,</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by IgG antibody levels against pertussis toxin (PT) that we detect in plasma 14 days post booster vaccinations."</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_12 =</span> <span class="fu">list</span>(</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_12"</span>,</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 1.2"</span>,</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by fold change of IgG antibody levels against pertussis toxin (PT) that we detect in plasma 14 days post booster vaccinations compared to titer values at day 0."</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_21 =</span> <span class="fu">list</span>(</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_21"</span>,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 2.1"</span>,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by predicted frequency of Monocytes on day 1 post boost after vaccination."</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_22 =</span> <span class="fu">list</span>(</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_22"</span>,</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 2.2"</span>,</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by fold change of predicted frequency of Monocytes on day 1 post booster vaccination compared to cell frequency values at day 0."</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_31 =</span> <span class="fu">list</span>(</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_31"</span>,</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 3.1"</span>,</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by predicted gene expression of CCL3 on day 3 post-booster vaccination."</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_32 =</span> <span class="fu">list</span>(</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_32"</span>,</span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 3.2"</span>,</span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by fold change of predicted gene expression of CCL3 on day 3 post booster vaccination compared to gene expression values at day 0."</span></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_41 =</span> <span class="fu">list</span>(</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_41"</span>,</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 4.1"</span>,</span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals based on their Th1/Th2 (IFN-g/IL-5) polarization ratio on Day 30 post-booster vaccination."</span></span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="loop" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Loop</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>get_cross_cohort_perf_combinations <span class="ot">&lt;-</span> <span class="cf">function</span>(model_df, meta_data, <span class="at">nobase=</span><span class="cn">FALSE</span>) {</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">all</span>(<span class="fu">c</span>(<span class="st">"target"</span>, <span class="st">"subject_id"</span>, <span class="st">"baseline"</span>) <span class="sc">%in%</span> <span class="fu">colnames</span>(model_df)))</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  model_dataset_df <span class="ot">&lt;-</span> model_df <span class="sc">%&gt;%</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                       dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>                       dplyr<span class="sc">::</span><span class="fu">distinct</span>(),</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">by=</span><span class="st">"subject_id"</span>)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  all_datasets <span class="ot">&lt;-</span> <span class="fu">unique</span>(model_dataset_df<span class="sc">$</span>dataset)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  cross_dataset_out <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(all_datasets, <span class="cf">function</span>(test_dataset) {</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># test_dataset &lt;- all_datasets[1]</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>    train_dataset <span class="ot">&lt;-</span> all_datasets[<span class="sc">!</span>all_datasets <span class="sc">%in%</span> test_dataset]</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>    train_df <span class="ot">&lt;-</span> model_dataset_df <span class="sc">%&gt;%</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(dataset <span class="sc">%in%</span> train_dataset) <span class="sc">%&gt;%</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(dataset, subject_id))</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>    test_df <span class="ot">&lt;-</span> model_dataset_df <span class="sc">%&gt;%</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(dataset <span class="sc">%in%</span> test_dataset) <span class="sc">%&gt;%</span></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(dataset, subject_id))</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (nobase) {</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>      train_df <span class="ot">&lt;-</span> train_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>baseline)</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>    model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">formula =</span> target <span class="sc">~</span> ., </span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>                            <span class="at">data =</span> train_df, </span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>                            <span class="at">num.trees =</span> <span class="dv">500</span>)</span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>    predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(model, test_df)<span class="sc">$</span>predictions</span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">tibble</span>(<span class="at">trainset =</span> <span class="fu">paste0</span>(train_dataset, <span class="at">collapse=</span><span class="st">"__"</span>),</span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>                   <span class="at">testset =</span> test_dataset,</span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>                   <span class="at">target =</span> test_df<span class="sc">$</span>target, </span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>                   <span class="at">prediction =</span> predictions,</span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a>                   <span class="at">baseline =</span> test_df<span class="sc">$</span>baseline,</span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a>                   <span class="at">testset_mean =</span> <span class="fu">mean</span>(test_df<span class="sc">$</span>target))</span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">%&gt;%</span></span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># cross_dataset_out %&gt;%</span></span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   ggplot() +</span></span>
<span id="cb12-42"><a href="#cb12-42" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   geom_point(aes(x=prediction, y=target)) +</span></span>
<span id="cb12-43"><a href="#cb12-43" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   ggdark::dark_mode(verbose=FALSE)</span></span>
<span id="cb12-44"><a href="#cb12-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-45"><a href="#cb12-45" aria-hidden="true" tabindex="-1"></a>  cross_dataset_out <span class="sc">%&gt;%</span></span>
<span id="cb12-46"><a href="#cb12-46" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(trainset, testset) <span class="sc">%&gt;%</span></span>
<span id="cb12-47"><a href="#cb12-47" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb12-48"><a href="#cb12-48" aria-hidden="true" tabindex="-1"></a>      <span class="at">test_mse =</span> <span class="fu">get_mse</span>(target, prediction),</span>
<span id="cb12-49"><a href="#cb12-49" aria-hidden="true" tabindex="-1"></a>      <span class="at">test_r2 =</span> <span class="fu">get_r2</span>(target, prediction),</span>
<span id="cb12-50"><a href="#cb12-50" aria-hidden="true" tabindex="-1"></a>      <span class="at">test_srho =</span> <span class="fu">get_spearman</span>(target, prediction),</span>
<span id="cb12-51"><a href="#cb12-51" aria-hidden="true" tabindex="-1"></a>      <span class="at">baseline_srho =</span> <span class="fu">get_spearman</span>(target, baseline),</span>
<span id="cb12-52"><a href="#cb12-52" aria-hidden="true" tabindex="-1"></a>      <span class="at">mse_same_cohort =</span> <span class="fu">get_mse</span>(target, testset_mean),</span>
<span id="cb12-53"><a href="#cb12-53" aria-hidden="true" tabindex="-1"></a>      <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb12-54"><a href="#cb12-54" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb12-55"><a href="#cb12-55" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-56"><a href="#cb12-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-57"><a href="#cb12-57" aria-hidden="true" tabindex="-1"></a>get_cross_cohort_perf_combinations_repeated <span class="ot">&lt;-</span> <span class="cf">function</span>(model_df, meta_data, </span>
<span id="cb12-58"><a href="#cb12-58" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">n_iter=</span><span class="dv">25</span>, <span class="at">nobase=</span><span class="cn">FALSE</span>) {</span>
<span id="cb12-59"><a href="#cb12-59" aria-hidden="true" tabindex="-1"></a>  <span class="co"># </span><span class="al">TODO</span></span>
<span id="cb12-60"><a href="#cb12-60" aria-hidden="true" tabindex="-1"></a>  df_long <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span>n_iter, <span class="cf">function</span>(iter) {</span>
<span id="cb12-61"><a href="#cb12-61" aria-hidden="true" tabindex="-1"></a>    <span class="fu">get_cross_cohort_perf_combinations</span>(<span class="at">model_df=</span>model_df, <span class="at">meta_data=</span>meta_data, </span>
<span id="cb12-62"><a href="#cb12-62" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">nobase=</span>nobase) <span class="sc">%&gt;%</span></span>
<span id="cb12-63"><a href="#cb12-63" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">iteration=</span>iter)</span>
<span id="cb12-64"><a href="#cb12-64" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">%&gt;%</span></span>
<span id="cb12-65"><a href="#cb12-65" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb12-66"><a href="#cb12-66" aria-hidden="true" tabindex="-1"></a>  df_out <span class="ot">&lt;-</span> df_long <span class="sc">%&gt;%</span></span>
<span id="cb12-67"><a href="#cb12-67" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(trainset, testset) <span class="sc">%&gt;%</span></span>
<span id="cb12-68"><a href="#cb12-68" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">test_srho_mean =</span> <span class="fu">mean</span>(test_srho),</span>
<span id="cb12-69"><a href="#cb12-69" aria-hidden="true" tabindex="-1"></a>                     <span class="at">test_srho_sd =</span> <span class="fu">sd</span>(test_srho),</span>
<span id="cb12-70"><a href="#cb12-70" aria-hidden="true" tabindex="-1"></a>                     <span class="at">baseline_srho_mean =</span> <span class="fu">mean</span>(baseline_srho),</span>
<span id="cb12-71"><a href="#cb12-71" aria-hidden="true" tabindex="-1"></a>                     <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb12-72"><a href="#cb12-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df_out)</span>
<span id="cb12-73"><a href="#cb12-73" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>permute_columns <span class="ot">&lt;-</span> <span class="cf">function</span>(mtx, <span class="at">seed=</span><span class="dv">42</span>) {</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (col_idx <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(mtx)) {</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    mtx[, col_idx] <span class="ot">&lt;-</span> <span class="fu">sample</span>(mtx[, col_idx], <span class="at">size=</span><span class="fu">nrow</span>(mtx), <span class="at">replace=</span><span class="cn">TRUE</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(mtx)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>tibble_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>metadata_covariates <span class="ot">&lt;-</span> <span class="fu">get_metadata_covariates</span>(meta_data)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>model_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"metadata"</span> <span class="ot">=</span> metadata_covariates,</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">"features"</span> <span class="ot">=</span> feature_df,</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">"features_permuted"</span> <span class="ot">=</span> <span class="fu">permute_columns</span>(feature_df),</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">"features_permuted_nobase"</span> <span class="ot">=</span> <span class="fu">permute_columns</span>(feature_df),</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">"metadata_plus_features"</span> <span class="ot">=</span> feature_df <span class="sc">%&gt;%</span> </span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(metadata_covariates, <span class="at">by=</span><span class="st">"subject_id"</span>)</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (task <span class="cf">in</span> task_meta) {</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># task &lt;- task_meta[[3]]</span></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (model_name <span class="cf">in</span> <span class="fu">names</span>(model_list)) {</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># model_name &lt;- "features_permuted"</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>    <span class="co">#print(model_name)</span></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    model_df <span class="ot">&lt;-</span> target_list[[task<span class="sc">$</span>name]] <span class="sc">%&gt;%</span></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">left_join</span>(model_list[[model_name]], <span class="at">by=</span><span class="st">"subject_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>      tidyr<span class="sc">::</span><span class="fu">drop_na</span>()</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (model_name <span class="sc">==</span> <span class="st">"features_permuted_nobase"</span>) {</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>      nobase <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>      nobase <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>    task_df <span class="ot">&lt;-</span> <span class="fu">get_cross_cohort_perf_combinations_repeated</span>(<span class="at">model_df=</span>model_df,</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">meta_data=</span>meta_data, </span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">nobase=</span>nobase) <span class="sc">%&gt;%</span></span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task =</span> task<span class="sc">$</span>name, <span class="at">model =</span> model_name) <span class="sc">%&gt;%</span></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="at">.cols=</span><span class="fu">c</span>(test_srho_mean, test_srho_sd, baseline_srho_mean),</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>                           <span class="at">.fns=</span><span class="sc">~</span><span class="fu">round</span>(.x, <span class="dv">2</span>))) </span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a>    tibble_list <span class="ot">&lt;-</span> rlist<span class="sc">::</span><span class="fu">list.append</span>(tibble_list, task_df)</span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(tibble_list) <span class="sc">%&gt;%</span></span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a>      <span class="at">task =</span> <span class="fu">factor</span>(task, <span class="at">levels=</span><span class="fu">paste0</span>(<span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">"task_"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>), <span class="at">each=</span><span class="dv">2</span>), <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="dv">4</span>))),</span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a>      <span class="at">model =</span> <span class="fu">factor</span>(model, <span class="at">levels=</span><span class="fu">names</span>(model_list))</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a>df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["trainset"],"name":[1],"type":["chr"],"align":["left"]},{"label":["testset"],"name":[2],"type":["chr"],"align":["left"]},{"label":["test_srho_mean"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["test_srho_sd"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["baseline_srho_mean"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["task"],"name":[6],"type":["fct"],"align":["left"]},{"label":["model"],"name":[7],"type":["fct"],"align":["left"]}],"data":[{"1":"2021_dataset","2":"2022_dataset","3":"-0.32","4":"0.02","5":"0.44","6":"task_11","7":"metadata"},{"1":"2022_dataset","2":"2021_dataset","3":"-0.33","4":"0.04","5":"0.64","6":"task_11","7":"metadata"},{"1":"2021_dataset","2":"2022_dataset","3":"-0.12","4":"0.07","5":"0.44","6":"task_11","7":"features"},{"1":"2022_dataset","2":"2021_dataset","3":"-0.13","4":"0.07","5":"0.64","6":"task_11","7":"features"},{"1":"2021_dataset","2":"2022_dataset","3":"-0.26","4":"0.11","5":"0.64","6":"task_11","7":"features_permuted"},{"1":"2022_dataset","2":"2021_dataset","3":"-0.22","4":"0.10","5":"0.65","6":"task_11","7":"features_permuted"},{"1":"2021_dataset","2":"2022_dataset","3":"-0.26","4":"0.13","5":"0.64","6":"task_11","7":"features_permuted_nobase"},{"1":"2022_dataset","2":"2021_dataset","3":"-0.24","4":"0.11","5":"0.65","6":"task_11","7":"features_permuted_nobase"},{"1":"2021_dataset","2":"2022_dataset","3":"-0.15","4":"0.10","5":"0.44","6":"task_11","7":"metadata_plus_features"},{"1":"2022_dataset","2":"2021_dataset","3":"-0.16","4":"0.07","5":"0.64","6":"task_11","7":"metadata_plus_features"},{"1":"2021_dataset","2":"2022_dataset","3":"0.38","4":"0.03","5":"-0.89","6":"task_12","7":"metadata"},{"1":"2022_dataset","2":"2021_dataset","3":"0.11","4":"0.02","5":"-0.73","6":"task_12","7":"metadata"},{"1":"2021_dataset","2":"2022_dataset","3":"0.30","4":"0.10","5":"-0.89","6":"task_12","7":"features"},{"1":"2022_dataset","2":"2021_dataset","3":"0.54","4":"0.07","5":"-0.73","6":"task_12","7":"features"},{"1":"2021_dataset","2":"2022_dataset","3":"0.17","4":"0.09","5":"-0.92","6":"task_12","7":"features_permuted"},{"1":"2022_dataset","2":"2021_dataset","3":"0.14","4":"0.08","5":"-0.66","6":"task_12","7":"features_permuted"},{"1":"2021_dataset","2":"2022_dataset","3":"0.13","4":"0.07","5":"-0.92","6":"task_12","7":"features_permuted_nobase"},{"1":"2022_dataset","2":"2021_dataset","3":"0.17","4":"0.07","5":"-0.66","6":"task_12","7":"features_permuted_nobase"},{"1":"2021_dataset","2":"2022_dataset","3":"0.35","4":"0.12","5":"-0.89","6":"task_12","7":"metadata_plus_features"},{"1":"2022_dataset","2":"2021_dataset","3":"0.54","4":"0.06","5":"-0.73","6":"task_12","7":"metadata_plus_features"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.54","4":"0.03","5":"0.55","6":"task_21","7":"metadata"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.61","4":"0.02","5":"0.88","6":"task_21","7":"metadata"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.56","4":"0.05","5":"0.81","6":"task_21","7":"metadata"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.44","4":"0.08","5":"0.55","6":"task_21","7":"features"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.76","4":"0.04","5":"0.88","6":"task_21","7":"features"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.81","4":"0.05","5":"0.84","6":"task_21","7":"features"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.34","4":"0.08","5":"0.60","6":"task_21","7":"features_permuted"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.05","4":"0.07","5":"0.89","6":"task_21","7":"features_permuted"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"NA","4":"NA","5":"NA","6":"task_21","7":"features_permuted"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.07","4":"0.10","5":"0.60","6":"task_21","7":"features_permuted_nobase"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"-0.13","4":"0.07","5":"0.89","6":"task_21","7":"features_permuted_nobase"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"NA","4":"NA","5":"NA","6":"task_21","7":"features_permuted_nobase"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.41","4":"0.07","5":"0.55","6":"task_21","7":"metadata_plus_features"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.76","4":"0.04","5":"0.88","6":"task_21","7":"metadata_plus_features"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.80","4":"0.06","5":"0.84","6":"task_21","7":"metadata_plus_features"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.34","4":"0.02","5":"-0.21","6":"task_22","7":"metadata"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"-0.26","4":"0.03","5":"-0.22","6":"task_22","7":"metadata"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"-0.08","4":"0.05","5":"-0.43","6":"task_22","7":"metadata"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.12","4":"0.07","5":"-0.21","6":"task_22","7":"features"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"-0.05","4":"0.06","5":"-0.22","6":"task_22","7":"features"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"-0.02","4":"0.11","5":"-0.31","6":"task_22","7":"features"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.02","4":"0.08","5":"-0.32","6":"task_22","7":"features_permuted"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.14","4":"0.09","5":"-0.01","6":"task_22","7":"features_permuted"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"NA","4":"NA","5":"NA","6":"task_22","7":"features_permuted"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.01","4":"0.08","5":"-0.32","6":"task_22","7":"features_permuted_nobase"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.19","4":"0.10","5":"-0.01","6":"task_22","7":"features_permuted_nobase"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"NA","4":"NA","5":"NA","6":"task_22","7":"features_permuted_nobase"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.11","4":"0.09","5":"-0.21","6":"task_22","7":"metadata_plus_features"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"-0.06","4":"0.07","5":"-0.22","6":"task_22","7":"metadata_plus_features"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"-0.04","4":"0.10","5":"-0.31","6":"task_22","7":"metadata_plus_features"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.15","4":"0.03","5":"0.53","6":"task_31","7":"metadata"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.46","4":"0.02","5":"0.57","6":"task_31","7":"metadata"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.15","4":"0.01","5":"0.27","6":"task_31","7":"metadata"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.06","4":"0.12","5":"0.53","6":"task_31","7":"features"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.16","4":"0.06","5":"0.57","6":"task_31","7":"features"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.36","4":"0.05","5":"0.27","6":"task_31","7":"features"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.35","4":"0.11","5":"0.56","6":"task_31","7":"features_permuted"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.10","4":"0.12","5":"0.39","6":"task_31","7":"features_permuted"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"-0.12","4":"0.10","5":"-0.26","6":"task_31","7":"features_permuted"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.31","4":"0.12","5":"0.56","6":"task_31","7":"features_permuted_nobase"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.08","4":"0.11","5":"0.39","6":"task_31","7":"features_permuted_nobase"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.18","4":"0.09","5":"-0.26","6":"task_31","7":"features_permuted_nobase"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.08","4":"0.11","5":"0.53","6":"task_31","7":"metadata_plus_features"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.16","4":"0.04","5":"0.57","6":"task_31","7":"metadata_plus_features"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.36","4":"0.06","5":"0.27","6":"task_31","7":"metadata_plus_features"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.25","4":"0.02","5":"-0.19","6":"task_32","7":"metadata"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.27","4":"0.02","5":"-0.43","6":"task_32","7":"metadata"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"-0.25","4":"0.02","5":"-0.33","6":"task_32","7":"metadata"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.16","4":"0.07","5":"-0.19","6":"task_32","7":"features"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"-0.02","4":"0.06","5":"-0.43","6":"task_32","7":"features"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.26","4":"0.06","5":"-0.33","6":"task_32","7":"features"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.27","4":"0.08","5":"0.08","6":"task_32","7":"features_permuted"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.20","4":"0.09","5":"-0.47","6":"task_32","7":"features_permuted"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.02","4":"0.15","5":"-0.64","6":"task_32","7":"features_permuted"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"0.26","4":"0.07","5":"0.08","6":"task_32","7":"features_permuted_nobase"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"0.11","4":"0.08","5":"-0.47","6":"task_32","7":"features_permuted_nobase"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"-0.05","4":"0.17","5":"-0.64","6":"task_32","7":"features_permuted_nobase"},{"1":"2020_dataset__2021_dataset","2":"2022_dataset","3":"-0.15","4":"0.06","5":"-0.19","6":"task_32","7":"metadata_plus_features"},{"1":"2020_dataset__2022_dataset","2":"2021_dataset","3":"-0.01","4":"0.07","5":"-0.43","6":"task_32","7":"metadata_plus_features"},{"1":"2021_dataset__2022_dataset","2":"2020_dataset","3":"0.28","4":"0.04","5":"-0.33","6":"task_32","7":"metadata_plus_features"},{"1":"2021_dataset","2":"2022_dataset","3":"0.46","4":"0.04","5":"0.74","6":"task_41","7":"metadata"},{"1":"2022_dataset","2":"2021_dataset","3":"0.53","4":"0.02","5":"0.55","6":"task_41","7":"metadata"},{"1":"2021_dataset","2":"2022_dataset","3":"-0.06","4":"0.05","5":"0.74","6":"task_41","7":"features"},{"1":"2022_dataset","2":"2021_dataset","3":"0.17","4":"0.07","5":"0.55","6":"task_41","7":"features"},{"1":"2021_dataset","2":"2022_dataset","3":"-0.08","4":"0.10","5":"0.64","6":"task_41","7":"features_permuted"},{"1":"2022_dataset","2":"2021_dataset","3":"0.07","4":"0.12","5":"0.61","6":"task_41","7":"features_permuted"},{"1":"2021_dataset","2":"2022_dataset","3":"-0.12","4":"0.08","5":"0.64","6":"task_41","7":"features_permuted_nobase"},{"1":"2022_dataset","2":"2021_dataset","3":"-0.12","4":"0.10","5":"0.61","6":"task_41","7":"features_permuted_nobase"},{"1":"2021_dataset","2":"2022_dataset","3":"-0.06","4":"0.06","5":"0.74","6":"task_41","7":"metadata_plus_features"},{"1":"2022_dataset","2":"2021_dataset","3":"0.15","4":"0.05","5":"0.55","6":"task_41","7":"metadata_plus_features"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>test_srho_sd) <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">c</span>(test_srho_mean, baseline_srho_mean),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">names_to=</span><span class="st">"metric"</span>, <span class="at">values_to=</span><span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>), <span class="at">color=</span><span class="st">"grey"</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span>task, <span class="at">x=</span>value, <span class="at">color=</span>testset, <span class="at">shape=</span>metric), <span class="at">alpha=</span><span class="fl">0.50</span>) <span class="sc">+</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Inverted geom defaults of fill and color/colour.
To change them back, use invert_geom_defaults().</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 8 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="validation_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb17" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Validation Test"</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Philipp Sven Lars Schäfer"</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "`r format(Sys.time(), '%d %B, %Y')`"</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> source</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="an">engine:</span><span class="co"> knitr</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="fu"># Packages</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(flextable)</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick)</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"generate_targets.R"</span>))</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"model.R"</span>))</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>celltype_meta <span class="ot">&lt;-</span> <span class="fu">read_celltype_meta</span>(input_dir)</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>gene_meta <span class="ot">&lt;-</span> <span class="fu">read_gene_meta_plus</span>(input_dir)</span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>protein_meta <span class="ot">&lt;-</span> <span class="fu">read_protein_meta</span>(input_dir)</span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a>meta_data <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>specimen_per_day <span class="ot">&lt;-</span> <span class="fu">get_specimen_per_day</span>(<span class="at">meta_data=</span>meta_data)</span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>RECOMPUTE <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (RECOMPUTE) {</span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"normalize_integrate.R"</span>))</span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a>  raw_experimental_data <span class="ot">&lt;-</span> <span class="fu">read_raw_experimental_data</span>(input_dir)</span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>  filtered_experimental_data <span class="ot">&lt;-</span> <span class="fu">filter_experimental_data</span>(</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">meta_data=</span>meta_data, </span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">experimental_data=</span>raw_experimental_data,</span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_meta=</span>gene_meta)</span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_rds</span>(filtered_experimental_data, </span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"filtered_experimental_data.RDS"</span>))</span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a>  normalized_experimental_data <span class="ot">&lt;-</span> <span class="fu">normalize_experimental_data</span>(</span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">meta_data=</span>meta_data, </span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">raw_experimental_data=</span>filtered_experimental_data,</span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_meta=</span>gene_meta)</span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_rds</span>(normalized_experimental_data, </span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"normalized_experimental_data.RDS"</span>))</span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a>  integrated_experimental_data <span class="ot">&lt;-</span> <span class="fu">integrate_experimental_data</span>(</span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">meta_data=</span>meta_data, </span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">normalized_experimental_data=</span>normalized_experimental_data)</span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_rds</span>(integrated_experimental_data, </span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a>            <span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"integrated_experimental_data.RDS"</span>))</span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a>  <span class="co"># use raw/filtered experimental data for computation of targets</span></span>
<span id="cb17-78"><a href="#cb17-78" aria-hidden="true" tabindex="-1"></a>  target_list <span class="ot">&lt;-</span> <span class="fu">generate_all_targets</span>(</span>
<span id="cb17-79"><a href="#cb17-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">meta_data=</span>meta_data, </span>
<span id="cb17-80"><a href="#cb17-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">experimental_data=</span>filtered_experimental_data, </span>
<span id="cb17-81"><a href="#cb17-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">experimental_data_settings=</span>experimental_data_settings, </span>
<span id="cb17-82"><a href="#cb17-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_meta=</span>gene_meta,</span>
<span id="cb17-83"><a href="#cb17-83" aria-hidden="true" tabindex="-1"></a>    <span class="at">protein_meta=</span>protein_meta</span>
<span id="cb17-84"><a href="#cb17-84" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb17-85"><a href="#cb17-85" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-86"><a href="#cb17-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_rds</span>(target_list, </span>
<span id="cb17-87"><a href="#cb17-87" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb17-88"><a href="#cb17-88" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"target_list.RDS"</span>))</span>
<span id="cb17-89"><a href="#cb17-89" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-90"><a href="#cb17-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(raw_experimental_data, filtered_experimental_data)</span>
<span id="cb17-91"><a href="#cb17-91" aria-hidden="true" tabindex="-1"></a>  experimental_data <span class="ot">&lt;-</span> integrated_experimental_data</span>
<span id="cb17-92"><a href="#cb17-92" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb17-93"><a href="#cb17-93" aria-hidden="true" tabindex="-1"></a>  experimental_data <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb17-94"><a href="#cb17-94" aria-hidden="true" tabindex="-1"></a>                                                            <span class="st">"integrated_experimental_data.RDS"</span>))</span>
<span id="cb17-95"><a href="#cb17-95" aria-hidden="true" tabindex="-1"></a>  experimental_data <span class="ot">&lt;-</span> experimental_data[<span class="sc">-</span><span class="fu">which</span>(<span class="fu">names</span>(experimental_data) <span class="sc">==</span></span>
<span id="cb17-96"><a href="#cb17-96" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">"pbmc_gene_expression_counts"</span>)]</span>
<span id="cb17-97"><a href="#cb17-97" aria-hidden="true" tabindex="-1"></a>  target_list <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb17-98"><a href="#cb17-98" aria-hidden="true" tabindex="-1"></a>                                           <span class="st">"target_list.RDS"</span>))</span>
<span id="cb17-99"><a href="#cb17-99" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-100"><a href="#cb17-100" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-101"><a href="#cb17-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-102"><a href="#cb17-102" aria-hidden="true" tabindex="-1"></a><span class="fu"># Conclusions</span></span>
<span id="cb17-103"><a href="#cb17-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-104"><a href="#cb17-104" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>For tasks 1.1, 2.1, 3.1, 4.1 the prediction based on baseline is pretty good</span>
<span id="cb17-105"><a href="#cb17-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-106"><a href="#cb17-106" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>For task 1.2, (-1) times the baseline seems to be a super good predictors</span>
<span id="cb17-107"><a href="#cb17-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-108"><a href="#cb17-108" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>In tasks 1.2, 2.2, 3.2 my models looks pretty shitty</span>
<span id="cb17-109"><a href="#cb17-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-110"><a href="#cb17-110" aria-hidden="true" tabindex="-1"></a><span class="fu"># Construct Feature Matrix</span></span>
<span id="cb17-111"><a href="#cb17-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-114"><a href="#cb17-114" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-115"><a href="#cb17-115" aria-hidden="true" tabindex="-1"></a>df_boruta <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="at">file =</span> <span class="fu">file.path</span>(input_dir, <span class="st">"prc_datasets"</span>, </span>
<span id="cb17-116"><a href="#cb17-116" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"df_boruta.RDS"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-117"><a href="#cb17-117" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">modality_feature =</span> <span class="fu">paste0</span>(modality, <span class="st">"_"</span>, feature))</span>
<span id="cb17-118"><a href="#cb17-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-119"><a href="#cb17-119" aria-hidden="true" tabindex="-1"></a>boruta_features <span class="ot">&lt;-</span> <span class="fu">unique</span>(df_boruta<span class="sc">$</span>modality_feature)</span>
<span id="cb17-120"><a href="#cb17-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-121"><a href="#cb17-121" aria-hidden="true" tabindex="-1"></a>experimental_predictors <span class="ot">&lt;-</span> <span class="fu">generate_wide_experimental_data</span>(<span class="at">experimental_data=</span>experimental_data,</span>
<span id="cb17-122"><a href="#cb17-122" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">impute=</span><span class="st">"median"</span>, </span>
<span id="cb17-123"><a href="#cb17-123" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">verbose=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-124"><a href="#cb17-124" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">imap</span>(<span class="cf">function</span>(df, modality) {</span>
<span id="cb17-125"><a href="#cb17-125" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(modality, <span class="st">"_"</span>, <span class="fu">colnames</span>(df))</span>
<span id="cb17-126"><a href="#cb17-126" aria-hidden="true" tabindex="-1"></a>    df</span>
<span id="cb17-127"><a href="#cb17-127" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb17-128"><a href="#cb17-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-129"><a href="#cb17-129" aria-hidden="true" tabindex="-1"></a>all_specimens <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(experimental_predictors, <span class="sc">~</span> <span class="fu">rownames</span>(.x)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-130"><a href="#cb17-130" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">reduce</span>(<span class="at">.x=</span>., <span class="at">.f=</span>dplyr<span class="sc">::</span>union)</span>
<span id="cb17-131"><a href="#cb17-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-132"><a href="#cb17-132" aria-hidden="true" tabindex="-1"></a>all_features <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(experimental_predictors, <span class="sc">~</span> <span class="fu">colnames</span>(.x)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-133"><a href="#cb17-133" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">reduce</span>(<span class="at">.x=</span>., <span class="at">.f=</span>dplyr<span class="sc">::</span>union)</span>
<span id="cb17-134"><a href="#cb17-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-135"><a href="#cb17-135" aria-hidden="true" tabindex="-1"></a>feature_mtx <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="at">data=</span><span class="cn">NA</span>, <span class="at">nrow=</span><span class="fu">length</span>(all_specimens), <span class="at">ncol=</span><span class="fu">length</span>(all_features))</span>
<span id="cb17-136"><a href="#cb17-136" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(feature_mtx) <span class="ot">&lt;-</span> all_specimens; <span class="fu">colnames</span>(feature_mtx) <span class="ot">&lt;-</span> all_features</span>
<span id="cb17-137"><a href="#cb17-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-138"><a href="#cb17-138" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (modality_data <span class="cf">in</span> experimental_predictors) {</span>
<span id="cb17-139"><a href="#cb17-139" aria-hidden="true" tabindex="-1"></a>  feature_mtx[<span class="fu">rownames</span>(modality_data), <span class="fu">colnames</span>(modality_data)] <span class="ot">&lt;-</span></span>
<span id="cb17-140"><a href="#cb17-140" aria-hidden="true" tabindex="-1"></a>    modality_data</span>
<span id="cb17-141"><a href="#cb17-141" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-142"><a href="#cb17-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-143"><a href="#cb17-143" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(boruta_features <span class="sc">%in%</span> <span class="fu">colnames</span>(feature_mtx)))</span>
<span id="cb17-144"><a href="#cb17-144" aria-hidden="true" tabindex="-1"></a>specimen_day_0 <span class="ot">&lt;-</span> <span class="fu">as.character</span>(specimen_per_day<span class="sc">$</span>day_0<span class="sc">$</span>specimen_id)[</span>
<span id="cb17-145"><a href="#cb17-145" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.character</span>(specimen_per_day<span class="sc">$</span>day_0<span class="sc">$</span>specimen_id) <span class="sc">%in%</span> <span class="fu">rownames</span>(feature_mtx)]</span>
<span id="cb17-146"><a href="#cb17-146" aria-hidden="true" tabindex="-1"></a>feature_mtx <span class="ot">&lt;-</span> feature_mtx[specimen_day_0, boruta_features]</span>
<span id="cb17-147"><a href="#cb17-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-148"><a href="#cb17-148" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(feature_mtx)</span>
<span id="cb17-149"><a href="#cb17-149" aria-hidden="true" tabindex="-1"></a>feature_mtx <span class="ot">&lt;-</span> feature_mtx[<span class="fu">rowMeans</span>(<span class="fu">is.na</span>(feature_mtx)) <span class="sc">&lt;</span> <span class="fl">0.5</span>, ]</span>
<span id="cb17-150"><a href="#cb17-150" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(feature_mtx)</span>
<span id="cb17-151"><a href="#cb17-151" aria-hidden="true" tabindex="-1"></a>feature_mtx <span class="ot">&lt;-</span> <span class="fu">t</span>(impute<span class="sc">::</span><span class="fu">impute.knn</span>(<span class="at">data=</span><span class="fu">t</span>(feature_mtx))<span class="sc">$</span>data)</span>
<span id="cb17-152"><a href="#cb17-152" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(feature_mtx)</span>
<span id="cb17-153"><a href="#cb17-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-154"><a href="#cb17-154" aria-hidden="true" tabindex="-1"></a>feature_df <span class="ot">&lt;-</span> feature_mtx <span class="sc">%&gt;%</span></span>
<span id="cb17-155"><a href="#cb17-155" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-156"><a href="#cb17-156" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-157"><a href="#cb17-157" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">specimen_id =</span> <span class="fu">as.numeric</span>(specimen_id)) <span class="sc">%&gt;%</span></span>
<span id="cb17-158"><a href="#cb17-158" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>((specimen_per_day<span class="sc">$</span>day_0 <span class="sc">%&gt;%</span> </span>
<span id="cb17-159"><a href="#cb17-159" aria-hidden="true" tabindex="-1"></a>                      dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, specimen_id, dataset)),</span>
<span id="cb17-160"><a href="#cb17-160" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-161"><a href="#cb17-161" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(specimen_id, dataset))</span>
<span id="cb17-162"><a href="#cb17-162" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="sc">!</span><span class="fu">any</span>(<span class="fu">is.na</span>(feature_df<span class="sc">$</span>subject_id)))</span>
<span id="cb17-163"><a href="#cb17-163" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-164"><a href="#cb17-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-165"><a href="#cb17-165" aria-hidden="true" tabindex="-1"></a><span class="fu"># Tasks</span></span>
<span id="cb17-166"><a href="#cb17-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-169"><a href="#cb17-169" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-170"><a href="#cb17-170" aria-hidden="true" tabindex="-1"></a>task_meta <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb17-171"><a href="#cb17-171" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_11 =</span> <span class="fu">list</span>(</span>
<span id="cb17-172"><a href="#cb17-172" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_11"</span>,</span>
<span id="cb17-173"><a href="#cb17-173" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 1.1"</span>,</span>
<span id="cb17-174"><a href="#cb17-174" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by IgG antibody levels against pertussis toxin (PT) that we detect in plasma 14 days post booster vaccinations."</span></span>
<span id="cb17-175"><a href="#cb17-175" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb17-176"><a href="#cb17-176" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_12 =</span> <span class="fu">list</span>(</span>
<span id="cb17-177"><a href="#cb17-177" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_12"</span>,</span>
<span id="cb17-178"><a href="#cb17-178" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 1.2"</span>,</span>
<span id="cb17-179"><a href="#cb17-179" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by fold change of IgG antibody levels against pertussis toxin (PT) that we detect in plasma 14 days post booster vaccinations compared to titer values at day 0."</span></span>
<span id="cb17-180"><a href="#cb17-180" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb17-181"><a href="#cb17-181" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_21 =</span> <span class="fu">list</span>(</span>
<span id="cb17-182"><a href="#cb17-182" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_21"</span>,</span>
<span id="cb17-183"><a href="#cb17-183" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 2.1"</span>,</span>
<span id="cb17-184"><a href="#cb17-184" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by predicted frequency of Monocytes on day 1 post boost after vaccination."</span></span>
<span id="cb17-185"><a href="#cb17-185" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb17-186"><a href="#cb17-186" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_22 =</span> <span class="fu">list</span>(</span>
<span id="cb17-187"><a href="#cb17-187" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_22"</span>,</span>
<span id="cb17-188"><a href="#cb17-188" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 2.2"</span>,</span>
<span id="cb17-189"><a href="#cb17-189" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by fold change of predicted frequency of Monocytes on day 1 post booster vaccination compared to cell frequency values at day 0."</span></span>
<span id="cb17-190"><a href="#cb17-190" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb17-191"><a href="#cb17-191" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_31 =</span> <span class="fu">list</span>(</span>
<span id="cb17-192"><a href="#cb17-192" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_31"</span>,</span>
<span id="cb17-193"><a href="#cb17-193" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 3.1"</span>,</span>
<span id="cb17-194"><a href="#cb17-194" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by predicted gene expression of CCL3 on day 3 post-booster vaccination."</span></span>
<span id="cb17-195"><a href="#cb17-195" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb17-196"><a href="#cb17-196" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_32 =</span> <span class="fu">list</span>(</span>
<span id="cb17-197"><a href="#cb17-197" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_32"</span>,</span>
<span id="cb17-198"><a href="#cb17-198" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 3.2"</span>,</span>
<span id="cb17-199"><a href="#cb17-199" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals by fold change of predicted gene expression of CCL3 on day 3 post booster vaccination compared to gene expression values at day 0."</span></span>
<span id="cb17-200"><a href="#cb17-200" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb17-201"><a href="#cb17-201" aria-hidden="true" tabindex="-1"></a>  <span class="at">task_41 =</span> <span class="fu">list</span>(</span>
<span id="cb17-202"><a href="#cb17-202" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"task_41"</span>,</span>
<span id="cb17-203"><a href="#cb17-203" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="st">"## Task 4.1"</span>,</span>
<span id="cb17-204"><a href="#cb17-204" aria-hidden="true" tabindex="-1"></a>    <span class="at">description =</span> <span class="st">"Rank the individuals based on their Th1/Th2 (IFN-g/IL-5) polarization ratio on Day 30 post-booster vaccination."</span></span>
<span id="cb17-205"><a href="#cb17-205" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb17-206"><a href="#cb17-206" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-207"><a href="#cb17-207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-208"><a href="#cb17-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-209"><a href="#cb17-209" aria-hidden="true" tabindex="-1"></a><span class="fu"># Loop</span></span>
<span id="cb17-210"><a href="#cb17-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-213"><a href="#cb17-213" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-214"><a href="#cb17-214" aria-hidden="true" tabindex="-1"></a>get_cross_cohort_perf_combinations <span class="ot">&lt;-</span> <span class="cf">function</span>(model_df, meta_data, <span class="at">nobase=</span><span class="cn">FALSE</span>) {</span>
<span id="cb17-215"><a href="#cb17-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">all</span>(<span class="fu">c</span>(<span class="st">"target"</span>, <span class="st">"subject_id"</span>, <span class="st">"baseline"</span>) <span class="sc">%in%</span> <span class="fu">colnames</span>(model_df)))</span>
<span id="cb17-216"><a href="#cb17-216" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-217"><a href="#cb17-217" aria-hidden="true" tabindex="-1"></a>  model_dataset_df <span class="ot">&lt;-</span> model_df <span class="sc">%&gt;%</span></span>
<span id="cb17-218"><a href="#cb17-218" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(meta_data <span class="sc">%&gt;%</span> </span>
<span id="cb17-219"><a href="#cb17-219" aria-hidden="true" tabindex="-1"></a>                       dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb17-220"><a href="#cb17-220" aria-hidden="true" tabindex="-1"></a>                       dplyr<span class="sc">::</span><span class="fu">distinct</span>(),</span>
<span id="cb17-221"><a href="#cb17-221" aria-hidden="true" tabindex="-1"></a>                     <span class="at">by=</span><span class="st">"subject_id"</span>)</span>
<span id="cb17-222"><a href="#cb17-222" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-223"><a href="#cb17-223" aria-hidden="true" tabindex="-1"></a>  all_datasets <span class="ot">&lt;-</span> <span class="fu">unique</span>(model_dataset_df<span class="sc">$</span>dataset)</span>
<span id="cb17-224"><a href="#cb17-224" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-225"><a href="#cb17-225" aria-hidden="true" tabindex="-1"></a>  cross_dataset_out <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(all_datasets, <span class="cf">function</span>(test_dataset) {</span>
<span id="cb17-226"><a href="#cb17-226" aria-hidden="true" tabindex="-1"></a>    <span class="co"># test_dataset &lt;- all_datasets[1]</span></span>
<span id="cb17-227"><a href="#cb17-227" aria-hidden="true" tabindex="-1"></a>    train_dataset <span class="ot">&lt;-</span> all_datasets[<span class="sc">!</span>all_datasets <span class="sc">%in%</span> test_dataset]</span>
<span id="cb17-228"><a href="#cb17-228" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-229"><a href="#cb17-229" aria-hidden="true" tabindex="-1"></a>    train_df <span class="ot">&lt;-</span> model_dataset_df <span class="sc">%&gt;%</span></span>
<span id="cb17-230"><a href="#cb17-230" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(dataset <span class="sc">%in%</span> train_dataset) <span class="sc">%&gt;%</span></span>
<span id="cb17-231"><a href="#cb17-231" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(dataset, subject_id))</span>
<span id="cb17-232"><a href="#cb17-232" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-233"><a href="#cb17-233" aria-hidden="true" tabindex="-1"></a>    test_df <span class="ot">&lt;-</span> model_dataset_df <span class="sc">%&gt;%</span></span>
<span id="cb17-234"><a href="#cb17-234" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(dataset <span class="sc">%in%</span> test_dataset) <span class="sc">%&gt;%</span></span>
<span id="cb17-235"><a href="#cb17-235" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(dataset, subject_id))</span>
<span id="cb17-236"><a href="#cb17-236" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-237"><a href="#cb17-237" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (nobase) {</span>
<span id="cb17-238"><a href="#cb17-238" aria-hidden="true" tabindex="-1"></a>      train_df <span class="ot">&lt;-</span> train_df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>baseline)</span>
<span id="cb17-239"><a href="#cb17-239" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb17-240"><a href="#cb17-240" aria-hidden="true" tabindex="-1"></a>    model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">formula =</span> target <span class="sc">~</span> ., </span>
<span id="cb17-241"><a href="#cb17-241" aria-hidden="true" tabindex="-1"></a>                            <span class="at">data =</span> train_df, </span>
<span id="cb17-242"><a href="#cb17-242" aria-hidden="true" tabindex="-1"></a>                            <span class="at">num.trees =</span> <span class="dv">500</span>)</span>
<span id="cb17-243"><a href="#cb17-243" aria-hidden="true" tabindex="-1"></a>    predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(model, test_df)<span class="sc">$</span>predictions</span>
<span id="cb17-244"><a href="#cb17-244" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">tibble</span>(<span class="at">trainset =</span> <span class="fu">paste0</span>(train_dataset, <span class="at">collapse=</span><span class="st">"__"</span>),</span>
<span id="cb17-245"><a href="#cb17-245" aria-hidden="true" tabindex="-1"></a>                   <span class="at">testset =</span> test_dataset,</span>
<span id="cb17-246"><a href="#cb17-246" aria-hidden="true" tabindex="-1"></a>                   <span class="at">target =</span> test_df<span class="sc">$</span>target, </span>
<span id="cb17-247"><a href="#cb17-247" aria-hidden="true" tabindex="-1"></a>                   <span class="at">prediction =</span> predictions,</span>
<span id="cb17-248"><a href="#cb17-248" aria-hidden="true" tabindex="-1"></a>                   <span class="at">baseline =</span> test_df<span class="sc">$</span>baseline,</span>
<span id="cb17-249"><a href="#cb17-249" aria-hidden="true" tabindex="-1"></a>                   <span class="at">testset_mean =</span> <span class="fu">mean</span>(test_df<span class="sc">$</span>target))</span>
<span id="cb17-250"><a href="#cb17-250" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">%&gt;%</span></span>
<span id="cb17-251"><a href="#cb17-251" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb17-252"><a href="#cb17-252" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-253"><a href="#cb17-253" aria-hidden="true" tabindex="-1"></a>  <span class="co"># cross_dataset_out %&gt;%</span></span>
<span id="cb17-254"><a href="#cb17-254" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   ggplot() +</span></span>
<span id="cb17-255"><a href="#cb17-255" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   geom_point(aes(x=prediction, y=target)) +</span></span>
<span id="cb17-256"><a href="#cb17-256" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   ggdark::dark_mode(verbose=FALSE)</span></span>
<span id="cb17-257"><a href="#cb17-257" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-258"><a href="#cb17-258" aria-hidden="true" tabindex="-1"></a>  cross_dataset_out <span class="sc">%&gt;%</span></span>
<span id="cb17-259"><a href="#cb17-259" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(trainset, testset) <span class="sc">%&gt;%</span></span>
<span id="cb17-260"><a href="#cb17-260" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(</span>
<span id="cb17-261"><a href="#cb17-261" aria-hidden="true" tabindex="-1"></a>      <span class="at">test_mse =</span> <span class="fu">get_mse</span>(target, prediction),</span>
<span id="cb17-262"><a href="#cb17-262" aria-hidden="true" tabindex="-1"></a>      <span class="at">test_r2 =</span> <span class="fu">get_r2</span>(target, prediction),</span>
<span id="cb17-263"><a href="#cb17-263" aria-hidden="true" tabindex="-1"></a>      <span class="at">test_srho =</span> <span class="fu">get_spearman</span>(target, prediction),</span>
<span id="cb17-264"><a href="#cb17-264" aria-hidden="true" tabindex="-1"></a>      <span class="at">baseline_srho =</span> <span class="fu">get_spearman</span>(target, baseline),</span>
<span id="cb17-265"><a href="#cb17-265" aria-hidden="true" tabindex="-1"></a>      <span class="at">mse_same_cohort =</span> <span class="fu">get_mse</span>(target, testset_mean),</span>
<span id="cb17-266"><a href="#cb17-266" aria-hidden="true" tabindex="-1"></a>      <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb17-267"><a href="#cb17-267" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb17-268"><a href="#cb17-268" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-269"><a href="#cb17-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-270"><a href="#cb17-270" aria-hidden="true" tabindex="-1"></a>get_cross_cohort_perf_combinations_repeated <span class="ot">&lt;-</span> <span class="cf">function</span>(model_df, meta_data, </span>
<span id="cb17-271"><a href="#cb17-271" aria-hidden="true" tabindex="-1"></a>                                                        <span class="at">n_iter=</span><span class="dv">25</span>, <span class="at">nobase=</span><span class="cn">FALSE</span>) {</span>
<span id="cb17-272"><a href="#cb17-272" aria-hidden="true" tabindex="-1"></a>  <span class="co"># </span><span class="al">TODO</span></span>
<span id="cb17-273"><a href="#cb17-273" aria-hidden="true" tabindex="-1"></a>  df_long <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span>n_iter, <span class="cf">function</span>(iter) {</span>
<span id="cb17-274"><a href="#cb17-274" aria-hidden="true" tabindex="-1"></a>    <span class="fu">get_cross_cohort_perf_combinations</span>(<span class="at">model_df=</span>model_df, <span class="at">meta_data=</span>meta_data, </span>
<span id="cb17-275"><a href="#cb17-275" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">nobase=</span>nobase) <span class="sc">%&gt;%</span></span>
<span id="cb17-276"><a href="#cb17-276" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">iteration=</span>iter)</span>
<span id="cb17-277"><a href="#cb17-277" aria-hidden="true" tabindex="-1"></a>  }) <span class="sc">%&gt;%</span></span>
<span id="cb17-278"><a href="#cb17-278" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb17-279"><a href="#cb17-279" aria-hidden="true" tabindex="-1"></a>  df_out <span class="ot">&lt;-</span> df_long <span class="sc">%&gt;%</span></span>
<span id="cb17-280"><a href="#cb17-280" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(trainset, testset) <span class="sc">%&gt;%</span></span>
<span id="cb17-281"><a href="#cb17-281" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">test_srho_mean =</span> <span class="fu">mean</span>(test_srho),</span>
<span id="cb17-282"><a href="#cb17-282" aria-hidden="true" tabindex="-1"></a>                     <span class="at">test_srho_sd =</span> <span class="fu">sd</span>(test_srho),</span>
<span id="cb17-283"><a href="#cb17-283" aria-hidden="true" tabindex="-1"></a>                     <span class="at">baseline_srho_mean =</span> <span class="fu">mean</span>(baseline_srho),</span>
<span id="cb17-284"><a href="#cb17-284" aria-hidden="true" tabindex="-1"></a>                     <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb17-285"><a href="#cb17-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df_out)</span>
<span id="cb17-286"><a href="#cb17-286" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-287"><a href="#cb17-287" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-288"><a href="#cb17-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-289"><a href="#cb17-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-292"><a href="#cb17-292" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-293"><a href="#cb17-293" aria-hidden="true" tabindex="-1"></a>permute_columns <span class="ot">&lt;-</span> <span class="cf">function</span>(mtx, <span class="at">seed=</span><span class="dv">42</span>) {</span>
<span id="cb17-294"><a href="#cb17-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb17-295"><a href="#cb17-295" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (col_idx <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(mtx)) {</span>
<span id="cb17-296"><a href="#cb17-296" aria-hidden="true" tabindex="-1"></a>    mtx[, col_idx] <span class="ot">&lt;-</span> <span class="fu">sample</span>(mtx[, col_idx], <span class="at">size=</span><span class="fu">nrow</span>(mtx), <span class="at">replace=</span><span class="cn">TRUE</span>)</span>
<span id="cb17-297"><a href="#cb17-297" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-298"><a href="#cb17-298" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(mtx)</span>
<span id="cb17-299"><a href="#cb17-299" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-300"><a href="#cb17-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-301"><a href="#cb17-301" aria-hidden="true" tabindex="-1"></a>tibble_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb17-302"><a href="#cb17-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-303"><a href="#cb17-303" aria-hidden="true" tabindex="-1"></a>metadata_covariates <span class="ot">&lt;-</span> <span class="fu">get_metadata_covariates</span>(meta_data)</span>
<span id="cb17-304"><a href="#cb17-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-305"><a href="#cb17-305" aria-hidden="true" tabindex="-1"></a>model_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb17-306"><a href="#cb17-306" aria-hidden="true" tabindex="-1"></a>  <span class="st">"metadata"</span> <span class="ot">=</span> metadata_covariates,</span>
<span id="cb17-307"><a href="#cb17-307" aria-hidden="true" tabindex="-1"></a>  <span class="st">"features"</span> <span class="ot">=</span> feature_df,</span>
<span id="cb17-308"><a href="#cb17-308" aria-hidden="true" tabindex="-1"></a>  <span class="st">"features_permuted"</span> <span class="ot">=</span> <span class="fu">permute_columns</span>(feature_df),</span>
<span id="cb17-309"><a href="#cb17-309" aria-hidden="true" tabindex="-1"></a>  <span class="st">"features_permuted_nobase"</span> <span class="ot">=</span> <span class="fu">permute_columns</span>(feature_df),</span>
<span id="cb17-310"><a href="#cb17-310" aria-hidden="true" tabindex="-1"></a>  <span class="st">"metadata_plus_features"</span> <span class="ot">=</span> feature_df <span class="sc">%&gt;%</span> </span>
<span id="cb17-311"><a href="#cb17-311" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(metadata_covariates, <span class="at">by=</span><span class="st">"subject_id"</span>)</span>
<span id="cb17-312"><a href="#cb17-312" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-313"><a href="#cb17-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-314"><a href="#cb17-314" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (task <span class="cf">in</span> task_meta) {</span>
<span id="cb17-315"><a href="#cb17-315" aria-hidden="true" tabindex="-1"></a>  <span class="co"># task &lt;- task_meta[[3]]</span></span>
<span id="cb17-316"><a href="#cb17-316" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-317"><a href="#cb17-317" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (model_name <span class="cf">in</span> <span class="fu">names</span>(model_list)) {</span>
<span id="cb17-318"><a href="#cb17-318" aria-hidden="true" tabindex="-1"></a>    <span class="co"># model_name &lt;- "features_permuted"</span></span>
<span id="cb17-319"><a href="#cb17-319" aria-hidden="true" tabindex="-1"></a>    <span class="co">#print(model_name)</span></span>
<span id="cb17-320"><a href="#cb17-320" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-321"><a href="#cb17-321" aria-hidden="true" tabindex="-1"></a>    model_df <span class="ot">&lt;-</span> target_list[[task<span class="sc">$</span>name]] <span class="sc">%&gt;%</span></span>
<span id="cb17-322"><a href="#cb17-322" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">left_join</span>(model_list[[model_name]], <span class="at">by=</span><span class="st">"subject_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-323"><a href="#cb17-323" aria-hidden="true" tabindex="-1"></a>      tidyr<span class="sc">::</span><span class="fu">drop_na</span>()</span>
<span id="cb17-324"><a href="#cb17-324" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-325"><a href="#cb17-325" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (model_name <span class="sc">==</span> <span class="st">"features_permuted_nobase"</span>) {</span>
<span id="cb17-326"><a href="#cb17-326" aria-hidden="true" tabindex="-1"></a>      nobase <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb17-327"><a href="#cb17-327" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb17-328"><a href="#cb17-328" aria-hidden="true" tabindex="-1"></a>      nobase <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb17-329"><a href="#cb17-329" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb17-330"><a href="#cb17-330" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb17-331"><a href="#cb17-331" aria-hidden="true" tabindex="-1"></a>    task_df <span class="ot">&lt;-</span> <span class="fu">get_cross_cohort_perf_combinations_repeated</span>(<span class="at">model_df=</span>model_df,</span>
<span id="cb17-332"><a href="#cb17-332" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">meta_data=</span>meta_data, </span>
<span id="cb17-333"><a href="#cb17-333" aria-hidden="true" tabindex="-1"></a>                                                           <span class="at">nobase=</span>nobase) <span class="sc">%&gt;%</span></span>
<span id="cb17-334"><a href="#cb17-334" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">task =</span> task<span class="sc">$</span>name, <span class="at">model =</span> model_name) <span class="sc">%&gt;%</span></span>
<span id="cb17-335"><a href="#cb17-335" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="fu">across</span>(<span class="at">.cols=</span><span class="fu">c</span>(test_srho_mean, test_srho_sd, baseline_srho_mean),</span>
<span id="cb17-336"><a href="#cb17-336" aria-hidden="true" tabindex="-1"></a>                           <span class="at">.fns=</span><span class="sc">~</span><span class="fu">round</span>(.x, <span class="dv">2</span>))) </span>
<span id="cb17-337"><a href="#cb17-337" aria-hidden="true" tabindex="-1"></a>    tibble_list <span class="ot">&lt;-</span> rlist<span class="sc">::</span><span class="fu">list.append</span>(tibble_list, task_df)</span>
<span id="cb17-338"><a href="#cb17-338" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-339"><a href="#cb17-339" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb17-340"><a href="#cb17-340" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-341"><a href="#cb17-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-342"><a href="#cb17-342" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(tibble_list) <span class="sc">%&gt;%</span></span>
<span id="cb17-343"><a href="#cb17-343" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb17-344"><a href="#cb17-344" aria-hidden="true" tabindex="-1"></a>      <span class="at">task =</span> <span class="fu">factor</span>(task, <span class="at">levels=</span><span class="fu">paste0</span>(<span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">"task_"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>), <span class="at">each=</span><span class="dv">2</span>), <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="dv">4</span>))),</span>
<span id="cb17-345"><a href="#cb17-345" aria-hidden="true" tabindex="-1"></a>      <span class="at">model =</span> <span class="fu">factor</span>(model, <span class="at">levels=</span><span class="fu">names</span>(model_list))</span>
<span id="cb17-346"><a href="#cb17-346" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb17-347"><a href="#cb17-347" aria-hidden="true" tabindex="-1"></a>df</span>
<span id="cb17-348"><a href="#cb17-348" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-349"><a href="#cb17-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-350"><a href="#cb17-350" aria-hidden="true" tabindex="-1"></a><span class="in">```{r fig.width=10, fig.height=6}</span></span>
<span id="cb17-351"><a href="#cb17-351" aria-hidden="true" tabindex="-1"></a><span class="in">df %&gt;%</span></span>
<span id="cb17-352"><a href="#cb17-352" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(-test_srho_sd) %&gt;%</span></span>
<span id="cb17-353"><a href="#cb17-353" aria-hidden="true" tabindex="-1"></a><span class="in">  tidyr::pivot_longer(cols=c(test_srho_mean, baseline_srho_mean),</span></span>
<span id="cb17-354"><a href="#cb17-354" aria-hidden="true" tabindex="-1"></a><span class="in">                      names_to="metric", values_to="value") %&gt;%</span></span>
<span id="cb17-355"><a href="#cb17-355" aria-hidden="true" tabindex="-1"></a><span class="in">  ggplot() +</span></span>
<span id="cb17-356"><a href="#cb17-356" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_vline(xintercept=c(0, 0.5), color="grey", linetype="dashed") +</span></span>
<span id="cb17-357"><a href="#cb17-357" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_point(aes(y=task, x=value, color=testset, shape=metric), alpha=0.50) +</span></span>
<span id="cb17-358"><a href="#cb17-358" aria-hidden="true" tabindex="-1"></a><span class="in">  ggdark::dark_mode(verbose=TRUE) +</span></span>
<span id="cb17-359"><a href="#cb17-359" aria-hidden="true" tabindex="-1"></a><span class="in">  facet_wrap(~model)</span></span>
<span id="cb17-360"><a href="#cb17-360" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>