<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.45">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Philipp Sven Lars Schäfer">
<meta name="dcterms.date" content="2024-09-14">

<title>Baseline Models V1 – PSLS @ 3rd CMI-PB Challenge</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">PSLS @ 3rd CMI-PB Challenge</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/psl-schaefer"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://x.com/psl_schaefer"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/in/philipp-sl-schaefer/"> <i class="bi bi-linkedin" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#packages" id="toc-packages" class="nav-link active" data-scroll-target="#packages"><span class="header-section-number">1</span> Packages</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2</span> Data</a></li>
  <li><a href="#task-1" id="toc-task-1" class="nav-link" data-scroll-target="#task-1"><span class="header-section-number">3</span> Task 1</a>
  <ul class="collapse">
  <li><a href="#a" id="toc-a" class="nav-link" data-scroll-target="#a"><span class="header-section-number">3.1</span> 1a</a></li>
  <li><a href="#b" id="toc-b" class="nav-link" data-scroll-target="#b"><span class="header-section-number">3.2</span> 1b</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Baseline Models V1</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Philipp Sven Lars Schäfer </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 14, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="packages" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Packages</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick) <span class="co"># formatting</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ranger)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>knit_hooks<span class="sc">$</span><span class="fu">set</span>(<span class="at">crop =</span> knitr<span class="sc">::</span>hook_pdfcrop) <span class="co"># formatting</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Data</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># loading</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>meta_df <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">&lt;-</span> <span class="fu">read_harmonized_experimental_data</span>(input_dir)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># filtering</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>meta_df <span class="ot">&lt;-</span> <span class="fu">filter_harmonized_meta_data</span>(<span class="at">meta_data=</span>meta_df, <span class="at">experimental_data=</span>data_list)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">&lt;-</span> <span class="fu">filter_harmonized_experimental_data</span>(<span class="at">meta_data=</span>meta_df, <span class="at">experimental_data=</span>data_list)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Removed 429 specimens from pbmc_cell_frequencybecause missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Removed 0 specimens from pbmc_gene_expressionbecause missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Removed 3989 specimens from plasma_antibody_levelsbecause missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Removed 0 specimens from plasma_cytokine_concentrations_by_legendplexbecause missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Removed 495 specimens from plasma_cytokine_concentrations_by_olinkbecause missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Removed 0 specimens from t_cell_activationbecause missing in meta data</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Removed 0 specimens from t_cell_polarizationbecause missing in meta data</code></pre>
</div>
</div>
</section>
<section id="task-1" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Task 1</h1>
<ul>
<li><p>10-fold CV</p></li>
<li><p>Random Forest</p></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>df_targets <span class="ot">&lt;-</span> meta_df <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">14</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(data_list<span class="sc">$</span>plasma_antibody_levels, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(isotype_antigen <span class="sc">==</span> <span class="st">"IgG_PT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, planned_day_relative_to_boost, MFI_normalised) <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"planned_day_relative_to_boost"</span>,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">values_from=</span><span class="st">"MFI_normalised"</span>, <span class="at">names_prefix =</span> <span class="st">"day_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">fc =</span> day_14 <span class="sc">/</span> day_0) <span class="sc">%&gt;%</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">drop_na</span>()</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>demo_covariates <span class="ot">&lt;-</span> meta_df <span class="sc">%&gt;%</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, biological_sex, ethnicity, </span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>                age_at_boost, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">infancy_vac_value=</span><span class="dv">1</span>, <span class="at">biological_sex_value=</span><span class="dv">1</span>, </span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">ethnicity_value=</span><span class="dv">1</span>, <span class="at">dataset_value=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>infancy_vac, <span class="at">values_from=</span>infancy_vac_value, </span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>                     <span class="at">names_prefix=</span><span class="st">"infancy_vac_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>biological_sex, <span class="at">values_from=</span>biological_sex_value,</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>                     <span class="at">names_prefix=</span><span class="st">"biological_sex_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>ethnicity, <span class="at">values_from=</span>ethnicity_value, </span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>                     <span class="at">names_prefix=</span><span class="st">"ethnicity_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>dataset, <span class="at">values_from=</span>dataset_value, </span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>                     <span class="at">names_prefix=</span><span class="st">"dataset_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="at">.fns =</span> <span class="sc">~</span><span class="fu">replace_na</span>(.,<span class="dv">0</span>)))</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>pbmc_covariates <span class="ot">&lt;-</span> meta_df <span class="sc">%&gt;%</span></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">inner_join</span>(data_list<span class="sc">$</span>pbmc_cell_frequency, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, cell_type_name, percent_live_cell) <span class="sc">%&gt;%</span></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"cell_type_name"</span>, <span class="at">values_from=</span><span class="st">"percent_live_cell"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="at">.fns =</span> <span class="sc">~</span><span class="fu">replace_na</span>(.,<span class="dv">0</span>)))</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>df_all <span class="ot">&lt;-</span> demo_covariates <span class="sc">%&gt;%</span></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">inner_join</span>(pbmc_covariates, <span class="at">by=</span><span class="st">"subject_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">inner_join</span>(df_targets, <span class="at">by=</span><span class="st">"subject_id"</span>)</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a>pbmc_feature_names <span class="ot">&lt;-</span> <span class="fu">unique</span>(data_list<span class="sc">$</span>pbmc_cell_frequency<span class="sc">$</span>cell_type_name)</span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>demo_feature_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(demo_covariates)[<span class="sc">!</span>(<span class="fu">colnames</span>(demo_covariates) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"subject_id"</span>))]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="a" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="a"><span class="header-section-number">3.1</span> 1a</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(demo_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(day_14), </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>rf_model</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Ranger result

Call:
 NULL 

Type:                             Regression 
Number of trees:                  500 
Sample size:                      71 
Number of independent variables:  13 
Mtry:                             3 
Target node size:                 5 
Variable importance mode:         permutation 
Splitrule:                        variance 
OOB prediction error (MSE):       16.2515 
R squared (OOB):                  0.4304134 </code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(pbmc_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(day_14), </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>rf_model</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Ranger result

Call:
 NULL 

Type:                             Regression 
Number of trees:                  500 
Sample size:                      71 
Number of independent variables:  40 
Mtry:                             6 
Target node size:                 5 
Variable importance mode:         permutation 
Splitrule:                        variance 
OOB prediction error (MSE):       24.20728 
R squared (OOB):                  0.1515772 </code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(demo_feature_names, pbmc_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(day_14), </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>rf_model</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Ranger result

Call:
 NULL 

Type:                             Regression 
Number of trees:                  500 
Sample size:                      71 
Number of independent variables:  52 
Mtry:                             7 
Target node size:                 5 
Variable importance mode:         permutation 
Splitrule:                        variance 
OOB prediction error (MSE):       21.49207 
R squared (OOB):                  0.2467406 </code></pre>
</div>
</div>
</section>
<section id="b" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="b"><span class="header-section-number">3.2</span> 1b</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(demo_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(fc), </span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>rf_model</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Ranger result

Call:
 NULL 

Type:                             Regression 
Number of trees:                  500 
Sample size:                      71 
Number of independent variables:  13 
Mtry:                             3 
Target node size:                 5 
Variable importance mode:         permutation 
Splitrule:                        variance 
OOB prediction error (MSE):       44.1938 
R squared (OOB):                  0.3437689 </code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(pbmc_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(fc), </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>rf_model</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Ranger result

Call:
 NULL 

Type:                             Regression 
Number of trees:                  500 
Sample size:                      71 
Number of independent variables:  40 
Mtry:                             6 
Target node size:                 5 
Variable importance mode:         permutation 
Splitrule:                        variance 
OOB prediction error (MSE):       58.40528 
R squared (OOB):                  0.1327435 </code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(demo_feature_names, pbmc_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(fc), </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>rf_model</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Ranger result

Call:
 NULL 

Type:                             Regression 
Number of trees:                  500 
Sample size:                      71 
Number of independent variables:  52 
Mtry:                             7 
Target node size:                 5 
Variable importance mode:         permutation 
Splitrule:                        variance 
OOB prediction error (MSE):       58.82191 
R squared (OOB):                  0.1265569 </code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>df_all <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>day_0, <span class="at">y=</span>fc)) <span class="sc">+</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>() <span class="sc">+</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using method = 'loess' and formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="model_1_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(jsonlite)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'jsonlite'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:purrr':

    flatten</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httr)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Base URL for the API</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>base_url <span class="ot">&lt;-</span> <span class="st">"https://www.cmi-pb.org/api/v5"</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co">#base_url &lt;- "https://www.cmi-pb.org/api"</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the endpoints you want to query</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>endpoints <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"/cell_type"</span>, <span class="st">"/gene"</span>)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through endpoints and download JSON files</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (endpoint <span class="cf">in</span> endpoints) {</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Construct full URL</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  url <span class="ot">&lt;-</span> <span class="fu">paste0</span>(base_url, endpoint)</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Make GET request</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>  response <span class="ot">&lt;-</span> <span class="fu">GET</span>(url)</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if the request was successful</span></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">status_code</span>(response) <span class="sc">==</span> <span class="dv">200</span>) {</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save the JSON content to a file</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>    json_content <span class="ot">&lt;-</span> <span class="fu">content</span>(response, <span class="at">as =</span> <span class="st">"text"</span>, <span class="at">encoding =</span> <span class="st">"UTF-8"</span>)</span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write</span>(json_content, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"output_"</span>, <span class="fu">basename</span>(endpoint), <span class="st">".json"</span>))</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Downloaded:"</span>, endpoint, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Failed to download"</span>, endpoint, <span class="st">"Status code:"</span>, <span class="fu">status_code</span>(response), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Downloaded: /cell_type 
Downloaded: /gene </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>celltype_info <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(jsonlite<span class="sc">::</span><span class="fu">read_json</span>(<span class="st">"output_cell_type.json"</span>), <span class="cf">function</span>(l) {</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(l)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(jsonlite<span class="sc">::</span><span class="fu">read_json</span>(<span class="st">"output_gene.json"</span>)) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>List of 63967
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000283891.1"
  ..$ gene_symbol              : chr "MIR628"
  ..$ gene_name                : chr "microRNA 628 [Source:HGNC Symbol;Acc:HGNC:32884]"
  ..$ synonyms                 : chr "MIRN628,hsa-mir-628"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "693213"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 55372940
  ..$ end                      : int 55373034
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 628 [Source:HGNC Symbol;Acc:HGNC:32884]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000251931.1"
  ..$ gene_symbol              : chr "RNU6-871P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 871, pseudogene [Source:HGNC Symbol;Acc:HGNC:47834]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479950"
  ..$ summary                  : NULL
  ..$ start                    : int 59450673
  ..$ end                      : int 59450772
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, U6 small nuclear 871, pseudogene [Source:HGNC Symbol;Acc:HGNC:47834]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207766.1"
  ..$ gene_symbol              : chr "MIR626"
  ..$ gene_name                : chr "microRNA 626 [Source:HGNC Symbol;Acc:HGNC:32882]"
  ..$ synonyms                 : chr "MIRN626,hsa-mir-626"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "693211"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 41691585
  ..$ end                      : int 41691678
  ..$ strand                   : int 1
  ..$ description              : chr "microRNA 626 [Source:HGNC Symbol;Acc:HGNC:32882]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000275323.3"
  ..$ gene_symbol              : chr "RPS9"
  ..$ gene_name                : chr "ribosomal protein S9 [Source:HGNC Symbol;Acc:HGNC:10442]"
  ..$ synonyms                 : chr "S9,S9"
  ..$ biotype                  : chr "protein_coding"
  ..$ ncbi_id                  : chr "6203"
  ..$ summary                  : chr "Ribosomes, the organelles that catalyze protein synthesis, consist of a small 40S subunit and a large 60S subun"| __truncated__
  ..$ start                    : int 54201473
  ..$ end                      : int 54208260
  ..$ strand                   : int 1
  ..$ description              : chr "ribosomal protein S9 [Source:HGNC Symbol;Acc:HGNC:10442]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000276678.1"
  ..$ gene_symbol              : chr "GHRLOS"
  ..$ gene_name                : chr "Ghrelin opposite strand RNA conserved region [Source:RFAM;Acc:RF02126]"
  ..$ synonyms                 : chr "NCRNA00068,GHRL-AS1"
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "100126793"
  ..$ summary                  : chr "This gene is an antisense gene of the ghrelin/obestatin prepropeptide gene. Alternatively spliced transcript va"| __truncated__
  ..$ start                    : int 10287413
  ..$ end                      : int 10287482
  ..$ strand                   : int 1
  ..$ description              : chr "Ghrelin opposite strand RNA conserved region [Source:RFAM;Acc:RF02126]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207260.1"
  ..$ gene_symbol              : chr "RNU6-35P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 35, pseudogene [Source:HGNC Symbol;Acc:HGNC:34279]"
  ..$ synonyms                 : chr "RNU6-35"
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "100873750"
  ..$ summary                  : NULL
  ..$ start                    : int 109992325
  ..$ end                      : int 109992431
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6 small nuclear 35, pseudogene [Source:HGNC Symbol;Acc:HGNC:34279]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000265993.1"
  ..$ gene_symbol              : chr "MIR5694"
  ..$ gene_name                : chr "microRNA 5694 [Source:HGNC Symbol;Acc:HGNC:43530]"
  ..$ synonyms                 : chr "hsa-mir-5694"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100847064"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 67441855
  ..$ end                      : int 67441930
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 5694 [Source:HGNC Symbol;Acc:HGNC:43530]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207185.1"
  ..$ gene_symbol              : chr "RNU6-1157P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 1157, pseudogene [Source:HGNC Symbol;Acc:HGNC:48120]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106481545"
  ..$ summary                  : NULL
  ..$ start                    : int 118593988
  ..$ end                      : int 118594093
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6 small nuclear 1157, pseudogene [Source:HGNC Symbol;Acc:HGNC:48120]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000283793.1"
  ..$ gene_symbol              : chr "MIR6861"
  ..$ gene_name                : chr "microRNA 6861 [Source:HGNC Symbol;Acc:HGNC:50129]"
  ..$ synonyms                 : chr "hsa-mir-6861"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "102465519"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 112163258
  ..$ end                      : int 112163321
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 6861 [Source:HGNC Symbol;Acc:HGNC:50129]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000201545.1"
  ..$ gene_symbol              : chr "RNU4-85P"
  ..$ gene_name                : chr "RNA, U4 small nuclear 85, pseudogene [Source:HGNC Symbol;Acc:HGNC:47021]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479597"
  ..$ summary                  : NULL
  ..$ start                    : int 19996803
  ..$ end                      : int 19996925
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U4 small nuclear 85, pseudogene [Source:HGNC Symbol;Acc:HGNC:47021]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000272439.1"
  ..$ gene_symbol              : chr "RNU6-91P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 91, pseudogene [Source:HGNC Symbol;Acc:HGNC:47054]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479606"
  ..$ summary                  : NULL
  ..$ start                    : int 99444306
  ..$ end                      : int 99444412
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6 small nuclear 91, pseudogene [Source:HGNC Symbol;Acc:HGNC:47054]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000241821.3"
  ..$ gene_symbol              : chr "RN7SL170P"
  ..$ gene_name                : chr "RNA, 7SL, cytoplasmic 170, pseudogene [Source:HGNC Symbol;Acc:HGNC:46186]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106479283"
  ..$ summary                  : NULL
  ..$ start                    : int 56688401
  ..$ end                      : int 56688695
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 7SL, cytoplasmic 170, pseudogene [Source:HGNC Symbol;Acc:HGNC:46186]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000265889.1"
  ..$ gene_symbol              : chr "Y_RNA"
  ..$ gene_name                : chr "Y RNA [Source:RFAM;Acc:RF00019]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 36285312
  ..$ end                      : int 36285413
  ..$ strand                   : int 1
  ..$ description              : chr "Y RNA [Source:RFAM;Acc:RF00019]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207194.1"
  ..$ gene_symbol              : chr "RNU6-1026P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 1026, pseudogene [Source:HGNC Symbol;Acc:HGNC:47989]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106480631"
  ..$ summary                  : NULL
  ..$ start                    : int 50582404
  ..$ end                      : int 50582511
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6 small nuclear 1026, pseudogene [Source:HGNC Symbol;Acc:HGNC:47989]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000283621.1"
  ..$ gene_symbol              : chr "MIR6759"
  ..$ gene_name                : chr "microRNA 6759 [Source:HGNC Symbol;Acc:HGNC:50063]"
  ..$ synonyms                 : chr "hsa-mir-6759"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "102466729"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 57748618
  ..$ end                      : int 57748682
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 6759 [Source:HGNC Symbol;Acc:HGNC:50063]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000264032.1"
  ..$ gene_symbol              : chr "MIR4491"
  ..$ gene_name                : chr "microRNA 4491 [Source:HGNC Symbol;Acc:HGNC:41808]"
  ..$ synonyms                 : chr "hsa-mir-4491"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100616330"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 111347757
  ..$ end                      : int 111347824
  ..$ strand                   : int 1
  ..$ description              : chr "microRNA 4491 [Source:HGNC Symbol;Acc:HGNC:41808]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207172.1"
  ..$ gene_symbol              : chr "RNU6-1162P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 1162, pseudogene [Source:HGNC Symbol;Acc:HGNC:48125]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106480078"
  ..$ summary                  : NULL
  ..$ start                    : int 63630678
  ..$ end                      : int 63630781
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6 small nuclear 1162, pseudogene [Source:HGNC Symbol;Acc:HGNC:48125]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000206629.1"
  ..$ gene_symbol              : chr "RNU1-63P"
  ..$ gene_name                : chr "RNA, U1 small nuclear 63, pseudogene [Source:HGNC Symbol;Acc:HGNC:48405]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106480664"
  ..$ summary                  : NULL
  ..$ start                    : int 67429591
  ..$ end                      : int 67429754
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, U1 small nuclear 63, pseudogene [Source:HGNC Symbol;Acc:HGNC:48405]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000283558.1"
  ..$ gene_symbol              : chr "hsa-mir-3158-1"
  ..$ gene_name                : chr "hsa-mir-3158-1 [Source:miRBase;Acc:MI0014186]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 101601417
  ..$ end                      : int 101601497
  ..$ strand                   : int -1
  ..$ description              : chr "hsa-mir-3158-1 [Source:miRBase;Acc:MI0014186]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000252735.1"
  ..$ gene_symbol              : chr "RNU6-528P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 528, pseudogene [Source:HGNC Symbol;Acc:HGNC:47491]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479800"
  ..$ summary                  : NULL
  ..$ start                    : int 33338867
  ..$ end                      : int 33338966
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6 small nuclear 528, pseudogene [Source:HGNC Symbol;Acc:HGNC:47491]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000252319.1"
  ..$ gene_symbol              : chr "Y_RNA"
  ..$ gene_name                : chr "Y RNA [Source:RFAM;Acc:RF00019]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 8720034
  ..$ end                      : int 8720132
  ..$ strand                   : int -1
  ..$ description              : chr "Y RNA [Source:RFAM;Acc:RF00019]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000201912.1"
  ..$ gene_symbol              : chr "RN7SKP189"
  ..$ gene_name                : chr "RNA, 7SK small nuclear pseudogene 189 [Source:HGNC Symbol;Acc:HGNC:45913]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106479180"
  ..$ summary                  : NULL
  ..$ start                    : int 145057108
  ..$ end                      : int 145057437
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 7SK small nuclear pseudogene 189 [Source:HGNC Symbol;Acc:HGNC:45913]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000274122.1"
  ..$ gene_symbol              : chr "Metazoa_SRP"
  ..$ gene_name                : chr "Metazoan signal recognition particle RNA [Source:RFAM;Acc:RF00017]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 28387160
  ..$ end                      : int 28387417
  ..$ strand                   : int 1
  ..$ description              : chr "Metazoan signal recognition particle RNA [Source:RFAM;Acc:RF00017]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000200162.1"
  ..$ gene_symbol              : chr "Y_RNA"
  ..$ gene_name                : chr "Y RNA [Source:RFAM;Acc:RF00019]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 153607377
  ..$ end                      : int 153607479
  ..$ strand                   : int 1
  ..$ description              : chr "Y RNA [Source:RFAM;Acc:RF00019]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000200163.1"
  ..$ gene_symbol              : chr "SNORD115-18"
  ..$ gene_name                : chr "small nucleolar RNA, C/D box 115-18 [Source:HGNC Symbol;Acc:HGNC:33037]"
  ..$ synonyms                 : chr "HBII-52-18"
  ..$ biotype                  : chr "snoRNA"
  ..$ ncbi_id                  : chr "100033456"
  ..$ summary                  : NULL
  ..$ start                    : int 25203227
  ..$ end                      : int 25203308
  ..$ strand                   : int 1
  ..$ description              : chr "small nucleolar RNA, C/D box 115-18 [Source:HGNC Symbol;Acc:HGNC:33037]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000276764.1"
  ..$ gene_symbol              : chr "Metazoa_SRP"
  ..$ gene_name                : chr "Metazoan signal recognition particle RNA [Source:RFAM;Acc:RF00017]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 18958147
  ..$ end                      : int 18958413
  ..$ strand                   : int -1
  ..$ description              : chr "Metazoan signal recognition particle RNA [Source:RFAM;Acc:RF00017]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000275994.1"
  ..$ gene_symbol              : chr "SNORA24"
  ..$ gene_name                : chr "small nucleolar RNA, H/ACA box 24 [Source:HGNC Symbol;Acc:HGNC:32614]"
  ..$ synonyms                 : chr "SNORA24A,ACA24"
  ..$ biotype                  : chr "snoRNA"
  ..$ ncbi_id                  : chr "677809"
  ..$ summary                  : NULL
  ..$ start                    : int 118279190
  ..$ end                      : int 118279320
  ..$ strand                   : int 1
  ..$ description              : chr "small nucleolar RNA, H/ACA box 24 [Source:HGNC Symbol;Acc:HGNC:32614]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000277950.1"
  ..$ gene_symbol              : chr "Metazoa_SRP"
  ..$ gene_name                : chr "Metazoan signal recognition particle RNA [Source:RFAM;Acc:RF00017]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 20070716
  ..$ end                      : int 20070965
  ..$ strand                   : int 1
  ..$ description              : chr "Metazoan signal recognition particle RNA [Source:RFAM;Acc:RF00017]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000222308.1"
  ..$ gene_symbol              : chr "RNA5SP198"
  ..$ gene_name                : chr "RNA, 5S ribosomal pseudogene 198 [Source:HGNC Symbol;Acc:HGNC:43098]"
  ..$ synonyms                 : chr "RN5S198"
  ..$ biotype                  : chr "rRNA"
  ..$ ncbi_id                  : chr "100873460"
  ..$ summary                  : NULL
  ..$ start                    : int 151875566
  ..$ end                      : int 151875692
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, 5S ribosomal pseudogene 198 [Source:HGNC Symbol;Acc:HGNC:43098]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000201965.1"
  ..$ gene_symbol              : chr "Y_RNA"
  ..$ gene_name                : chr "Y RNA [Source:RFAM;Acc:RF00019]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 38125292
  ..$ end                      : int 38125394
  ..$ strand                   : int -1
  ..$ description              : chr "Y RNA [Source:RFAM;Acc:RF00019]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000252264.1"
  ..$ gene_symbol              : chr "RNU7-153P"
  ..$ gene_name                : chr "RNA, U7 small nuclear 153 pseudogene [Source:HGNC Symbol;Acc:HGNC:45687]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479082"
  ..$ summary                  : NULL
  ..$ start                    : int 14319455
  ..$ end                      : int 14319516
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U7 small nuclear 153 pseudogene [Source:HGNC Symbol;Acc:HGNC:45687]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000200121.2"
  ..$ gene_symbol              : chr "Y_RNA"
  ..$ gene_name                : chr "Y RNA [Source:RFAM;Acc:RF00019]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 175176493
  ..$ end                      : int 175176586
  ..$ strand                   : int -1
  ..$ description              : chr "Y RNA [Source:RFAM;Acc:RF00019]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000251770.1"
  ..$ gene_symbol              : chr "RNA5SP486"
  ..$ gene_name                : chr "RNA, 5S ribosomal pseudogene 486 [Source:HGNC Symbol;Acc:HGNC:43386]"
  ..$ synonyms                 : chr "RN5S486"
  ..$ biotype                  : chr "rRNA"
  ..$ ncbi_id                  : chr "100873728"
  ..$ summary                  : NULL
  ..$ start                    : int 47873191
  ..$ end                      : int 47873273
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 5S ribosomal pseudogene 486 [Source:HGNC Symbol;Acc:HGNC:43386]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000202217.1"
  ..$ gene_symbol              : chr "RN7SKP47"
  ..$ gene_name                : chr "RNA, 7SK small nuclear pseudogene 47 [Source:HGNC Symbol;Acc:HGNC:45771]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106481974"
  ..$ summary                  : NULL
  ..$ start                    : int 74799895
  ..$ end                      : int 74800222
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 7SK small nuclear pseudogene 47 [Source:HGNC Symbol;Acc:HGNC:45771]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000244748.3"
  ..$ gene_symbol              : chr "RN7SL153P"
  ..$ gene_name                : chr "RNA, 7SL, cytoplasmic 153, pseudogene [Source:HGNC Symbol;Acc:HGNC:46169]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106480490"
  ..$ summary                  : NULL
  ..$ start                    : int 72314397
  ..$ end                      : int 72314688
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, 7SL, cytoplasmic 153, pseudogene [Source:HGNC Symbol;Acc:HGNC:46169]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000251972.1"
  ..$ gene_symbol              : chr "RNU6-123P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 123, pseudogene [Source:HGNC Symbol;Acc:HGNC:47086]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479620"
  ..$ summary                  : NULL
  ..$ start                    : int 25943044
  ..$ end                      : int 25943145
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, U6 small nuclear 123, pseudogene [Source:HGNC Symbol;Acc:HGNC:47086]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000212607.1"
  ..$ gene_symbol              : chr "SNORA3B"
  ..$ gene_name                : chr "small nucleolar RNA, H/ACA box 3B [Source:HGNC Symbol;Acc:HGNC:32638]"
  ..$ synonyms                 : chr "ACA3-2,SNORA45B,SNORA45"
  ..$ biotype                  : chr "snoRNA"
  ..$ ncbi_id                  : chr "677826"
  ..$ summary                  : chr "Small nucleolar RNAs (snoRNAs) are small noncoding RNAs involved in RNA processing. Box H/ACA snoRNAs, such as "| __truncated__
  ..$ start                    : int 8685439
  ..$ end                      : int 8685569
  ..$ strand                   : int 1
  ..$ description              : chr "small nucleolar RNA, H/ACA box 3B [Source:HGNC Symbol;Acc:HGNC:32638]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000283290.1"
  ..$ gene_symbol              : chr "U6"
  ..$ gene_name                : chr "U6 spliceosomal RNA [Source:RFAM;Acc:RF00026]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 24352057
  ..$ end                      : int 24352157
  ..$ strand                   : int -1
  ..$ description              : chr "U6 spliceosomal RNA [Source:RFAM;Acc:RF00026]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207321.1"
  ..$ gene_symbol              : chr "RNU6-160P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 160, pseudogene [Source:HGNC Symbol;Acc:HGNC:47123]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479637"
  ..$ summary                  : NULL
  ..$ start                    : int 153331622
  ..$ end                      : int 153331728
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6 small nuclear 160, pseudogene [Source:HGNC Symbol;Acc:HGNC:47123]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000273541.1"
  ..$ gene_symbol              : chr "DLEU1_2"
  ..$ gene_name                : chr "Deleted in lymphocytic leukemia 1 conserved region 2 [Source:RFAM;Acc:RF02104]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 50104833
  ..$ end                      : int 50105174
  ..$ strand                   : int 1
  ..$ description              : chr "Deleted in lymphocytic leukemia 1 conserved region 2 [Source:RFAM;Acc:RF02104]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000253792.1"
  ..$ gene_symbol              : chr "AC025437.5"
  ..$ gene_name                : NULL
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "lincRNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 158173601
  ..$ end                      : int 158176422
  ..$ strand                   : int -1
  ..$ description              : NULL
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000284185.1"
  ..$ gene_symbol              : chr "MIR219A2"
  ..$ gene_name                : chr "microRNA 219a-2 [Source:HGNC Symbol;Acc:HGNC:31598]"
  ..$ synonyms                 : chr "MIRN219-2,MIR219-2,hsa-mir-219-2,MIRN219-2,MIR219-2,hsa-mir-219-2"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "407003"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 128392618
  ..$ end                      : int 128392714
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 219a-2 [Source:HGNC Symbol;Acc:HGNC:31598]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000263858.1"
  ..$ gene_symbol              : chr "MIR4769"
  ..$ gene_name                : chr "microRNA 4769 [Source:HGNC Symbol;Acc:HGNC:41694]"
  ..$ synonyms                 : chr "hsa-mir-4769"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100616147"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 47587429
  ..$ end                      : int 47587505
  ..$ strand                   : int 1
  ..$ description              : chr "microRNA 4769 [Source:HGNC Symbol;Acc:HGNC:41694]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000283409.1"
  ..$ gene_symbol              : chr "MIR3662"
  ..$ gene_name                : chr "microRNA 3662 [Source:HGNC Symbol;Acc:HGNC:38983]"
  ..$ synonyms                 : chr "hsa-mir-3662"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100500880"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 134979338
  ..$ end                      : int 134979432
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 3662 [Source:HGNC Symbol;Acc:HGNC:38983]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000264357.1"
  ..$ gene_symbol              : chr "MIR4648"
  ..$ gene_name                : chr "microRNA 4648 [Source:HGNC Symbol;Acc:HGNC:41560]"
  ..$ synonyms                 : chr "hsa-mir-4648"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100616116"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 2527074
  ..$ end                      : int 2527145
  ..$ strand                   : int 1
  ..$ description              : chr "microRNA 4648 [Source:HGNC Symbol;Acc:HGNC:41560]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000264249.1"
  ..$ gene_symbol              : chr "MIR3912"
  ..$ gene_name                : chr "microRNA 3912 [Source:HGNC Symbol;Acc:HGNC:38923]"
  ..$ synonyms                 : chr "hsa-mir-3912"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100500831"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 171386656
  ..$ end                      : int 171386760
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 3912 [Source:HGNC Symbol;Acc:HGNC:38923]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000239822.3"
  ..$ gene_symbol              : chr "RN7SL754P"
  ..$ gene_name                : chr "RNA, 7SL, cytoplasmic 754, pseudogene [Source:HGNC Symbol;Acc:HGNC:46770]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106479497"
  ..$ summary                  : NULL
  ..$ start                    : int 430239
  ..$ end                      : int 430504
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, 7SL, cytoplasmic 754, pseudogene [Source:HGNC Symbol;Acc:HGNC:46770]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000274046.1"
  ..$ gene_symbol              : chr "Y_RNA"
  ..$ gene_name                : chr "Y RNA [Source:RFAM;Acc:RF00019]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 7092616
  ..$ end                      : int 7092716
  ..$ strand                   : int 1
  ..$ description              : chr "Y RNA [Source:RFAM;Acc:RF00019]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000199879.1"
  ..$ gene_symbol              : chr "RNU1-120P"
  ..$ gene_name                : chr "RNA, U1 small nuclear 120, pseudogene [Source:HGNC Symbol;Acc:HGNC:48462]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106481627"
  ..$ summary                  : NULL
  ..$ start                    : int 148263476
  ..$ end                      : int 148263632
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U1 small nuclear 120, pseudogene [Source:HGNC Symbol;Acc:HGNC:48462]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000221763.1"
  ..$ gene_symbol              : chr "MIR1289-1"
  ..$ gene_name                : chr "microRNA 1289-1 [Source:HGNC Symbol;Acc:HGNC:35281]"
  ..$ synonyms                 : chr "MIRN1289-1,hsa-mir-1289-1"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100302125"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 35453954
  ..$ end                      : int 35454097
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 1289-1 [Source:HGNC Symbol;Acc:HGNC:35281]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000202081.1"
  ..$ gene_symbol              : chr "RNU6-1280P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 1280, pseudogene [Source:HGNC Symbol;Acc:HGNC:48243]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106481584"
  ..$ summary                  : NULL
  ..$ start                    : int 85651522
  ..$ end                      : int 85651628
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, U6 small nuclear 1280, pseudogene [Source:HGNC Symbol;Acc:HGNC:48243]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000278618.1"
  ..$ gene_symbol              : chr "MIR8069-2"
  ..$ gene_name                : chr "microRNA 8069-2 [Source:HGNC Symbol;Acc:HGNC:50836]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "103504736"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 13724189
  ..$ end                      : int 13724274
  ..$ strand                   : int 1
  ..$ description              : chr "microRNA 8069-2 [Source:HGNC Symbol;Acc:HGNC:50836]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000264075.1"
  ..$ gene_symbol              : chr "MIR4783"
  ..$ gene_name                : chr "microRNA 4783 [Source:HGNC Symbol;Acc:HGNC:41874]"
  ..$ synonyms                 : chr "hsa-mir-4783"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100616187"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 127423537
  ..$ end                      : int 127423618
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 4783 [Source:HGNC Symbol;Acc:HGNC:41874]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000252076.1"
  ..$ gene_symbol              : chr "RN7SKP196"
  ..$ gene_name                : chr "RNA, 7SK small nuclear pseudogene 196 [Source:HGNC Symbol;Acc:HGNC:45920]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106479182"
  ..$ summary                  : NULL
  ..$ start                    : int 58869184
  ..$ end                      : int 58869528
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 7SK small nuclear pseudogene 196 [Source:HGNC Symbol;Acc:HGNC:45920]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000263510.1"
  ..$ gene_symbol              : chr "MIR4497"
  ..$ gene_name                : chr "microRNA 4497 [Source:HGNC Symbol;Acc:HGNC:41737]"
  ..$ synonyms                 : chr "hsa-mir-4497"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100616454"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 109833348
  ..$ end                      : int 109833436
  ..$ strand                   : int 1
  ..$ description              : chr "microRNA 4497 [Source:HGNC Symbol;Acc:HGNC:41737]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000266555.1"
  ..$ gene_symbol              : chr "MIR3145"
  ..$ gene_name                : chr "microRNA 3145 [Source:HGNC Symbol;Acc:HGNC:38327]"
  ..$ synonyms                 : chr "hsa-mir-3145"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100423001"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 138435213
  ..$ end                      : int 138435294
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 3145 [Source:HGNC Symbol;Acc:HGNC:38327]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000264986.1"
  ..$ gene_symbol              : chr "MIR5092"
  ..$ gene_name                : chr "microRNA 5092 [Source:HGNC Symbol;Acc:HGNC:43467]"
  ..$ synonyms                 : chr "hsa-mir-5092"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100847039"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 125151465
  ..$ end                      : int 125151552
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 5092 [Source:HGNC Symbol;Acc:HGNC:43467]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207924.3"
  ..$ gene_symbol              : chr "MIR196A2"
  ..$ gene_name                : chr "microRNA 196a-2 [Source:HGNC Symbol;Acc:HGNC:31568]"
  ..$ synonyms                 : chr "MIRN196A2,MIRN196-2,hsa-mir-196a-2,hsa-mir-196-2"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "406973"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 53991738
  ..$ end                      : int 53991847
  ..$ strand                   : int 1
  ..$ description              : chr "microRNA 196a-2 [Source:HGNC Symbol;Acc:HGNC:31568]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000243015.2"
  ..$ gene_symbol              : chr "RN7SL737P"
  ..$ gene_name                : chr "RNA, 7SL, cytoplasmic 737, pseudogene [Source:HGNC Symbol;Acc:HGNC:46753]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106481123"
  ..$ summary                  : NULL
  ..$ start                    : int 93509487
  ..$ end                      : int 93509768
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 7SL, cytoplasmic 737, pseudogene [Source:HGNC Symbol;Acc:HGNC:46753]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000202515.1"
  ..$ gene_symbol              : chr "VTRNA1-3"
  ..$ gene_name                : chr "vault RNA 1-3 [Source:HGNC Symbol;Acc:HGNC:12656]"
  ..$ synonyms                 : chr "HVG3,hvg-3,vtRNA1-3,VR3,VAULTRC3"
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "56662"
  ..$ summary                  : NULL
  ..$ start                    : int 140726158
  ..$ end                      : int 140726246
  ..$ strand                   : int 1
  ..$ description              : chr "vault RNA 1-3 [Source:HGNC Symbol;Acc:HGNC:12656]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000252305.1"
  ..$ gene_symbol              : chr "SNORA74"
  ..$ gene_name                : chr "Small nucleolar RNA SNORA74 [Source:RFAM;Acc:RF00090]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snoRNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 14177131
  ..$ end                      : int 14177310
  ..$ strand                   : int 1
  ..$ description              : chr "Small nucleolar RNA SNORA74 [Source:RFAM;Acc:RF00090]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207798.1"
  ..$ gene_symbol              : chr "MIR216A"
  ..$ gene_name                : chr "microRNA 216a [Source:HGNC Symbol;Acc:HGNC:31593]"
  ..$ synonyms                 : chr "MIRN216A,MIRN216,hsa-mir-216a,hsa-mir-216"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "406998"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 55988950
  ..$ end                      : int 55989059
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 216a [Source:HGNC Symbol;Acc:HGNC:31593]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000268674.2"
  ..$ gene_symbol              : chr "FAM231D"
  ..$ gene_name                : chr "family with sequence similarity 231 member D [Source:NCBI gene;Acc:644634]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "protein_coding"
  ..$ ncbi_id                  : chr "644634"
  ..$ summary                  : NULL
  ..$ start                    : int 35407
  ..$ end                      : int 35916
  ..$ strand                   : int 1
  ..$ description              : chr "family with sequence similarity 231 member D [Source:NCBI gene;Acc:644634]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000264975.1"
  ..$ gene_symbol              : chr "MIR4431"
  ..$ gene_name                : chr "microRNA 4431 [Source:HGNC Symbol;Acc:HGNC:41738]"
  ..$ synonyms                 : chr "hsa-mir-4431"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100616431"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 52702522
  ..$ end                      : int 52702615
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 4431 [Source:HGNC Symbol;Acc:HGNC:41738]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000201216.1"
  ..$ gene_symbol              : chr "Y_RNA"
  ..$ gene_name                : chr "Y RNA [Source:RFAM;Acc:RF00019]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 102950467
  ..$ end                      : int 102950564
  ..$ strand                   : int -1
  ..$ description              : chr "Y RNA [Source:RFAM;Acc:RF00019]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000277845.1"
  ..$ gene_symbol              : chr "RN7SKP175"
  ..$ gene_name                : chr "RNA, 7SK small nuclear pseudogene 175 [Source:HGNC Symbol;Acc:HGNC:45899]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106479173"
  ..$ summary                  : NULL
  ..$ start                    : int 143542110
  ..$ end                      : int 143542398
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 7SK small nuclear pseudogene 175 [Source:HGNC Symbol;Acc:HGNC:45899]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000266663.1"
  ..$ gene_symbol              : chr "MIR3169"
  ..$ gene_name                : chr "microRNA 3169 [Source:HGNC Symbol;Acc:HGNC:38380]"
  ..$ synonyms                 : chr "hsa-mir-3169"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100422973"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 61199798
  ..$ end                      : int 61199880
  ..$ strand                   : int -1
  ..$ description              : chr "microRNA 3169 [Source:HGNC Symbol;Acc:HGNC:38380]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000283664.1"
  ..$ gene_symbol              : chr "MIR4679-1"
  ..$ gene_name                : chr "microRNA 4679-1 [Source:HGNC Symbol;Acc:HGNC:41612]"
  ..$ synonyms                 : chr "hsa-mir-4679-1"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100616128"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 89063336
  ..$ end                      : int 89063410
  ..$ strand                   : int 1
  ..$ description              : chr "microRNA 4679-1 [Source:HGNC Symbol;Acc:HGNC:41612]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207122.1"
  ..$ gene_symbol              : chr "RNU6-591P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 591, pseudogene [Source:HGNC Symbol;Acc:HGNC:47554]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479827"
  ..$ summary                  : NULL
  ..$ start                    : int 39081180
  ..$ end                      : int 39081285
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, U6 small nuclear 591, pseudogene [Source:HGNC Symbol;Acc:HGNC:47554]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000212238.1"
  ..$ gene_symbol              : chr "RNA5SP244"
  ..$ gene_name                : chr "RNA, 5S ribosomal pseudogene 244 [Source:HGNC Symbol;Acc:HGNC:43144]"
  ..$ synonyms                 : chr "RN5S244"
  ..$ biotype                  : chr "rRNA"
  ..$ ncbi_id                  : chr "100885864"
  ..$ summary                  : NULL
  ..$ start                    : int 129756266
  ..$ end                      : int 129756377
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 5S ribosomal pseudogene 244 [Source:HGNC Symbol;Acc:HGNC:43144]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000201489.1"
  ..$ gene_symbol              : chr "Y_RNA"
  ..$ gene_name                : chr "Y RNA [Source:RFAM;Acc:RF00019]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 7537096
  ..$ end                      : int 7537196
  ..$ strand                   : int -1
  ..$ description              : chr "Y RNA [Source:RFAM;Acc:RF00019]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000277994.1"
  ..$ gene_symbol              : chr "HOTAIR_3"
  ..$ gene_name                : chr "HOTAIR conserved region 3 [Source:RFAM;Acc:RF01906]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 53965965
  ..$ end                      : int 53966061
  ..$ strand                   : int 1
  ..$ description              : chr "HOTAIR conserved region 3 [Source:RFAM;Acc:RF01906]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000206917.1"
  ..$ gene_symbol              : chr "RNU1-52P"
  ..$ gene_name                : chr "RNA, U1 small nuclear 52, pseudogene [Source:HGNC Symbol;Acc:HGNC:48394]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106480164"
  ..$ summary                  : NULL
  ..$ start                    : int 58677574
  ..$ end                      : int 58677737
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, U1 small nuclear 52, pseudogene [Source:HGNC Symbol;Acc:HGNC:48394]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000206722.1"
  ..$ gene_symbol              : chr "RNU6-1074P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 1074, pseudogene [Source:HGNC Symbol;Acc:HGNC:48037]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106480635"
  ..$ summary                  : NULL
  ..$ start                    : int 139256657
  ..$ end                      : int 139256760
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, U6 small nuclear 1074, pseudogene [Source:HGNC Symbol;Acc:HGNC:48037]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000222950.1"
  ..$ gene_symbol              : chr "RN7SKP262"
  ..$ gene_name                : chr "RNA, 7SK small nuclear pseudogene 262 [Source:HGNC Symbol;Acc:HGNC:45986]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106479212"
  ..$ summary                  : NULL
  ..$ start                    : int 25779287
  ..$ end                      : int 25779569
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, 7SK small nuclear pseudogene 262 [Source:HGNC Symbol;Acc:HGNC:45986]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000244677.3"
  ..$ gene_symbol              : chr "RN7SL706P"
  ..$ gene_name                : chr "RNA, 7SL, cytoplasmic 706, pseudogene [Source:HGNC Symbol;Acc:HGNC:46722]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106479480"
  ..$ summary                  : NULL
  ..$ start                    : int 68149617
  ..$ end                      : int 68149908
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, 7SL, cytoplasmic 706, pseudogene [Source:HGNC Symbol;Acc:HGNC:46722]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000277922.1"
  ..$ gene_symbol              : chr "FTX_5"
  ..$ gene_name                : chr "FTX transcript, XIST regulator conserved region 5 [Source:RFAM;Acc:RF02122]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 74293357
  ..$ end                      : int 74293574
  ..$ strand                   : int 1
  ..$ description              : chr "FTX transcript, XIST regulator conserved region 5 [Source:RFAM;Acc:RF02122]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000222154.1"
  ..$ gene_symbol              : chr "RN7SKP218"
  ..$ gene_name                : chr "RNA, 7SK small nuclear pseudogene 218 [Source:HGNC Symbol;Acc:HGNC:45942]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106480900"
  ..$ summary                  : NULL
  ..$ start                    : int 53490148
  ..$ end                      : int 53490439
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, 7SK small nuclear pseudogene 218 [Source:HGNC Symbol;Acc:HGNC:45942]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000277752.1"
  ..$ gene_symbol              : chr "DLEU2_5"
  ..$ gene_name                : chr "Deleted in lymphocytic leukemia 2 conserved region 5 [Source:RFAM;Acc:RF02109]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 50075570
  ..$ end                      : int 50075653
  ..$ strand                   : int 1
  ..$ description              : chr "Deleted in lymphocytic leukemia 2 conserved region 5 [Source:RFAM;Acc:RF02109]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000276253.1"
  ..$ gene_symbol              : chr "AC243790.1"
  ..$ gene_name                : NULL
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "protein_coding"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 30361040
  ..$ end                      : int 30361438
  ..$ strand                   : int 1
  ..$ description              : NULL
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000221216.1"
  ..$ gene_symbol              : chr "RNU6ATAC27P"
  ..$ gene_name                : chr "RNA, U6atac small nuclear 27, pseudogene [Source:HGNC Symbol;Acc:HGNC:46926]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106481167"
  ..$ summary                  : NULL
  ..$ start                    : int 28481362
  ..$ end                      : int 28481480
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6atac small nuclear 27, pseudogene [Source:HGNC Symbol;Acc:HGNC:46926]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000200831.1"
  ..$ gene_symbol              : chr "SNORD36B"
  ..$ gene_name                : chr "small nucleolar RNA, C/D box 36B [Source:HGNC Symbol;Acc:HGNC:10164]"
  ..$ synonyms                 : chr "U36b,RNU36B"
  ..$ biotype                  : chr "snoRNA"
  ..$ ncbi_id                  : chr "26814"
  ..$ summary                  : NULL
  ..$ start                    : int 133350095
  ..$ end                      : int 133350168
  ..$ strand                   : int 1
  ..$ description              : chr "small nucleolar RNA, C/D box 36B [Source:HGNC Symbol;Acc:HGNC:10164]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000278237.1"
  ..$ gene_symbol              : chr "NCRUPAR_2"
  ..$ gene_name                : chr "Non-protein coding RNA, upstream of F2R/PAR1 conserved region 2 [Source:RFAM;Acc:RF02155]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 76712763
  ..$ end                      : int 76712861
  ..$ strand                   : int 1
  ..$ description              : chr "Non-protein coding RNA, upstream of F2R/PAR1 conserved region 2 [Source:RFAM;Acc:RF02155]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000271852.1"
  ..$ gene_symbol              : chr "SNORD11B"
  ..$ gene_name                : chr "small nucleolar RNA, C/D box 11B [Source:HGNC Symbol;Acc:HGNC:33571]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snoRNA"
  ..$ ncbi_id                  : chr "100113392"
  ..$ summary                  : NULL
  ..$ start                    : int 202291332
  ..$ end                      : int 202291421
  ..$ strand                   : int 1
  ..$ description              : chr "small nucleolar RNA, C/D box 11B [Source:HGNC Symbol;Acc:HGNC:33571]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000201980.1"
  ..$ gene_symbol              : chr "SNORA62"
  ..$ gene_name                : chr "Small nucleolar RNA SNORA62/SNORA6 family [Source:RFAM;Acc:RF00091]"
  ..$ synonyms                 : chr "E2-1,RNU108,E2,RNE2"
  ..$ biotype                  : chr "snoRNA"
  ..$ ncbi_id                  : chr "6044"
  ..$ summary                  : NULL
  ..$ start                    : int 4756645
  ..$ end                      : int 4756797
  ..$ strand                   : int 1
  ..$ description              : chr "Small nucleolar RNA SNORA62/SNORA6 family [Source:RFAM;Acc:RF00091]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000251756.1"
  ..$ gene_symbol              : chr "RNA5SP385"
  ..$ gene_name                : chr "RNA, 5S ribosomal pseudogene 385 [Source:HGNC Symbol;Acc:HGNC:43285]"
  ..$ synonyms                 : chr "RN5S385"
  ..$ biotype                  : chr "rRNA"
  ..$ ncbi_id                  : chr "100873640"
  ..$ summary                  : NULL
  ..$ start                    : int 51692752
  ..$ end                      : int 51692861
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 5S ribosomal pseudogene 385 [Source:HGNC Symbol;Acc:HGNC:43285]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000244307.3"
  ..$ gene_symbol              : chr "RN7SL395P"
  ..$ gene_name                : chr "RNA, 7SL, cytoplasmic 395, pseudogene [Source:HGNC Symbol;Acc:HGNC:46411]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106481031"
  ..$ summary                  : NULL
  ..$ start                    : int 144785298
  ..$ end                      : int 144785588
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 7SL, cytoplasmic 395, pseudogene [Source:HGNC Symbol;Acc:HGNC:46411]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000199880.1"
  ..$ gene_symbol              : chr "RNU6-888P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 888, pseudogene [Source:HGNC Symbol;Acc:HGNC:47851]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479959"
  ..$ summary                  : NULL
  ..$ start                    : int 138123011
  ..$ end                      : int 138123111
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, U6 small nuclear 888, pseudogene [Source:HGNC Symbol;Acc:HGNC:47851]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000276800.1"
  ..$ gene_symbol              : chr "Metazoa_SRP"
  ..$ gene_name                : chr "Metazoan signal recognition particle RNA [Source:RFAM;Acc:RF00017]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 21463963
  ..$ end                      : int 21464279
  ..$ strand                   : int 1
  ..$ description              : chr "Metazoan signal recognition particle RNA [Source:RFAM;Acc:RF00017]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000201811.1"
  ..$ gene_symbol              : chr "SNORA71"
  ..$ gene_name                : chr "Small nucleolar RNA SNORA71 [Source:RFAM;Acc:RF00056]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snoRNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 38442036
  ..$ end                      : int 38442169
  ..$ strand                   : int 1
  ..$ description              : chr "Small nucleolar RNA SNORA71 [Source:RFAM;Acc:RF00056]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000222616.1"
  ..$ gene_symbol              : chr "RN7SKP27"
  ..$ gene_name                : chr "RNA, 7SK small nuclear pseudogene 27 [Source:HGNC Symbol;Acc:HGNC:45751]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106480841"
  ..$ summary                  : NULL
  ..$ start                    : int 22861552
  ..$ end                      : int 22861884
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, 7SK small nuclear pseudogene 27 [Source:HGNC Symbol;Acc:HGNC:45751]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000275405.1"
  ..$ gene_symbol              : chr "U1"
  ..$ gene_name                : chr "U1 spliceosomal RNA [Source:RFAM;Acc:RF00003]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 21861
  ..$ end                      : int 22024
  ..$ strand                   : int -1
  ..$ description              : chr "U1 spliceosomal RNA [Source:RFAM;Acc:RF00003]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000207115.1"
  ..$ gene_symbol              : chr "RNU6-364P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 364, pseudogene [Source:HGNC Symbol;Acc:HGNC:47327]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106481293"
  ..$ summary                  : NULL
  ..$ start                    : int 96341140
  ..$ end                      : int 96341246
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6 small nuclear 364, pseudogene [Source:HGNC Symbol;Acc:HGNC:47327]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000252622.1"
  ..$ gene_symbol              : chr "RNU6-881P"
  ..$ gene_name                : chr "RNA, U6 small nuclear 881, pseudogene [Source:HGNC Symbol;Acc:HGNC:47844]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : chr "106479956"
  ..$ summary                  : NULL
  ..$ start                    : int 131126990
  ..$ end                      : int 131127091
  ..$ strand                   : int 1
  ..$ description              : chr "RNA, U6 small nuclear 881, pseudogene [Source:HGNC Symbol;Acc:HGNC:47844]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000238942.1"
  ..$ gene_symbol              : chr "SNORD2"
  ..$ gene_name                : chr "small nucleolar RNA, C/D box 2 [Source:HGNC Symbol;Acc:HGNC:32678]"
  ..$ synonyms                 : chr "snR39B,R39B"
  ..$ biotype                  : chr "snoRNA"
  ..$ ncbi_id                  : chr "619567"
  ..$ summary                  : NULL
  ..$ start                    : int 186784796
  ..$ end                      : int 186784864
  ..$ strand                   : int 1
  ..$ description              : chr "small nucleolar RNA, C/D box 2 [Source:HGNC Symbol;Acc:HGNC:32678]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000264909.1"
  ..$ gene_symbol              : chr "U1"
  ..$ gene_name                : chr "U1 spliceosomal RNA [Source:RFAM;Acc:RF00003]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "snRNA"
  ..$ ncbi_id                  : NULL
  ..$ summary                  : NULL
  ..$ start                    : int 32479741
  ..$ end                      : int 32479887
  ..$ strand                   : int -1
  ..$ description              : chr "U1 spliceosomal RNA [Source:RFAM;Acc:RF00003]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000222783.1"
  ..$ gene_symbol              : chr "RN7SKP212"
  ..$ gene_name                : chr "RNA, 7SK small nuclear pseudogene 212 [Source:HGNC Symbol;Acc:HGNC:45936]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106481750"
  ..$ summary                  : NULL
  ..$ start                    : int 130811768
  ..$ end                      : int 130812095
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, 7SK small nuclear pseudogene 212 [Source:HGNC Symbol;Acc:HGNC:45936]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000281810.1"
  ..$ gene_symbol              : chr "MIR3972"
  ..$ gene_name                : chr "microRNA 3972 [Source:HGNC Symbol;Acc:HGNC:41876]"
  ..$ synonyms                 : chr "hsa-mir-3972"
  ..$ biotype                  : chr "miRNA"
  ..$ ncbi_id                  : chr "100616188"
  ..$ summary                  : chr "microRNAs (miRNAs) are short (20-24 nt) non-coding RNAs that are involved in post-transcriptional regulation of"| __truncated__
  ..$ start                    : int 17277889
  ..$ end                      : int 17277975
  ..$ strand                   : int 1
  ..$ description              : chr "microRNA 3972 [Source:HGNC Symbol;Acc:HGNC:41876]"
 $ :List of 11
  ..$ versioned_ensembl_gene_id: chr "ENSG00000242769.3"
  ..$ gene_symbol              : chr "RN7SL432P"
  ..$ gene_name                : chr "RNA, 7SL, cytoplasmic 432, pseudogene [Source:HGNC Symbol;Acc:HGNC:46448]"
  ..$ synonyms                 : NULL
  ..$ biotype                  : chr "misc_RNA"
  ..$ ncbi_id                  : chr "106481041"
  ..$ summary                  : NULL
  ..$ start                    : int 114697627
  ..$ end                      : int 114697924
  ..$ strand                   : int -1
  ..$ description              : chr "RNA, 7SL, cytoplasmic 432, pseudogene [Source:HGNC Symbol;Acc:HGNC:46448]"
  [list output truncated]</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(jsonlite<span class="sc">::</span><span class="fu">read_json</span>(<span class="st">"output_gene.json"</span>), <span class="cf">function</span>(l) {</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">tibble</span>(</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">versioned_ensembl_gene_id =</span> l[[<span class="st">"versioned_ensembl_gene_id"</span>]],</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_symbol =</span> l[[<span class="st">"gene_symbol"</span>]]</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb35" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Baseline Models V1"</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Philipp Sven Lars Schäfer"</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "`r format(Sys.time(), '%d %B, %Y')`"</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> source</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="an">engine:</span><span class="co"> knitr</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="fu"># Packages</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(tidyverse)</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ggdark)</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(factoextra)</span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(FactoMineR)</span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(magick) <span class="co"># formatting</span></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">library</span>(ranger)</span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>knit_hooks<span class="sc">$</span><span class="fu">set</span>(<span class="at">crop =</span> knitr<span class="sc">::</span>hook_pdfcrop) <span class="co"># formatting</span></span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"read_data.R"</span>))</span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"src"</span>, <span class="st">"colors.R"</span>))</span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data</span></span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a>input_dir <span class="ot">=</span> <span class="fu">file.path</span>(<span class="st">".."</span>, <span class="st">"data"</span>)</span>
<span id="cb35-34"><a href="#cb35-34" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-35"><a href="#cb35-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-38"><a href="#cb35-38" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-39"><a href="#cb35-39" aria-hidden="true" tabindex="-1"></a><span class="co"># loading</span></span>
<span id="cb35-40"><a href="#cb35-40" aria-hidden="true" tabindex="-1"></a>meta_df <span class="ot">&lt;-</span> <span class="fu">read_harmonized_meta_data</span>(input_dir)</span>
<span id="cb35-41"><a href="#cb35-41" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">&lt;-</span> <span class="fu">read_harmonized_experimental_data</span>(input_dir)</span>
<span id="cb35-42"><a href="#cb35-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-43"><a href="#cb35-43" aria-hidden="true" tabindex="-1"></a><span class="co"># filtering</span></span>
<span id="cb35-44"><a href="#cb35-44" aria-hidden="true" tabindex="-1"></a>meta_df <span class="ot">&lt;-</span> <span class="fu">filter_harmonized_meta_data</span>(<span class="at">meta_data=</span>meta_df, <span class="at">experimental_data=</span>data_list)</span>
<span id="cb35-45"><a href="#cb35-45" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">&lt;-</span> <span class="fu">filter_harmonized_experimental_data</span>(<span class="at">meta_data=</span>meta_df, <span class="at">experimental_data=</span>data_list)</span>
<span id="cb35-46"><a href="#cb35-46" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-47"><a href="#cb35-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-48"><a href="#cb35-48" aria-hidden="true" tabindex="-1"></a><span class="fu"># Task 1</span></span>
<span id="cb35-49"><a href="#cb35-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-50"><a href="#cb35-50" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>10-fold CV</span>
<span id="cb35-51"><a href="#cb35-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-52"><a href="#cb35-52" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Random Forest</span>
<span id="cb35-53"><a href="#cb35-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-56"><a href="#cb35-56" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-57"><a href="#cb35-57" aria-hidden="true" tabindex="-1"></a>df_targets <span class="ot">&lt;-</span> meta_df <span class="sc">%&gt;%</span></span>
<span id="cb35-58"><a href="#cb35-58" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">14</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb35-59"><a href="#cb35-59" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(data_list<span class="sc">$</span>plasma_antibody_levels, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-60"><a href="#cb35-60" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(isotype_antigen <span class="sc">==</span> <span class="st">"IgG_PT"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-61"><a href="#cb35-61" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, planned_day_relative_to_boost, MFI_normalised) <span class="sc">%&gt;%</span></span>
<span id="cb35-62"><a href="#cb35-62" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"planned_day_relative_to_boost"</span>,</span>
<span id="cb35-63"><a href="#cb35-63" aria-hidden="true" tabindex="-1"></a>                     <span class="at">values_from=</span><span class="st">"MFI_normalised"</span>, <span class="at">names_prefix =</span> <span class="st">"day_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-64"><a href="#cb35-64" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">fc =</span> day_14 <span class="sc">/</span> day_0) <span class="sc">%&gt;%</span></span>
<span id="cb35-65"><a href="#cb35-65" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">drop_na</span>()</span>
<span id="cb35-66"><a href="#cb35-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-67"><a href="#cb35-67" aria-hidden="true" tabindex="-1"></a>demo_covariates <span class="ot">&lt;-</span> meta_df <span class="sc">%&gt;%</span></span>
<span id="cb35-68"><a href="#cb35-68" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, infancy_vac, biological_sex, ethnicity, </span>
<span id="cb35-69"><a href="#cb35-69" aria-hidden="true" tabindex="-1"></a>                age_at_boost, dataset) <span class="sc">%&gt;%</span></span>
<span id="cb35-70"><a href="#cb35-70" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb35-71"><a href="#cb35-71" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">infancy_vac_value=</span><span class="dv">1</span>, <span class="at">biological_sex_value=</span><span class="dv">1</span>, </span>
<span id="cb35-72"><a href="#cb35-72" aria-hidden="true" tabindex="-1"></a>                <span class="at">ethnicity_value=</span><span class="dv">1</span>, <span class="at">dataset_value=</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-73"><a href="#cb35-73" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>infancy_vac, <span class="at">values_from=</span>infancy_vac_value, </span>
<span id="cb35-74"><a href="#cb35-74" aria-hidden="true" tabindex="-1"></a>                     <span class="at">names_prefix=</span><span class="st">"infancy_vac_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-75"><a href="#cb35-75" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>biological_sex, <span class="at">values_from=</span>biological_sex_value,</span>
<span id="cb35-76"><a href="#cb35-76" aria-hidden="true" tabindex="-1"></a>                     <span class="at">names_prefix=</span><span class="st">"biological_sex_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-77"><a href="#cb35-77" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>ethnicity, <span class="at">values_from=</span>ethnicity_value, </span>
<span id="cb35-78"><a href="#cb35-78" aria-hidden="true" tabindex="-1"></a>                     <span class="at">names_prefix=</span><span class="st">"ethnicity_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-79"><a href="#cb35-79" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span>dataset, <span class="at">values_from=</span>dataset_value, </span>
<span id="cb35-80"><a href="#cb35-80" aria-hidden="true" tabindex="-1"></a>                     <span class="at">names_prefix=</span><span class="st">"dataset_"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-81"><a href="#cb35-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="at">.fns =</span> <span class="sc">~</span><span class="fu">replace_na</span>(.,<span class="dv">0</span>)))</span>
<span id="cb35-82"><a href="#cb35-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-83"><a href="#cb35-83" aria-hidden="true" tabindex="-1"></a>pbmc_covariates <span class="ot">&lt;-</span> meta_df <span class="sc">%&gt;%</span></span>
<span id="cb35-84"><a href="#cb35-84" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(planned_day_relative_to_boost <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb35-85"><a href="#cb35-85" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">inner_join</span>(data_list<span class="sc">$</span>pbmc_cell_frequency, <span class="at">by=</span><span class="st">"specimen_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-86"><a href="#cb35-86" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(subject_id, cell_type_name, percent_live_cell) <span class="sc">%&gt;%</span></span>
<span id="cb35-87"><a href="#cb35-87" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"cell_type_name"</span>, <span class="at">values_from=</span><span class="st">"percent_live_cell"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-88"><a href="#cb35-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="at">.fns =</span> <span class="sc">~</span><span class="fu">replace_na</span>(.,<span class="dv">0</span>)))</span>
<span id="cb35-89"><a href="#cb35-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-90"><a href="#cb35-90" aria-hidden="true" tabindex="-1"></a>df_all <span class="ot">&lt;-</span> demo_covariates <span class="sc">%&gt;%</span></span>
<span id="cb35-91"><a href="#cb35-91" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">inner_join</span>(pbmc_covariates, <span class="at">by=</span><span class="st">"subject_id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-92"><a href="#cb35-92" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">inner_join</span>(df_targets, <span class="at">by=</span><span class="st">"subject_id"</span>)</span>
<span id="cb35-93"><a href="#cb35-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-94"><a href="#cb35-94" aria-hidden="true" tabindex="-1"></a>pbmc_feature_names <span class="ot">&lt;-</span> <span class="fu">unique</span>(data_list<span class="sc">$</span>pbmc_cell_frequency<span class="sc">$</span>cell_type_name)</span>
<span id="cb35-95"><a href="#cb35-95" aria-hidden="true" tabindex="-1"></a>demo_feature_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(demo_covariates)[<span class="sc">!</span>(<span class="fu">colnames</span>(demo_covariates) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"subject_id"</span>))]</span>
<span id="cb35-96"><a href="#cb35-96" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-97"><a href="#cb35-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-98"><a href="#cb35-98" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1a</span></span>
<span id="cb35-99"><a href="#cb35-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-102"><a href="#cb35-102" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-103"><a href="#cb35-103" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb35-104"><a href="#cb35-104" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-105"><a href="#cb35-105" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(demo_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb35-106"><a href="#cb35-106" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-107"><a href="#cb35-107" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(day_14), </span>
<span id="cb35-108"><a href="#cb35-108" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb35-109"><a href="#cb35-109" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-110"><a href="#cb35-110" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb35-111"><a href="#cb35-111" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb35-112"><a href="#cb35-112" aria-hidden="true" tabindex="-1"></a>rf_model</span>
<span id="cb35-113"><a href="#cb35-113" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-114"><a href="#cb35-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-117"><a href="#cb35-117" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-118"><a href="#cb35-118" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb35-119"><a href="#cb35-119" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-120"><a href="#cb35-120" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(pbmc_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb35-121"><a href="#cb35-121" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-122"><a href="#cb35-122" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(day_14), </span>
<span id="cb35-123"><a href="#cb35-123" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb35-124"><a href="#cb35-124" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-125"><a href="#cb35-125" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb35-126"><a href="#cb35-126" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb35-127"><a href="#cb35-127" aria-hidden="true" tabindex="-1"></a>rf_model</span>
<span id="cb35-128"><a href="#cb35-128" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-129"><a href="#cb35-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-132"><a href="#cb35-132" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-133"><a href="#cb35-133" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb35-134"><a href="#cb35-134" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-135"><a href="#cb35-135" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(demo_feature_names, pbmc_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb35-136"><a href="#cb35-136" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-137"><a href="#cb35-137" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(day_14), </span>
<span id="cb35-138"><a href="#cb35-138" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb35-139"><a href="#cb35-139" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-140"><a href="#cb35-140" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb35-141"><a href="#cb35-141" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb35-142"><a href="#cb35-142" aria-hidden="true" tabindex="-1"></a>rf_model</span>
<span id="cb35-143"><a href="#cb35-143" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-144"><a href="#cb35-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-145"><a href="#cb35-145" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1b</span></span>
<span id="cb35-146"><a href="#cb35-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-149"><a href="#cb35-149" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-150"><a href="#cb35-150" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb35-151"><a href="#cb35-151" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-152"><a href="#cb35-152" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(demo_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb35-153"><a href="#cb35-153" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-154"><a href="#cb35-154" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(fc), </span>
<span id="cb35-155"><a href="#cb35-155" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb35-156"><a href="#cb35-156" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-157"><a href="#cb35-157" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb35-158"><a href="#cb35-158" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb35-159"><a href="#cb35-159" aria-hidden="true" tabindex="-1"></a>rf_model</span>
<span id="cb35-160"><a href="#cb35-160" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-161"><a href="#cb35-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-164"><a href="#cb35-164" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-165"><a href="#cb35-165" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb35-166"><a href="#cb35-166" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-167"><a href="#cb35-167" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(pbmc_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb35-168"><a href="#cb35-168" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-169"><a href="#cb35-169" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(fc), </span>
<span id="cb35-170"><a href="#cb35-170" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb35-171"><a href="#cb35-171" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-172"><a href="#cb35-172" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb35-173"><a href="#cb35-173" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb35-174"><a href="#cb35-174" aria-hidden="true" tabindex="-1"></a>rf_model</span>
<span id="cb35-175"><a href="#cb35-175" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-176"><a href="#cb35-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-179"><a href="#cb35-179" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-180"><a href="#cb35-180" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb35-181"><a href="#cb35-181" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> ranger<span class="sc">::</span><span class="fu">ranger</span>(<span class="at">x =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-182"><a href="#cb35-182" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">all_of</span>(<span class="fu">c</span>(demo_feature_names, pbmc_feature_names, <span class="st">"day_0"</span>))),</span>
<span id="cb35-183"><a href="#cb35-183" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-184"><a href="#cb35-184" aria-hidden="true" tabindex="-1"></a>                             dplyr<span class="sc">::</span><span class="fu">pull</span>(fc), </span>
<span id="cb35-185"><a href="#cb35-185" aria-hidden="true" tabindex="-1"></a>                           <span class="at">importance =</span> <span class="st">"permutation"</span></span>
<span id="cb35-186"><a href="#cb35-186" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb35-187"><a href="#cb35-187" aria-hidden="true" tabindex="-1"></a><span class="co">#rf_model$variable.importance</span></span>
<span id="cb35-188"><a href="#cb35-188" aria-hidden="true" tabindex="-1"></a>rf_model<span class="sc">$</span>call <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb35-189"><a href="#cb35-189" aria-hidden="true" tabindex="-1"></a>rf_model</span>
<span id="cb35-190"><a href="#cb35-190" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-191"><a href="#cb35-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-194"><a href="#cb35-194" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-195"><a href="#cb35-195" aria-hidden="true" tabindex="-1"></a>df_all <span class="sc">%&gt;%</span></span>
<span id="cb35-196"><a href="#cb35-196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>day_0, <span class="at">y=</span>fc)) <span class="sc">+</span></span>
<span id="cb35-197"><a href="#cb35-197" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb35-198"><a href="#cb35-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>() <span class="sc">+</span></span>
<span id="cb35-199"><a href="#cb35-199" aria-hidden="true" tabindex="-1"></a>  ggdark<span class="sc">::</span><span class="fu">dark_mode</span>(<span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb35-200"><a href="#cb35-200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-201"><a href="#cb35-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-204"><a href="#cb35-204" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb35-205"><a href="#cb35-205" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb35-206"><a href="#cb35-206" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(jsonlite)</span>
<span id="cb35-207"><a href="#cb35-207" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httr)</span>
<span id="cb35-208"><a href="#cb35-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-209"><a href="#cb35-209" aria-hidden="true" tabindex="-1"></a><span class="co"># Base URL for the API</span></span>
<span id="cb35-210"><a href="#cb35-210" aria-hidden="true" tabindex="-1"></a>base_url <span class="ot">&lt;-</span> <span class="st">"https://www.cmi-pb.org/api/v5"</span></span>
<span id="cb35-211"><a href="#cb35-211" aria-hidden="true" tabindex="-1"></a><span class="co">#base_url &lt;- "https://www.cmi-pb.org/api"</span></span>
<span id="cb35-212"><a href="#cb35-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-213"><a href="#cb35-213" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the endpoints you want to query</span></span>
<span id="cb35-214"><a href="#cb35-214" aria-hidden="true" tabindex="-1"></a>endpoints <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"/cell_type"</span>, <span class="st">"/gene"</span>)</span>
<span id="cb35-215"><a href="#cb35-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-216"><a href="#cb35-216" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through endpoints and download JSON files</span></span>
<span id="cb35-217"><a href="#cb35-217" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (endpoint <span class="cf">in</span> endpoints) {</span>
<span id="cb35-218"><a href="#cb35-218" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Construct full URL</span></span>
<span id="cb35-219"><a href="#cb35-219" aria-hidden="true" tabindex="-1"></a>  url <span class="ot">&lt;-</span> <span class="fu">paste0</span>(base_url, endpoint)</span>
<span id="cb35-220"><a href="#cb35-220" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-221"><a href="#cb35-221" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Make GET request</span></span>
<span id="cb35-222"><a href="#cb35-222" aria-hidden="true" tabindex="-1"></a>  response <span class="ot">&lt;-</span> <span class="fu">GET</span>(url)</span>
<span id="cb35-223"><a href="#cb35-223" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-224"><a href="#cb35-224" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if the request was successful</span></span>
<span id="cb35-225"><a href="#cb35-225" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">status_code</span>(response) <span class="sc">==</span> <span class="dv">200</span>) {</span>
<span id="cb35-226"><a href="#cb35-226" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save the JSON content to a file</span></span>
<span id="cb35-227"><a href="#cb35-227" aria-hidden="true" tabindex="-1"></a>    json_content <span class="ot">&lt;-</span> <span class="fu">content</span>(response, <span class="at">as =</span> <span class="st">"text"</span>, <span class="at">encoding =</span> <span class="st">"UTF-8"</span>)</span>
<span id="cb35-228"><a href="#cb35-228" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write</span>(json_content, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"output_"</span>, <span class="fu">basename</span>(endpoint), <span class="st">".json"</span>))</span>
<span id="cb35-229"><a href="#cb35-229" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Downloaded:"</span>, endpoint, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb35-230"><a href="#cb35-230" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb35-231"><a href="#cb35-231" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Failed to download"</span>, endpoint, <span class="st">"Status code:"</span>, <span class="fu">status_code</span>(response), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb35-232"><a href="#cb35-232" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb35-233"><a href="#cb35-233" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-234"><a href="#cb35-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-235"><a href="#cb35-235" aria-hidden="true" tabindex="-1"></a>celltype_info <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(jsonlite<span class="sc">::</span><span class="fu">read_json</span>(<span class="st">"output_cell_type.json"</span>), <span class="cf">function</span>(l) {</span>
<span id="cb35-236"><a href="#cb35-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>(l)</span>
<span id="cb35-237"><a href="#cb35-237" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb35-238"><a href="#cb35-238" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb35-239"><a href="#cb35-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-240"><a href="#cb35-240" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(jsonlite<span class="sc">::</span><span class="fu">read_json</span>(<span class="st">"output_gene.json"</span>)) <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span>
<span id="cb35-241"><a href="#cb35-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-242"><a href="#cb35-242" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(jsonlite<span class="sc">::</span><span class="fu">read_json</span>(<span class="st">"output_gene.json"</span>), <span class="cf">function</span>(l) {</span>
<span id="cb35-243"><a href="#cb35-243" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">tibble</span>(</span>
<span id="cb35-244"><a href="#cb35-244" aria-hidden="true" tabindex="-1"></a>    <span class="at">versioned_ensembl_gene_id =</span> l[[<span class="st">"versioned_ensembl_gene_id"</span>]],</span>
<span id="cb35-245"><a href="#cb35-245" aria-hidden="true" tabindex="-1"></a>    <span class="at">gene_symbol =</span> l[[<span class="st">"gene_symbol"</span>]]</span>
<span id="cb35-246"><a href="#cb35-246" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb35-247"><a href="#cb35-247" aria-hidden="true" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb35-248"><a href="#cb35-248" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>()</span>
<span id="cb35-249"><a href="#cb35-249" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb35-250"><a href="#cb35-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-251"><a href="#cb35-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-252"><a href="#cb35-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-253"><a href="#cb35-253" aria-hidden="true" tabindex="-1"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>